{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import libraries\n",
    "import pandas as pd\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "# disable warnings, use w caution\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "# project specific libs\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import pathlib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "# set project path\n",
    "path = '/Users/KevinBu/Desktop/clemente_lab/Projects/twinsra/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "##### MP4 Mapping and Taxa Tables #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>species</th>\n",
       "      <th>k  Archaea|p  Candidatus Thermoplasmatota|c  Thermoplasmata|o  Methanomassiliicoccales|f  Methanomassiliicoccaceae|g  Methanomassiliicoccaceae unclassified|s  Methanomassiliicoccaceae unclassified SGB380</th>\n",
       "      <th>k  Archaea|p  Euryarchaeota|c  Methanobacteria|o  Methanobacteriales|f  Methanobacteriaceae|g  Methanobrevibacter|s  Methanobrevibacter smithii</th>\n",
       "      <th>k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Actinomyces|s  Actinomyces SGB17154</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Fannyhessea|s  Fannyhessea vaginae</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  GGB79819|s  GGB79819 SGB4702</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  GGB9832|s  GGB9832 SGB15478</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Schaalia|s  Schaalia turicensis</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Bifidobacteriales|f  Bifidobacteriaceae|g  Bifidobacterium|s  Bifidobacterium adolescentis</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Bifidobacteriales|f  Bifidobacteriaceae|g  Bifidobacterium|s  Bifidobacterium animalis</th>\n",
       "      <th>...</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Enterobacterales|f  Enterobacteriaceae|g  Klebsiella|s  Klebsiella variicola</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Moraxellales|f  Moraxellaceae|g  Acinetobacter|s  Acinetobacter SGB14792</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Moraxellales|f  Moraxellaceae|g  GGB72452|s  GGB72452 SGB4052</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Pasteurellales|f  Pasteurellaceae|g  Haemophilus|s  Haemophilus SGB47419</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Pasteurellales|f  Pasteurellaceae|g  Haemophilus|s  Haemophilus parainfluenzae</th>\n",
       "      <th>k  Bacteria|p  Synergistetes|c  Synergistia|o  Synergistales|f  Synergistaceae|g  GGB9469|s  GGB9469 SGB14863</th>\n",
       "      <th>k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1787|f  FGB1787|g  GGB4745|s  GGB4745 SGB6570</th>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia muciniphila</th>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp BIOML A40</th>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp KLE1798</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>0.0</td>\n",
       "      <td>431.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4350.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>3299.0</td>\n",
       "      <td>157290.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12451.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7046.0</td>\n",
       "      <td>2143.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9971.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>129889.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>780.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17148.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>45062.0</td>\n",
       "      <td>9347.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>194671.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 690 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "species  k  Archaea|p  Candidatus Thermoplasmatota|c  Thermoplasmata|o  Methanomassiliicoccales|f  Methanomassiliicoccaceae|g  Methanomassiliicoccaceae unclassified|s  Methanomassiliicoccaceae unclassified SGB380  \\\n",
       "500A                                                   0.0                                                                                                                                                             \n",
       "501A                                                3299.0                                                                                                                                                             \n",
       "502A                                                   0.0                                                                                                                                                             \n",
       "503A                                                   0.0                                                                                                                                                             \n",
       "504A                                                   0.0                                                                                                                                                             \n",
       "\n",
       "species  k  Archaea|p  Euryarchaeota|c  Methanobacteria|o  Methanobacteriales|f  Methanobacteriaceae|g  Methanobrevibacter|s  Methanobrevibacter smithii  \\\n",
       "500A                                                 431.0                                                                                                 \n",
       "501A                                              157290.0                                                                                                 \n",
       "502A                                                   0.0                                                                                                 \n",
       "503A                                                   0.0                                                                                                 \n",
       "504A                                                   0.0                                                                                                 \n",
       "\n",
       "species  k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899  \\\n",
       "500A                                                   0.0                                                                                                   \n",
       "501A                                                   0.0                                                                                                   \n",
       "502A                                                   0.0                                                                                                   \n",
       "503A                                                   0.0                                                                                                   \n",
       "504A                                               17148.0                                                                                                   \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Actinomyces|s  Actinomyces SGB17154  \\\n",
       "500A                                                   0.0                                                                              \n",
       "501A                                                   0.0                                                                              \n",
       "502A                                                   0.0                                                                              \n",
       "503A                                                   0.0                                                                              \n",
       "504A                                                   0.0                                                                              \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Fannyhessea|s  Fannyhessea vaginae  \\\n",
       "500A                                                   0.0                                                                             \n",
       "501A                                                   0.0                                                                             \n",
       "502A                                                   0.0                                                                             \n",
       "503A                                                 780.0                                                                             \n",
       "504A                                                   0.0                                                                             \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  GGB79819|s  GGB79819 SGB4702  \\\n",
       "500A                                                   0.0                                                                       \n",
       "501A                                                   0.0                                                                       \n",
       "502A                                                   0.0                                                                       \n",
       "503A                                                   0.0                                                                       \n",
       "504A                                                   0.0                                                                       \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  GGB9832|s  GGB9832 SGB15478  \\\n",
       "500A                                                   0.0                                                                      \n",
       "501A                                                   0.0                                                                      \n",
       "502A                                                   0.0                                                                      \n",
       "503A                                                   0.0                                                                      \n",
       "504A                                                   0.0                                                                      \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Schaalia|s  Schaalia turicensis  \\\n",
       "500A                                                   0.0                                                                          \n",
       "501A                                                   0.0                                                                          \n",
       "502A                                                   0.0                                                                          \n",
       "503A                                                   0.0                                                                          \n",
       "504A                                                   0.0                                                                          \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Bifidobacteriales|f  Bifidobacteriaceae|g  Bifidobacterium|s  Bifidobacterium adolescentis  \\\n",
       "500A                                                4350.0                                                                                              \n",
       "501A                                               12451.0                                                                                              \n",
       "502A                                                   0.0                                                                                              \n",
       "503A                                                   0.0                                                                                              \n",
       "504A                                               45062.0                                                                                              \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Bifidobacteriales|f  Bifidobacteriaceae|g  Bifidobacterium|s  Bifidobacterium animalis  \\\n",
       "500A                                                   0.0                                                                                          \n",
       "501A                                                   0.0                                                                                          \n",
       "502A                                                   0.0                                                                                          \n",
       "503A                                                   0.0                                                                                          \n",
       "504A                                                9347.0                                                                                          \n",
       "\n",
       "species  ...  \\\n",
       "500A     ...   \n",
       "501A     ...   \n",
       "502A     ...   \n",
       "503A     ...   \n",
       "504A     ...   \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Enterobacterales|f  Enterobacteriaceae|g  Klebsiella|s  Klebsiella variicola  \\\n",
       "500A                                                   0.0                                                                                      \n",
       "501A                                                   0.0                                                                                      \n",
       "502A                                                   0.0                                                                                      \n",
       "503A                                                   0.0                                                                                      \n",
       "504A                                                   0.0                                                                                      \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Moraxellales|f  Moraxellaceae|g  Acinetobacter|s  Acinetobacter SGB14792  \\\n",
       "500A                                                   0.0                                                                                  \n",
       "501A                                                7046.0                                                                                  \n",
       "502A                                                   0.0                                                                                  \n",
       "503A                                                   0.0                                                                                  \n",
       "504A                                                   0.0                                                                                  \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Moraxellales|f  Moraxellaceae|g  GGB72452|s  GGB72452 SGB4052  \\\n",
       "500A                                                   0.0                                                                       \n",
       "501A                                                2143.0                                                                       \n",
       "502A                                                   0.0                                                                       \n",
       "503A                                                   0.0                                                                       \n",
       "504A                                                   0.0                                                                       \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Pasteurellales|f  Pasteurellaceae|g  Haemophilus|s  Haemophilus SGB47419  \\\n",
       "500A                                                   0.0                                                                                  \n",
       "501A                                                   0.0                                                                                  \n",
       "502A                                                   0.0                                                                                  \n",
       "503A                                                   0.0                                                                                  \n",
       "504A                                                   0.0                                                                                  \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Pasteurellales|f  Pasteurellaceae|g  Haemophilus|s  Haemophilus parainfluenzae  \\\n",
       "500A                                                   0.0                                                                                        \n",
       "501A                                                9971.0                                                                                        \n",
       "502A                                                   0.0                                                                                        \n",
       "503A                                                   0.0                                                                                        \n",
       "504A                                                   0.0                                                                                        \n",
       "\n",
       "species  k  Bacteria|p  Synergistetes|c  Synergistia|o  Synergistales|f  Synergistaceae|g  GGB9469|s  GGB9469 SGB14863  \\\n",
       "500A                                                   0.0                                                               \n",
       "501A                                                   0.0                                                               \n",
       "502A                                                   0.0                                                               \n",
       "503A                                                   0.0                                                               \n",
       "504A                                                   0.0                                                               \n",
       "\n",
       "species  k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1787|f  FGB1787|g  GGB4745|s  GGB4745 SGB6570  \\\n",
       "500A                                                   0.0                                             \n",
       "501A                                                   0.0                                             \n",
       "502A                                                   0.0                                             \n",
       "503A                                                   0.0                                             \n",
       "504A                                                   0.0                                             \n",
       "\n",
       "species  k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia muciniphila  \\\n",
       "500A                                                   0.0                                                                                       \n",
       "501A                                              129889.0                                                                                       \n",
       "502A                                                   0.0                                                                                       \n",
       "503A                                                   0.0                                                                                       \n",
       "504A                                                   0.0                                                                                       \n",
       "\n",
       "species  k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp BIOML A40  \\\n",
       "500A                                                   0.0                                                                                        \n",
       "501A                                                   0.0                                                                                        \n",
       "502A                                                   0.0                                                                                        \n",
       "503A                                                   0.0                                                                                        \n",
       "504A                                              194671.0                                                                                        \n",
       "\n",
       "species  k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp KLE1798  \n",
       "500A                                                   0.0                                                                                     \n",
       "501A                                                   0.0                                                                                     \n",
       "502A                                                   0.0                                                                                     \n",
       "503A                                                   0.0                                                                                     \n",
       "504A                                                   0.0                                                                                     \n",
       "\n",
       "[5 rows x 690 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#### take metaphlan file and drop samples 516 and 517\n",
    "df_mp = pd.read_csv(path + 'inputs/metaphlan_taxa_table.tsv', sep='\\t', index_col=0)\n",
    "df_mp = df_mp.T\n",
    "# check counts df_mp.sum(axis=1)\n",
    "df_mp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Separate</th>\n",
       "      <th>SomeOtherInfo</th>\n",
       "      <th>Together</th>\n",
       "      <th>TwinPair</th>\n",
       "      <th>AliquotID</th>\n",
       "      <th>AliquotWeight</th>\n",
       "      <th>AliquotWeightUnit</th>\n",
       "      <th>SpecimenBodySite</th>\n",
       "      <th>...</th>\n",
       "      <th>StudyType</th>\n",
       "      <th>SubjectType</th>\n",
       "      <th>BirthYear</th>\n",
       "      <th>HostSubjectId</th>\n",
       "      <th>Nationality</th>\n",
       "      <th>Sex</th>\n",
       "      <th>SpecimenType</th>\n",
       "      <th>UberonCodeType</th>\n",
       "      <th>Weight</th>\n",
       "      <th>WeightDateCollected</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>#q2:types</th>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>...</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>GTTATGACGGAT</td>\n",
       "      <td>CCGGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Limit 45 Characters</td>\n",
       "      <td>NaN</td>\n",
       "      <td>500A.Fecal</td>\n",
       "      <td>T2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fecal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>CCAATGATAAGC</td>\n",
       "      <td>CCGGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>All</td>\n",
       "      <td>NaN</td>\n",
       "      <td>501A.Fecal</td>\n",
       "      <td>T2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>501</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fecal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>TCAATGACCGCA</td>\n",
       "      <td>CCGGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>All</td>\n",
       "      <td>NaN</td>\n",
       "      <td>502A.Fecal</td>\n",
       "      <td>T3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fecal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>CTCTCATATGCT</td>\n",
       "      <td>CCGGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>All</td>\n",
       "      <td>NaN</td>\n",
       "      <td>503A.Fecal</td>\n",
       "      <td>T3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fecal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BarcodeSequence    LinkerPrimerSequence             Separate  \\\n",
       "#SampleID                                                                \n",
       "#q2:types     categorical             categorical          categorical   \n",
       "500A         GTTATGACGGAT  CCGGACTACHVGGGTWTCTAAT  Limit 45 Characters   \n",
       "501A         CCAATGATAAGC  CCGGACTACHVGGGTWTCTAAT                  All   \n",
       "502A         TCAATGACCGCA  CCGGACTACHVGGGTWTCTAAT                  All   \n",
       "503A         CTCTCATATGCT  CCGGACTACHVGGGTWTCTAAT                  All   \n",
       "\n",
       "          SomeOtherInfo     Together     TwinPair    AliquotID AliquotWeight  \\\n",
       "#SampleID                                                                      \n",
       "#q2:types   categorical  categorical  categorical  categorical   categorical   \n",
       "500A                NaN   500A.Fecal           T2          NaN           NaN   \n",
       "501A                NaN   501A.Fecal           T2          NaN           NaN   \n",
       "502A                NaN   502A.Fecal           T3          NaN           NaN   \n",
       "503A                NaN   503A.Fecal           T3          NaN           NaN   \n",
       "\n",
       "          AliquotWeightUnit SpecimenBodySite  ...    StudyType  SubjectType  \\\n",
       "#SampleID                                     ...                             \n",
       "#q2:types       categorical      categorical  ...  categorical  categorical   \n",
       "500A                    NaN              NaN  ...          NaN        Human   \n",
       "501A                    NaN              NaN  ...          NaN        Human   \n",
       "502A                    NaN              NaN  ...          NaN        Human   \n",
       "503A                    NaN              NaN  ...          NaN        Human   \n",
       "\n",
       "             BirthYear HostSubjectId  Nationality          Sex SpecimenType  \\\n",
       "#SampleID                                                                     \n",
       "#q2:types  categorical   categorical  categorical  categorical  categorical   \n",
       "500A               NaN           500          NaN          NaN        fecal   \n",
       "501A               NaN           501          NaN          NaN        fecal   \n",
       "502A               NaN           502          NaN          NaN        fecal   \n",
       "503A               NaN           503          NaN          NaN        fecal   \n",
       "\n",
       "          UberonCodeType       Weight WeightDateCollected  \n",
       "#SampleID                                                  \n",
       "#q2:types    categorical  categorical         categorical  \n",
       "500A                 NaN          NaN                 NaN  \n",
       "501A                 NaN          NaN                 NaN  \n",
       "502A                 NaN          NaN                 NaN  \n",
       "503A                 NaN          NaN                 NaN  \n",
       "\n",
       "[5 rows x 93 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rename without Fecal suffix for ensemble fastq's\n",
    "# grab mapping file, partition samples and controls\n",
    "df_map = pd.read_csv(path + 'inputs/Qiime2_0_KB/qiime_mapping_file_noctrl_no182183.tsv', sep='\\t', index_col=0)\n",
    "df_map = df_map.reset_index()\n",
    "new_sid = [x.replace('.Fecal','') for x in list(df_map['#SampleID'].values)]\n",
    "df_map['#SampleID'] = new_sid\n",
    "df_map = df_map.set_index('#SampleID')\n",
    "df_map.to_csv(path + 'inputs/Qiime2_1_KB/qiime_mapping_file_noctrl_no182183_ensemble.tsv', sep='\\t')\n",
    "\n",
    "# create new ensemble mapping file for Q2\n",
    "df_map = pd.read_csv(path + 'inputs/qiime_mapping_file_noctrl_no182183_ensemble.tsv', sep='\\t', index_col=0)\n",
    "df_map = df_map.drop(['516A','517A'])\n",
    "df_map.to_csv(path + 'inputs/qiime_mapping_file_noctrl_no182183_no516517_ensemble.tsv', sep='\\t', index=True)\n",
    "df_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>species</th>\n",
       "      <th>k  Archaea|p  Candidatus Thermoplasmatota|c  Thermoplasmata|o  Methanomassiliicoccales|f  Methanomassiliicoccaceae|g  Methanomassiliicoccaceae unclassified|s  Methanomassiliicoccaceae unclassified SGB380</th>\n",
       "      <th>k  Archaea|p  Euryarchaeota|c  Methanobacteria|o  Methanobacteriales|f  Methanobacteriaceae|g  Methanobrevibacter|s  Methanobrevibacter smithii</th>\n",
       "      <th>k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Actinomyces|s  Actinomyces SGB17154</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Fannyhessea|s  Fannyhessea vaginae</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  GGB79819|s  GGB79819 SGB4702</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  GGB9832|s  GGB9832 SGB15478</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Schaalia|s  Schaalia turicensis</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Bifidobacteriales|f  Bifidobacteriaceae|g  Bifidobacterium|s  Bifidobacterium adolescentis</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Bifidobacteriales|f  Bifidobacteriaceae|g  Bifidobacterium|s  Bifidobacterium animalis</th>\n",
       "      <th>...</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Enterobacterales|f  Enterobacteriaceae|g  Klebsiella|s  Klebsiella variicola</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Moraxellales|f  Moraxellaceae|g  Acinetobacter|s  Acinetobacter SGB14792</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Moraxellales|f  Moraxellaceae|g  GGB72452|s  GGB72452 SGB4052</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Pasteurellales|f  Pasteurellaceae|g  Haemophilus|s  Haemophilus SGB47419</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Pasteurellales|f  Pasteurellaceae|g  Haemophilus|s  Haemophilus parainfluenzae</th>\n",
       "      <th>k  Bacteria|p  Synergistetes|c  Synergistia|o  Synergistales|f  Synergistaceae|g  GGB9469|s  GGB9469 SGB14863</th>\n",
       "      <th>k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1787|f  FGB1787|g  GGB4745|s  GGB4745 SGB6570</th>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia muciniphila</th>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp BIOML A40</th>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp KLE1798</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000299</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>0.000239</td>\n",
       "      <td>0.011409</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000903</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000511</td>\n",
       "      <td>0.000155</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000723</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009421</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00129</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003391</td>\n",
       "      <td>0.000703</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014649</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 690 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "species  k  Archaea|p  Candidatus Thermoplasmatota|c  Thermoplasmata|o  Methanomassiliicoccales|f  Methanomassiliicoccaceae|g  Methanomassiliicoccaceae unclassified|s  Methanomassiliicoccaceae unclassified SGB380  \\\n",
       "500A                                              0.000000                                                                                                                                                             \n",
       "501A                                              0.000239                                                                                                                                                             \n",
       "502A                                              0.000000                                                                                                                                                             \n",
       "503A                                              0.000000                                                                                                                                                             \n",
       "504A                                              0.000000                                                                                                                                                             \n",
       "\n",
       "species  k  Archaea|p  Euryarchaeota|c  Methanobacteria|o  Methanobacteriales|f  Methanobacteriaceae|g  Methanobrevibacter|s  Methanobrevibacter smithii  \\\n",
       "500A                                              0.000030                                                                                                 \n",
       "501A                                              0.011409                                                                                                 \n",
       "502A                                              0.000000                                                                                                 \n",
       "503A                                              0.000000                                                                                                 \n",
       "504A                                              0.000000                                                                                                 \n",
       "\n",
       "species  k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899  \\\n",
       "500A                                               0.00000                                                                                                   \n",
       "501A                                               0.00000                                                                                                   \n",
       "502A                                               0.00000                                                                                                   \n",
       "503A                                               0.00000                                                                                                   \n",
       "504A                                               0.00129                                                                                                   \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Actinomyces|s  Actinomyces SGB17154  \\\n",
       "500A                                                   0.0                                                                              \n",
       "501A                                                   0.0                                                                              \n",
       "502A                                                   0.0                                                                              \n",
       "503A                                                   0.0                                                                              \n",
       "504A                                                   0.0                                                                              \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Fannyhessea|s  Fannyhessea vaginae  \\\n",
       "500A                                              0.000000                                                                             \n",
       "501A                                              0.000000                                                                             \n",
       "502A                                              0.000000                                                                             \n",
       "503A                                              0.000075                                                                             \n",
       "504A                                              0.000000                                                                             \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  GGB79819|s  GGB79819 SGB4702  \\\n",
       "500A                                                   0.0                                                                       \n",
       "501A                                                   0.0                                                                       \n",
       "502A                                                   0.0                                                                       \n",
       "503A                                                   0.0                                                                       \n",
       "504A                                                   0.0                                                                       \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  GGB9832|s  GGB9832 SGB15478  \\\n",
       "500A                                                   0.0                                                                      \n",
       "501A                                                   0.0                                                                      \n",
       "502A                                                   0.0                                                                      \n",
       "503A                                                   0.0                                                                      \n",
       "504A                                                   0.0                                                                      \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Schaalia|s  Schaalia turicensis  \\\n",
       "500A                                                   0.0                                                                          \n",
       "501A                                                   0.0                                                                          \n",
       "502A                                                   0.0                                                                          \n",
       "503A                                                   0.0                                                                          \n",
       "504A                                                   0.0                                                                          \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Bifidobacteriales|f  Bifidobacteriaceae|g  Bifidobacterium|s  Bifidobacterium adolescentis  \\\n",
       "500A                                              0.000299                                                                                              \n",
       "501A                                              0.000903                                                                                              \n",
       "502A                                              0.000000                                                                                              \n",
       "503A                                              0.000000                                                                                              \n",
       "504A                                              0.003391                                                                                              \n",
       "\n",
       "species  k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Bifidobacteriales|f  Bifidobacteriaceae|g  Bifidobacterium|s  Bifidobacterium animalis  \\\n",
       "500A                                              0.000000                                                                                          \n",
       "501A                                              0.000000                                                                                          \n",
       "502A                                              0.000000                                                                                          \n",
       "503A                                              0.000000                                                                                          \n",
       "504A                                              0.000703                                                                                          \n",
       "\n",
       "species  ...  \\\n",
       "500A     ...   \n",
       "501A     ...   \n",
       "502A     ...   \n",
       "503A     ...   \n",
       "504A     ...   \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Enterobacterales|f  Enterobacteriaceae|g  Klebsiella|s  Klebsiella variicola  \\\n",
       "500A                                                   0.0                                                                                      \n",
       "501A                                                   0.0                                                                                      \n",
       "502A                                                   0.0                                                                                      \n",
       "503A                                                   0.0                                                                                      \n",
       "504A                                                   0.0                                                                                      \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Moraxellales|f  Moraxellaceae|g  Acinetobacter|s  Acinetobacter SGB14792  \\\n",
       "500A                                              0.000000                                                                                  \n",
       "501A                                              0.000511                                                                                  \n",
       "502A                                              0.000000                                                                                  \n",
       "503A                                              0.000000                                                                                  \n",
       "504A                                              0.000000                                                                                  \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Moraxellales|f  Moraxellaceae|g  GGB72452|s  GGB72452 SGB4052  \\\n",
       "500A                                              0.000000                                                                       \n",
       "501A                                              0.000155                                                                       \n",
       "502A                                              0.000000                                                                       \n",
       "503A                                              0.000000                                                                       \n",
       "504A                                              0.000000                                                                       \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Pasteurellales|f  Pasteurellaceae|g  Haemophilus|s  Haemophilus SGB47419  \\\n",
       "500A                                                   0.0                                                                                  \n",
       "501A                                                   0.0                                                                                  \n",
       "502A                                                   0.0                                                                                  \n",
       "503A                                                   0.0                                                                                  \n",
       "504A                                                   0.0                                                                                  \n",
       "\n",
       "species  k  Bacteria|p  Proteobacteria|c  Gammaproteobacteria|o  Pasteurellales|f  Pasteurellaceae|g  Haemophilus|s  Haemophilus parainfluenzae  \\\n",
       "500A                                              0.000000                                                                                        \n",
       "501A                                              0.000723                                                                                        \n",
       "502A                                              0.000000                                                                                        \n",
       "503A                                              0.000000                                                                                        \n",
       "504A                                              0.000000                                                                                        \n",
       "\n",
       "species  k  Bacteria|p  Synergistetes|c  Synergistia|o  Synergistales|f  Synergistaceae|g  GGB9469|s  GGB9469 SGB14863  \\\n",
       "500A                                                   0.0                                                               \n",
       "501A                                                   0.0                                                               \n",
       "502A                                                   0.0                                                               \n",
       "503A                                                   0.0                                                               \n",
       "504A                                                   0.0                                                               \n",
       "\n",
       "species  k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1787|f  FGB1787|g  GGB4745|s  GGB4745 SGB6570  \\\n",
       "500A                                                   0.0                                             \n",
       "501A                                                   0.0                                             \n",
       "502A                                                   0.0                                             \n",
       "503A                                                   0.0                                             \n",
       "504A                                                   0.0                                             \n",
       "\n",
       "species  k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia muciniphila  \\\n",
       "500A                                              0.000000                                                                                       \n",
       "501A                                              0.009421                                                                                       \n",
       "502A                                              0.000000                                                                                       \n",
       "503A                                              0.000000                                                                                       \n",
       "504A                                              0.000000                                                                                       \n",
       "\n",
       "species  k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp BIOML A40  \\\n",
       "500A                                              0.000000                                                                                        \n",
       "501A                                              0.000000                                                                                        \n",
       "502A                                              0.000000                                                                                        \n",
       "503A                                              0.000000                                                                                        \n",
       "504A                                              0.014649                                                                                        \n",
       "\n",
       "species  k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp KLE1798  \n",
       "500A                                                   0.0                                                                                     \n",
       "501A                                                   0.0                                                                                     \n",
       "502A                                                   0.0                                                                                     \n",
       "503A                                                   0.0                                                                                     \n",
       "504A                                                   0.0                                                                                     \n",
       "\n",
       "[5 rows x 690 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in metaphlan taxa table\n",
    "# both appear to be abs\n",
    "df_otu = pd.read_csv(path + 'outputs/jobs00/metaphlan_taxa_table_abs.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "# create taxonomy file\n",
    "df_taxa = df_otu.reset_index()\n",
    "\n",
    "# copy species column\n",
    "df_taxa.iloc[:,1] = df_taxa['species'].values\n",
    "\n",
    "# rename columns\n",
    "df_taxa = df_taxa.rename({'species': 'Feature ID','500A': 'Taxon', '501A': 'Confidence'},axis='columns')\n",
    "\n",
    "# set confidence level\n",
    "df_taxa['Confidence'] = 0.99\n",
    "\n",
    "# remove remaining columns\n",
    "df_taxa = df_taxa.iloc[:,:3]\n",
    "\n",
    "# set index\n",
    "df_taxa = df_taxa.set_index('Feature ID')\n",
    "\n",
    "# format taxa column\n",
    "df_taxa['Taxon'] = df_taxa['Taxon'].map(lambda x: x.replace(' ','_'))\n",
    "df_taxa['Taxon'] = df_taxa['Taxon'].map(lambda x: x.replace('|','; '))\n",
    "#df_taxa['Taxon'] = df_taxa['Taxon'].map(lambda x: 'g__' + x)\n",
    "#df_taxa['Taxon'] = df_taxa['Taxon'].map(lambda x: x.replace(' ', ';s__'))\n",
    "\n",
    "# export as tsv\n",
    "df_taxa.to_csv(path + 'outputs/jobs03/df_taxonomy.tsv', sep='\\t')\n",
    "\n",
    "# transpose and normalize\n",
    "df_otu = df_otu.T\n",
    "df_otu = df_otu.div(df_otu.sum(axis=1),axis=0)\n",
    "\n",
    "# grab top 100\n",
    "sorted_means = df_otu.mean(axis=0).sort_values(ascending=False)\n",
    "top_100 = sorted_means[:100].index.values\n",
    "df_otu_top100 = df_otu.loc[:,top_100]\n",
    "df_otu_top100.to_csv(path + 'inputs/df_otu_top100.tsv' ,sep='\\t')\n",
    "\n",
    "df_otu.to_csv(path + 'inputs/df_otu_mp4.tsv' ,sep='\\t')\n",
    "df_otu.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TwinPair</th>\n",
       "      <th>Diagnosis</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>T2</td>\n",
       "      <td>RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>T2</td>\n",
       "      <td>Unaffected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>T3</td>\n",
       "      <td>RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>T3</td>\n",
       "      <td>Unaffected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>T4</td>\n",
       "      <td>RA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          TwinPair   Diagnosis\n",
       "#SampleID                     \n",
       "500A            T2          RA\n",
       "501A            T2  Unaffected\n",
       "502A            T3          RA\n",
       "503A            T3  Unaffected\n",
       "504A            T4          RA"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# grab mapping file, partition samples and controls\n",
    "df_map = pd.read_csv(path + 'inputs/qiime_mapping_file_noctrl_no182183_no516517_ensemble.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "# get first row for any future operations\n",
    "first_row = df_map.reset_index().iloc[0,]\n",
    "first_row_df = pd.DataFrame(first_row).T\n",
    "df_map = df_map.iloc[1:,]\n",
    "\n",
    "# remove separate and together columns\n",
    "df_map = df_map.drop(['Separate','Together'], axis=1)\n",
    "\n",
    "# subset on useful columns\n",
    "# TwinPair and Diagnosis are useful columns\n",
    "df_map = df_map.loc[:,['TwinPair','Diagnosis']]\n",
    "\n",
    "df_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RA_mean</th>\n",
       "      <th>UA_mean</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Species</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Methanomassiliicoccaceae unclassified SGB380</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Methanobrevibacter smithii</th>\n",
       "      <td>0.000582</td>\n",
       "      <td>0.001441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nitrosopumilus SGB14899</th>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.000072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Actinomyces SGB17154</th>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fannyhessea vaginae</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000009</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 RA_mean   UA_mean\n",
       "Species                                                           \n",
       "  Methanomassiliicoccaceae unclassified SGB380  0.000000  0.000030\n",
       "  Methanobrevibacter smithii                    0.000582  0.001441\n",
       "  Nitrosopumilus SGB14899                       0.000361  0.000072\n",
       "  Actinomyces SGB17154                          0.000003  0.000000\n",
       "  Fannyhessea vaginae                           0.000000  0.000009"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# q2R for stacked barplot\n",
    "RA = list(df_map[df_map['Diagnosis'] == 'RA'].index.values)\n",
    "UA = list(df_map[df_map['Diagnosis'] == 'Unaffected'].index.values)\n",
    "\n",
    "df_q2R = df_otu.copy().T\n",
    "df_q2R['UA_mean'] = df_q2R[UA].mean(axis=1)\n",
    "df_q2R['RA_mean'] = df_q2R[RA].mean(axis=1)\n",
    "df_q2R = df_q2R[['RA_mean','UA_mean']]\n",
    "df_q2R.index.name = 'Species'\n",
    "df_q2R.index = df_q2R.index.map(lambda x: x.split('|s')[1])\n",
    "df_q2R.to_csv(path + 'inputs/df_q2R.tsv', sep='\\t')\n",
    "df_q2R.head()\n",
    "# df_q2R\n",
    "#df_q2R.sort_values(by='RA_mean',ascending=False).iloc[0:10,:]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "##### Alpha and Beta Div #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.4605094942007545 0.6591148429056253\n",
      "17.0 0.9453125\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjcAAAGwCAYAAABVdURTAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAAA4sUlEQVR4nO3deXhU9cH28XuyLyRhTwJkAZEtrILKIloKgoAIwuMK4gYUpbKLRaqABrEVFC0tGIiyWaFvQcq+iASqgmXzETWEnWAMRLaEJZlJMvP+wcPUMQkmw5AzOfl+rmuucn5nmXtsljtntTgcDocAAABMwsfoAAAAAJ5EuQEAAKZCuQEAAKZCuQEAAKZCuQEAAKZCuQEAAKZCuQEAAKZSKcuNw+FQTk6OuMUPAADmUynLzcWLFxUREaGLFy8aHQUAAHhYpSw3AADAvCg3AADAVCg3AADAVCg3AADAVCg3AADAVCg3AADAVCg3AADAVCg3AADAVCg3AADAVCg3AADAVCg3AADAVCg3AADAVPyMDgDcLDabTRs3blRaWpri4+PVu3dvBQcHGx0LAHCTWRwOh8PoEOUtJydHERERys7OVnh4uNFxcBOcO3dOgwcP1qFDh5xjdevW1eLFi1W3bl0DkwEAbjYOS8GUZs+e7VJsJCkjI0N//vOfDUoEACgvHJZCifLy8pSenm50DLds2LCh2PFPP/1UBw8eLOc0nhMbG6ugoCCjYwCAV+OwFIelSnTw4EENGzbM6BhuycrKUmFhYZFxi8WiqKgoAxJ5RlJSkho1amR0DADwapQbyk2JKvKemyVLlmjZsmVFxu+77z6NGDHCgESewZ4bAPh1lBvKjSlZrVaNGjVKW7dudY4lJCRo4cKFCgsLMzAZAOBmo9xQbkztu+++07Zt27RkyRItWLCAQzoAUAkYfrVURkaGBg0apBo1aigkJEStW7fWnj17rrvOtm3b1LZtWwUFBalBgwaaO3duOaVFRZOQkKBu3bopICDA6CgAgHJi6NVS58+fV6dOndSlSxetX79etWvX1pEjR1S1atUS1zl27Jh69eqloUOHasmSJfriiy/0/PPPq1atWhowYED5hQcAAF7J0HLzpz/9STExMfrwww+dY/Hx8dddZ+7cuYqNjdWsWbMkSU2bNtXu3bs1Y8aMEsuN1WqV1Wp1Tufk5NxwdgAA4J0MPSy1atUqtWvXTg899JBq166tNm3aaN68edddZ8eOHerevbvLWI8ePbR7927l5+cXu8706dMVERHhfMXExHjsMwAAAO9iaLk5evSo5syZo1tvvVUbN27U8OHDNXLkSC1atKjEdU6dOqXIyEiXscjISBUUFOjMmTPFrjNx4kRlZ2c7XydPnvTo5wAAAN7D0MNSdrtd7dq10xtvvCFJatOmjb777jvNmTNHgwcPLnE9i8XiMn3tgq9fjl8TGBiowMBAD6UGAADezNA9N9HR0WrWrJnLWNOmTa9747ioqCidOnXKZSwrK0t+fn6qUaPGTckJAAAqDkPLTadOnZSWluYydvDgQcXFxZW4TocOHbR582aXsU2bNqldu3by9/e/KTkBAEDFYWi5GTNmjHbu3Kk33nhDhw8f1t///nclJSW53B5/4sSJLoeohg8frhMnTmjs2LFKTU3VBx98oOTkZI0fP96IjwAAALyMoeXm9ttv1yeffKKPP/5YzZs31+uvv65Zs2Zp4MCBzmUyMzNdDlPVr19f69atU0pKilq3bq3XX39d7733Hve4AQAAknj8Ao9fqASuPd2cJ2oDQOVg+OMXAAAAPIlyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATIVyAwAATMXQcjNlyhRZLBaXV1RUVInLp6SkFFneYrHowIED5ZgaAAB4Mz+jAyQkJOjTTz91Tvv6+v7qOmlpaQoPD3dO16pV66ZkAwAAFY/h5cbPz++6e2uKU7t2bVWtWrXUy1utVlmtVud0Tk5Omd4PAABUHIafc3Po0CHVqVNH9evX16OPPqqjR4/+6jpt2rRRdHS0unbtqq1bt/7q8tOnT1dERITzFRMT44noAADACxlabu68804tWrRIGzdu1Lx583Tq1Cl17NhRZ8+eLXb56OhoJSUlafny5VqxYoUaN26srl27avv27dd9n4kTJyo7O9v5Onny5M34OAAAwAtYHA6Hw+gQ11y+fFm33HKLJkyYoLFjx5ZqnT59+shisWjVqlWlfp+cnBxFREQoOzvb5dwdmNPBgwc1bNgwJSUlqVGjRkbHAQDcZIYflvq50NBQtWjRQocOHSr1Ou3bty/T8gAAwNy8qtxYrValpqYqOjq61Ovs27evTMsDAABzM/RqqfHjx6tPnz6KjY1VVlaWEhMTlZOToyeffFLS1XNlMjIytGjRIknSrFmzFB8fr4SEBNlsNi1ZskTLly/X8uXLjfwYAADAixhabn744Qc99thjOnPmjGrVqqX27dtr586diouLkyRlZmYqPT3dubzNZtP48eOVkZGh4OBgJSQkaO3aterVq5dRHwEAAHgZrzqhuLxwQnHlwgnFAFC5eNU5NwAAADeKcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAAEyFcgMAqJDOnz+vAwcO6KeffjI6CryMn9EBAAAoC7vdrlWrVul///d/nWONGzfWgAED5O/vb2AyeAv23AAAKpQvvvjCpdhIUlpamrZs2WJQIngb9twAQCWUl5en9PR0o2O4ZdeuXcWO7927Vw0aNCjnNJ4TGxuroKAgo2OYAuUGACqh9PR0DRs2zOgYbunYsaMCAwOLjFut1gr7mSQpKSlJjRo1MjqGKVBuAKASio2NVVJSktEx3PLVV1/p6NGjRcZjYmIq7GeSrv5/As+g3ABAJRQUFFRh9xJERUXpww8/1IULF5xjQUFB6tevn2rWrGlcMHgNt04onjJlik6cOOHpLAAAFFFQUKBDhw4pLS1NNptN4eHhGj58uHr27KlbbrlFhw8fVq9evSg2cHJrz83q1auVmJioe+65R88++6z69+/PSVAAAI87duyY/vnPf+rKlSuSpMDAQD3wwANq1qyZ7rjjDlWtWlXz588v9hwcVF5u7bnZs2eP9u7dq5YtW2rMmDGKjo7Wc889V+IZ7AAAlJXNZtM//vEPZ7GRrp40vGLFCl28eNHAZPB2bt/npmXLlnrnnXeUkZGhDz74QBkZGerUqZNatGihd999V9nZ2Z7MCbjl0KFD2rZtm/Lz842OAqCM0tLSlJeXV2S8sLBQ3377rQGJUFHc8E387Ha7bDabrFarHA6Hqlevrjlz5igmJkbLli3zREagzGw2m0aOHKn7779fM2bM0JkzZ/TKK6/o0qVLRkcDUErX+6PEZrOVYxJUNG6Xmz179uj3v/+9oqOjNWbMGLVp00apqanatm2bDhw4oMmTJ2vkyJGezAqUWlJSkjZu3Ogy9vXXX2vmzJkGJQJQVg0bNpSPT/G/piIjI7V27Vpt3rxZTZo00fnz58s5HbyZxeFwOMq6UsuWLZWamqru3btr6NCh6tOnj3x9fV2W+emnnxQZGSm73e6xsJ6Sk5OjiIgIZWdnKzw83Og4uAnuvffeYu++WqVKFe3Zs8eARADc8eWXX2rz5s0uYy1btlRaWpqsVqtzzMfHR4MHD1ZcXFx5R4QXcutqqYceekjPPPOM6tatW+IytWrV8spig8rh5z/0SjMOlMXp06c5r7Cc1KxZU/fdd59OnDghh8OhevXq6cCBA0W+l+12u9auXatu3boZlBSSFBERocjISKNjuFduXnnlFee/r+34sVgsnkkEeECXLl20dOnSIuO//e1vDUgDMzl9+rQGPTFY+TaKslE6deqkgICAIuNZWVkV+vELZuAfEKglixcZXnDcvkNxcnKy3nnnHR06dEiSdOutt2r06NEaMmSIx8IB7nrhhRf01Vdf6dixY86xmjVr6sUXXzQwFcwgOztb+TarchvcI3tQhNFxKqVcy2UFqOiRgTy7jy43e8CARJAkn7xs6eg2ZWdnV8xy88orr+idd97RCy+8oA4dOkiSduzYoTFjxuj48eNKTEz0aEigrGrWrKmVK1dq3bp12rFjh1JSUvTXv/5VMTExRkeDSdiDImQP5Y64RjhaGKg2/meKjB8rqCp7aNXyDwSv41a5mTNnjubNm6fHHnvMOfbAAw+oZcuWeuGFFyg38ApBQUHq37+/mjdvrl27dikkJMToSAA84GRBmAKthWoYkK0Ai135DouO5YfrcD570nCVW+WmsLBQ7dq1KzLetm1bFRQU3HAoAACu53B+VR3ND1ewpVB5Dl8V3vht22Aibn01DBo0SHPmzCkynpSUpIEDB95wKAAAfo1dPrrs8KfYoIgbOqF406ZNat++vSRp586dOnnypAYPHqyxY8c6l3v77bdvPCUAAEApuVVuvv32W912222SpCNHjki6el+bWrVquTzvg8vDAQBAeXOr3GzdutXTOQAAADzC7cNS1/zwww+yWCzXvVtxZcQdTL3HiRMnXP4XxvKWO5gCMC+3yo3dbldiYqJmzpzpfMpyWFiYxo0bp0mTJpX4oLPKgjuYlh+Hw6ErV64oLy9PDodDQUFBCg0NLfaQ6LRp0wxIiF/yljuYAjAvt8rNpEmTlJycrDfffFOdOnWSw+HQF198oSlTpigvL6/S/xLhDqbl58q3W1SQc8o5nZ+fryv+EQpp3YtzvryQN93BFIB5uVVuFi5cqPnz5+uBB/57m+tWrVqpbt26ev755yt9ubmGO5jeXIXZp1Rw+nDR8XM/KP/KRfnVbmBAKgCA0dwqN+fOnVOTJk2KjDdp0kTnzp274VBAadjPZZQ873yGHBFRyk/fJ3tOliyh1eUf10Y+weHlmBAAYAS3To5p1aqVZs+eXWR89uzZatWq1Q2HAkrDEhxW8kxfP13ZNk+21BQVZHyv/IOf60rKPBVmny6/gAAAQ7i15+bPf/6zevfurU8//VQdOnSQxWLRl19+qZMnT2rdunWezggUy7d2Q1lCq8tx2XVvoSUwVIUXTsuRd8l1hfw82VK3Krj9o+WYEgBQ3tzac3PPPffo4MGDevDBB3XhwgWdO3dO/fv3V1pamjp37uzpjECxLD4+Cu7wmHxr1XeO+VSPUVCHgbKfPV7sOoU/HS2ndAAAo5R5z01+fr66d++u999/nxOHYTifkKoK7vC4HLYrcjgc8gkMvTrDP1iyXi66gn9w+QYEAJS7Mu+58ff317fffuuRy2ynTJkii8Xi8oqKirruOtu2bVPbtm0VFBSkBg0aaO7cuTecAxWfJSDkv8VGkn9c62KX84/lnDAAMDu3DksNHjxYycnJHgmQkJCgzMxM52v//v0lLnvs2DH16tVLnTt31r59+/Tyyy9r5MiRWr58uUeywDz8G9wh//i2kuXal7hFfvWaK6Dx3YbmAgDcfG6dUGyz2TR//nxt3rxZ7dq1U2hoqMv8sjwJ3M/P71f31lwzd+5cxcbGatasWZKkpk2bavfu3ZoxY4YGDBhQ6veE+VksFgW2vE/+je6S/dIZ+YRW5zJwAKgkbvip4AcPHryhAIcOHVKdOnUUGBioO++8U2+88YYaNCj+5ms7duxQ9+7dXcZ69Oih5ORk5efny9/fv9j1rFarrNb/PgohJyfnhjKj4vAJqiKfoCpGxwAAlCNDnwp+5513atGiRWrUqJFOnz6txMREdezYUd99951q1KhRZPlTp04VuWV7ZGSkCgoKdObMGUVHRxf7PtOnT9fUqVM9khkAAHg3t865eeaZZ3Tx4sUi45cvX9YzzzxT6u307NlTAwYMUIsWLdStWzetXbtW0tXHO5TklycyOxyOYsd/buLEicrOzna+Tp48WeqMAACgYnGr3CxcuFC5ublFxnNzc7Vo0SK3w4SGhqpFixY6dOhQsfOjoqJ06tQpl7GsrCz5+fkVu6fnmsDAQIWHh7u8AACAOZXpsFROTo4cDoccDocuXryooKAg57zCwkKtW7dOtWvXdjuM1WpVampqiTcC7NChg1avXu0ytmnTJrVr167E820AAEDlUqZyU7VqVef9aBo1alRkvsViKdO5LePHj1efPn0UGxurrKwsJSYmKicnR08++aSkq4eTMjIynHuDhg8frtmzZ2vs2LEaOnSoduzYoeTkZH388cdl+RgAAMDEylRutm7dKofDod/+9rdavny5qlev7pwXEBCguLg41alTp9Tb++GHH/TYY4/pzJkzqlWrltq3b6+dO3cqLi5OkpSZman09HTn8vXr19e6des0ZswY/fWvf1WdOnX03nvvcRk4AABwKlO5ueeeeyRdvZleTEyMfHzcOmXHaenSpdedv2DBgmIz7N2794beFwAAmJdbl4LHxcXpwoUL+s9//qOsrCzZ7XaX+YMHD/ZIOAAAgLJyq9ysXr1aAwcO1OXLlxUWFuZyGbbFYqHcAAAAw7h1XGncuHHOe91cuHBB58+fd77OnTvn6YwAAACl5la5ycjI0MiRIxUSEuLpPAAAADfErXLTo0cP7d6929NZAAAAbphb59z07t1bL774or7//nu1aNGiyA30HnjgAY+EAwAAKCu3ys3QoUMlSa+99lqReRaLRYWFhTeWCgAAwE1ulZtfXvoNAADgLW7sLnyS8vLyPJEDAADAI9wqN4WFhXr99ddVt25dValSRUePHpUkvfLKK0pOTvZoQAAAgLJwq9xMmzZNCxYs0J///GcFBAQ4x1u0aKH58+d7LBwAAEBZuVVuFi1apKSkJA0cOFC+vr7O8ZYtW+rAgQMeCwcAAFBWbt/Er2HDhkXG7Xa78vPzbzgUAACAu9wqNwkJCfr3v/9dZPz//b//pzZt2txwKAAAAHe5dSn45MmT9cQTTygjI0N2u10rVqxQWlqaFi1apDVr1ng6IwAAQKm5teemT58+WrZsmdatWyeLxaJXX31VqampWr16te69915PZwQAACg1t/bcSFefL9WjR4/rLvPxxx/rgQceUGhoqLtvAwAAUCY3fBO/6/nd736n06dP38y3AAAAcHFTy43D4biZmwcAACjippYbAACA8ka5AQAApkK5AQAApkK5AQAApnJTy01cXJz8/f1v5lsAAAC4cPs+N5Jks9mUlZUlu93uMh4bGytJ+vbbb29k8wAAAGXmVrk5dOiQnnnmGX355Zcu4w6HQxaLRYWFhR4JBwAAUFZulZunnnpKfn5+WrNmjaKjo2WxWDydCwAAwC1ulZuvv/5ae/bsUZMmTTydBwAA4Ia4dUJxs2bNdObMGU9nAQAAuGFulZs//elPmjBhglJSUnT27Fnl5OS4vAAAAIzi1mGpbt26SZK6du3qMs4JxQAAwGhulZutW7d6OgcAAIBHuFVu7rnnHk/nAAAA8Ai3b+J34cIFJScnKzU1VRaLRc2aNdMzzzyjiIgIT+YDAAAoE7dOKN69e7duueUWvfPOOzp37pzOnDmjt99+W7fccov27t3r6YwAAACl5taemzFjxuiBBx7QvHnz5Od3dRMFBQUaMmSIRo8ere3bt3s0JAAAQGm5VW52797tUmwkyc/PTxMmTFC7du08Fg4AAKCs3DosFR4ervT09CLjJ0+eVFhY2A2HAgAAcJdb5eaRRx7Rs88+q2XLlunkyZP64YcftHTpUg0ZMkSPPfaYpzMCAACUmluHpWbMmCGLxaLBgweroKBAkuTv76/nnntOb775pkcDAgAAlIVb5SYgIEDvvvuupk+friNHjsjhcKhhw4YKCQnxdD4AAIAycfs+N5IUEhKiFi1aeCoLAADADXOr3Fy+fFlvvvmmtmzZoqysLNntdpf5R48e9Ug4AACAsnKr3AwZMkTbtm3TE088oejoaFksFk/nAgAAcItb5Wb9+vVau3atOnXq5Ok8puKTe8HoCIBX4XsCQHlwq9xUq1ZN1atX93QW0wk+xp2aAQAob26Vm9dff12vvvqqFi5cyBVS15Fb/27Zg6saHQPwGj65Fyj9AG46t8rNzJkzdeTIEUVGRio+Pl7+/v4u83l45lX24Kqyh9Y0OgYAAJWKW+WmX79+Ho4BAADgGW6Vm8mTJ3s6BwAAgEfc0E38bDZbsfe5iY2NvaFQAAAA7nKr3Bw8eFDPPvusvvzyS5dxh8Mhi8WiwsJCj4QDAAAoK7fKzdNPPy0/Pz+tWbOGm/gBAACv4la5+frrr7Vnzx41adLE03kAAABuiI87KzVr1kxnzpzxdBYAAIAb5la5+dOf/qQJEyYoJSVFZ8+eVU5OjssLAADAKG6Vm27dumnnzp3q2rWrateurWrVqqlatWqqWrWqqlWr5laQ6dOny2KxaPTo0SUuk5KSIovFUuR14MABt94TAACYj1vn3GzdutWjIXbt2qWkpCS1bNmyVMunpaUpPDzcOV2rVi2P5gEAABWXW+Xmnnvu8ViAS5cuaeDAgZo3b54SExNLtU7t2rVVtWpVj2UAAADm4dZhqWuuXLmiAwcO6JtvvnF5lcWIESPUu3dvdevWrdTrtGnTRtHR0eratWup9iJZrVbOCwIAoJJwa8/NTz/9pKefflrr168vdn5pb+K3dOlS7d27V7t27SrV8tHR0UpKSlLbtm1ltVq1ePFide3aVSkpKbr77rtLXG/69OmaOnVqqd4DAABUbG6Vm9GjR+v8+fPauXOnunTpok8++USnT59WYmKiZs6cWaptnDx5UqNGjdKmTZsUFBRUqnUaN26sxo0bO6c7dOigkydPasaMGdctNxMnTtTYsWOd0zk5OYqJiSnVewIAgIrFrXLz2Wef6V//+pduv/12+fj4KC4uTvfee6/Cw8M1ffp09e7d+1e3sWfPHmVlZalt27bOscLCQm3fvl2zZ8+W1WqVr6/vr26nffv2WrJkyXWXCQwMVGBg4K9/MAAAUOG5VW4uX76s2rVrS5KqV6+un376SY0aNVKLFi20d+/eUm2ja9eu2r9/v8vY008/rSZNmuill14qVbGRpH379ik6OrpsHwAAAJiWW+WmcePGSktLU3x8vFq3bq33339f8fHxmjt3bqmLRlhYmJo3b+4yFhoaqho1ajjHJ06cqIyMDC1atEiSNGvWLMXHxyshIUE2m01LlizR8uXLtXz5cnc+BgCggvKVXTF+lxTua9Mlu79O5ldRvkr3RzHMz+1zbjIzMyVJkydPVo8ePfTRRx8pICBACxYs8Fi4zMxMpaenO6dtNpvGjx+vjIwMBQcHKyEhQWvXrlWvXr089p4AAO8WaCnQXcGZCvUpcI419M/WF7lRuuwIMDAZvIVb5WbgwIHOf7dp00bHjx/XgQMHFBsbq5o1a7odJiUlxWX6l0VpwoQJmjBhgtvbBwBUfE0CzrsUG0kK8ilUs8Dz2pUXaVAqeBO3ys0vhYSE6LbbbvPEpgAAuK5I39wSxq+UcxJ4K7fKTWFhoRYsWKAtW7YoKytLdrvdZf5nn33mkXAAAPxSgXwkFb2fWsGN3ZcWJuJWuRk1apQWLFig3r17q3nz5rJYLJ7OBQBAsU7mV1HTwPPFjgOSm+Vm6dKl+sc//sGJvACAcnc4P0JVfGyq53dZ1/62PlUQogO2asYGg9dwq9wEBASoYcOGns4CAMCvcsiifdbaSrPlK9zn6qXgl7hKCj/j1gHKcePG6d1335XD4fB0HgAASuWKw1+nCkMpNiii1Htu+vfv7zL92Wefaf369UpISJC/v7/LvBUrVngmHQAAQBmVutxERES4TD/44IMeDwMAAHCjSl1uPvzwQ+e/c3NzZbfbFRoaKkk6fvy4Vq5cqaZNm6pHjx6eTwkAAFBKbp1z07dvXy1evFiSdOHCBbVv314zZ85Uv379NGfOHI8GBAAAKAu3ys3evXvVuXNnSdI///lPRUZG6sSJE1q0aJHee+89jwYEAAAoC7fKzZUrVxQWFiZJ2rRpk/r37y8fHx+1b99eJ06c8GhAAACAsnDrPjcNGzbUypUr9eCDD2rjxo0aM2aMJCkrK0vh4eEeDQgA3sgn94LREQCv4k3fE26Vm1dffVWPP/64xowZo65du6pDhw6Sru7FadOmjUcDAoA3Cj623egIAErgVrn5n//5H911113KzMxUq1atnONdu3blEnEAlUJu/btlD65qdAzAa/jkXvCa0u9WuZGkqKgoRUVFuYzdcccdNxwIACoCe3BV2UNrGh0DQDF4PjwAADAVyg0AADAVyg0AADAVyg0AADAVyg0AADAVyg0AADAVty8FBwDAKH4qVKz/JYX72HTJ7q/0/DDZ5Gt0LHgJyg0AoEIJshToruAfFeJT6By7xT9bX+RG65IjwMBk8BYclgIAVCiNA867FBtJCvSxq1ngOYMSwdtQbgAAFUqkb26x47VLGEflQ7kBAFQo+SX86irgVxr+D18JAIAK5WR+lTKNo/Kh3AAAKpQj+RE6kV9FDsfVaYdD+rEgRKm2asYGg9fgaikAQIXikEX/a62lg7ZqCvOx6bLdX5cd/kbHgheh3AAAKqRch59yC/k1hqL4qoCpFV74UfacLPmE1pBvjRij4wAAygHlBqbkKMxX3u4VKjx92DnmUz1GwXc+LIt/kIHJAAA3GycUw5Rsh750KTaSZD93UtbvPzMoEQBPCrXkq3nAWXUMzlSrwJ8U5mMzOhK8COUGplTww7cljH9XzkkAeFqYj013h/yoBgE5qumbpzj/S7o7+EdV98kzOhq8BOUG5mQvLH7cUcI4gAqjccB5+VvsLmO+FoeaBJ43KBG8DeUGpuQX3bj48ahG5ZwEgKdV97EWO16DPTf4P5QbmFJA487yCavlMmYJiVBAs64GJQLgKVaHb7HjeSWMo/LhaimYkiUgRMH3PKuCzAPOS8H96jaVxZcbfQEV3bH8cLX2PVNk/Hh+uAFp4I0oNzAti4+v/OsmSHUTjI4CwIPSC8IUYC1Uw4BsBVjsKnBYdCw/XIfyI4yOBi9BuQEAVDiH86vqaH64gi0FynP4qZCzLPAzlBsAQIVkl48uOwKMjgEvRLkBADf45GUbHQHwKt70PUG5AYAyiIiIkH9AoHR0m9FRAK/jHxCoiAjjz32i3ABAGURGRmrJ4kXKzvaev1IrsxMnTmjatGmaNGmS4uLijI5T6UVERCgyMtLoGJQbACiryMhIr/gBjv+Ki4tTo0bcpBNXcXo5AAAwFcoNAAAwFcoNAAAwFcoNAAAwFcoNAAAwFcoNAAAwFcoNAAAwFcoNAAAwFW7iB1Ow511UQfo3cuRdlE/1GPnVaSKLj6/RsQAABqDcoMIrPJuu3J3LpELb1YHje5R/rJ6COzwui5+/seEAAOXOaw5LTZ8+XRaLRaNHj77uctu2bVPbtm0VFBSkBg0aaO7cueUTEF7L+s36/xab/2M//4Pyj+8xKBEAwEheUW527dqlpKQktWzZ8rrLHTt2TL169VLnzp21b98+vfzyyxo5cqSWL19eTknhbeyXL8h+8Uyx8wpOHyrnNAAAb2D4YalLly5p4MCBmjdvnhITE6+77Ny5cxUbG6tZs2ZJkpo2bardu3drxowZGjBgQDmkLRufPJ4afNNZr5Q4y0eSz+Xiiw+MwfcEPCU9PV2ff/65MjIy1KpVK2VmZvLgTDgZXm5GjBih3r17q1u3br9abnbs2KHu3bu7jPXo0UPJycnKz8+Xv3/x51dYrVZZrVbndE5Ozo0Hv46IiAj5BwRKR7fd1PfBVfmBgS7//14TbDuvgh2LVVBQID8/P4WEhJT4NYLy4x8QqIiICKNjoAJLT0/XwoULZbfbJUnVq1dXSkqKIiMj1aRJE4PTwRsYWm6WLl2qvXv3ateuXaVa/tSpU4qMjHQZi4yMVEFBgc6cOaPo6Ohi15s+fbqmTp16w3lLKzIyUksWL1J2Nn+lloezZ88qMTFRhw8fliT5+vqqS5cu2rFjh65cubpnx2azKT8/X4mJiWrWrJmRcSu9iIiIIt/HQFls377dWWx+btu2bZQbSDKw3Jw8eVKjRo3Spk2bFBQUVOr1LBaLy7TD4Sh2/OcmTpyosWPHOqdzcnIUExNTxsRlExkZyQ/wcrR27Vrt27dPWVlZat26taZNm6bLly+7LJOfn6/ly5erX79+xoQE4BGnTp0q0zgqH8PKzZ49e5SVlaW2bds6xwoLC7V9+3bNnj1bVqtVvr6u9ymJiooq8sWblZUlPz8/1ahRo8T3CgwMVGBgoGc/ALxOmzZtnP/es6f4K6V2795dXnEA3CTVq1cv8sfLtXFAMvBqqa5du2r//v36+uuvna927dpp4MCB+vrrr4sUG0nq0KGDNm/e7DK2adMmtWvXjnMp4KJmzZplGgdQcXTs2LFM46h8DCs3YWFhat68ucsrNDRUNWrUUPPmzSVdPZw0ePBg5zrDhw/XiRMnNHbsWKWmpuqDDz5QcnKyxo8fb9THgJcaOHBgseOPP/54OScB4GlNmjRR//79nX+s5Obmql27di5HAlC5ecV9bkqSmZmp9PR053T9+vW1bt06paSkqHXr1nr99df13nvveeVl4DDWww8/rFGjRiksLEzS1XOy+vXrp9/97ncGJwPgCS1atNCIESP08MMPa+fOnbr11luNjgQvYnFcOyO3EsnJyVFERISys7MVHh5udBzcRLm5ufryyy81depUzZ8/n/tgACZz8OBBDRs2TElJSXx/w8mr99wANyo4OFgxMTHy8eFLHQAqC37iAwAAU6HcAAAAU6HcAAAAU6HcAAAAU6HcAAAAU6HcAAAAU6HcwNQuXryow4cPF/sEYQCAOVFuYEoOh0PvvPOOOnfurDFjxuj06dOaM2eOCgoKjI4GALjJKDcwpb///e+aO3eucnNznWPr1q3TnDlzDEwFACgPfkYHgPfKy8tzebZXRbJo0aJix5csWaIePXqUcxrPiY2NVVBQkNExAMCrUW5QovT0dA0bNszoGG45ffp0seMXLlyosJ9JEs/PAYBSoNygRLGxsUpKSjI6hlveeustbd++vch4q1atlJiYaEAiz4iNjTU6AgB4PcoNShQUFFRh9xJMnDhR+/fv1/nz551jISEh+uMf/1hhPxMAoHQoNzClBg0aaOXKlfroo4904MAB1a9fXwMHDlRcXJzR0QAANxnlBqYVFRWlcePGGR0DAFDOuBQcAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYCuUGAACYip/RAQAAKKuCggKlpqYqNTVVUVFRKigoMDoSvAjlBgBQoVy5ckULFizQTz/9JElq2rSp1q1bpzp16qhq1arGhoNX4LAUAKBCSUlJcRabay5fvqzNmzcblAjehj03AFAJ5eXlKT093egYbvn222+LHT9w4IAOHjxYzmk8JzY2VkFBQUbHMAXKDQBUQunp6Ro2bJjRMdzSvn17BQcHFxm32WwV9jNJUlJSkho1amR0DFOwOBwOh9EhyltOTo4iIiKUnZ2t8PBwo+MAQLmryHtuvvnmG3333XdFxhs2bKjbb7/dgESewZ4bzzF0z82cOXM0Z84cHT9+XJKUkJCgV199VT179ix2+ZSUFHXp0qXIeGpqqpo0aXIzowKAqQQFBVXYvQQNGjRQfn6+yyGo2NhYDRgwgHIASQaXm3r16unNN99Uw4YNJUkLFy5U3759tW/fPiUkJJS4Xlpamssel1q1at30rAAA7+Dn56fHHntMmZmZOn36tGrUqKGYmBijY8GLeN1hqerVq+utt97Ss88+W2TetT0358+fL9PlflarVVar1Tmdk5OjmJgYDksBAGBCXnMpeGFhoZYuXarLly+rQ4cO1122TZs2io6OVteuXbV169Zf3fb06dMVERHhfNHwAQAwL8P33Ozfv18dOnRQXl6eqlSpor///e/q1atXscumpaVp+/btatu2raxWqxYvXqy5c+cqJSVFd999d4nvwZ4bAAAqD8PLjc1mU3p6ui5cuKDly5dr/vz52rZtm5o1a1aq9fv06SOLxaJVq1aV+j25WgoAAPMy/LBUQECAGjZsqHbt2mn69Olq1aqV3n333VKv3759ex06dOgmJgQAABWJ4eXmlxwOh8shpF+zb98+RUdH38REAACgIjH0UvCXX35ZPXv2VExMjC5evKilS5cqJSVFGzZskCRNnDhRGRkZWrRokSRp1qxZio+PV0JCgmw2m5YsWaLly5dr+fLlRn4MAADgRQwtN6dPn9YTTzyhzMxMRUREqGXLltqwYYPuvfdeSVJmZqbLHTRtNpvGjx+vjIwMBQcHKyEhQWvXri3xBGQAAFD5GH5CsRE4oRgAAPPyunNuAAAAbgTlBgAAmArlBgAAmIqhJxQb5dppRjk5OQYnAQAAZRUWFiaLxVLi/EpZbi5evChJPGMKAIAK6NcuCKqUV0vZ7Xb9+OOPv9r8YA7XniV28uRJro4DTIbv78qJPTfF8PHxUb169YyOgXIWHh7ODz/ApPj+xs9xQjEAADAVyg0AADAVyg1MLzAwUJMnT1ZgYKDRUQB4GN/fKE6lPKEYAACYF3tuAACAqVBuAACAqVBuAACAqVBuAABe58CBA2rfvr2CgoLUunXrEse8icVi0cqVK42OAVFuYBJPPfWULBaLLBaL/Pz8FBsbq+eee07nz593WS43N1fVqlVT9erVlZuba1BaoGL6zW9+o9GjRxcZX7lypcfv9j558mSFhoYqLS1NW7ZsKXHsRlFIzIlyA9O47777lJmZqePHj2v+/PlavXq1nn/+eZdlli9frubNm6tZs2ZasWKFQUkB/JojR47orrvuUlxcnGrUqFHiGFAcyg1MIzAwUFFRUapXr566d++uRx55RJs2bXJZJjk5WYMGDdKgQYOUnJxsUFLAvKZMmaLWrVtr8eLFio+PV0REhB599FHnA4slacOGDbrrrrtUtWpV1ahRQ/fff7+OHDninG+xWLRnzx699tprslgsmjJlSrFjkpSRkaFHHnlE1apVU40aNdS3b18dP37cJdMHH3yghIQEBQYGKjo6Wr///e8lSfHx8ZKkBx98UBaLxTktSatXr1bbtm0VFBSkBg0aaOrUqSooKHDOP3TokO6++24FBQWpWbNm2rx5s2f/Q+KGUG5gSkePHtWGDRvk7+/vHDty5Ih27Nihhx9+WA8//LC+/PJLHT161MCUgDkdOXJEK1eu1Jo1a7RmzRpt27ZNb775pnP+5cuXNXbsWO3atUtbtmyRj4+PHnzwQdntdklSZmamEhISNG7cOGVmZmr8+PHFjl25ckVdunRRlSpVtH37dn3++eeqUqWK7rvvPtlsNknSnDlzNGLECA0bNkz79+/XqlWr1LBhQ0nSrl27JEkffvihMjMzndMbN27UoEGDNHLkSH3//fd6//33tWDBAk2bNk3S1Ycv9+/fX76+vtq5c6fmzp2rl156qdz++6IUHIAJPPnkkw5fX19HaGioIygoyCHJIcnx9ttvO5d5+eWXHf369XNO9+3b1zFp0iQj4gIV0j333OMYNWpUkfFPPvnEce3XyeTJkx0hISGOnJwc5/wXX3zRceedd5a43aysLIckx/79+51jrVq1ckyePNlluV+OJScnOxo3buyw2+3OMavV6ggODnZs3LjR4XA4HHXq1Lnu97kkxyeffOIy1rlzZ8cbb7zhMrZ48WJHdHS0w+FwODZu3Ojw9fV1nDx50jl//fr1xW4LxmDPDUyjS5cu+vrrr/XVV1/phRdeUI8ePfTCCy9IkgoLC7Vw4UINGjTIufygQYO0cOFCFRYWGhUZMKX4+HiFhYU5p6Ojo5WVleWcPnLkiB5//HE1aNBA4eHhql+/viQpPT29TO+zZ88eHT58WGFhYapSpYqqVKmi6tWrKy8vT0eOHFFWVpZ+/PFHde3atczbfe2115zbrFKlioYOHarMzExduXJFqampio2NVb169ZzrdOjQoUzvgZvLz+gAgKeEhoY6dze/99576tKli6ZOnarXX39dGzdudB6b/7nCwkJt2rRJPXv2NCIyUKGEh4crOzu7yPiFCxcUHh7unP754WDp6jk01w45SVKfPn0UExOjefPmqU6dOrLb7WrevLnzUFJp2e12tW3bVh999FGRebVq1ZKPj3t/v9vtdk2dOlX9+/cvMi8oKEiOYp5a5OmrxXBjKDcwrcmTJ6tnz5567rnnlJycrEcffVSTJk1yWebNN99UcnIy5QYohSZNmmj9+vVFxnft2qXGjRuXahtnz55Vamqq3n//fXXu3FmS9Pnnn7uV57bbbtOyZctUu3Ztl3L1c/Hx8dqyZYu6dOlS7Hx/f/8ie29vu+02paWlOf9Y+qVmzZopPT1dP/74o+rUqSNJ2rFjh1ufATcH5Qam9Zvf/EYJCQmaNm2aVq9erVWrVql58+Yuyzz55JPq3bu3fvrpJ9WqVcugpEDF8Pzzz2v27NnOE3SDg4O1efNmJScna/HixaXaxrWrmpKSkhQdHa309HT94Q9/cCvPwIED9dZbb6lv37567bXXVK9ePaWnp2vFihV68cUXVa9ePU2ZMkXDhw9X7dq11bNnT128eFFffPGF85D1tfLTqVMnBQYGqlq1anr11Vd1//33KyYmRg899JB8fHz0zTffaP/+/UpMTFS3bt3UuHFjDR48WDNnzlROTk6RP5xgLM65gamNHTtWSUlJys/PL/a4e5cuXRQWFlbqH8xAZRYfH69///vfOnLkiLp3767bb79dCxYs0IIFC/TQQw+Vahs+Pj5aunSp9uzZo+bNm2vMmDF666233MoTEhKi7du3KzY2Vv3791fTpk31zDPPKDc317kn58knn9SsWbP0t7/9TQkJCbr//vt16NAh5zZmzpypzZs3KyYmRm3atJEk9ejRQ2vWrNHmzZt1++23q3379nr77bcVFxfn/AyffPKJrFar7rjjDg0ZMsR5JRW8g8VR3MFDAACACoo9NwAAwFQoNwAAwFQoNwAAwFQoNwAAwFQoNwAAwFQoNwAAwFQoNwAAwFQoNwAAwFQoNwAMYbFYtHLlSqNjlElKSoosFosuXLhgdBQA10G5AeBRTz31lCwWiywWi/z9/RUZGal7771XH3zwgcuToTMzMyvcA0s7duyozMxMRUREGB0FwHVQbgB43H333afMzEwdP35c69evV5cuXTRq1Cjdf//9KigokCRFRUUpMDDQ4KRlExAQoKioKFksFqOjALgOyg0AjwsMDFRUVJTq1q2r2267TS+//LL+9a9/af369VqwYIGkooelXnrpJTVq1EghISFq0KCBXnnlFeXn57tsNzExUbVr11ZYWJiGDBmiP/zhD2rdurVz/lNPPaV+/fppxowZio6OVo0aNTRixAiX7Zw/f16DBw9WtWrVFBISop49e7o8SPHEiRPq06ePqlWrptDQUCUkJGjdunWSih6Wut6yAIzjZ3QAAJXDb3/7W7Vq1UorVqzQkCFDiswPCwvTggULVKdOHe3fv19Dhw5VWFiYJkyYIEn66KOPNG3aNP3tb39Tp06dtHTpUs2cOVP169d32c7WrVsVHR2trVu36vDhw3rkkUfUunVrDR06VNLVAnTo0CGtWrVK4eHheumll9SrVy99//338vf314gRI2Sz2bR9+3aFhobq+++/V5UqVYr9TGVZFkD5odwAKDdNmjTRN998U+y8P/7xj85/x8fHa9y4cVq2bJmz3PzlL3/Rs88+q6efflqS9Oqrr2rTpk26dOmSy3aqVaum2bNny9fXV02aNFHv3r21ZcsWDR061FlqvvjiC3Xs2FHS1dIUExOjlStX6qGHHlJ6eroGDBigFi1aSJIaNGhQ4ucpy7IAyg+HpQCUG4fDUeL5Kv/85z911113KSoqSlWqVNErr7yi9PR05/y0tDTdcccdLuv8clqSEhIS5Ovr65yOjo5WVlaWJCk1NVV+fn668847nfNr1Kihxo0bKzU1VZI0cuRIJSYmqlOnTpo8eXKJZaysywIoP5QbAOUmNTW1yGEkSdq5c6ceffRR9ezZU2vWrNG+ffs0adIk2Ww2l+V+WYwcDkeRbfn7+xdZ59pVWsUtf2382raHDBmio0eP6oknntD+/fvVrl07/eUvfyl2vbIsC6D8UG4AlIvPPvtM+/fv14ABA4rM++KLLxQXF6dJkyapXbt2uvXWW3XixAmXZRo3bqz//Oc/LmO7d+8uU4ZmzZqpoKBAX331lXPs7NmzOnjwoJo2beoci4mJ0fDhw7VixQqNGzdO8+bNK3GbZVkWQPngnBsAHme1WnXq1CkVFhbq9OnT2rBhg6ZPn677779fgwcPLrJ8w4YNlZ6erqVLl+r222/X2rVr9cknn7gs88ILL2jo0KFq166dOnbsqGXLlumbb74p03kut956q/r27auhQ4fq/fffV1hYmP7whz+obt266tu3ryRp9OjR6tmzpxo1aqTz58/rs88+cyk+P1eWZQGUH8oNAI/bsGGDoqOj5efnp2rVqqlVq1Z677339OSTT8rHp+gO4759+2rMmDH6/e9/L6vVqt69e+uVV17RlClTnMsMHDhQR48e1fjx45WXl6eHH35YTz31VJG9Ob/mww8/dN5zx2az6e6779a6deuch7MKCws1YsQI/fDDDwoPD9d9992nd955p9htlWVZAOXH4ijpIDQAeLl7771XUVFRWrx4sdFRAHgR9twAqBCuXLmiuXPnqkePHvL19dXHH3+sTz/9VJs3bzY6GgAvw54bABVCbm6u+vTpo71798pqtapx48b64x//qP79+xsdDYCXodwAAABT4VJwAABgKpQbAABgKpQbAABgKpQbAABgKpQbAABgKpQbAABgKpQbAABgKpQbAABgKv8fQNBzrdcMSIEAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# paired shannon diversity difference from 0 \n",
    "# this is from shannon_significance.qzv -> download as tsv\n",
    "df_alpha = pd.read_csv(path + 'outputs/jobs02/metadata_shannon.tsv', sep='\\t', index_col=0)\n",
    "# df_alpha = pd.read_csv(path + 'outputs/jobs02/metadata_chao1.tsv', sep='\\t', index_col=0)\n",
    "# df_alpha = pd.read_csv(path + 'outputs/jobs02/metadata_observed.tsv', sep='\\t', index_col=0)\n",
    "# df_alpha = pd.read_csv(path + 'outputs/jobs02/metadata_evenness.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "alpha_metric = 'shannon_entropy'\n",
    "# alpha_metric = 'pielou_evenness' # n.s. except for diff alpha\n",
    "# alpha_metric = 'observed_features' # n.s.\n",
    "# alpha_metric = 'chao1' # n.s.\n",
    "\n",
    "\n",
    "\n",
    "# drop unnecessary rows and cols\n",
    "df_alpha = df_alpha.drop('#q2:types')\n",
    "df_alpha = df_alpha.drop('Separate', axis=1)\n",
    "\n",
    "# get paired per twin pair\n",
    "pair_to_diff = {}\n",
    "for p in list(df_alpha.TwinPair.values):\n",
    "    df = df_alpha[df_alpha['TwinPair'] == p]\n",
    "    alpha_UA = float(df[df['Diagnosis'] == 'Unaffected'][alpha_metric].values)\n",
    "    alpha_RA = float(df[df['Diagnosis'] == 'RA'][alpha_metric].values)\n",
    "    pair_to_diff[p] = alpha_RA - alpha_UA\n",
    "\n",
    "df_paired_alpha = pd.DataFrame.from_dict(pair_to_diff, orient='index', columns=[alpha_metric + '_diff'])\n",
    "\n",
    "# one-sided t-test, n.s.; RA-UA values \n",
    "t, p = scipy.stats.ttest_1samp(df_paired_alpha[alpha_metric + '_diff'],popmean=0)\n",
    "print(t, p)\n",
    "\n",
    "m, p = scipy.stats.wilcoxon(df_paired_alpha[alpha_metric + '_diff'])\n",
    "print(m, p)\n",
    "\n",
    "# boxplot of differences\n",
    "#sns.boxplot(data=df_paired_alpha)\n",
    "#sns.swarmplot(data=df_paired_alpha, palette='dark:grey')\n",
    "#sns.despine()\n",
    "\n",
    "# separate\n",
    "df_alpha[alpha_metric] = df_alpha[alpha_metric].map(lambda x: float(x))\n",
    "ax = sns.boxplot(data=df_alpha, x='Diagnosis', y=alpha_metric)\n",
    "sns.swarmplot(data=df_alpha, x='Diagnosis', y=alpha_metric, palette='dark:grey')\n",
    "sns.despine()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>PC9</th>\n",
       "      <th>PC10</th>\n",
       "      <th>PC11</th>\n",
       "      <th>PC12</th>\n",
       "      <th>PC13</th>\n",
       "      <th>PC14</th>\n",
       "      <th>PC15</th>\n",
       "      <th>PC16</th>\n",
       "      <th>TwinPair</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>sib_02</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>-0.064242</td>\n",
       "      <td>0.204422</td>\n",
       "      <td>0.004331</td>\n",
       "      <td>0.0605</td>\n",
       "      <td>-0.104939</td>\n",
       "      <td>-0.191733</td>\n",
       "      <td>-0.062688</td>\n",
       "      <td>-0.144355</td>\n",
       "      <td>0.216428</td>\n",
       "      <td>0.173113</td>\n",
       "      <td>0.053189</td>\n",
       "      <td>0.075915</td>\n",
       "      <td>0.00359</td>\n",
       "      <td>-0.034984</td>\n",
       "      <td>-0.025506</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>T2</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>CorrectDx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>-0.232621</td>\n",
       "      <td>-0.057078</td>\n",
       "      <td>-0.131391</td>\n",
       "      <td>0.044585</td>\n",
       "      <td>0.020252</td>\n",
       "      <td>-0.152454</td>\n",
       "      <td>0.05447</td>\n",
       "      <td>-0.073076</td>\n",
       "      <td>-0.123425</td>\n",
       "      <td>-0.096711</td>\n",
       "      <td>-0.124822</td>\n",
       "      <td>0.036416</td>\n",
       "      <td>0.092423</td>\n",
       "      <td>-0.100855</td>\n",
       "      <td>-0.065866</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>T3</td>\n",
       "      <td>RA</td>\n",
       "      <td>MisDx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>-0.271498</td>\n",
       "      <td>-0.158031</td>\n",
       "      <td>-0.053222</td>\n",
       "      <td>0.026877</td>\n",
       "      <td>0.086599</td>\n",
       "      <td>-0.095532</td>\n",
       "      <td>0.038394</td>\n",
       "      <td>0.054045</td>\n",
       "      <td>-0.0398</td>\n",
       "      <td>0.12725</td>\n",
       "      <td>0.106447</td>\n",
       "      <td>-0.177049</td>\n",
       "      <td>0.10352</td>\n",
       "      <td>0.007377</td>\n",
       "      <td>0.054464</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>T3</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>MisDx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>-0.111142</td>\n",
       "      <td>-0.222525</td>\n",
       "      <td>-0.105657</td>\n",
       "      <td>0.199485</td>\n",
       "      <td>0.006978</td>\n",
       "      <td>-0.037698</td>\n",
       "      <td>0.046442</td>\n",
       "      <td>0.312272</td>\n",
       "      <td>0.087048</td>\n",
       "      <td>-0.059049</td>\n",
       "      <td>0.075261</td>\n",
       "      <td>0.124197</td>\n",
       "      <td>-0.030614</td>\n",
       "      <td>0.009814</td>\n",
       "      <td>-0.016471</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>T4</td>\n",
       "      <td>RA</td>\n",
       "      <td>CorrectDx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505A</th>\n",
       "      <td>-0.201807</td>\n",
       "      <td>-0.003872</td>\n",
       "      <td>-0.022754</td>\n",
       "      <td>-0.262292</td>\n",
       "      <td>0.09817</td>\n",
       "      <td>0.087379</td>\n",
       "      <td>-0.128997</td>\n",
       "      <td>0.034522</td>\n",
       "      <td>-0.004889</td>\n",
       "      <td>0.035228</td>\n",
       "      <td>-0.080164</td>\n",
       "      <td>0.128241</td>\n",
       "      <td>-0.008776</td>\n",
       "      <td>-0.060466</td>\n",
       "      <td>0.106393</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>T4</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>CorrectDx</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           PC1       PC2       PC3       PC4       PC5       PC6       PC7  \\\n",
       "501A -0.064242  0.204422  0.004331    0.0605 -0.104939 -0.191733 -0.062688   \n",
       "502A -0.232621 -0.057078 -0.131391  0.044585  0.020252 -0.152454   0.05447   \n",
       "503A -0.271498 -0.158031 -0.053222  0.026877  0.086599 -0.095532  0.038394   \n",
       "504A -0.111142 -0.222525 -0.105657  0.199485  0.006978 -0.037698  0.046442   \n",
       "505A -0.201807 -0.003872 -0.022754 -0.262292   0.09817  0.087379 -0.128997   \n",
       "\n",
       "           PC8       PC9      PC10      PC11      PC12      PC13      PC14  \\\n",
       "501A -0.144355  0.216428  0.173113  0.053189  0.075915   0.00359 -0.034984   \n",
       "502A -0.073076 -0.123425 -0.096711 -0.124822  0.036416  0.092423 -0.100855   \n",
       "503A  0.054045   -0.0398   0.12725  0.106447 -0.177049   0.10352  0.007377   \n",
       "504A  0.312272  0.087048 -0.059049  0.075261  0.124197 -0.030614  0.009814   \n",
       "505A  0.034522 -0.004889  0.035228 -0.080164  0.128241 -0.008776 -0.060466   \n",
       "\n",
       "          PC15 PC16 TwinPair   Diagnosis     sib_02  \n",
       "501A -0.025506 -0.0       T2  Unaffected  CorrectDx  \n",
       "502A -0.065866 -0.0       T3          RA      MisDx  \n",
       "503A  0.054464 -0.0       T3  Unaffected      MisDx  \n",
       "504A -0.016471 -0.0       T4          RA  CorrectDx  \n",
       "505A  0.106393 -0.0       T4  Unaffected  CorrectDx  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# process beta ordination for R plotting\n",
    "df_pc = pd.read_csv(path + 'outputs/jobs02/Q2_outputs/ordination.txt', sep='\\t', skiprows=9)\n",
    "\n",
    "# set index to first sample and drop last two metadata rows\n",
    "df_pc = df_pc.set_index('500A') \n",
    "df_pc = df_pc.iloc[:-2,:]\n",
    "\n",
    "# create new row from column names and replace old col names\n",
    "df_pc.loc['500A'] = df_pc.columns.values  # adding a row\n",
    "df_pc.columns = ['PC' + str(i+1) for i in range(len(df_pc.columns))]\n",
    "df_pc.index.name = 'SampleID'\n",
    "\n",
    "# merge with metadata\n",
    "df_pc = pd.concat([df_pc, df_map], axis=1)\n",
    "df_pc['sib_02'] = df_pc['TwinPair'].map(lambda x: 'MisDx' if x == 'T3' else 'CorrectDx')\n",
    "df_pc.to_csv(path + 'outputs/jobs02/bray_curtis_pcoa.tsv', sep='\\t')\n",
    "df_pc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.8205185439560694 0.14804199847741845\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Axes: xlabel='category', ylabel='distance'>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjcAAAGxCAYAAACeKZf2AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAABEvElEQVR4nO3de1yUdd7/8feAnESBlBU84SEPCUSuuJi6Wq2urbUlspTtdjRNve3k2s9SvBNlU2q3g1a3Gmvqqu2dWTibrtu92OahdZeIsBAtLU1MQRMTTFIUrt8f3nA7zgwCDnMNM6/n4zGPje9c11yfYS+c93y/1/X9WgzDMAQAAOAl/MwuAAAAwJUINwAAwKsQbgAAgFch3AAAAK9CuAEAAF6FcAMAALwK4QYAAHgVwg0AAPAqrcwuwN1qamp05MgRtW3bVhaLxexyAABAAxiGoVOnTqlTp07y86u/b8bnws2RI0fUtWtXs8sAAABNcOjQIXXp0qXebXwu3LRt21bShV9OWFiYydUAAICGqKioUNeuXes+x+vjc+GmdigqLCyMcAMAQAvTkEtKuKAYAAB4FcINAADwKoQbAADgVQg3AADAqxBuAACAVyHcAAAAr0K4AQAAXoVwAwAAvArhBgAAeBXCDQAA8CqEGwAA4FUINwC8gtVqVVJSkkJDQ5WUlCSr1Wp2SQBMQrgB0OJZrVaNHTtWeXl5qqysVF5enlJSUgg4gI/yuVXBATSPM2fOqLi42JRjz5kzx67NMAylp6crNjbWhIqkmJgYBQcHm3JswNcRbgC4RHFxsSZNmmTKsYuKihy279q1y7SasrKy1KdPH1OODc9ktVq1YMECFRUVKS4uTmlpaUpOTja7LK9kMQzDMLsId6qoqFB4eLjKy8sVFhZmdjmA1zCz5yY1NVWFhYV27QkJCVq3bp0JFdFzA1u1Q6cXs1gsys7OJuA0UGM+vwk3AFo8q9WqlJQUXfzPmcVi0fr16zVmzBgTK4PZ3B26N2/erKVLl+rLL79Ur169NGXKFI0cOdJjAnhLDt2Em3oQbgDvZLValZ6erl27dik+Pl4ZGRkEG2jv3r1uG5o8fvy4wyHSuLg47dmzRzU1NXbP+fn5adiwYe4oT1LLHi5tzOc319wA8ArJycmKjY3VpEmTWvQ/4HCtmJgYZWVlueVYqampDtvPnz+v0NBQnTp1yu65+Ph4t9UnXfh9+ALCDQDAawUHB7st6H711VcO2w8dOqQePXpo9+7ddkOnGRkZBPFmwDw3AAC4QFxcnMP2Xr16KTIyUq+++qrNRJNcE9Z8CDcAALhAWlqaLBaLTZvFYtHkyZMlSSNHjlRubq6+//575ebmEmyaEeEGAAAXSE5OVnZ2tk3vzFNPPaWlS5dq+/btSk1NZdZsN+GaGwAAXCQ5Oblu3ppL57YpLCxUSkoKc9u4AT03AAA0gwULFti1GYahzMxME6rxLYQbAACagbNlQZy1w3UINwAANANnd085a4frEG4AAGgGzu6eSktLM6ki30G4AQCgGdTePZWQkCA/Pz8lJCQwt42bcLcUAADNhGVBzEHPDQAAbmK1Wm3mwWHem+ZBuAEA4Ao1JLRs3rxZY8eOVV5eniorK5WXl6eUlBQCTjMg3AAAcAVqJ+u7XGhZunSp3b7Me9M8CDcAAFyBhk7W9+WXXzrcn3lvXM/0cLN48WL16NFDwcHBSkxM1Pbt2+vd/r/+67/Ur18/hYSEqG/fvlq1apWbKgUAwF5DJ+vr1auXw+2Y98b1TA03a9eu1bRp0zR79mwVFBRo2LBhGj16tIqLix1uv2TJEs2aNUtz585VUVGR5s2bp4cfflgbNmxwc+UAAFzQ0Mn6pkyZwrw3bmIxDMMw6+CDBg3SgAEDtGTJkrq2fv36KTk52eEY5JAhQzR06FD94Q9/qGubNm2aPv74Y3344YcNOmZFRYXCw8NVXl6usLCwK38TADzG3r17ueUWbme1WpWSkqKLP04tFovWr18vwzA0Z84cFRUVKS4uTrfeeqv+8Y9/1P2clpbGvDcN1JjPb9PmuamqqlJ+fr5mzpxp0z5q1Cjt2LHD4T5nz55VcHCwTVtISIg++ugjnTt3TgEBAQ73OXv2bN3PFRUVLqgeAIALaifry8zMtAkthmHYrQq+a9cuVgV3A9OGpY4fP67q6mpFRUXZtEdFRam0tNThPjfffLOWLVum/Px8GYahjz/+WMuXL9e5c+d0/Phxh/tkZmYqPDy87tG1a1eXvxcAgG9LTk5Wbm6uvv/+e+Xm5mrMmDGsCm4i0y8ovnT80TAMu7ZaTz/9tEaPHq3rr79eAQEBGjNmjB544AFJkr+/v8N9Zs2apfLy8rrHoUOHXFo/AACOsCq4eUwLN5GRkfL397frpTl27Jhdb06tkJAQLV++XJWVlfr6669VXFys7t27q23btoqMjHS4T1BQkMLCwmweAAA0N1YFN49p4SYwMFCJiYnKycmxac/JydGQIUPq3TcgIEBdunSRv7+/3nzzTf3yl7+Un5/pnVAAANRhVXDzmJoIpk+frmXLlmn58uXas2ePfvvb36q4uFhTpkyRdGFI6b777qvbfu/evVqzZo327dunjz76SHfddZd27drlcFwTAAAzNXVVcNafunKmhptx48Zp4cKFysjIUP/+/bVt2zZt2rRJ3bp1kySVlJTYzHlTXV2tF154Qdddd51+/vOf68yZM9qxY4e6d+9u0jsAAMC55ORkrVu3TsOGDdO6devqgo2zANPQpRxQP9NuBa81depUTZ061eFzK1eutPm5X79+KigocENVAAA0jNVq1YIFC2xuA6/vVu/aAFOrNsBkZ2fXe4cVt483nOnhBgDgvY4ePary8nKzy2g2mzdv1sMPP1z3c21QefXVVzVy5EhJ0sGDB23+d86cOXavYxiG0tPTtW/fPofH2bVrl/bu3evq8t0uPDzc6U1DrkS4AQA0i6NHj+qee+/Tuaqzl9+4hfrkk0/s2gzD0MyZMzVgwACb9vnz50tyfiv4rl27FBoa6vA5f39/TZo06QqrNV9AYJDWrF7V7AGHcAOgRXI0FBAbG2t2WbhIeXm5zlWd1Q89b1BNcLjZ5TSL0//8l+P2H87qdOzt+u5AoUo+eV9nThxVcLsodRwwQsGR+1V5zH7OtZDILuow4Gc69d6fJF28MpJFHYam6nSP+OZ5E27id6Zc2r9V5eXlhBsAuJSzaxZeffVVE6uCMzXB4aoJdTwXWUsXHNlZlaUHHLR30YkjB7X/vZV1bZXHDumr9/6kqKRbVHnsG10aYKKGjFV4rwHqGRSm0tyNOlN2WMHtOyv6+tsU3muAapr93XgPwg2AFsfZRZevvfaarrrqKhMqgq+KHnSb9v/lFV0aVKKvv02l/37XwR6GThXvUc8xj9oFmIheF4axInonKqJ3ojvK91qEGwAtjrNrFr788kv95Cc/cXM18GURvROdBpWvNy5xuM+ZssMEmGZGuAHQ4sTFxSkvL8+uvVevXiZUA1/nLKg4HbJq39kdZfk01iwA0OI4m9a+vLxc27dvV2pqKpOewXTRg26TdOlC0BeGrNC8CDcAWpzaae1rZ3jt3bu3DMPQwYMHVVNTo8LCQmZ1helqh6xaR/eUX0CQWkf3VM/kx+qurUHzYVgKQIuUnJxcN2NrUlKS3fPM6gpPwLU15qDnBkCL5+wCY2ftALwb4QZAixcXF9eodgDejXADwOM4WzHZGWcXGKelpbn8WMClTu7L1+dr5mrnwof0+Zq5Orkv3+ySfB7hBoBHqZ19OC8vT5WVlXWzD9cXOmovME5ISJCfn58SEhK0fv16jRkzxuXHAi52cl++9v/lZVWWHlDN+SpVlh7Q/r+8UhdwCD7mINwA8CjOZh/OzMysd7/k5GStW7dOw4YN07p162yCjbPemaYeC6hVmrvBQauh0tyNlw0+aD6EGwAepb6Lg5syhFRf7wwXIuNKnTl+2HF72eF6gw+aF+EGgEdxdhFwp06dmjSEVF/vDBci40oFRzqebTi4fed6gw+aF+EGgEdxdnGwIxcPIVmtVqWmptrNUFxf78yVXIgMSPXPQlxf8EHzItwA8CiXzj6clJSk9evX6/Bhx992a4erxo4dq8LCQrsZiuvrnXF2rMtdiAzUqm8WYpZfMA8zFANe4ujRoyovLze7DJeIjY3V6tWrbdquvvpqFRYW2m179dVXa86cOXbthmEoPT1dkydP1scffyzDMOqes1gsuv/++7V3716Hx9q7d6+L3om5wsPDFRUVZXYZXs/ZLMT1rRiO5kW4AbzA0aNHdc+99+lc1VmzS2k2NTU1Dturq6u1Z88eh8/t2rVLb731lmJjY1VcXKzTp08rNDRUMTExeuutt/TWW281Z8mmCwgM0prVqwg4JmL5BXMQbgAvUF5ernNVZ/VDzxtUExxudjnNIkTS1V0LVfrJP/TDiVKFtItW9IARCukRr+BvF6ry2CH7fSK76HTs7QqR1PeS5067o2gT+Z0pl/ZvVXl5OeEGPodwA3iRmuBw1YRGml1GswmPv0nh8TfZtNVIih48Vvv/8ook46JnLIoaMtarfx/wfCf35as0d4POHD+s4MjOih50Gz05bsAFxQBavPou6gTMwiR+5qHnBoBX4NoGeJr6JvHjXG1e9NwAANAMmMTPPIQbAACaAZP4mYdwgyZpyho/AOBLmMTPPIQbNFp9CxECAC7gQnfzcEFxC3XmzBkVFxebcuz6ZoONjY01oSIpJiZGwcHBphwbAOpnSIZx0f+iuRFuWqji4mJNmjTJlGM7W4hw165dptWUlZWlPn36mHJsAHCk9lbwWrW3gvcc8yh3SzUzwk0LFRMTo6ysLFOOnZqa6nCNn/j4eNNqiomJMeW4AOAMt4Kbh3DTQgUHB5vWU5GRkaGUlBS7hQgzMjLoPQGA/8Wt4ObhgmI0WnJysrKzs5WQkCA/Pz8lJCRo/fr1GjNmjNmlAYDH4FZw89BzgyZJTk5WbGysJk2axPUucCvW6kFLET3oNodrnnErePOj5wZAi8FaPWhJuBXcPPTcAGgxuEATLQ1rnpmDcAOgxajvAk2GqwDUYlgKQIvh7ALNVqERDFcBqEO4AdBiOFur59KWCy4MVwHwPYQbAC2Gsws0z33/ncPtmU8E8E1ccwOgRXF0gWZwZGdVlh6w25b5RADfRM8NgBbP2XAV84kAvolwA6DFYz4RABdjWAqAV2A+EQC16LkBAABehXADAAC8CuEGAAB4FcINAADwKqaHm8WLF6tHjx4KDg5WYmKitm/fXu/2b7zxhq677jq1bt1aHTt21Pjx41VWVuamagEAgKczNdysXbtW06ZN0+zZs1VQUKBhw4Zp9OjRKi4udrj9hx9+qPvuu08TJkxQUVGR1q1bp7y8PE2cONHNlQMAAE9larh58cUXNWHCBE2cOFH9+vXTwoUL1bVrVy1ZssTh9v/+97/VvXt3PfbYY+rRo4d++tOfavLkyfr444/dXDkAAPBUpoWbqqoq5efna9SoUTbto0aN0o4dOxzuM2TIEH3zzTfatGmTDMPQ0aNH9fbbb+vWW291epyzZ8+qoqLC5gEAALyXaeHm+PHjqq6uVlRUlE17VFSUSktLHe4zZMgQvfHGGxo3bpwCAwMVHR2tiIgIvfLKK06Pk5mZqfDw8LpH165dXfo+AACAZzH9gmKLxXY9GMMw7Npq7d69W4899pjmzJmj/Px8vffeezpw4ICmTJni9PVnzZql8vLyusehQ4dcWj8A9zq5L1+fr5mrnQsf0udr5urkvnyzSwLgYUxbfiEyMlL+/v52vTTHjh2z682plZmZqaFDh2rGjBmSpISEBIWGhmrYsGF65pln1LFjR7t9goKCFBQU5Po3AMDtTu7L1/6/vFz3c2XpAe3/yyvqOebRepdeOLkvX6W5G3Tm+GEFR3ZW9KDbWKoB8GKm9dwEBgYqMTFROTk5Nu05OTkaMmSIw30qKyvl52dbsr+/v6QLPT4AvFtp7gYHrYZKczc63ac2EFWWHlDN+aq6QESPD+C9TF04c/r06br33ns1cOBADR48WFlZWSouLq4bZpo1a5YOHz6sVatWSZJuu+02PfTQQ1qyZIluvvlmlZSUaNq0aUpKSlKnTp3MfCsA3ODM8cOO28sOO+2dqS8Q0XvjHn4/nDS7BHgAd54HpoabcePGqaysTBkZGSopKVF8fLw2bdqkbt26SZJKSkps5rx54IEHdOrUKb366qt64oknFBERoZ/97Gd67rnnzHoLANwoOLKzKksP2LUHhEY4Ha6qLxDBPUIObDO7BPgYU8ONJE2dOlVTp051+NzKlSvt2h599FE9+uijzVwVAE8UPeg27f/LK5IuHoa2yJCjYekLvTPOAlFw+87NVSYu8UOP4aoJiTC7DJjM74eTbgu6pocbAGioiN6J6jnmUZXmbtSZssMKbt9Z0dffpq83Op7480zZYXW/ZbLDQBR9/W1uqRlSTUiEakIjzS4DPoRwA6BFieidaHetTH29M84CUUSvAe4qGYCbEW4AtHjOhqvaxvTT52vm1l1k3P2WyVxEDPgA0yfxA4ArVds70zq6p/wCgtQ6uqeikm7R0Y/+yi3ggA+i5waAV7h0uOrzNXMdbMUt4IAvoOcGgFfiFnDAd9FzA3gRX50s7bsDhSr55H2dOXFUwe2i1HHACAW3i1LlMfu15EKuipLf6eMmVOlevnouABLhBvAqvjhZ2vHjx/VVUVHdz5XHDumr91aqa9euqnSwffcftVXo7nfdVyAAtyPcAF7EFydLO/DOQoftJ85adPUvHlDpJ//QDydKFdIuWtEDRiikR7xOu7dEU7hzwjTA0xBuAC/ii5OlnTlx1GH7D98dVXj8TQqPv8mmvcYdRQEwFRcUA2jRgiMdL6PA8gqA7yLcAGjRogfdJslySSvLKwC+jHADoEVzNIFfz+THWF4B8GFccwOgxaudlK80d4POHP9Gpf9+VzIMJusDfBQ9NwBavJP78rX/Ly+z1AIASYQbAF6gNHeDg9YLSy0A8D2EGzSJ1WpVamqqtm/frtTUVFmtVrNLgg9jqQUAFyPcoNGsVqvGjh2rwsJC1dTUqLCwUCkpKQQcmIbbwQFcjHCDRluwYIFdm2EYyszMNKEagNvBAdgi3KDRii5ax6ch7UBz43ZwABfjVnA0WlxcnPLy8hy2A2aJ6J3Ird/wOCf35f/vFAWHFRzZWdGDbuM8dQN6btBoaWlpslhshwAsFovS0tJMqggAPA9TFJiHcINGS05OVnZ2thISEuTn56eEhAStX79eY8aMMbs0APAYTFFgHoal0CTJycmKjY3VpEmTlJWVpT59+phdEgB4FKYoMA89NwAANAOmKDAP4QYAgGbAFAXmIdwAANAMmKLAPFxzAwBAM2GKAnPQcwMAALwK4QYAAHgVwg0AAPAqhBsAAOBVuKC4iY4ePary8nKzyzDVwYMHbf7Xl4WHhysqKsrsMgAAItw0ydGjR3XPvffpXNVZs0vxCPPnzze7BNMFBAZpzepVBBwA8ACEmyYoLy/Xuaqz+qHnDaoJDje7HJjM70y5tH+rysvLCTeAD2Llb89DuLkCNcHhqgmNNLsMU/DHDAD/t/J3rdqVv3uOeZR/E01EuEGj8ccMABfUt/J3RO9EvgiahHCDRrvcHzPQXPiggKepb+Vvvgiah1vB0Wj1/TEDzaX2g6Ky9IBqzlfVfVCc3JdvdmnwYfWt/F3fF0E0L8INGq2+P2aguVzug+Lkvnx9vmaudi58SJ+vmUvogVvUt/I3XwTNQ7hBo9X3xww0l4Z0/9OrA3erb+Vvvgiah2tu0Gi1f8yluRt1puywgtt3VvT1tymi1wCzS4MXC47srMrSA/btl+n+59oGNDdnK39HD7pN+//yiiTjola+CLoD4QZN4uyPGWgu9X1QfL1xicN96P6HmfgiaB7CDYAWob4Pivp6dQAz8UXQHIQbAC0G3f8AGoILigG0ePVd1AnA99BzA8ArOOrVYdI/wDcRbtAkfGjA0zE7LOC7GJZCozGnCFoCZocFfNcVhZuqqip98cUXOn/+fJNfY/HixerRo4eCg4OVmJio7du3O932gQcekMVisXvExcU1+fhoPD400BIwOyzgu5oUbiorKzVhwgS1bt1acXFxKi4uliQ99thjevbZZxv8OmvXrtW0adM0e/ZsFRQUaNiwYRo9enTd611q0aJFKikpqXscOnRI7dq10x133NGUt4Em4kMDnsbR0gvMDgv4riaFm1mzZunTTz/Vli1bFBwcXNc+cuRIrV27tsGv8+KLL2rChAmaOHGi+vXrp4ULF6pr165assTxhFzh4eGKjo6ue3z88cf67rvvNH78+Ka8DTQRHxrwJM6GSdvGxIplQgDf1KRwY7Va9eqrr+qnP/2pLJb/+8cjNjZWX331VYNeo6qqSvn5+Ro1apRN+6hRo7Rjx44Gvcbrr7+ukSNHqlu3bk63OXv2rCoqKmweuDKsLQVP4myY9FTxHm4PB3xUk+6W+vbbb9WhQwe79tOnT9uEnfocP35c1dXVioqKsmmPiopSaWnpZfcvKSnR3/72N/35z3+ud7vMzEzNmzevQTWhYZhSHJ6kvmFSZoeFGbib1HxN6rn5yU9+or/+9a91P9cGmj/+8Y8aPHhwo17r0jBkGEaDAtLKlSsVERGh5OTkerebNWuWysvL6x6HDh1qVH1wLKJ3oq65J139H8/SNfekE2xgGoZJ4Um4m9QzNKnnJjMzU7/4xS+0e/dunT9/XosWLVJRUZH+9a9/aevWrQ16jcjISPn7+9v10hw7dsyuN+dShmFo+fLluvfeexUYGFjvtkFBQQoKCmpQTY3l98PJZnldtCycB+Zi6QV4Elao9wxNCjdDhgzRP//5Tz3//PO6+uqr9fe//10DBgzQv/71L1177bUNeo3AwEAlJiYqJydHY8eOrWvPycnRmDFj6t1369at+vLLLzVhwoSmlO8yIQe2mXp8AAyTwrNwN6lnaPIMxddee63+9Kc/XdHBp0+frnvvvVcDBw7U4MGDlZWVpeLiYk2ZMkXShSGlw4cPa9WqVTb7vf766xo0aJDi4+Ov6PhX6ocew1UTEmFqDTCf3w8nCbomq/1GfOE6h29U+u93JcPgmzLcjhXqPUOTws2mTZvk7++vm2++2ab9f/7nf1RTU6PRo0c36HXGjRunsrIyZWRkqKSkRPHx8dq0aVPd3U8lJSV2c96Ul5frnXfe0aJFi5pSukvVhESoJjTS7DIAn8dSC/AU9Q2TcqGx+zQp3MycOdPhZH2GYWjmzJkNDjeSNHXqVE2dOtXhcytXrrRrCw8PV2VlZYNfH82DP1J4Eq5zgKdwNkwqwyCAu1GTws2+ffsUGxtr137NNdfoyy+/vOKi4Nn4luy5/M6Um12CKZxe53D8G/mdPu7majyDr54LnsDRFASfr5nrYEsCeHNpUrgJDw/X/v371b17d5v2L7/8UqGhoa6oCx6Mb8meJzw8XAGBQdL+ht2t6G1CQ4J06lSVw/bQ3e+aUJFnCAgMUnh4uNll+BxHPdtcaOxeTQo3t99+u6ZNm6b169fr6quvlnQh2DzxxBO6/fbbXVogPA9/pJ4nKipKa1avUnm5939b37x5s5YuXaovv/xSvXr10pQpU3TnnXfqkUcekWH833UOFotFzz77rEaOHGliteYKDw+/7NQacC1nPduBET9S1cljdttzoXHzaFK4+cMf/qBf/OIXuuaaa9SlSxdJ0jfffKNhw4bp+eefd2mB8DzcDeCZoqKivP6DzGq16uGHH677ubCwUI888oiys7OVnZ2t9PR07dq1S/Hx8crIyLjstBKAqznr2bbIogvL1jAfkzs0eVhqx44dysnJ0aeffqqQkBAlJCRo+PDhrq4PHohJ02CWBQsW2LUZhqHMzEzl5uYqNjZWkyZNUlZWlvr06WNChfB1znq2z50+yXxMbtTkeW4sFotGjRplt/AlvB+TpsEsRUVFTtutVqvmzJmjoqIipaamKiMj47LLswCuVl/PNmuduU+Tw83777+v999/X8eOHVNNTY3Nc8uXL7/iwuDZ+COFGeLi4pSXl2fX3qlTJ5uZzgsLC5WSkqLs7GwCDtyKnm3P0KSFM+fNm6dRo0bp/fff1/Hjx/Xdd9/ZPACgOaSlpdktrGuxWGwuJK5VO1wFuFNtz3br6J7yCwhSYESUgiI66OuNS/T5mrksoOkmTeq5Wbp0qVauXKl7773X1fUAgFPJycnKzs5WZmamioqKFBcXp7S0NP3mN79xuL2zYSygOdX2bDMnmHmaFG6qqqo0ZMgQV9cCAJeVnJxsN9TkbLgqLi7OTVUB9pgTzDxNGpaaOHGi/vznP7u6FgBoEmfDVWlpaSZVBDAnmJma1HNz5swZZWVlafPmzUpISFBAQIDN8y+++KJLigOAhqgdrmKeG8/kq0tBBLeLUuWxQ3btIVdF+eSyIO48D5oUbj777DP1799fkrRr1y6b5y799gQA7pCcnMw8Nx7G15cF6fGjMBXZT0qs7j9q67PLgrhrSZAmhZsPPvjA1XUAQKNYrVYtWLDA5sJiRwv6wjy+tCyIM5s3b9bLL7+sffv2qXfv3nrsscdYEsQNM6k3eZ4bADCL1Wq1mdcmLy9PKSkpevXVV02sCo74wrIg9enTp49Gjhxp16PoKJwzJ5PrNDnc5OXlad26dSouLlZVle1qvNnZ2VdcGAA442wZhtdee01XXXWVCRUBDecsnDPppOs06W6pN998U0OHDtXu3bu1fv16nTt3Trt379Y//vEPt4ylAfBtzuav+fLLL91cCdB49a2RBtdoUrhZsGCBXnrpJW3cuFGBgYFatGiR9uzZozvvvFMxMTGurhEAbDibv6ZXr15urgS+zmq1KikpSaGhoUpKSpLVar3sPvWtkQbXaFK4+eqrr3TrrbdKkoKCgnT69GlZLBb99re/VVZWlksLBIBLOZrXRpLKy8u1fft2paamNuhDBrgStcNLeXl5qqysrBteuvjcs1qtSk1NtTkvnYVzJp10nSaFm3bt2unUqVOSpM6dO9fdDn7y5ElVVla6rjoAcKB2Xpvab8y9e/eWJB08eFA1NTV1C2cScNCcLje8VBt+CgsLbc7LESNGMOlkM2tSuBk2bJhycnIkSXfeeacef/xxPfTQQ/r1r3+tESNGuLRAAHAkOTlZubm5+v777xUREWH3PNcwoLldbnjJWfj5xz/+YRPOk5KStH79eiaddKEm3S316quv6syZM5KkWbNmKSAgQB9++KFSUlL09NNPu7RAeKaT+/JVmrtBZ44fVnBkZ0UPuo21UmAarmGAGS63pll956WjNdLgOk0elurUqdOFF/Dz05NPPql3331XL774Irdh+oDalW4rSw+o5nxV3Uq3J/flm10afBTXMMAMl1vTjPPSPE0KN/7+/jp2zH5O6bKyMvn7+19xUfBs9a10C5iBhTNhhkuv/bp0eInz0jxNCjeGYThsP3v2rAIDA6+oIHg+VrqFp6n9kElISJCfn58SEhK4hgFucfG1X7m5uTbnHOeleRp1zc3LL78s6ULyXLZsmdq0aVP3XHV1tbZt26ZrrrnGtRXC4wRHdlZl6QH79vadTagGuICFM+GJOC/N0ahw89JLL0m60HOzdOlSmyGowMBAde/eXUuXLnVthfA40YNu0/6/vCLp4h48i6Kvv82skgAAqNOocHPgwIVv6zfddJOys7O5eNhHRfROVM8xj6o0d6POlB1WcPvOir7+NkX0GmB2aQAANO1W8A8++MDm5+rqahUWFqpbt24EHh8R0TuRW78BAB6pSRcUT5s2Ta+//rqkC8Fm+PDhGjBggLp27aotW7a4sj4AAIBGaVK4Wbduna677jpJ0oYNG/T111/r888/17Rp0zR79myXFggAANAYTQo3ZWVlio6OliRt2rRJd9xxh/r06aMJEyaosLDQpQUCAAA0RpPCTVRUlHbv3q3q6mq99957GjlypCSpsrKSSfwAAPhfjlYFR/NrUrgZP3687rzzTsXHx8tisejnP/+5JCk3N5d5bgAAkPNVwQk4za9Jd0vNnTtX8fHxOnTokO644w4FBQVJurAsw8yZM11aIAAALZGzVcEzMzNZNLOZNSncSFJqaqpd2/33339FxQAA4C1Yrd48DQ43L7/8siZNmqTg4OC6ZRiceeyxx664MAAAWrK4uDjl5eU5bEfzanC4eemll3T33XcrODi4bhkGRywWC+EGAODz0tLSlJKSYrPYNKuCu0eDw03t0guX/jcAALBXuyp4enq6du3apfj4eGVkZLAquBs0ONxMnz69QdtZLBa98MILTS4IAABvwarg5mhwuCkoKLD5OT8/X9XV1erbt68kae/evfL391diIusNAXA/q9WqOXPmqKioSKmpqcrIyOCOFMBHNTjcXLxY5osvvqi2bdvqT3/6U91Cmd99953Gjx+vYcOGub5KAKhH7XwitWrnE8nOzibgAD6oSZP4vfDCC8rMzLRZAfyqq67SM888w5AUALerbz4RAL6nSeGmoqJCR48etWs/duyYTp06dcVFAUBjMJ8IgIs1KdyMHTtW48eP19tvv61vvvlG33zzjd5++21NmDBBKSkprq4RAOrlbN4Q5hMBfFOTws3SpUt166236p577lG3bt3UrVs33X333Ro9erQWL17s6hoBoF5paWmyWCw2bcwnAviuJoWb1q1ba/HixSorK1NBQYE++eQTnThxQosXL1ZoaKirawSAetXOJ5KQkCA/Pz8lJCRo/fr1zCcC+Kgmry0lSaGhoUpISHBVLQDQZMwnAqBWk3puAAAAPNUV9dy4wuLFi/WHP/xBJSUliouL08KFC+udK+fs2bPKyMjQmjVrVFpaqi5dumj27Nl68MEH3Vj1BX5nyt1+THgezgMA8Cymhpu1a9dq2rRpWrx4sYYOHarXXntNo0eP1u7duxUTE+NwnzvvvFNHjx7V66+/rl69eunYsWM6f/68W+sODw9XQGCQtH+rW48LzxUQGKTw8HCzywAAyORw8+KLL2rChAmaOHGiJGnhwoX6n//5Hy1ZssTh5Fvvvfeetm7dqv3796tdu3aSpO7du7uzZElSVFSU1qxepfJy3/7GfvDgQc2fP1+zZ89Wt27dzC7HVOHh4YqKijK7DACATAw3VVVVys/P18yZM23aR40apR07djjc591339XAgQP1+9//XqtXr1ZoaKhuv/12/e53v1NISIjDfc6ePauzZ8/W/VxRUeGS+qOiovgw+1/dunXj4k0AgMcwLdwcP35c1dXVdgEhKipKpaWlDvfZv3+/PvzwQwUHB2v9+vU6fvy4pk6dqhMnTmj58uUO98nMzNS8efNcXj8AAPBMpt8tdenEW4Zh2LXVqqmpkcVi0RtvvKGkpCTdcsstevHFF7Vy5Ur98MMPDveZNWuWysvL6x6HDh1y+XsAAACew7Sem8jISPn7+9v10hw7dszpcE/Hjh3VuXNnmws3+/XrJ8Mw9M0336h37952+wQFBSkoKMi1xQMAAI9lWs9NYGCgEhMTlZOTY9Oek5OjIUOGONxn6NChOnLkiL7//vu6tr1798rPz09dunRp1noBeDar1arU1FRt375dqampslqtZpcEwCSmDktNnz5dy5Yt0/Lly7Vnzx799re/VXFxsaZMmSLpwpDSfffdV7f9b37zG7Vv317jx4/X7t27tW3bNs2YMUMPPvig0wuKAXg/q9WqsWPHqrCwUDU1NSosLFRKSgoBB/BRpt4KPm7cOJWVlSkjI0MlJSWKj4/Xpk2b6m4rLikpUXFxcd32bdq0UU5Ojh599FENHDhQ7du315133qlnnnnGrLcA4H+dOXPG5u/VnebMmWPXZhiG0tPTFRsba0JFUkxMjIKDg005NuDrTJ+heOrUqZo6darD51auXGnXds0119gNZQEwX3FxsSZNmmTKsYuKihy279q1y7SaWN8KMI/p4QaAd4iJiVFWVpYpx05NTVVhYaFde3x8vGk1OZtlHUDzI9wAcIng4GDTeioyMjKUkpIiwzDq2iwWizIyMug9AXyQ6fPcAMCVSk5OVnZ2tpKSkhQaGqqkpCStX79eY8aMMbs0ACYg3ADwCsnJycrNzdX333+v3Nxcgg08AlMUmINwAwBAM2CKAvMQbgAAaAYLFiywazMMQ5mZmSZU41sINwAANANnUxQ4a4frEG4AAGgGcXFxjWqH6xBuAABoBmlpabJYLDZtFotFaWlpJlXkOwg3AAA0g9opChISEuTn56eEhASmKHATJvEDAKCZJCcnKzY2VpMmTWJJDjei5wYAAHgVwg0AAPAqhBsAAOBVCDcAAMCrEG4AAIBXIdwAAACvQrgBAABehXADAAC8CuEGAAB4FcINAADwKoQbAADgVQg3AADAqxBuAACAVyHcAAAAr0K4AQAAXoVwAwAAvArhBgAAeBXCDQAA8CqEGwAA4FUINwAAwKsQbgAAgFch3AAAAK9CuAEAAF6FcAMAALwK4QYAAHgVwg0AAPAqhBsAAOBVCDcAAMCrEG4AAIBXIdwAAACvQrgBAABehXADAAC8CuEGAAB4FcINAADwKoQbAADgVQg3AADAqxBuAACAVyHcAAAAr2J6uFm8eLF69Oih4OBgJSYmavv27U633bJliywWi93j888/d2PFAADAk5kabtauXatp06Zp9uzZKigo0LBhwzR69GgVFxfXu98XX3yhkpKSukfv3r3dVDEAAPB0poabF198URMmTNDEiRPVr18/LVy4UF27dtWSJUvq3a9Dhw6Kjo6ue/j7+7upYgAA4OlMCzdVVVXKz8/XqFGjbNpHjRqlHTt21Lvvj3/8Y3Xs2FEjRozQBx98UO+2Z8+eVUVFhc0DAAB4L9PCzfHjx1VdXa2oqCib9qioKJWWljrcp2PHjsrKytI777yj7Oxs9e3bVyNGjNC2bducHiczM1Ph4eF1j65du7r0fQAAAM/SyuwCLBaLzc+GYdi11erbt6/69u1b9/PgwYN16NAhPf/88xo+fLjDfWbNmqXp06fX/VxRUUHAAQDAi5nWcxMZGSl/f3+7Xppjx47Z9ebU5/rrr9e+ffucPh8UFKSwsDCbBwAA8F6mhZvAwEAlJiYqJyfHpj0nJ0dDhgxp8OsUFBSoY8eOri4PAAC0UKYOS02fPl333nuvBg4cqMGDBysrK0vFxcWaMmWKpAtDSocPH9aqVaskSQsXLlT37t0VFxenqqoqrVmzRu+8847eeecdM98GAADwIKaGm3HjxqmsrEwZGRkqKSlRfHy8Nm3apG7dukmSSkpKbOa8qaqq0v/7f/9Phw8fVkhIiOLi4vTXv/5Vt9xyi1lvAQAAeBjTLyieOnWqpk6d6vC5lStX2vz85JNP6sknn3RDVQAAoKUyffkFAAAAVyLcAAAAr0K4AQAAXoVwAwAAvArhBgAAeBXCDQAA8CqEGzSJ1WpVamqqtm/frtTUVFmtVrNLAgBAEuEGTWC1WjV27FgVFhaqpqZGhYWFSklJIeAAADwC4QaNtmDBArs2wzCUmZlpQjUAANgi3KDRioqKGtUOAL6KIXxzEG7QaHFxcY1qBwBfxBC+eQg3aLS0tDRZLBabNovForS0NJMqAgDPwxC+eQg3aLTk5GRlZ2crISFBfn5+SkhI0Pr16zVmzBizSwMAj8EQvnkIN2iS5ORkrVu3TsOGDdO6desINgBwCYbwzUO4AQCgGTCEbx7CDQAAzYAhfPO0MrsAAAC8VXJysmJjYzVp0iRlZWWpT58+ZpfkE+i5AQAAXoVwAwAAvArhBk3CrJsAAE9FuEGjMesmAMCTcUFxC3XmzBkVFxebcuw5c+bYtRmGofT0dMXGxppQkRQTE6Pg4GBTjg0AzlitVs2ZM0dFRUVKTU1VRkaGkpOTzS7L61kMwzDMLsKdKioqFB4ervLycoWFhZldTpPt3btXkyZNMuXY27dvV01NjV27n5+fhg0bZkJF4i4EAB6ntpf7YhaLRdnZ2QScJmjM5zfhpoUys+cmNTVVhYWFdu0JCQlat26dCRXRcwPAMf6ttNWS/60k3NTDW8KNmaxWq1JSUnTxqWOxWJicCoDHoZfbVkvu5Sbc1INw4xpWq1WZmZkqKipSXFyc0tLSCDYAPA49N7boufFShBsAgDvQy+1ajfn85lZwAACaQe3aUklJSQoNDVVSUhLBxk3ouQEAAB6PnhsAAOCzCDcAAMCrEG4AAIBXIdwAAACvQrgBAABehXADAAC8CuEGAAB4FcINAADwKoQbAADgVQg3AADAqxBuAACAVyHcAAAAr0K4AQAAXoVwAwAAvArhBgAAeBXCDQAA8CqEGwAA4FUINwAAwKsQbgAAgFch3AAAAK9ierhZvHixevTooeDgYCUmJmr79u0N2u+f//ynWrVqpf79+zdvgQAAoEUxNdysXbtW06ZN0+zZs1VQUKBhw4Zp9OjRKi4urne/8vJy3XfffRoxYoSbKgUAAC2FxTAMw6yDDxo0SAMGDNCSJUvq2vr166fk5GRlZmY63e+uu+5S79695e/vL6vVqp07dzb4mBUVFQoPD1d5ebnCwsKupHwAAOAmjfn8Nq3npqqqSvn5+Ro1apRN+6hRo7Rjxw6n+61YsUJfffWV0tPTG3Scs2fPqqKiwuYBAAC8l2nh5vjx46qurlZUVJRNe1RUlEpLSx3us2/fPs2cOVNvvPGGWrVq1aDjZGZmKjw8vO7RtWvXK64dAAB4LtMvKLZYLDY/G4Zh1yZJ1dXV+s1vfqN58+apT58+DX79WbNmqby8vO5x6NChK64ZAAB4roZ1fzSDyMhI+fv72/XSHDt2zK43R5JOnTqljz/+WAUFBXrkkUckSTU1NTIMQ61atdLf//53/exnP7PbLygoSEFBQc3zJgAAgMcxrecmMDBQiYmJysnJsWnPycnRkCFD7LYPCwtTYWGhdu7cWfeYMmWK+vbtq507d2rQoEHuKh0AAHgw03puJGn69Om69957NXDgQA0ePFhZWVkqLi7WlClTJF0YUjp8+LBWrVolPz8/xcfH2+zfoUMHBQcH27UDAADfZWq4GTdunMrKypSRkaGSkhLFx8dr06ZN6tatmySppKTksnPeAAAAXMzUeW7MwDw3AAC0PI35/Da15wYAXKW6ulqfffaZTpw4oXbt2ikhIUH+/v5mlwXABIQbAC3etm3btHjxYpu7L6OjozV16lQNHz7cxMoAmMH0eW4A4Eps27ZN6enpOn/+vMrKypSXl6eysjKdP39e6enp2rZtm9klAnAzrrkB0GJVV1fr7rvv1vnz5/XOO+/YPGexWJSSkqKAgACtWbOGISqghWsRa0sBwJX67LPPVFpaqi+++MLuOcMwtHfvXpWUlOizzz4zoToAZiHcAGixTpw4IUnav3+/w+dr22u3A+AbCDcAWqx27dpJknr27Onw+dr22u0A+AbCDYAWKyEhQdHR0erbt6/dgrsWi0V9+vRRx44dlZCQYFKFAMxAuAHQYvn7+2vq1KkqKytTSkqKrr32WoWGhuraa69VSkqKysrK9B//8R9cTAz4GMINgBZt+PDhmjdvnlq1aqV27dpp4MCBateunQICAjRv3jzmuQF8ELeCA/AKzFAMeDeWXwDgc/z9/fXjH//Y7DIAeACGpQAAgFch3AAAAK9CuAEAAF6FcAMAALwK4QYAAHgVwg0AAPAqhBsAAOBVCDcAAMCrEG4AAIBX8bkZimtXm6ioqDC5EgAA0FC1n9sNWTXK58LNqVOnJEldu3Y1uRIAANBYp06dUnh4eL3b+NzCmTU1NTpy5Ijatm0ri8VidjktWkVFhbp27apDhw6xCCk8AuckPBHnpWsYhqFTp06pU6dO8vOr/6oan+u58fPzU5cuXcwuw6uEhYXxBwuPwjkJT8R5eeUu12NTiwuKAQCAVyHcAAAAr0K4QZMFBQUpPT1dQUFBZpcCSOKchGfivHQ/n7ugGAAAeDd6bgAAgFch3AAAAK9CuDHJjTfeqGnTppldRrNYuXKlIiIizC7DxpYtW2SxWHTy5EmzS2kxvPkcdRfOO9finHQNXzgvCTcmyc7O1u9+97sGbfv111/LYrFo586dzVuUXHPSjxs3Tnv37nVdUS4wZMgQlZSUNHiOBHjuOVp7rNpHeHi4rr/+em3YsMHh9gsWLJC/v7+effbZZq8NzctTz8n6/t3s37+/5s6da9fOedm8CDcmadeundq2bevS16yqqnLp6zVVSEiIOnToYHYZNgIDAxUdHe10Vurq6mrV1NS4uSrP5unn6ObNm1VSUqLc3FwlJSXpV7/6lXbt2mW33YoVK/Tkk09q+fLlLjs2zOHp52RjcF42L8KNSS7uXu3evbsWLFigBx98UG3btlVMTIyysrLqtu3Ro4ck6cc//rEsFotuvPFGSdIDDzyg5ORkZWZmqlOnTurTp48kac2aNRo4cKDatm2r6Oho/eY3v9GxY8cuW9PXX3+tm266SZJ01VVXyWKx6IEHHtCGDRsUERFR9+G/c+dOWSwWzZgxo27fyZMn69e//rUk+2GpuXPnqn///lq9erW6d++u8PBw3XXXXXXrfDXkd/XII4/okUceUUREhNq3b6///M//tFk87XLv+dJvVrU1bty4UbGxsQoKCtLBgwcbVI+v8MRz9GLt27dXdHS0rrnmGs2fP1/nzp3TBx98YLPN1q1b9cMPPygjI0OnT5/Wtm3bGnWMJUuW6Oqrr1ZgYKD69u2r1atX2zxvsVi0bNkyjR07Vq1bt1bv3r317rvvOnyt06dPKywsTG+//bZN+4YNGxQaGtrgvwdf5unnZENxXjY/wo2HeOGFFzRw4EAVFBRo6tSp+o//+A99/vnnkqSPPvpI0v99U83Ozq7b7/3339eePXuUk5OjjRs3SrrwTeR3v/udPv30U1mtVh04cEAPPPDAZWvo2rWr3nnnHUnSF198oZKSEi1atEjDhw/XqVOnVFBQIOnCH2ZkZKS2bt1at++WLVt0ww03OH3tr776SlarVRs3btTGjRu1devWRnXH/ulPf1KrVq2Um5url19+WS+99JKWLVtW93xT3nNlZaUyMzO1bNkyFRUVeVxvk6fxhHPUkXPnzumPf/yjJCkgIMDmuddff12//vWvFRAQoF//+td6/fXXG/y669ev1+OPP64nnnhCu3bt0uTJkzV+/Hi7ADVv3jzdeeed+uyzz3TLLbfo7rvv1okTJ+xeLzQ0VHfddZdWrFhh075ixQqlpqa6vEfCF3jqOXk5nJduYMAUN9xwg/H4448bhmEY3bp1M+65556652pqaowOHToYS5YsMQzDMA4cOGBIMgoKCmxe4/777zeioqKMs2fP1nusjz76yJBknDp16rJ1ffDBB4Yk47vvvrNpHzBggPH8888bhmEYycnJxvz5843AwECjoqLCKCkpMSQZe/bsMQzDMFasWGGEh4fX7Zuenm60bt3aqKioqGubMWOGMWjQoMvWYxgXflf9+vUzampq6tqeeuopo1+/fk73ufQ9X/q+VqxYYUgydu7c2aAafJGnnqO1xwoJCTFCQ0MNPz8/Q5LRvXt3o6ysrG678vJyo3Xr1nX/HxcUFBitW7c2ysvLG/L2jSFDhhgPPfSQTdsdd9xh3HLLLXU/SzL+8z//s+7n77//3rBYLMbf/vY3wzDsz7vc3FzD39/fOHz4sGEYhvHtt98aAQEBxpYtWxpUk6/z1HPS2b+bhmEY1113nZGenl73M+ele9Bz4yESEhLq/ttisSg6OrpBXaLXXnutAgMDbdoKCgo0ZswYdevWTW3btq3rji0uLm5yfTfeeKO2bNkiwzC0fft2jRkzRvHx8frwww/1wQcfKCoqStdcc43T/bt3727zDaBjx46N6vK9/vrrba6XGTx4sPbt26fq6mpJTXvPgYGBNr931M/TztG1a9eqoKBA7777rnr16qVly5apXbt2dc//+c9/Vs+ePXXddddJunBhZ8+ePfXmm2826PX37NmjoUOH2rQNHTpUe/bssWm7+PcSGhqqtm3bOv29JCUlKS4uTqtWrZIkrV69WjExMRo+fHiDaoItTzsnG4Lz0j0INx7i0u50i8XSoAtcQ0NDbX4+ffq0Ro0apTZt2mjNmjXKy8vT+vXrJV3ZhXM33nijtm/frk8//VR+fn6KjY3VDTfcoK1bt152SEpq+vtriKa+55CQEKcXGMOep52jXbt2Ve/evXXrrbdq2bJlGjdunM0/3suXL1dRUZFatWpV9ygqKmrUEMCl54dhGHZtjf29TJw4sW4IYMWKFRo/fjznYRN5yjlZu9J3eXm53XMnT560uUuT89I9CDctQO03jNpeivp8/vnnOn78uJ599lkNGzZM11xzTaN6SJwdq/a6m4ULF+qGG26QxWLRDTfcoC1btjQo3Fypf//733Y/9+7dW/7+/lf8nnHl3HmOOnLDDTcoPj5e8+fPlyQVFhbq448/1pYtW7Rz5866x7Zt25SXl+fwrqpL9evXTx9++KFN244dO9SvX78rqvWee+5RcXGxXn75ZRUVFen++++/oteDY+48J3v37i0/Pz/l5eXZtJeUlOjw4cPq27evJM5Ld2pldgG4vA4dOigkJETvvfeeunTpouDgYKfztcTExCgwMFCvvPKKpkyZol27djV4XghJ6tatmywWizZu3KhbbrlFISEhatOmjcLDw9W/f3+tWbNGixYtknQh8Nxxxx06d+5cXRduczl06JCmT5+uyZMn65NPPtErr7yiF154QdKVv2dcOXeeo8488cQTuuOOO/Tkk0/q9ddfV1JSksNu9cGDB+v111/XSy+9VO/rzZgxQ3feeacGDBigESNGaMOGDcrOztbmzZuvqM6rrrpKKSkpmjFjhkaNGqUuXbpc0evBMXeek23bttXkyZP1xBNPqFWrVrruuut05MgRzZ49W/369dOoUaMkifPSjei5aQFatWqll19+Wa+99po6deqkMWPGON32Rz/6kVauXKl169YpNjZWzz77rJ5//vkGH6tz586aN2+eZs6cqaioKD3yyCN1z910002qrq6uCzJXXXWVYmNj9aMf/eiKvzVczn333acffvhBSUlJevjhh/Xoo49q0qRJkq78PePKufMcdeaXv/ylunfvrvnz52vNmjX61a9+5XC7X/3qV1qzZs1lhxuSk5O1aNEi/eEPf1BcXJxee+01rVixwiVBfsKECaqqqtKDDz54xa8Fx9x9Tr700kuaOHGi0tLSFBcXp7vvvls9evTQ3//+d7Vq1UpVVVWcl27EquDweDfeeKP69++vhQsXml0K4BJvvPGGHn/8cR05csTuwlbALN50XjIsBQBuUllZqQMHDigzM1OTJ09u8R8g8A7eeF4yLOVjpkyZojZt2jh8TJkyxe31FBcXO62nTZs2Lr8NE57PXedoXFyc0+O88cYbLjvOxX7/+9+rf//+ioqK0qxZs5rlGHA9d/67yXnpGgxL+Zhjx46poqLC4XNhYWFun6X3/Pnz+vrrr50+3717d7VqRQejL3HXOXrw4EGdO3fO4XNRUVEtd2ZWuJw7/93kvHQNwg0AAPAqDEsBAACvQrgBAABehXADAAC8CuEGAAB4FcINAADwKoQbAB5p7ty56t+/v9llAGiBCDcA0ADO5h4B4HkINwCaTU1NjZ577jn16tVLQUFBiomJ0fz58yVJTz31lPr06aPWrVurZ8+eevrpp+sCxMqVKzVv3jx9+umnslgsslgsWrlypSSpvLxckyZNUocOHRQWFqaf/exn+vTTT22O+8wzz6hDhw5q27atJk6cqJkzZ9r0AtXU1CgjI0NdunRRUFCQ+vfvr/fee6/u+a+//loWi0VvvfWWbrzxRgUHBysrK0thYWF6++23bY61YcMGhYaG6tSpU83wGwTQFIQbAM1m1qxZeu655/T0009r9+7d+vOf/6yoqChJUtu2bbVy5Urt3r1bixYt0h//+Ee99NJLkqRx48bpiSeeUFxcnEpKSlRSUqJx48bJMAzdeuutKi0t1aZNm5Sfn68BAwZoxIgROnHihKQLi//Nnz9fzz33nPLz8xUTE6MlS5bY1LVo0SK98MILev755/XZZ5/p5ptv1u233659+/bZbPfUU0/pscce0549ezR27FjdddddWrFihc02K1asUGpqKjPHAp7EAIBmUFFRYQQFBRl//OMfG7T973//eyMxMbHu5/T0dOO6666z2eb99983wsLCjDNnzti0X3311cZrr71mGIZhDBo0yHj44Ydtnh86dKjNa3Xq1MmYP3++zTY/+clPjKlTpxqGYRgHDhwwJBkLFy602SY3N9fw9/c3Dh8+bBiGYXz77bdGQECAsWXLlga9RwDuQc8NgGaxZ88enT17ViNGjHD4/Ntvv62f/vSnio6OVps2bfT0009fdqHU/Px8ff/992rfvr3NgoIHDhzQV199JUn64osvlJSUZLPfxT9XVFToyJEjGjp0qM02Q4cO1Z49e2zaBg4caPc6cXFxWrVqlSRp9erViomJ0fDhw+utG4B7sSIhgGYREhLi9Ll///vfuuuuuzRv3jzdfPPNCg8P15tvvqkXXnih3tesqalRx44dtWXLFrvnIiIi6v7bYrHYPGc4WELP0TaXtoWGhtrtN3HiRL366quaOXOmVqxYofHjx9vtB8Bc9NwAaBa9e/dWSEiI3n//fbvn/vnPf6pbt26aPXu2Bg4cqN69e+vgwYM22wQGBqq6utqmbcCAASotLVWrVq3Uq1cvm0dkZKQkqW/fvvroo49s9vv444/r/jssLEydOnXShx9+aLPNjh071K9fv8u+r3vuuUfFxcV6+eWXVVRUpPvvv/+y+wBwL3puADSL4OBgPfXUU3ryyScVGBiooUOH6ttvv1VRUZF69eql4uJivfnmm/rJT36iv/71r1q/fr3N/t27d9eBAwe0c+dOdenSRW3bttXIkSM1ePBgJScn67nnnlPfvn115MgRbdq0ScnJyRo4cKAeffRRPfTQQxo4cKCGDBmitWvX6rPPPlPPnj3rXnvGjBlKT0/X1Vdfrf79+2vFihXauXOn3njjjcu+r6uuukopKSmaMWOGRo0apS5durj8dwfgCpl90Q8A71VdXW0888wzRrdu3YyAgAAjJibGWLBggWEYhjFjxgyjffv2Rps2bYxx48YZL730khEeHl6375kzZ4xf/epXRkREhCHJWLFihWEYFy5UfvTRR41OnToZAQEBRteuXY27777bKC4urts3IyPDiIyMNNq0aWM8+OCDxmOPPWZcf/31NnXNmzfP6Ny5sxEQEGBcd911xt/+9re652svKC4oKHD4vt5//31DkvHWW2+57pcFwGUshuFgMBoAvMjPf/5zRUdHa/Xq1S55vTfeeEOPP/64jhw5osDAQJe8JgDXYVgKgFeprKzU0qVLdfPNN8vf31///d//rc2bNysnJ8clr33gwAFlZmZq8uTJBBvAQ3FBMQCvYrFYtGnTJg0bNkyJiYnasGGD3nnnHY0cOfKKX/v3v/+9+vfvr6ioKM2aNcsF1QJoDgxLAQAAr0LPDQAA8CqEGwAA4FUINwAAwKsQbgAAgFch3AAAAK9CuAEAAF6FcAMAALwK4QYAAHgVwg0AAPAq/x9w3ebMITIPuwAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# paired beta div: unweighted distances to compare inter-twin vs inter-disease distances\n",
    "# paired shannon diversity difference from 0 \n",
    "df_beta = pd.read_csv(path + 'outputs/jobs02/Q2_outputs/bray_curtis_distance_matrix.tsv',\n",
    "                      sep='\\t', index_col=0)\n",
    "\n",
    "# grab twin to pair dict\n",
    "pair_to_ids = {}\n",
    "for p in list(df_alpha.TwinPair.values):\n",
    "    df = df_alpha[df_alpha['TwinPair'] == p]\n",
    "    id_UA = str(df[df['Diagnosis'] == 'Unaffected'].index.values[0])\n",
    "    id_RA = str(df[df['Diagnosis'] == 'RA'].index.values[0])\n",
    "    pair_to_ids[p] = (id_UA, id_RA)\n",
    "\n",
    "# get distances for each twin pair per beta div matrix    \n",
    "pair_to_dist = {}\n",
    "for p in list(df_alpha.TwinPair.values):\n",
    "    id_UA, id_RA = pair_to_ids[p]\n",
    "    pair_to_dist[p] = df_beta.loc[id_UA, id_RA]\n",
    "\n",
    "df_paired_beta = pd.DataFrame.from_dict(pair_to_dist, orient='index', columns=['Bray_Curtis'])\n",
    "\n",
    "# grab inter RA distances\n",
    "# this is from bray_curtis_significance_Diagnosis.qzv -> download as tsv\n",
    "df_raw = pd.read_csv(path + 'outputs/jobs02/bray_curtis_raw_data.tsv', \n",
    "                     sep='\\t', index_col=0)\n",
    "df_RA = df_raw[df_raw['Group1'] == 'RA']\n",
    "df_RA = df_RA[df_RA['Group2'] == 'RA']\n",
    "df_UA = df_raw[df_raw['Group1'] == 'Unaffected']\n",
    "df_UA = df_UA[df_UA['Group2'] == 'Unaffected']\n",
    "\n",
    "# compare distances\n",
    "inter_twin = df_paired_beta['Bray_Curtis'].values\n",
    "inter_RA = df_RA['Distance'].values\n",
    "inter_UA = df_UA['Distance'].values\n",
    "\n",
    "u, p = scipy.stats.kruskal(inter_UA, inter_RA, inter_twin)\n",
    "print(u, p)\n",
    "\n",
    "u, p = scipy.stats.mannwhitneyu(inter_twin, inter_RA)\n",
    "# print(u, p)\n",
    "\n",
    "t, p = scipy.stats.ttest_ind(inter_twin, inter_RA)\n",
    "#print(t, p)\n",
    "\n",
    "t, p = scipy.stats.ttest_ind(inter_UA, inter_RA)\n",
    "# print(t, p)\n",
    "\n",
    "f, p = scipy.stats.f_oneway(inter_UA, inter_RA, inter_twin)\n",
    "# print(f, p)\n",
    "\n",
    "category = ['intra_twin_pair']*len(inter_twin) + ['intra_RA_only']*len(inter_RA) + ['intra_UA_only']*len(inter_UA)\n",
    "distances = list(inter_twin) + list(inter_RA) + list(inter_UA)\n",
    "df_dist = pd.DataFrame(data=np.array([category,distances]).T, columns=['category','distance'])\n",
    "df_dist['distance'] = df_dist['distance'].astype(float)\n",
    "df_dist.to_csv(path + 'outputs/jobs02/intra_beta_dist.tsv',sep='\\t')\n",
    "                     \n",
    "sns.boxplot(data=df_dist, x='category', y='distance')\n",
    "sns.swarmplot(data=df_dist, x='category', y='distance', color='black')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.8010989010989249 0.669951839482939\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Axes: xlabel='category', ylabel='distance'>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjcAAAGxCAYAAACeKZf2AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAABBHElEQVR4nO3deXxU1f3/8fcEQhJCFhZJwhIWWSShAQGjSFksFL/SL80Q4latlqrATwQVFSRW0bQQrQuLVjGKUEWrRZKpILWCZVNpGhEwBFTQSCgkIFgSIIQAub8/fGS+DJnsM3MnN6/n45FHnXPvnfsZeiHvOffcc2yGYRgCAACwiACzCwAAAPAkwg0AALAUwg0AALAUwg0AALAUwg0AALAUwg0AALAUwg0AALAUwg0AALCUlmYX4GsVFRU6dOiQwsLCZLPZzC4HAADUgWEYOnHihDp16qSAgJr7ZppduDl06JC6du1qdhkAAKABDhw4oC5dutS4T7MLN2FhYZJ+/MMJDw83uRoAAFAXJSUl6tq1q/P3eE2aXbipvBUVHh5OuAEAoImpy5ASBhQDAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAsASHw6HExESFhoYqMTFRDofD7JIAmIRwA6DJczgcmjBhgnJyclRaWqqcnBwlJycTcIBmqtmtCg7AO8rKylRQUGDKuR977LEqbYZhaO7cuYqLizOhIik2NlbBwcGmnBto7gg3ADyioKBAkydPNuXceXl5btt37dplWk0ZGRnq06ePKecGmjvCDQCPiI2NVUZGhinnTklJUW5ubpX2/v37m1ZTbGysKecFQLgB4CHBwcGm9VSkpaUpOTlZhmE422w2m9LS0ug9AZohBhQDaPLsdrsyMzOVkJCggIAAJSQkKCsrS0lJSWaXBsAE9NwAsAS73a64uDhNnjyZ8S5AM0fPDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDYAmyeFwKDExUaGhoUpMTJTD4TC7JAB+grWlADQ5DodDEyZMcL7OyclRcnKyXnjhBROrAuAv6LkB0OTMnz+/SpthGHr55ZdNqAaAvyHcAGhy8vLy3Lbv27fPx5UA8EeEGwBNTnx8vNv2Xr16+bgSAP6IcAOgyUlNTZXNZqvSXlxcrC1btiglJYUBxkAzRrgB0OTY7XZlZmY6n5bq3bu3JGn//v2qqKhQbm6ukpOTCThAM0W4AdAk2e12ZWdn6+TJk4qMjKyy3TAMpaen+74wAKYj3ABo8qobYFxdOwBrI9wAaPKqG2BcXTsAazM13KSnp+uKK65QWFiYOnbsKLvdrq+++qrW4zZt2qTBgwcrODhYPXv21JIlS3xQLQB/5W6Asc1mU2pqqkkVATCTqeFm06ZNmjZtmv71r39p3bp1OnfunMaOHatTp05Ve0x+fr7GjRun4cOHa/v27UpNTdWMGTO0atUqH1YOwJ9UDjBOSEhQQECAEhISlJWVpaSkJLNLA2ACm2EYhtlFVPr+++/VsWNHbdq0SSNGjHC7z+zZs/Xee+9pz549zrapU6dq586d2rp1a63nKCkpUUREhIqLixUeHu6x2gGY7+uvv9bkyZOVkZGhPn36mF0OAA+qz+9vvxpzU1xcLElq165dtfts3bpVY8eOdWm79tpr9dlnn+ns2bNerQ8AAPg/v1k40zAMzZw5Uz/96U/Vv3//avcrKipSVFSUS1tUVJTOnTuno0ePKiYmxmXbmTNndObMGefrkpISzxYOAAD8it/03Nxzzz364osv9Je//KXWfS8eOFh5Z83djKXp6emKiIhw/nTt2tUzBQMAAL/kF+Fm+vTpeu+997RhwwZ16dKlxn2jo6NVVFTk0nbkyBG1bNlS7du3r7L/nDlzVFxc7Pw5cOCAR2sHAAD+xdTbUoZhaPr06crKytLGjRvVo0ePWo8ZOnSoVq9e7dL24YcfasiQIQoMDKyyf1BQkIKCgjxWMwAA8G+m9txMmzZNK1as0FtvvaWwsDAVFRWpqKhIp0+fdu4zZ84c3Xbbbc7XU6dO1f79+zVz5kzt2bNHr732mpYuXaoHH3zQjI8AAAD8jKnh5qWXXlJxcbFGjRqlmJgY588777zj3KewsFAFBQXO1z169NDatWu1ceNGDRw4UL///e+1ePFiTZw40YyPAAAA/Izpt6Vqs3z58iptI0eO1Oeff+6FigAAQFPnFwOKAQAAPIVwAwAALIVwAwAALIVwAwAALIVwAwAALIVwAwAALIVwAwAALIVwAwAALIVwAwAALIVwAwAALIVwAwAALIVwA8ASHA6HUlJStGXLFqWkpMjhcJhdEgCTEG4ANHkOh0MTJkxQbm6uKioqlJubq+TkZAIO0EwRbgA0efPnz6/SZhiG0tPTTagGgNkINwCavLy8vHq1A7A2wg2AJi8+Pr5e7QCsjXADoMlLTU2VzWZzabPZbEpNTTWpIgBmItwAaPLsdrsyMzOVkJCggIAAJSQkKCsrS0lJSWaXBsAELc0uAAA8wW63Ky4uTpMnT1ZGRob69OljdkkATELPDQAAsBTCDQAAsBTCDRrE4XAoMTFRoaGhSkxMZLI0AIDfINyg3ipng83JyVFpaalycnKYDRYA4DcYUNxElZWVqaCgwJRzP/bYY1XaDMPQ3LlzFRcXZ0JFUmxsrIKDg005NwDAvxBumqiCggJNnjzZlHNXN+vrrl27TKuJp2MAAJUIN01UbGysMjIyTDl3SkqKcnNzq7T379/ftJpiY2NNOS8AwP8Qbpqo4OBg03oq0tLSlJycLMMwnG02m01paWn0ngAATMeAYtQbs8ECAPwZPTdoEGaDBQD4K3puAACApRBuAACApRBuAACApRBuAACApRBuAACApRBuAACApRBuAACApRBuAACApRBuAADwEofDocTERIWGhioxMVEOh8PskpoFwg0AAF7gcDg0YcIE5eTkqLS0VDk5OUpOTibg+ADLLwAALKusrEwFBQWmnPuxxx6r0mYYhubOnau4uDgTKpJiY2MVHBxsyrl9iXADALCsgoICTZ482ZRz5+XluW3ftWuXaTU1l7UACTcAAMuKjY1VRkaGKedOSUlRbm5ulfb+/fubVlNsbKwp5/U1wg0AwLKCg4NN66lIS0tTcnKyDMNwttlsNqWlpTWL3hMzMaAYAAAvsNvtyszMVEJCggICApSQkKCsrCwlJSWZXZrl0XMDAICX2O12xcXFafLkyc1mvIs/oOcGAAAPYE4b/0HPDQAAjVQ5p02lyjltMjMzTXvsuzmj5wYAgEaaP39+lTbDMJSenm5CNSDcoEEcDodSUlK0ZcsWpaSk0P0KoFmrbk6b6trhXYQb1Ftl92tubq4qKiqUm5vLlOIAmrX4+Ph6tcO7CDeoN7pfAcBVamqqbDabS5vNZtPPfvYzerlNQLhBvdH9CgCuKue0ufBpqdmzZ+vJJ5+kl9sEhBvUG92vAFCV3W5Xdna2Tp48qezsbH300UdV9qGX2zcIN6i36rpfU1NTTaoIAPwPvdzmIdyg3phSHABqRy+3eQg3aBC73a6VK1dq+PDhWrlyJcEGAC5CL7d5CDcAAHgBvdzmYfkFAAC8hIUzzUHPDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBTCDQAAsBRTw83mzZs1fvx4derUSTabTQ6Ho8b9N27cKJvNVuXnyy+/9E3BAADA75m6/MKpU6c0YMAATZo0SRMnTqzzcV999ZXCw8Odry+55BJvlAcAAJogU8PNddddp+uuu67ex3Xs2FGRkZGeLwgAADR5TXLMzeWXX66YmBiNHj1aGzZsqHHfM2fOqKSkxOUHAABYV5MKNzExMcrIyNCqVauUmZmpvn37avTo0dq8eXO1x6SnpysiIsL507VrVx9WDAAAfM3U21L11bdvX/Xt29f5eujQoTpw4ICeeeYZjRgxwu0xc+bM0cyZM52vS0pKCDgAAFhYk+q5ceeqq67S3r17q90eFBSk8PBwlx8AAGBdTT7cbN++XTExMWaXAQAA/ISpt6VOnjypffv2OV/n5+drx44dateunWJjYzVnzhwdPHhQr7/+uiRp4cKF6t69u+Lj41VeXq4VK1Zo1apVWrVqlVkfAQAA+BlTw81nn32ma665xvm6cmzM7bffruXLl6uwsFAFBQXO7eXl5XrwwQd18OBBhYSEKD4+Xu+//77GjRvn89oBAIB/MjXcjBo1SoZhVLt9+fLlLq9nzZqlWbNmebkqAADQlDX5MTcAAAAXItwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLIdwAAABLaVILZ/qTw4cPq7i42OwyTLV//36X/23OIiIiFBUVZXYZAAARbhrk8OHDuvXXt+ls+RmzS/EL8+bNM7sE0wW2CtKKN14n4ACAHyDcNEBxcbHOlp/R6Z4jVREcYXY5MFlAWbH07SYVFxcTbgDADxBuGqEiOEIVoR3MLgMAAFyAAcUAAMBSCDcAAMBSCDcAAMBSCDcAAMBSCDcAAMBSeFoKDXJ87zYVZa9W2dGDCu7QWdFXjldk78FmlwUAAD03qL/je7fp278tVmlRvirOlau0KF/f/u15Hd+7zezSAAAg3KD+irJXu2k1VJS9xue1AABwMcIN6q3s6EH37cfctwMA4EuEG9RbcIfO7tvbu28HfMHhcCglJUVbtmxRSkqKHA6H2SUBMAnhBvUWfeV4SbaLWm2Kvmq8GeUAcjgcmjBhgnJzc1VRUaHc3FwlJycTcIBminCDeovsPVg9k6ardXRPBQQGqXV0T/W0z1Bkr0Fml4Zmav78+VXaDMNQenq6CdUAMBuPgqNBInsP5tFv+I28vLx6tQOwNnpuADR58fHx9WoHYG2EGwBNXmpqqmw213FgNptNqampJlUEwEyNCjfl5eX66quvdO7cOU/VAwD1ZrfblZmZqYSEBAUEBCghIUFZWVlKSkoyuzQAJmhQuCktLdUdd9yh1q1bKz4+XgUFBZKkGTNm6Mknn/RogQBQF3a7XStXrtTw4cO1cuVKgg38AlMUmKNB4WbOnDnauXOnNm7cqODgYGf7mDFj9M4773isOAAAmiqmKDBPg8KNw+HQCy+8oJ/+9Kcu97nj4uL0zTffeKw4AACaKqYoME+Dws3333+vjh07Vmk/depUlUF9AAA0R0xRYJ4GhZsrrrhC77//vvN1ZaB55ZVXNHToUM9UBgBAE8YUBeZp0CR+6enp+p//+R/t3r1b586d06JFi5SXl6etW7dq06ZNnq4RAIAmJzU1VcnJyTIMw9nGFAW+0aCem6uvvlqffPKJSktLdemll+rDDz9UVFSUtm7dqsGDmbUWAACmKDBPg5df+MlPfqI///nPnqwFAABLsdvtiouL0+TJk5WRkaE+ffqYXVKz0KCem7Vr1+of//hHlfZ//OMf+vvf/97oogAAaGocDocSExMVGhqqxMREHvk2UYPCzcMPP6zz589XaTcMQw8//HCjiwIAb+MXETypck6bnJwclZaWKicnhzltTNSgcLN3717FxcVVab/sssu0b9++RhcFAN7ELyJ4GnPa+JcGhZuIiAh9++23Vdr37dun0NDQRhcFAJ5QXe8Mv4jgacxp418aFG5++ctf6r777nOZjXjfvn164IEH9Mtf/tJjxQFAQ9XUO8MvIngac9r4lwaFm6efflqhoaG67LLL1KNHD/Xo0UP9+vVT+/bt9cwzz3i6RgBwqutYmZp6Z/hFBE9LTU2tMkM/c9qYp0GPgkdEROjTTz/VunXrtHPnToWEhCghIUEjRozwdH0A6ujw4cMqLi42uwyvWr9+vaZNm+Z8Xdkb88ILL2jMmDHav3+/JGn//v3atWuX2/fYtWuXnn76aX322WdVJle7/fbb9fXXX3v3Q/hQRESEoqKizC6jWaic0yY9PV15eXmKj49XamqqDMNQSkqK8vLylJKSorS0NNntdrPLtTybceHf7magpKREERERKi4uVnh4eIPe4+uvv9bkyZN1Ku6Xqgjt4OEK0dQEnDqq0N3vmTqHxeHDh3Xrr2/T2fIzppzfVz7//HOdOHGiSntYWJgGDRpUr32PHj2qgoICnTp1SqGhoYqNjVWHDtb6+xzYKkgr3nidgGOSylujF7LZbMrMzCTgNEB9fn83eBK/jz76SB999JGOHDmiiooKl22vvfZaQ98WQAMUFxfrbPkZne45UhXBEWaX4zWnPtnqvv30Gf0npIcKP/9IZT8cVnC7KIX3GqIT2zdKuvD7m00dh6XoVI/+CpHU9+L38U7ZpggoK5a+3aTi4mLCjUlqujVKuPGuBoWbJ554QmlpaRoyZIhiYmJYCRzwExXBEZbuTQzu0FmlRflV2lu2aatvPljufF165IBKj/xHUYnjdKJgj8qOHVRw+86Kvmq8InoNUkWVdwA8j4Hr5mlQuFmyZImWL1+uX//6156uBwCqFX3leH37t+d1cW+M+69Xhk4U7NFlt871SW3AxeLj45WTk+O2Hd7VoKelysvLdfXVV3u6FgCoUWTvweqZNF2to3sqIDBIraN7qqd9hs6e/K/b/cuOHfRxhcD/4Qkq8zQo3Nx555166623PF0LANQqsvdgXXbrXA28N0OX3TpXkb0GKbhDZ7f7Brd33w74AquCm6dBt6XKysqUkZGh9evXKyEhQYGBgS7bn3vuOY8UBwB1Ud3tquirxptVEiCJVcHN0qBw88UXX2jgwIGSVGUuCQYXA/C1yttVRdlrXAYPR/YaVPvBACynQeFmw4YNnq4DABolsvdgRfYebHYZAPxAg8bcAAAA+KsGT+KXk5OjlStXqqCgQOXl5S7bMjMzG10YAABAQzSo5+btt9/WsGHDtHv3bmVlZens2bPavXu3/vnPfyoiwrqzowIAAP/XoJ6b+fPna8GCBZo2bZrCwsK0aNEi9ejRQ1OmTFFMTIynawSAKo7v3aai7NUqO3pQwR06K/rK8Yy5ASCpgT0333zzjX7xi19IkoKCgnTq1CnZbDbdf//9ysjI8GiBAHCx43u36du/LVZpUb4qzpWrtChf3/7teR3fu83s0gD4gQaFm3bt2jlX2+3cubPzcfDjx4+rtLTUc9UBgBtF2avdtBoqyl7j81qAmjgcDqWkpGjLli1KSUmRw+Ewu6RmoUHhZvjw4Vq3bp0k6YYbbtC9996ru+66SzfffLNGjx7t0QIB4GJlR90vq8ByC/AnDodDEyZMUG5urioqKpSbm6vk5GQCjg80aMzNCy+8oLKyMknSnDlzFBgYqI8//ljJycl69NFHPVogAFysutXBq1tugfE5MMP8+fOrtBmGofT0dNntdt8X1Iw0+LZUp06dfnyDgADNmjVL7733np577jm1bdvWowUCwMWirxwvVVkL3P1yC4zPgVny8vLq1Q7PaVC4adGihY4cOVKl/dixY2rRokWjiwKAmrhbHTwq8Rcq+td72rHwLn254nFneGF8DswSHx9fr3Z4ToNuSxmG4bb9zJkzatWqVaMKAoC6uHC5hcremUqVvTM9k6YzPgemSU1NVXJyssvvTJvNptTUVBOrah7qFW4WL/7xHw+bzaZXX31Vbdq0cW47f/68Nm/erMsuu8yzFQJALWrqnanv+BzAU+x2uzIzMzV37lzt2rVL/fv3V1pampKSkswuzfLqFW4WLFgg6ceemyVLlrjcgmrVqpW6d++uJUuWeLZCAKhFTb0z3cdN0bd/e17ShT3O7sfnAJ5mt9sVFxenyZMnKyMjQ3369DG7pGahXuEmP//Hbz/XXHONMjMzGTwMwGvq84RTTb0zleNzirLXqOzYQQW376zoq8Yrstcgb38ESDp8+LCKi4vNLsNU+/fvd/7v+vXrtWTJEu3bt0+9evXS1KlTNWbMGJMr9J2IiAhFRUV5/TwNGnOzYcMGl9fnz59Xbm6uunXrRuAB0Gg1jaFxF3Cirxxfbe9M1ZD0vwQbHzl8+LBu/fVtOlt+xuxS/ML999/v8qRUbm6upk2bpvj4eHXo0MHEynwnsFWQVrzxutcDToPCzX333aef/OQnuuOOO3T+/HmNGDFCW7duVevWrbVmzRqNGjWqTu+zefNmPf3009q2bZsKCwuVlZVV67P/mzZt0syZM5WXl6dOnTpp1qxZmjp1akM+BgA/VdMYGnfhprreGRlGvUISPKu4uFhny8/odM+RqghmUeX8VQvdtn/3/QmFjPitb4sxQUBZsfTtJhUXF/tnuFm5cqVuvfVWSdLq1av13Xff6csvv9Trr7+uRx55RJ988kmd3ufUqVMaMGCAJk2apIkTJ9a6f35+vsaNG6e77rpLK1as0CeffKK7775bl1xySZ2OB9A0NOQJpwufnqr05YrH3exZfUiCd1QER6gitHn0TNSk7IfDbttP//cwfz4e1qBwc+zYMUVHR0uS1q5dq+uvv159+vTRHXfc4Xyiqi6uu+46XXfddXXef8mSJYqNjdXChQslSf369dNnn32mZ555hnADWIinnnDiMXD4E57c850GhZuoqCjt3r1bMTEx+uCDD/Tiiy9KkkpLS706id/WrVs1duxYl7Zrr71WS5cu1dmzZxUYGOi1cwNNQcDp42aX4BExA0fqmw++08VjaMJjuuvL13+nsh8OK7hdlGIGjVbbHj+p9n2C20Wp9MiBKu0hbaMUcOqox+v2J1a5FvxdfQa+1zQ2DJ7VoHAzadIk3XDDDYqJiZHNZtPPf/5zSVJ2drZX57kpKiqqcp8uKipK586d09GjRxUTE1PlmDNnzujMmf8bzFZSUuK1+gCzheRvNrsEjwiVFBwfp4KCAp06dUqhoaGKjIzUge3/dO5TeuSAvvlguXMw5tGjR132j42NVY9LwpVXdTJ1db8kTKG73/PdB4Il1WXg+8XhJypxnE4U7OHJPS9rULh5/PHH1b9/fx04cEDXX3+9goKCJP24LMPDDz/s0QIvZrO5ridTOfPjxe2V0tPT9cQTT3i1JsBfnO4xQhUhkWaX4REhkvpe8Hp3DYMxy7peoW/yNjnbTpw4oby83br0f27XpV2vUNHn/9TpH4oU0i5a0YNGK6RHf53yavXmCzh93DJh11/VNvDdXfgpLfqOAe0+0KBwI0kpKSlV2m6//fZGFVOb6OhoFRUVubQdOXJELVu2VPv27d0eM2fOHM2cOdP5uqSkRF27dvVqnYBZKkIiLTswsabBmIU7NrnZYqhwx2ZddutcRfS/xmVLhRfqQ/NT25iu+j71B8+pc7hZvHixJk+erODg4FoHDc+YMaPRhbkzdOhQrV7terF8+OGHGjJkSLXjbYKCgpw9SwCarpoGY5Yd/Y/bYxg4DG+qbYAwA9rNU+dws2DBAt1yyy0KDg52LsPgjs1mq3O4OXnypPbt2+d8nZ+frx07dqhdu3aKjY3VnDlzdPDgQb3++uuSpKlTp+qFF17QzJkzddddd2nr1q1aunSp/vKXv9T1YwBoomoajFn0r/d4CgU+V9sAYZ6OMk+dw03l0gsX/3djfPbZZ7rmmv/rLq68fXT77bdr+fLlKiwsVEFBgXN7jx49tHbtWt1///3605/+pE6dOmnx4sU8Bg40AzUuo2AYPIUCn6ttaQ+ejjJPncPNheNWamKz2fTss8/Wad9Ro0a5LAV/seXLl1dpGzlypD7//PM6vT8Aa3E3UV9lO+tHwQzVXZOV27guzVHncLN9+3aX19u2bdP58+fVt++PzzN8/fXXatGihQYPZpAUAN+r6ZcM4Ev1mfsG3lHncHPhYpnPPfecwsLC9Oc//9m5UOZ///tfTZo0ScOHD/d8lQAANAH1XfQV3hHQkIOeffZZpaenu6wA3rZtW/3hD3+o8y0pAACspqbHv+E7DQo3JSUlOny46pwTR44c0YkTJxpdFAAATRGPf/uHBoWbCRMmaNKkSXr33Xf1n//8R//5z3/07rvv6o477lBycrKnawQAoEkI7uD+MW8e//atBs1QvGTJEj344IO69dZbdfbs2R/fqGVL3XHHHXr66ac9WiAAAP7swgHEgW3autmDx799rUHhpnXr1nrxxRf19NNP65tvvpFhGOrVq5dCQ0M9XR8AAH7r4gHEZ47/OGSjVWSUzp06rpahkbJJ+m7NSzw55UMNui1VKTQ0VAkJCRowYADBBgDQ7LgfQCy1DA5V93FTVH78sM4cP6yKc+XOJ6eO793m4yqbn0aFGwAAmrOaBhDz5JR5GrwqOACYob4TpDGhGryJBV39Ez03AJqMyvENpUX5dermr+/+QH1FXzleku2i1h8HEPPklHkINwCajPp283NbAN5WuX5U6+ieCggMUuvonuppn6HIXoNqDD7wLm5LAWgy6jtBGhOqwRdY0NX/EG4ANBk1jW9wN7ampv0BX2BBV3NwWwpAk1FdN39YbD+3Y2vCYuPc7s9tAcDaCDcAmozqxjecKNjtZm9DJwr2VDseAoB1cVsKQJPirpv/uzUvud237NhBbgsAzRA9NwCaPB65BXAhwg2AJo9HbgFciHADoMmraa4RAM0PY24AWAJjawBUItwAAOAlrG1mDm5LAQDgBaxtZh7CDQAAXsDaZuYh3AAA4AWsbWYewg0AAF7A/EvmIdwAAOAFzL9kHp6WAgB4VcDp42aXYIp2nbrJ9j+3q+jzf+r0D0UKaRet6EGj1TYmVjp11OzyfM6X1wHhBgDgVSH5m80uwTShkrr06yGpx48Np7+Vdn9rZknNAuEGAOBVp3uMUEVIpNll+Mx/83NV+PlHKvvhsILbRSlm0Gi17fETs8syXcDp4z4LuoQbAIBXVYREqiK0g9ll+MTxvdv07QfLna9LjxzQNx/8WT2TpjN5nw8xoBgAAA9hbhv/QLgBAMBDmNvGPxBuAADwEOa28Q+MuQEAoBEuXBwzsE1bN3swt42vEW4AAGigysUxK505fliS1CoySudOHVfL0EjZJH235iVWBfchbksBANBA7gcQSy2DQ9V93BSVHz+sM8cPsyq4jxFuAABooJoGEPPklHkINwAANFBNA4h5cso8hBsAABqopsUxeXLKPIQbAAAaKLL3YPVMmq7W0T0VEBik1tE91dM+Q5G9BrEquIl4WgoAgEaI7D3Y7RNQlcGnKHuNyo4dVHD7zoq+arwiew0yocrmhXADAICXVBd84F3clgIAAJZCuAEAAJZCuAEAAJZCuAEAAJZCuAEAAJbC01IAAHjJhSuGs3Cm79BzAwCAF1SuGF5alM/CmT5GuAEAwAtYONM8hBsAALyAhTPNQ7gBAMALWDjTPIQbAAC8gIUzzUO4AQDAC2paMRzexaPgAAB4CQtnmoOeGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmmh5sXX3xRPXr0UHBwsAYPHqwtW7ZUu+/GjRtls9mq/Hz55Zc+rBgAAPgzU8PNO++8o/vuu0+PPPKItm/fruHDh+u6665TQUFBjcd99dVXKiwsdP707t3bRxUDAAB/Z2q4ee6553THHXfozjvvVL9+/bRw4UJ17dpVL730Uo3HdezYUdHR0c6fFi1a+KhiAADg70wLN+Xl5dq2bZvGjh3r0j527Fh9+umnNR57+eWXKyYmRqNHj9aGDRtq3PfMmTMqKSlx+QEAANZlWrg5evSozp8/r6ioKJf2qKgoFRUVuT0mJiZGGRkZWrVqlTIzM9W3b1+NHj1amzdvrvY86enpioiIcP507drVo58DAAD4l5ZmF2Cz2VxeG4ZRpa1S37591bdvX+froUOH6sCBA3rmmWc0YsQIt8fMmTNHM2fOdL4uKSkh4AAAYGGm9dx06NBBLVq0qNJLc+TIkSq9OTW56qqrtHfv3mq3BwUFKTw83OUHAABYl2k9N61atdLgwYO1bt06TZgwwdm+bt06JSUl1fl9tm/frpiYGG+UiBoc37tNRdmrVXb0oII7dFb0leMV2Xuw2WUBAGDubamZM2fq17/+tYYMGaKhQ4cqIyNDBQUFmjp1qqQfbykdPHhQr7/+uiRp4cKF6t69u+Lj41VeXq4VK1Zo1apVWrVqlZkfo9k5vnebvv3bYufr0qJ8ffu359UzaToBB0Czw5c9/2NquLnxxht17NgxpaWlqbCwUP3799fatWvVrVs3SVJhYaHLnDfl5eV68MEHdfDgQYWEhCg+Pl7vv/++xo0bZ9ZHaJaKsle7aTVUlL2Gv9AAmhW+7Pkn0wcU33333br77rvdblu+fLnL61mzZmnWrFk+qAo1KTt60H37MfftAGBVtX3Zo1fHHKYvv4CmJ7hDZ/ft7d23A4BV1fRlr7JXp7QoXxXnyp29Osf3bvNxlc0P4Qb1Fn3leEkXP65vU/RV480oBwBMU9OXvZp6deBdhBvUW2TvweqZNF2to3sqIDBIraN7qqd9hiJ7DTK7NADwqZq+7HEL3zymj7lB0xTZezD3jQE0e5Vf9oqy16js2EEFt++s6KvGK7LXIAV36KzSovwqx3AL3/sINwAANEJ1X/airxyvb//2vCTjglZu4fsCt6UAAPACbuGbh54bAAC8hFv45qDnBgAAWArhBgAAWArhBgAAWArhBgAAWArhBgAAWArhBgAAWArhBgAAWArhBgAAWArhBgAAWArhBgAAWArLLwCwhON7t6koe7XKjh5UcIfOir5yPNPeA80U4aYRAk4fN7sE+AGuA/Md37tN3/5tsfN1aVG+vv3b8+qZNJ2AAzRDhJtGCMnfbHYJACQVZa9202qoKHsN4QZohgg3jXC6xwhVhESaXQZMFnD6OEHXZGVHD7pvP+a+HYC1EW4aoSIkUhWhHcwuA2j2gjt0VmlRftX29p1NqAaA2XhaCkCTF33leEm2i1ptir5qvBnlADAZ4QZAkxfZe7B6Jk1X6+ieCggMUuvonuppn6HIXoPMLg2ACbgtBcASInsPZvAwAEn03AAAAIuh5wYAgAaqbfJIJpc0Bz03AAA0QOXkkaVF+ao4V+6cPPL43m112g7voecGAOBVAWXFZpfgFUVbs9y0Gjq81aF2nbrVur258eV1QLgBAHhFRESEAlsFSd9uMrsUr6hu8sjTR/+j0N3v1bq9OQpsFaSIiAivn4dwAwDwiqioKK1443UVF1uz5yYlJUW5ublV2vv376+MjIxatzdHERERioqK8vp5CDcAAK+JioryyS8zM6SlpSk5OVmGYTjbbDab0tLS1KdPn1q3w3sYUAwAQAPY7XZlZmYqMTFRoaGhSkxMVFZWlpKSkly2JyQkKCAgQAkJCS7b4T303AAA0EB2u112u73G7XFxcZo8ebIyMjLosfERem4AAICl0HMDwBKYLA1AJXpuADR5TJYG4EL03KBafBNGU1GUvdpNq6Gi7DVcs0AzRM8N3OKbMJqS6iZLKzvmvh2AtdFzA7fq+k2Y3h3/YtVp7msT3C5KpUcOVGkPaRulgFNHTajIfM31WvA1h8Oh+fPnKy8vT/Hx8UpNTXV5esrhcOixxx5TXl6eUlJSlJaWVuPTVfAMwg3cqss34crenUqVvTs9k6YTcHzM6tPc16bHJeHKO1K1vfslYc12mnvJd1PdN1cOh0MTJkxwvs7JyVFycrIyMzNlt9urbM/NzXXZDu8h3MCt4A6dVVqUX7W9fWfnfzPOwX9YfZp7SVq/fr2WLFmiffv2qVevXpo6darGjBnjsn3x4sXau3evevfurRkzZrhsb458NdV9czV//vwqbYZhKD09XXa7vdbt8B7CTSNYuds3ZuBIffPBd5KMC1ptihk40tnNX23vztH/NKtbAf5yHVh5mnuHw6Fp06Y5X+fm5uqee+5x+Qbcp08fjRkzhsnS4DN5eXk1tte2Hd5DuGmA5nALIFRScHycCgoKdOrUKQUFBckwDOV/+GcdCQ1VbGysQkOCdOJEedVjQ4Ka3a0Auv+9i2/A8Efx8fHKyclx216X7fAewk0DNIdbABdav369y7fmEydOaPfu3brhhhv0zjvvuOxrs9n05JNPNrvbAXT/e1dt34ArB3Xu2rVLLVu21Pr16+m5gdelpqa6XRgzNTW1TtvhRUYzU1xcbEgyiouLzS6lybjiiisM/Xh/yuUnISHBiI+PNxISEozQ0FAjMTHRcDgcZpcLC6ruGkxMTDSysrKqtNtsNiMrK8vsstEMZGVlGYmJidX+G5iVlWUkJCQYAQEBRkJCAv9GNkJ9fn/bDMMwqkYe6yopKVFERISKi4sVHh5udjlNQmhoqEpLS6u0t27dWldccQXjG+B1DofD7TfgrKwszZs3z23Xf2JiorKzs31ZJuDW119/zVgwD6jP728m8UOtqrs/3KtXLx9XgubKbrcrMzNTiYmJCg0NVWJiorKyspSUlFSnQZsOh8PlWIfD4aPKAZiBcINapaamymazubTZbDZNmTLFpIrQHNntdmVnZ+vkyZPKzs5WUlKSpOrDd2V75VwjOTk5Ki0tdc5FQsABrItwg1q5+9Y8e/ZsLVmyRFu2bFFKSgq/KOAT7npgqgvflYM2a3rSCoA1MeYG9XbxrJvSj79MmHUT3lTTdSdJ6enp2rVrl1q0aKEnn3xSd999t6Tqx4yFhobq5MmT3i8czR5jbjyjPr+/eRS8iSorK1NBQYEp537ssceqtBmGoblz5youLs6EiqTY2FgFBwebcm74Rk09MJUDhyvX8FmyZIk6deoku93OXCNAM0S4aaIKCgo0efJkU85d3QDOXbt2mVYT34isr6aBwzWt4cNcI0DzQ7hpomJjY5WRkWHKuVNSUpSbm1ulvX///qbVFBsba8p54Ts19cDU1quTmZmp9PR0l5WbKwckA7Aewk0TFRwcbFpPRVpamttvwmlpafSewGtq6oH51a9+5faYyt4eu93OeDD4ncqZtS8M3VynnsHTUqi3muYcAbylpuuutsfBAbM4HA6lpKRUebKUKQq8i6elADR5Nc1gTOiGWWp6wm/+/PnMrF1PzFAMoFmp7NVJSEhQQECAEhISCDYwXU1jweoyszYajnADwBLsdrtWrlyp4cOHa+XKlQQbmK6mAMOtVO8i3AAA0AjVrV1WU4CpbWZtNA7hBgCABqppYHBNAYYHM7yLcAMAQAPVNK6mtrFg1S0Gi8Yj3ABoMqrr/gfMUpeBwZVP8RmGoWb2gLJpCDcAmgTmBYE/qmlcTeU1m5ubq4qKCueyIFyz3sc8NwCahMTExBrnBXE4HM6FM+Pj45WWlsZsr/D6IsPr16/XPffcU2WOpRdeeEFLlixxu1RNQkKCVq5c6bWaatKUFxmuz+9vwg2AJiE0NFSlpaVu21esWFHtZGkEnObt66+/9vqCvkePHlVBQYFOnTql0NBQxcbGqkOHDtqyZYsqKiqq7B8QEKDhw4d7tabqNOVFhptUuHnxxRf19NNPq7CwUPHx8Vq4cGGN/6dv2rRJM2fOVF5enjp16qRZs2Zp6tSpdT4f4QbwDm9/Q65uwdaEhAQZhsE3ZLjl7euyJjVds1yX9Vef39+mLpz5zjvv6L777tOLL76oYcOG6eWXX9Z1112n3bt3u13lOT8/X+PGjdNdd92lFStW6JNPPtHdd9+tSy65RBMnTjThEwCoVFBQ4NVvyO6+AUvS+fPntWfPHrfbdu3a5fVv7dVpyt+QrYRFhpsnU3turrzySg0aNEgvvfSSs61fv36y2+1KT0+vsv/s2bP13nvvufxDNnXqVO3cuVNbt26t0znpuQG8wxffkNevX6+XX35Z+/btU69evTRlyhSNGTOGb8jwWw6Hw7ncQuXkfTzy3TBN4rZUeXm5WrdurZUrV7rcK7/33nu1Y8cObdq0qcoxI0aM0OWXX65FixY527KysnTDDTeotLRUgYGBVY45c+aMzpw543xdUlKirl27Em4AC2HhTMD6msTCmUePHtX58+cVFRXl0h4VFaWioiK3xxQVFbnd/9y5czp69KjbY9LT0xUREeH86dq1q2c+AAC/wWyvAC5k6pgbSVWmpjYMo0pbbfu7a680Z84czZw50/m6sucGgLXY7XaejAIgycRw06FDB7Vo0aJKL82RI0eq9M5Uio6Odrt/y5Yt1b59e7fHBAUFKSgoyDNFAwAAv2fabalWrVpp8ODBWrdunUv7unXrdPXVV7s9ZujQoVX2//DDDzVkyBC3420AAEDzY+ryCzNnztSrr76q1157TXv27NH999+vgoIC57w1c+bM0W233ebcf+rUqdq/f79mzpypPXv26LXXXtPSpUv14IMPmvURAACAnzF1zM2NN96oY8eOKS0tTYWFherfv7/Wrl2rbt26SZIKCwtdHi3t0aOH1q5dq/vvv19/+tOf1KlTJy1evJg5bgAAgJPpMxT7GvPcAADQ9DSJR8EBAAC8gXADAAAshXADAAAshXADAAAshXADAAAshXADAAAsxfS1pXyt8sn3kpISkysBAAB1Vfl7uy4z2DS7cHPixAlJYvFMAACaoBMnTigiIqLGfZrdJH4VFRU6dOiQwsLCalx9HLWrXGH9wIEDTIgIv8A1CX/EdekZhmHoxIkT6tSpkwICah5V0+x6bgICAtSlSxezy7CU8PBw/sLCr3BNwh9xXTZebT02lRhQDAAALIVwAwAALIVwgwYLCgrS3LlzFRQUZHYpgCSuSfgnrkvfa3YDigEAgLXRcwMAACyFcAMAACyFcGOSUaNG6b777jO7DK9Yvny5IiMjzS7DxcaNG2Wz2XT8+HGzS2kyrHyN+grXnWdxTXpGc7guCTcmyczM1O9///s67fvdd9/JZrNpx44d3i1Knrnob7zxRn399deeK8oDrr76ahUWFtZ5jgT47zVaea7Kn4iICF111VVavXq12/3nz5+vFi1a6Mknn/R6bfAuf70ma/p3c+DAgXr88certHNdehfhxiTt2rVTWFiYR9+zvLzco+/XUCEhIerYsaPZZbho1aqVoqOjq52V+vz586qoqPBxVf7N36/R9evXq7CwUNnZ2UpMTNTEiRO1a9euKvstW7ZMs2bN0muvveaxc8Mc/n5N1gfXpXcRbkxyYfdq9+7dNX/+fP32t79VWFiYYmNjlZGR4dy3R48ekqTLL79cNptNo0aNkiT95je/kd1uV3p6ujp16qQ+ffpIklasWKEhQ4YoLCxM0dHR+tWvfqUjR47UWtN3332na665RpLUtm1b2Ww2/eY3v9Hq1asVGRnp/OW/Y8cO2Ww2PfTQQ85jp0yZoptvvllS1dtSjz/+uAYOHKg33nhD3bt3V0REhG666SbnOl91+bO65557dM899ygyMlLt27fX7373O5fF02r7zBd/s6qscc2aNYqLi1NQUJD2799fp3qaC3+8Ri/Uvn17RUdH67LLLtO8efN09uxZbdiwwWWfTZs26fTp00pLS9OpU6e0efPmep3jpZde0qWXXqpWrVqpb9++euONN1y222w2vfrqq5owYYJat26t3r1767333nP7XqdOnVJ4eLjeffddl/bVq1crNDS0zn8fmjN/vybriuvS+wg3fuLZZ5/VkCFDtH37dt199936f//v/+nLL7+UJP373/+W9H/fVDMzM53HffTRR9qzZ4/WrVunNWvWSPrxm8jvf/977dy5Uw6HQ/n5+frNb35Taw1du3bVqlWrJElfffWVCgsLtWjRIo0YMUInTpzQ9u3bJf34F7NDhw7atGmT89iNGzdq5MiR1b73N998I4fDoTVr1mjNmjXatGlTvbpj//znP6tly5bKzs7W4sWLtWDBAr366qvO7Q35zKWlpUpPT9err76qvLw8v+tt8jf+cI26c/bsWb3yyiuSpMDAQJdtS5cu1c0336zAwEDdfPPNWrp0aZ3fNysrS/fee68eeOAB7dq1S1OmTNGkSZOqBKgnnnhCN9xwg7744guNGzdOt9xyi3744Ycq7xcaGqqbbrpJy5Ytc2lftmyZUlJSPN4j0Rz46zVZG65LHzBgipEjRxr33nuvYRiG0a1bN+PWW291bquoqDA6duxovPTSS4ZhGEZ+fr4hydi+fbvLe9x+++1GVFSUcebMmRrP9e9//9uQZJw4caLWujZs2GBIMv773/+6tA8aNMh45plnDMMwDLvdbsybN89o1aqVUVJSYhQWFhqSjD179hiGYRjLli0zIiIinMfOnTvXaN26tVFSUuJse+ihh4wrr7yy1noM48c/q379+hkVFRXOttmzZxv9+vWr9piLP/PFn2vZsmWGJGPHjh11qqE58tdrtPJcISEhRmhoqBEQEGBIMrp3724cO3bMuV9xcbHRunVr5//H27dvN1q3bm0UFxfX5eMbV199tXHXXXe5tF1//fXGuHHjnK8lGb/73e+cr0+ePGnYbDbj73//u2EYVa+77Oxso0WLFsbBgwcNwzCM77//3ggMDDQ2btxYp5qaO3+9Jqv7d9MwDGPAgAHG3Llzna+5Ln2Dnhs/kZCQ4Pxvm82m6OjoOnWJ/uQnP1GrVq1c2rZv366kpCR169ZNYWFhzu7YgoKCBtc3atQobdy4UYZhaMuWLUpKSlL//v318ccfa8OGDYqKitJll11W7fHdu3d3+QYQExNTry7fq666ymW8zNChQ7V3716dP39eUsM+c6tWrVz+3FEzf7tG33nnHW3fvl3vvfeeevXqpVdffVXt2rVzbn/rrbfUs2dPDRgwQNKPAzt79uypt99+u07vv2fPHg0bNsylbdiwYdqzZ49L24V/LqGhoQoLC6v2zyUxMVHx8fF6/fXXJUlvvPGGYmNjNWLEiDrVBFf+dk3WBdelbxBu/MTF3ek2m61OA1xDQ0NdXp86dUpjx45VmzZttGLFCuXk5CgrK0tS4wbOjRo1Slu2bNHOnTsVEBCguLg4jRw5Ups2bar1lpTU8M9XFw39zCEhIdUOMEZV/naNdu3aVb1799YvfvELvfrqq7rxxhtd/vF+7bXXlJeXp5YtWzp/8vLy6nUL4OLrwzCMKm31/XO58847nbcAli1bpkmTJnEdNpC/XJOVK30XFxdX2Xb8+HGXpzS5Ln2DcNMEVH7DqOylqMmXX36po0eP6sknn9Tw4cN12WWX1auHpLpzVY67WbhwoUaOHCmbzaaRI0dq48aNdQo3jfWvf/2ryuvevXurRYsWjf7MaDxfXqPujBw5Uv3799e8efMkSbm5ufrss8+0ceNG7dixw/mzefNm5eTkuH2q6mL9+vXTxx9/7NL26aefql+/fo2q9dZbb1VBQYEWL16svLw83X777Y16P7jny2uyd+/eCggIUE5Ojkt7YWGhDh48qL59+0riuvSllmYXgNp17NhRISEh+uCDD9SlSxcFBwdXO19LbGysWrVqpeeff15Tp07Vrl276jwvhCR169ZNNptNa9as0bhx4xQSEqI2bdooIiJCAwcO1IoVK7Ro0SJJPwae66+/XmfPnnV24XrLgQMHNHPmTE2ZMkWff/65nn/+eT377LOSGv+Z0Xi+vEar88ADD+j666/XrFmztHTpUiUmJrrtVh86dKiWLl2qBQsW1Ph+Dz30kG644QYNGjRIo0eP1urVq5WZman169c3qs62bdsqOTlZDz30kMaOHasuXbo06v3gni+vybCwME2ZMkUPPPCAWrZsqQEDBujQoUN65JFH1K9fP40dO1aSuC59iJ6bJqBly5ZavHixXn75ZXXq1ElJSUnV7nvJJZdo+fLlWrlypeLi4vTkk0/qmWeeqfO5OnfurCeeeEIPP/ywoqKidM899zi3XXPNNTp//rwzyLRt21ZxcXG65JJLGv2toTa33XabTp8+rcTERE2bNk3Tp0/X5MmTJTX+M6PxfHmNVud///d/1b17d82bN08rVqzQxIkT3e43ceJErVixotbbDXa7XYsWLdLTTz+t+Ph4vfzyy1q2bJlHgvwdd9yh8vJy/fa3v230e8E9X1+TCxYs0J133qnU1FTFx8frlltuUY8ePfThhx+qZcuWKi8v57r0IVYFh98bNWqUBg4cqIULF5pdCuARb775pu69914dOnSoysBWwCxWui65LQUAPlJaWqr8/Hylp6drypQpTf4XCKzBitclt6WamalTp6pNmzZuf6ZOnerzegoKCqqtp02bNh5/DBP+z1fXaHx8fLXnefPNNz12ngv98Y9/1MCBAxUVFaU5c+Z45RzwPF/+u8l16Rnclmpmjhw5opKSErfbwsPDfT5L77lz5/Tdd99Vu7179+5q2ZIOxubEV9fo/v37dfbsWbfboqKimu7MrPA4X/67yXXpGYQbAABgKdyWAgAAlkK4AQAAlkK4AQAAlkK4AQAAlkK4AQAAlkK4AeCXHn/8cQ0cONDsMgA0QYQbAKiD6uYeAeB/CDcAvKaiokJPPfWUevXqpaCgIMXGxmrevHmSpNmzZ6tPnz5q3bq1evbsqUcffdQZIJYvX64nnnhCO3fulM1mk81m0/LlyyVJxcXFmjx5sjp27Kjw8HD97Gc/086dO13O+4c//EEdO3ZUWFiY7rzzTj388MMuvUAVFRVKS0tTly5dFBQUpIEDB+qDDz5wbv/uu+9ks9n017/+VaNGjVJwcLAyMjIUHh6ud9991+Vcq1evVmhoqE6cOOGFP0EADUG4AeA1c+bM0VNPPaVHH31Uu3fv1ltvvaWoqChJUlhYmJYvX67du3dr0aJFeuWVV7RgwQJJ0o033qgHHnhA8fHxKiwsVGFhoW688UYZhqFf/OIXKioq0tq1a7Vt2zYNGjRIo0eP1g8//CDpx8X/5s2bp6eeekrbtm1TbGysXnrpJZe6Fi1apGeffVbPPPOMvvjiC1177bX65S9/qb1797rsN3v2bM2YMUN79uzRhAkTdNNNN2nZsmUu+yxbtkwpKSnMHAv4EwMAvKCkpMQICgoyXnnllTrt/8c//tEYPHiw8/XcuXONAQMGuOzz0UcfGeHh4UZZWZlL+6WXXmq8/PLLhmEYxpVXXmlMmzbNZfuwYcNc3qtTp07GvHnzXPa54oorjLvvvtswDMPIz883JBkLFy502Sc7O9to0aKFcfDgQcMwDOP77783AgMDjY0bN9bpMwLwDXpuAHjFnj17dObMGY0ePdrt9nfffVc//elPFR0drTZt2ujRRx+tdaHUbdu26eTJk2rfvr3LgoL5+fn65ptvJElfffWVEhMTXY678HVJSYkOHTqkYcOGuewzbNgw7dmzx6VtyJAhVd4nPj5er7/+uiTpjTfeUGxsrEaMGFFj3QB8ixUJAXhFSEhItdv+9a9/6aabbtITTzyha6+9VhEREXr77bf17LPP1vieFRUViomJ0caNG6tsi4yMdP63zWZz2Wa4WULP3T4Xt4WGhlY57s4779QLL7yghx9+WMuWLdOkSZOqHAfAXPTcAPCK3r17KyQkRB999FGVbZ988om6deumRx55REOGDFHv3r21f/9+l31atWql8+fPu7QNGjRIRUVFatmypXr16uXy06FDB0lS37599e9//9vluM8++8z53+Hh4erUqZM+/vhjl30+/fRT9evXr9bPdeutt6qgoECLFy9WXl6ebr/99lqPAeBb9NwA8Irg4GDNnj1bs2bNUqtWrTRs2DB9//33ysvLU69evVRQUKC3335bV1xxhd5//31lZWW5HN+9e3fl5+drx44d6tKli8LCwjRmzBgNHTpUdrtdTz31lPr27atDhw5p7dq1stvtGjJkiKZPn6677rpLQ4YM0dVXX6133nlHX3zxhXr27Ol874ceekhz587VpZdeqoEDB2rZsmXasWOH3nzzzVo/V9u2bZWcnKyHHnpIY8eOVZcuXTz+Zwegkcwe9APAus6fP2/84Q9/MLp162YEBgYasbGxxvz58w3DMIyHHnrIaN++vdGmTRvjxhtvNBYsWGBEREQ4jy0rKzMmTpxoREZGGpKMZcuWGYbx40Dl6dOnG506dTICAwONrl27GrfccotRUFDgPDYtLc3o0KGD0aZNG+O3v/2tMWPGDOOqq65yqeuJJ54wOnfubAQGBhoDBgww/v73vzu3Vw4o3r59u9vP9dFHHxmSjL/+9a+e+8MC4DE2w3BzMxoALOTnP/+5oqOj9cYbb3jk/d58803de++9OnTokFq1auWR9wTgOdyWAmAppaWlWrJkia699lq1aNFCf/nLX7R+/XqtW7fOI++dn5+v9PR0TZkyhWAD+CkGFAOwFJvNprVr12r48OEaPHiwVq9erVWrVmnMmDGNfu8//vGPGjhwoKKiojRnzhwPVAvAG7gtBQAALIWeGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCmEGwAAYCn/HyWTQZIroT0IAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# paired alpha div: unweighted distances to compare inter-twin vs inter-disease distances\n",
    "# paired shannon diversity difference from 0 \n",
    "# grab twin to pair dict\n",
    "# get distances for each twin pair per beta div matrix    \n",
    "pair_to_dist = {}\n",
    "for p in list(df_alpha.TwinPair.values):\n",
    "    id_UA, id_RA = pair_to_ids[p]\n",
    "    pair_to_dist[p] = df_alpha.loc[id_RA,alpha_metric] - df_alpha.loc[id_UA,alpha_metric]\n",
    "    \n",
    "df_alpha_inter = pd.DataFrame.from_dict(pair_to_dist, orient='index', columns=[alpha_metric])\n",
    "\n",
    "# grab inter RA distances\n",
    "df_RA = df_alpha[df_alpha['Diagnosis'] == 'RA']\n",
    "RA_divs = list(df_RA[alpha_metric].values)\n",
    "RA_diffs = [np.abs(x - y) for i,x in enumerate(RA_divs) for j,y in enumerate(RA_divs) if i < j]\n",
    "\n",
    "# grab inter UA distances\n",
    "df_UA = df_alpha[df_alpha['Diagnosis'] == 'Unaffected']\n",
    "UA_divs = list(df_UA[alpha_metric].values)\n",
    "UA_diffs = [np.abs(x - y) for i,x in enumerate(UA_divs) for j,y in enumerate(UA_divs) if i < j]\n",
    "\n",
    "# compare distances\n",
    "inter_twin = np.abs(df_alpha_inter[alpha_metric].values)\n",
    "inter_RA = list(RA_diffs)\n",
    "inter_UA = list(UA_diffs)\n",
    "\n",
    "u, p = scipy.stats.kruskal(inter_UA, inter_RA, inter_twin)\n",
    "print(u, p)\n",
    "\n",
    "u, p = scipy.stats.mannwhitneyu(inter_twin, inter_RA)\n",
    "# print(u, p)\n",
    "\n",
    "t, p = scipy.stats.ttest_ind(inter_twin, inter_RA)\n",
    "# print(t, p)\n",
    "\n",
    "f, p = scipy.stats.f_oneway(inter_UA, inter_RA, inter_twin)\n",
    "# print(f, p)\n",
    "\n",
    "category = ['intra_twin_pair']*len(inter_twin) + ['intra_RA_only']*len(inter_RA) + ['intra_UA_only']*len(inter_UA)\n",
    "distances = list(inter_twin) + list(inter_RA) + list(inter_UA)\n",
    "df_dist = pd.DataFrame(data=np.array([category,distances]).T, columns=['category','distance'])\n",
    "df_dist['distance'] = df_dist['distance'].astype(float)\n",
    "df_dist.to_csv(path + 'outputs/jobs02/intra_alpha_dist.tsv',sep='\\t')\n",
    "\n",
    "sns.boxplot(data=df_dist, x='category', y='distance')\n",
    "sns.swarmplot(data=df_dist, x='category', y='distance', color='black')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "##### MTX Usage #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10.0 0.5714285714285714\n",
      "0.7847404067976351 0.4624554060605416\n",
      "0.09587007560175183 0.08939238572334704\n",
      "1.7347641269240992 0.23586737222279144\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAkAAAAGwCAYAAABB4NqyAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAABC7klEQVR4nO3df1xW9f3/8ecFCijGReoEVERTUZH8BSI/hrUszK0S/Zi0Cst0zNlMs6yYqcksZj/8WVqsEt2ckalkn8giZ/5I54yBGZpS2lC70GkJmgoF5/uHX69P1y40RPCA53G/3c5tXe/zPm9e7/K4J+9zrnNshmEYAgAAsBAPswsAAAC40ghAAADAcghAAADAcghAAADAcghAAADAcghAAADAcghAAADAcghA1TAMQ2VlZeIRSQAAXJ0IQNU4efKk7Ha7Tp48aXYpAACgHhCAAACA5RCAAACA5RCAAACA5ZgegBYtWqROnTrJx8dHERER2rx580X7L1++XL1791bz5s0VFBSk0aNH6/jx4879mZmZstlsbtvZs2freyoAAKCRMDUAZWVladKkSZo6dary8/MVHx+vIUOGqLi4uNr+W7Zs0ahRozRmzBgVFhZq5cqV2rFjh8aOHevSz8/PTw6Hw2Xz8fG5ElMCAACNgKkBaM6cORozZozGjh2rHj16aN68eQoODtbixYur7f+Pf/xDHTt21EMPPaROnTrp5z//uX7729/qk08+celns9kUGBjosgEAAJxnWgCqqKhQXl6eEhISXNoTEhK0devWao+JjY3VoUOHlJOTI8MwdOTIEb311lv61a9+5dLv1KlTCgkJUfv27XXbbbcpPz//orWUl5errKzMZQMAAFcv0wLQsWPHVFlZqYCAAJf2gIAAlZSUVHtMbGysli9frqSkJHl5eSkwMFD+/v5auHChs0/37t2VmZmptWvXasWKFfLx8VFcXJyKioouWEt6errsdrtzCw4OrptJAgCABsn0m6BtNpvLZ8Mw3NrO2717tx566CFNnz5deXl5WrdunQ4cOKBx48Y5+0RHR+vee+9V7969FR8frzfffFOhoaEuIem/paamqrS01LkdPHiwbiYHAAAapCZm/eDWrVvL09PTbbXn6NGjbqtC56WnpysuLk5TpkyRJPXq1Uu+vr6Kj4/XrFmzFBQU5HaMh4eH+vfvf9EVIG9vb3l7e1/GbAAAQGNi2gqQl5eXIiIilJub69Kem5ur2NjYao85ffq0PDxcS/b09JSkC763yzAMFRQUVBuOAACANZm2AiRJkydPVnJysiIjIxUTE6OMjAwVFxc7L2mlpqbq8OHDWrZsmSTp9ttv129+8xstXrxYgwcPlsPh0KRJkxQVFaW2bdtKkmbOnKno6Gh17dpVZWVlWrBggQoKCvTSSy+ZNk8AANCwmBqAkpKSdPz4caWlpcnhcCg8PFw5OTkKCQmRJDkcDpdnAt1///06efKkXnzxRT3yyCPy9/fXTTfdpNmzZzv7nDhxQikpKSopKZHdblffvn21adMmRUVFXfH5AQCAhslmXOjakYWVlZXJbrertLRUfn5+ZpcDAKiF7OxsPfPMMyosLFTPnj31hz/8QYmJiWaXhQaCAFQNAhAANG7Z2dkaNmyYS5vNZtPq1asJQZBk8iUwAEDDdfbs2Qu+mqihmz59ulubYRiaMWOGwsLCTKiobnTo0IFXO9URAhAAoFrFxcVKSUkxu4xaKSwsrLb9s88+a7RzkqSMjAyFhoaaXcZVgUtg1eASGAA07hWgESNGaNeuXW7tvXr10sqVK02oqG6wAlR3CEDVIAABQOOWnZ2t4cOHuzwjzmazac2aNRo6dKiJlaGhMP1VGAAA1LXExEStXr1avXr1koeHh3r16kX4gQtWgKrBChAAXB327dunlJQU7p2BG1aAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5ZgegBYtWqROnTrJx8dHERER2rx580X7L1++XL1791bz5s0VFBSk0aNH6/jx4y59Vq1apbCwMHl7eyssLExr1qypzykAAIBGxtQAlJWVpUmTJmnq1KnKz89XfHy8hgwZouLi4mr7b9myRaNGjdKYMWNUWFiolStXaseOHRo7dqyzz7Zt25SUlKTk5GTt3LlTycnJGjlypLZv336lpgUAABo4m2EYhlk/fMCAAerXr58WL17sbOvRo4cSExOVnp7u1v/555/X4sWL9eWXXzrbFi5cqGeffVYHDx6UJCUlJamsrEzvvfees8+tt96qa6+9VitWrKi2jvLycpWXlzs/l5WVKTg4WKWlpfLz87vseQIAzLFv3z6lpKQoIyNDoaGhZpeDBsS0FaCKigrl5eUpISHBpT0hIUFbt26t9pjY2FgdOnRIOTk5MgxDR44c0VtvvaVf/epXzj7btm1zG3Pw4MEXHFOS0tPTZbfbnVtwcPBlzAwAADR0pgWgY8eOqbKyUgEBAS7tAQEBKikpqfaY2NhYLV++XElJSfLy8lJgYKD8/f21cOFCZ5+SkpJLGlOSUlNTVVpa6tzOryYBAICrk+k3QdtsNpfPhmG4tZ23e/duPfTQQ5o+fbry8vK0bt06HThwQOPGjav1mJLk7e0tPz8/lw0AAFy9mpj1g1u3bi1PT0+3lZmjR4+6reCcl56erri4OE2ZMkWS1KtXL/n6+io+Pl6zZs1SUFCQAgMDL2lMAABgPaatAHl5eSkiIkK5ubku7bm5uYqNja32mNOnT8vDw7VkT09PSedWeSQpJibGbcwPPvjggmMCAADrMW0FSJImT56s5ORkRUZGKiYmRhkZGSouLnZe0kpNTdXhw4e1bNkySdLtt9+u3/zmN1q8eLEGDx4sh8OhSZMmKSoqSm3btpUkTZw4UQMHDtTs2bM1dOhQvf322/rwww+1ZcsW0+YJAAAaFlMDUFJSko4fP660tDQ5HA6Fh4crJydHISEhkiSHw+HyTKD7779fJ0+e1IsvvqhHHnlE/v7+uummmzR79mxnn9jYWL3xxht68sknNW3aNHXu3FlZWVkaMGDAFZ8fAABomEx9DlBDVVZWJrvdznOAAKCR4zlAuBDTvwUGAABwpRGAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAAACA5RCAYGnZ2dmKioqSr6+voqKilJ2dbXZJAIArgAAEy8rOztawYcO0Y8cOnT59Wjt27NDw4cMJQQBgAU3MLgCN29mzZ1VcXGx2GbUyffp0tzbDMDRjxgyFhYWZUFHd6NChg3x8fMwuAwAaNAIQLktxcbFSUlLMLqNWCgsLq23/7LPPGu2cJCkjI0OhoaFmlwEADRoBCJelQ4cOysjIMLuMWhkxYoR27drl1h4eHt5o5ySd+28CALg4AhAui4+PT6NdbUhLS9Pw4cNlGIazzWazKS0trdHOCQBQM6bfBL1o0SJ16tRJPj4+ioiI0ObNmy/Y9/7775fNZnPbevbs6eyTmZlZbZ+zZ89eiemgEUlMTNTq1avVq1cveXh4qFevXlqzZo2GDh1qdmkAgHpmagDKysrSpEmTNHXqVOXn5ys+Pl5Dhgy54E218+fPl8PhcG4HDx5Uy5Ytdeedd7r08/Pzc+nncDi4KRTVSkxM1MqVKxUfH6+VK1cSfgDAIkwNQHPmzNGYMWM0duxY9ejRQ/PmzVNwcLAWL15cbX+73a7AwEDn9sknn+jbb7/V6NGjXfrZbDaXfoGBgVdiOgAAoJEwLQBVVFQoLy9PCQkJLu0JCQnaunVrjcZ47bXXdPPNNyskJMSl/dSpUwoJCVH79u112223KT8//6LjlJeXq6yszGUDAABXL9MC0LFjx1RZWamAgACX9oCAAJWUlPzk8Q6HQ++9957Gjh3r0t69e3dlZmZq7dq1WrFihXx8fBQXF6eioqILjpWeni673e7cgoODazcpAADQKJh+E7TNZnP5bBiGW1t1MjMz5e/vr8TERJf26Oho3Xvvverdu7fi4+P15ptvKjQ0VAsXLrzgWKmpqSotLXVuBw8erNVcAABA42Da1+Bbt24tT09Pt9Weo0ePuq0K/TfDMPT6668rOTlZXl5eF+3r4eGh/v37X3QFyNvbW97e3jUvHgAANGqmrQB5eXkpIiJCubm5Lu25ubmKjY296LEbN27UF198oTFjxvzkzzEMQwUFBQoKCrqsegEAwNXD1AchTp48WcnJyYqMjFRMTIwyMjJUXFyscePGSTp3aerw4cNatmyZy3GvvfaaBgwYoPDwcLcxZ86cqejoaHXt2lVlZWVasGCBCgoK9NJLL12ROQEAgIbP1ACUlJSk48ePKy0tTQ6HQ+Hh4crJyXF+q8vhcLg9E6i0tFSrVq3S/Pnzqx3zxIkTSklJUUlJiex2u/r27atNmzYpKiqq3ucDAAAaB9NfhTF+/HiNHz++2n2ZmZlubXa7XadPn77geHPnztXcuXPrqrx6deTIEZWWlppdhuX9+9//dvlfmMtut//kfYAAcLlMD0BWdeTIEd2bPErfV5SbXQr+v6efftrsEiCpqZe3/vqXZYQgAPWKAGSS0tJSfV9RrjPX3aAqH7vZ5QANgsfZUmn/RpWWlhKAANQrApDJqnzsqvJtbXYZAABYiukPQgQAALjSCEAAAMByCEAAAMByCEAAAMByCEAAAMByCEAAAMByCEAAAMByCEAAAMByeBAiLO1EUZ5Ktr+js8cOy6d1OwUOuF3+XSPMLgsAUM9YAYJlnSjK0/63F+h0yQFV/VCh0yUHtP/thTpRlGd2aQCAesYKkMk8zpwwuwTLKtm2pppWQ0e2Zatl25ArXg84HwBcOQQgkzU7sMnsEizr7LHD1bafOXZIvrvXXuFqAABXEgHIZGc6DVRVM3+zy7Aknz37dfroQbf2Zq3b67uwO0yoCB5nTvBLAYArggBksqpm/rwN3iSBMcO0/+2FkowftdoUEDuM/yYAcJXjJmhYln/XCF03dIKaB14nj6beah54na5LfEj+XfqZXRoAoJ6xAgRL8+8awdfeAcCCWAECAACWQwACAACWQwACAACWQwACAACWQwACAACWQwACAACWQwACAACWQwACAACWQwACAACWQwACAFyVsrOzNWLECG3evFkjRoxQdna22SWhASEAAQCuOtnZ2Ro2bJh27dqlqqoq7dq1S8OHDycEwYl3gQFAPThy5IhKS0vNLsOypk+f7tZmGIZmzJihsLAwEyqCJNntdgUEBJhdhiQCEADUuSNHjuje5FH6vqLc7FIsq7CwsNr2zz77TCkpKVe4GpzX1Mtbf/3LsgYRgghAAFDHSktL9X1Fuc5cd4OqfOxml2NJPnv26/TRg27tzVq313dhd5hQETzOlkr7N6q0tJQAJEmLFi3Sc889J4fDoZ49e2revHmKj4+vtu/999+vpUuXurWHhYW5pP1Vq1Zp2rRp+vLLL9W5c2c9/fTTGjZsWL3NAQCqU+VjV5Vva7PLsKTAmGHa//ZCScaPWm0KiB3GfxNIMvkm6KysLE2aNElTp05Vfn6+4uPjNWTIEBUXF1fbf/78+XI4HM7t4MGDatmype68805nn23btikpKUnJycnauXOnkpOTNXLkSG3fvv1KTQsAYDL/rhG6bugENQ+8Th5NvdU88Dpdl/iQ/Lv0M7s0NBCmrgDNmTNHY8aM0dixYyVJ8+bN0/vvv6/FixcrPT3drb/dbpfd/n/LydnZ2fr22281evRoZ9u8efN0yy23KDU1VZKUmpqqjRs3at68eVqxYkW1dZSXl6u8/P+u1ZeVldXJ/AAA5vHvGiH/rhFml4EGyrQVoIqKCuXl5SkhIcGlPSEhQVu3bq3RGK+99ppuvvlmhYSEONu2bdvmNubgwYMvOmZ6erozXNntdgUHB1/CTAAAQGNjWgA6duyYKisr3W6ECggIUElJyU8e73A49N577zlXj84rKSm55DFTU1NVWlrq3A4edL9xDgAAXD1MvwnaZrO5fDYMw62tOpmZmfL391diYuJlj+nt7S1vb++aFQwAABo901aAWrduLU9PT7eVmaNHj/7k1+MMw9Drr7+u5ORkeXl5uewLDAys1ZgAAMA6TAtAXl5eioiIUG5urkt7bm6uYmNjL3rsxo0b9cUXX2jMmDFu+2JiYtzG/OCDD35yTAAAYB2mXgKbPHmykpOTFRkZqZiYGGVkZKi4uFjjxo2TdO7enMOHD2vZsmUux7322msaMGCAwsPD3cacOHGiBg4cqNmzZ2vo0KF6++239eGHH2rLli1XZE4AAKDhMzUAJSUl6fjx40pLS5PD4VB4eLhycnKc3+pyOBxuzwQqLS3VqlWrNH/+/GrHjI2N1RtvvKEnn3xS06ZNU+fOnZWVlaUBAwbU+3wAAEDjYPpN0OPHj9f48eOr3ZeZmenWZrfbdfr06YuOOWLECI0YMaIuygMAAFchU58EDQAAYAYCEAAAsBzTL4FZncfZUrNLABoMzgfUpRNFeSrZ/o7OHjssn9btFDjgdl6NAScCkEnsdruaenlL+zeaXQrQoDT18nZ55x9QGyeK8rT/7QXOz6dLDmj/2wt13dAJhCBIIgCZJiAgQH/9yzKVlvIbr9n+/e9/6+mnn9bUqVNd3isHc9jt9qvmwaUeZ06YXYJllWxbU02roSPbstWyLee5GRra+UAAMlFAQMBV8xf91SAkJEShoaFml4GrSLMDm8wuwbLOHjtcbfuZY4fku3vtFa4GDREBCADqyZlOA1XVzN/sMizJZ89+nT7q/mLrZq3b67uwO0yoCB5nTjSoXwoIQABQT6qa+avKt7XZZVhSYMww7X97oSTjR602BcQO478JJPE1eADAVci/a4SuGzpBzQOvk0dTbzUPvE7XJT4k/y79zC4NDQQrQACAq5J/1wi+8YULYgUIAABYDgEIAABYzmUFoIqKCu3du1c//PBDXdUDAABQ72oVgE6fPq0xY8aoefPm6tmzp4qLiyVJDz30kP70pz/VaYEAAAB1rVYBKDU1VTt37tRHH30kHx8fZ/vNN9+srKysOisOAACgPtTqW2DZ2dnKyspSdHS0bDabsz0sLExffvllnRUHAABQH2q1AvSf//xHbdq0cWv/7rvvXAIRAABAQ1SrANS/f3+9++67zs/nQ8+f//xnxcTE1E1lAAAA9aRWl8DS09N16623avfu3frhhx80f/58FRYWatu2bdq4cWNd1wgAAFCnarUCFBsbq48//linT59W586d9cEHHyggIEDbtm1TRARP3QQAAA1brV+Fcf3112vp0qV1WQsAAMAVUasVoJycHL3//vtu7e+//77ee++9yy4KAACgPtUqAD3xxBOqrKx0azcMQ0888cRlFwUAAFCfahWAioqKFBYW5tbevXt3ffHFF5ddFAAAQH2qVQCy2+3av3+/W/sXX3whX1/fyy4KAACgPtUqAN1xxx2aNGmSy1Ofv/jiCz3yyCO644476qw4AACA+lCrAPTcc8/J19dX3bt3V6dOndSpUyf16NFDrVq10vPPP1/XNQIAANSpWn0N3m63a+vWrcrNzdXOnTvVrFkz9erVSwMHDqzr+gAAAOpcrZ8DZLPZlJCQoISEhLqsBwAAoN7VOgCtX79e69ev19GjR1VVVeWy7/XXX7/swgAAAOpLrQLQzJkzlZaWpsjISAUFBfEGeAAA0KjUKgC9/PLLyszMVHJycl3XAwAAUO9q9S2wiooKxcbG1kkBixYtUqdOneTj46OIiAht3rz5ov3Ly8s1depUhYSEyNvbW507d3a55JaZmSmbzea2nT17tk7qBQAAjV+tVoDGjh2rv/3tb5o2bdpl/fCsrCxNmjRJixYtUlxcnF555RUNGTJEu3fvVocOHao9ZuTIkTpy5Ihee+01denSRUePHtUPP/zg0sfPz0979+51afPx8bmsWgEAwNWjVgHo7NmzysjI0IcffqhevXqpadOmLvvnzJlTo3HmzJmjMWPGaOzYsZKkefPm6f3339fixYuVnp7u1n/dunXauHGj9u/fr5YtW0qSOnbs6NbPZrMpMDDwEmcFK8rOztb06dNVWFioESNGKC0tTYmJiWaXBQCoZ7W6BPbpp5+qT58+8vDw0Geffab8/HznVlBQUKMxKioqlJeX5/Y1+oSEBG3durXaY9auXavIyEg9++yzateunUJDQ/Xoo4/qzJkzLv1OnTqlkJAQtW/fXrfddpvy8/MvWkt5ebnKyspcNlz9srOzNWzYMO3atUtVVVXatWuXhg8fruzsbLNLAwDUs1qtAG3YsOGyf/CxY8dUWVmpgIAAl/aAgACVlJRUe8z+/fu1ZcsW+fj4aM2aNTp27JjGjx+vb775xnkfUPfu3ZWZmanrr79eZWVlmj9/vuLi4rRz50517dq12nHT09M1c+bMy56TFZ09e1bFxcVml1Er06dPd2szDEMzZsyo9mW/jUWHDh245AsAP6HWzwGqK//9FXrDMC74tfqqqirZbDYtX75cdrtd0rnLaCNGjNBLL72kZs2aKTo6WtHR0c5j4uLi1K9fPy1cuFALFiyodtzU1FRNnjzZ+bmsrEzBwcGXOzVLKC4uVkpKitll1EphYWG17Z999lmjnZMkZWRkKDQ01OwyAKBBq3UA2rFjh1auXKni4mJVVFS47Fu9evVPHt+6dWt5enq6rfYcPXrUbVXovKCgILVr184ZfiSpR48eMgxDhw4dqnaFx8PDQ/3791dRUdEFa/H29pa3t/dP1gx3HTp0UEZGhtll1MqIESO0a9cut/bw8PBGOydJF/wCAQDg/9QqAL3xxhsaNWqUEhISlJubq4SEBBUVFamkpETDhg2r0RheXl6KiIhQbm6uyzG5ubkaOnRotcfExcVp5cqVOnXqlFq0aCFJ2rdvnzw8PNS+fftqjzEMQwUFBbr++usvcZaoCR8fn0a72pCWlqbhw4fLMAxnm81mU1paWqOdEwCgZmp1E/QzzzyjuXPn6n//93/l5eWl+fPna8+ePRo5cuQl/fY5efJkvfrqq3r99de1Z88ePfzwwyouLta4ceMknbs0NWrUKGf/u+++W61atdLo0aO1e/dubdq0SVOmTNEDDzygZs2aSTr3lOr3339f+/fvV0FBgcaMGaOCggLnmMB5iYmJWr16taKiouTr66uoqCitWbPmggEcAHD1qNUK0Jdffqlf/epXks5dPvruu+9ks9n08MMP66abbqrxDcVJSUk6fvy40tLS5HA4FB4erpycHIWEhEiSHA6Hyw22LVq0UG5uriZMmKDIyEi1atVKI0eO1KxZs5x9Tpw4oZSUFJWUlMhut6tv377atGmToqKiajNVXOUSExP52jsAWFCtAlDLli118uRJSVK7du302Wef6frrr9eJEyd0+vTpSxpr/PjxGj9+fLX7MjMz3dq6d++u3NzcC443d+5czZ0795JqAAAA1lKrABQfH6/c3Fxdf/31GjlypCZOnKi///3vys3N1aBBg+q6RgAAgDpVqwD04osvOt+tlZqaqqZNm2rLli0aPnz4Zb8eAwAAoL7V+hLYeR4eHnrsscf02GOP1VlRAAAA9alW3wLz9PTU0aNH3dqPHz8uT0/Pyy4KAACgPtUqAP34uSk/Vl5eLi8vr8sqCAAAoL5d0iWw86+SsNlsevXVV50PI5SkyspKbdq0Sd27d6/bCgEAAOrYJQWg818vNwxDL7/8ssvlLi8vL3Xs2FEvv/xy3VYIAABQxy4pAB04cECS9Itf/EKrV6/WtddeWy9FAQAA1Kda3QO0YcMGl/BTWVmpgoICffvtt3VWGAAAQH2pVQCaNGmSXnvtNUnnws/AgQPVr18/BQcH66OPPqrL+gAAAOpcrQLQypUr1bt3b0nSO++8o6+++kqff/65Jk2apKlTp9ZpgQAAAHWtVgHo+PHjCgwMlCTl5OTozjvvVGhoqMaMGaNdu3bVaYEAAAB1rVYBKCAgQLt371ZlZaXWrVunm2++WZJ0+vRpHoQIAAAavFq9CmP06NEaOXKkgoKCZLPZdMstt0iStm/fznOAAABAg1erAPTUU08pPDxcBw8e1J133ilvb29J516R8cQTT9RpgQAAAHWtVgFIkkaMGOHWdt99911WMQAAAFdCjQPQggULlJKSIh8fH+crMS7koYceuuzCAAAA6kuNA9DcuXN1zz33yMfHx/lKjOrYbDYCEAAAaNBqHIDOvwbjv/8ZAACgsalxAJo8eXKN+tlsNr3wwgu1LggAAKC+1TgA5efnu3zOy8tTZWWlunXrJknat2+fPD09FRERUbcVAgAA1LEaB6ANGzY4/3nOnDm65pprtHTpUudLUb/99luNHj1a8fHxdV8lAABAHarVk6BfeOEFpaenu7wR/tprr9WsWbO4/AUAABq8WgWgsrIyHTlyxK396NGjOnny5GUXBQAAUJ9qFYCGDRum0aNH66233tKhQ4d06NAhvfXWWxozZoyGDx9e1zUCAADUqVo9Cfrll1/Wo48+qnvvvVfff//9uYGaNNGYMWP03HPP1WmBAAAAda1WAah58+ZatGiRnnvuOX355ZcyDENdunSRr69vXdcHAABQ52r9LjBJ8vX1Va9eveqqFgAAgCuiVvcAAQAANGYEIAAAYDkEIAAAYDkEIAAAYDmmB6BFixapU6dO8vHxUUREhDZv3nzR/uXl5Zo6dapCQkLk7e2tzp076/XXX3fps2rVKoWFhcnb21thYWFas2ZNfU4BAAA0MqYGoKysLE2aNElTp05Vfn6+4uPjNWTIEBUXF1/wmJEjR2r9+vV67bXXtHfvXq1YsULdu3d37t+2bZuSkpKUnJysnTt3Kjk5WSNHjtT27duvxJQAAEAjYDMMwzDrhw8YMED9+vXT4sWLnW09evRQYmKi0tPT3fqvW7dOd911l/bv36+WLVtWO2ZSUpLKysr03nvvOdtuvfVWXXvttVqxYkWN6iorK5Pdbldpaan8/PwucVYArG7fvn1KSUnRd2F3qMq3tdnlAA2Cx3fH5Lt7rTIyMhQaGmp2OeatAFVUVCgvL08JCQku7QkJCdq6dWu1x6xdu1aRkZF69tln1a5dO4WGhurRRx/VmTNnnH22bdvmNubgwYMvOKZ07rJaWVmZywYAAK5el/UgxMtx7NgxVVZWKiAgwKU9ICBAJSUl1R6zf/9+bdmyRT4+PlqzZo2OHTum8ePH65tvvnHeB1RSUnJJY0pSenq6Zs6ceZkzAgAAjYXpN0HbbDaXz4ZhuLWdV1VVJZvNpuXLlysqKkq//OUvNWfOHGVmZrqsAl3KmJKUmpqq0tJS53bw4MHLmBEAAGjoTFsBat26tTw9Pd1WZo4ePeq2gnNeUFCQ2rVrJ7vd7mzr0aOHDMPQoUOH1LVrVwUGBl7SmJLk7e0tb2/vy5gNAABoTExbAfLy8lJERIRyc3Nd2nNzcxUbG1vtMXFxcfr666916tQpZ9u+ffvk4eGh9u3bS5JiYmLcxvzggw8uOCYAALAeUy+BTZ48Wa+++qpef/117dmzRw8//LCKi4s1btw4SecuTY0aNcrZ/+6771arVq00evRo7d69W5s2bdKUKVP0wAMPqFmzZpKkiRMn6oMPPtDs2bP1+eefa/bs2frwww81adIkM6YIAAAaINMugUnnvrJ+/PhxpaWlyeFwKDw8XDk5OQoJCZEkORwOl2cCtWjRQrm5uZowYYIiIyPVqlUrjRw5UrNmzXL2iY2N1RtvvKEnn3xS06ZNU+fOnZWVlaUBAwZc8fkBAICGydTnADVUPAcIwOXgOUCAO54DBAAAYDICEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBwCEAAAsBzTA9CiRYvUqVMn+fj4KCIiQps3b75g348++kg2m81t+/zzz519MjMzq+1z9uzZKzEdAADQCDQx84dnZWVp0qRJWrRokeLi4vTKK69oyJAh2r17tzp06HDB4/bu3Ss/Pz/n55/97Gcu+/38/LR3716XNh8fn7otHgAANFqmBqA5c+ZozJgxGjt2rCRp3rx5ev/997V48WKlp6df8Lg2bdrI39//gvttNpsCAwNrXEd5ebnKy8udn8vKymp8LAAAaHxMuwRWUVGhvLw8JSQkuLQnJCRo69atFz22b9++CgoK0qBBg7Rhwwa3/adOnVJISIjat2+v2267Tfn5+RcdLz09XXa73bkFBwdf+oQAAECjYVoAOnbsmCorKxUQEODSHhAQoJKSkmqPCQoKUkZGhlatWqXVq1erW7duGjRokDZt2uTs0717d2VmZmrt2rVasWKFfHx8FBcXp6KiogvWkpqaqtLSUud28ODBupkkAABokEy9BCadu1z1Y4ZhuLWd161bN3Xr1s35OSYmRgcPHtTzzz+vgQMHSpKio6MVHR3t7BMXF6d+/fpp4cKFWrBgQbXjent7y9vb+3KnAgAAGgnTVoBat24tT09Pt9Weo0ePuq0KXUx0dPRFV3c8PDzUv3//i/YBAADWYloA8vLyUkREhHJzc13ac3NzFRsbW+Nx8vPzFRQUdMH9hmGooKDgon0AAIC1mHoJbPLkyUpOTlZkZKRiYmKUkZGh4uJijRs3TtK5e3MOHz6sZcuWSTr3LbGOHTuqZ8+eqqio0F//+letWrVKq1atco45c+ZMRUdHq2vXriorK9OCBQtUUFCgl156yZQ5AgCAhsfUAJSUlKTjx48rLS1NDodD4eHhysnJUUhIiCTJ4XCouLjY2b+iokKPPvqoDh8+rGbNmqlnz55699139ctf/tLZ58SJE0pJSVFJSYnsdrv69u2rTZs2KSoq6orPDwAANEw2wzAMs4toaMrKymS321VaWurywEUAqIl9+/YpJSVF34XdoSrf1maXAzQIHt8dk+/utcrIyFBoaKjZ5Zj/KgwAAIArjQAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAshwAEAAAsx/QAtGjRInXq1Ek+Pj6KiIjQ5s2bL9j3o48+ks1mc9s+//xzl36rVq1SWFiYvL29FRYWpjVr1tT3NAAAQCNiagDKysrSpEmTNHXqVOXn5ys+Pl5DhgxRcXHxRY/bu3evHA6Hc+vatatz37Zt25SUlKTk5GTt3LlTycnJGjlypLZv317f0wEAAI2EqQFozpw5GjNmjMaOHasePXpo3rx5Cg4O1uLFiy96XJs2bRQYGOjcPD09nfvmzZunW265RampqerevbtSU1M1aNAgzZs3r55nAwAAGgvTAlBFRYXy8vKUkJDg0p6QkKCtW7de9Ni+ffsqKChIgwYN0oYNG1z2bdu2zW3MwYMHX3TM8vJylZWVuWwAAODqZVoAOnbsmCorKxUQEODSHhAQoJKSkmqPCQoKUkZGhlatWqXVq1erW7duGjRokDZt2uTsU1JSckljSlJ6errsdrtzCw4OvoyZAQCAhq6J2QXYbDaXz4ZhuLWd161bN3Xr1s35OSYmRgcPHtTzzz+vgQMH1mpMSUpNTdXkyZOdn8vKyghBAABcxUxbAWrdurU8PT3dVmaOHj3qtoJzMdHR0SoqKnJ+DgwMvOQxvb295efn57IBAICrl2kByMvLSxEREcrNzXVpz83NVWxsbI3Hyc/PV1BQkPNzTEyM25gffPDBJY0JAACubqZeAps8ebKSk5MVGRmpmJgYZWRkqLi4WOPGjZN07tLU4cOHtWzZMknnvuHVsWNH9ezZUxUVFfrrX/+qVatWadWqVc4xJ06cqIEDB2r27NkaOnSo3n77bX344YfasmWLKXMEAAANj6kBKCkpScePH1daWpocDofCw8OVk5OjkJAQSZLD4XB5JlBFRYUeffRRHT58WM2aNVPPnj317rvv6pe//KWzT2xsrN544w09+eSTmjZtmjp37qysrCwNGDDgis8PAAA0TDbDMAyzi2hoysrKZLfbVVpayv1AAC7Zvn37lJKSou/C7lCVb2uzywEaBI/vjsl391plZGQoNDTU7HLMfxUGAADAlUYAAgAAlkMAAgAAlkMAAgAAlkMAAgAAlkMAAgAAlkMAAgAAlkMAAgAAlkMAAgAAlkMAAgAAlkMAAgAAlkMAAgAAlmPq2+ABAKgvJ4ryVLL9HZ09dlg+rdspcMDt8u8aYXZZaCBYAQIAXHVOFOVp/9sLdLrkgKp+qNDpkgPa//ZCnSjKM7s0NBCsAAFAPfE4W2p2CZZVsm1NNa2GjmzLVsu2IVe8HjS884EABAB1zG63q6mXt7R/o9mlWNbZY4erbT9z7JB8d6+9wtXgvKZe3rLb7WaXIUmyGYZhmF1EQ1NWVia73a7S0lL5+fmZXQ6ARujIkSMqLW1Yv/FayYgRI7Rr1y639l69emnlypUmVATp3C8HAQEBZpchiQBULQIQADRu2dnZGj58uH78f3E2m01r1qzR0KFDTawMDQU3QQMArjqJiYlavXq1evXqJQ8PD/Xq1YvwAxesAFWDFSAAuDrs27dPKSkpysjIUGhoqNnloAFhBQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAcFXKzs7WiBEjtHnzZo0YMULZ2dlml4QGhJehVoOXoQJA45adna1hw4a5tNlsNq1evVqJiYnmFIUGpYnZBSxatEjPPfecHA6HevbsqXnz5ik+Pv4nj/v44491ww03KDw8XAUFBc72zMxMjR492q3/mTNn5OPjU5elA8BV7ezZsyouLja7jFqZPn26W5thGJoxY4bCwsJMqKhudOjQgf8vqyOmBqCsrCxNmjRJixYtUlxcnF555RUNGTJEu3fvVocOHS54XGlpqUaNGqVBgwbpyJEjbvv9/Py0d+9elzb+wADApSkuLlZKSorZZdRKYWFhte2fffZZo52TJGVkZCg0NNTsMq4Kpl4CGzBggPr166fFixc723r06KHExESlp6df8Li77rpLXbt2laenp7Kzs91WgCZNmqQTJ07UuI7y8nKVl5c7P5eVlSk4OJhLYAAsrTGvAI0YMUK7du1ya+/Vq5dWrlxpQkV1gxWgumPaClBFRYXy8vL0xBNPuLQnJCRo69atFzxuyZIl+vLLL/XXv/5Vs2bNqrbPqVOnFBISosrKSvXp00d//OMf1bdv3wuOmZ6erpkzZ9ZuIgBwlfLx8Wm0qw1paWkaPny4fvw7vs1mU1paWqOdE+qWad8CO3bsmCorKxUQEODSHhAQoJKSkmqPKSoq0hNPPKHly5erSZPqs1v37t2VmZmptWvXasWKFfLx8VFcXJyKioouWEtqaqpKS0ud28GDB2s/MQCA6RITE7V69WpFRUXJ19dXUVFRWrNmjYYOHWp2aWggTL8J2mazuXw2DMOtTZIqKyt19913a+bMmRdN79HR0YqOjnZ+jouLU79+/bRw4UItWLCg2mO8vb3l7e1dyxkAABqixMREvvGFCzItALVu3Vqenp5uqz1Hjx51WxWSpJMnT+qTTz5Rfn6+fv/730uSqqqqZBiGmjRpog8++EA33XST23EeHh7q37//RVeAAACAtZh2CczLy0sRERHKzc11ac/NzVVsbKxbfz8/P+3atUsFBQXObdy4cerWrZsKCgo0YMCAan+OYRgqKChQUFBQvcwDAAA0PqZeAps8ebKSk5MVGRmpmJgYZWRkqLi4WOPGjZN07t6cw4cPa9myZfLw8FB4eLjL8W3atJGPj49L+8yZMxUdHa2uXbuqrKxMCxYsUEFBgV566aUrOjcAANBwmRqAkpKSdPz4caWlpcnhcCg8PFw5OTkKCQmRJDkcjkv+CuaJEyeUkpKikpIS2e129e3bV5s2bVJUVFR9TAEAADRCvAqjGrwKAwCAqxsvQwUAAJZDAAIAAJZDAAIAAJZDAAIAAJZDAAIAAJZDAAIAAJZj+rvAGqLzTwYoKyszuRIAAHCprrnmmmrfK/pjBKBqnDx5UpIUHBxsciUAAOBS1eQ5fjwIsRpVVVX6+uuva5Qg0fiVlZUpODhYBw8e5MGXwFWG89uaWAGqJQ8PD7Vv397sMnCF+fn58RckcJXi/MZ/4yZoAABgOQQgAABgOQQgWJ63t7dmzJghb29vs0sBUMc4v3Eh3AQNAAAshxUgAABgOQQgAABgOQQgAABgOQQgAABgOQQg1Mj9998vm82mP/3pTy7t2dnZdfq07C+++EKjR49W+/bt5e3trU6dOunXv/61Pvnkkzr7GXXtq6++ks1mU0FBQY2Peeqpp2Sz2XTrrbe67Xv22Wdls9l04403SpI6duwom812we3GG2/U119/rZYtW2rBggUuY23fvl1NmzZVbm7u5UwRqFZJSYkmTJig6667Tt7e3goODtbtt9+u9evXm13aBdlsNmVnZ9e4f2Zmpmw2m3r06OG2780335TNZlPHjh0lSTfeeONFz9WOHTvq1KlT6ty5syZPnuwy1ldffSU/Pz+9+uqrlzM9XAICEGrMx8dHs2fP1rffflsv43/yySeKiIjQvn379Morr2j37t1as2aNunfvrkceeaTW41ZWVqqqqsqtvaKi4nLKvWxBQUHasGGDDh065NK+ZMkSdejQwfl5x44dcjgccjgcWrVqlSRp7969zrbVq1erbdu2WrBggVJTU1VUVCRJOnPmjO677z6NHTtWt9xyy5WbGCzhq6++UkREhP7+97/r2Wef1a5du7Ru3Tr94he/0IMPPlirMQ3D0A8//ODWbva56uvrq6NHj2rbtm0u7a+//rrLubp69WrnefnPf/5TkvThhx8623bs2KEWLVpoyZIlWrhwoTZv3izp3LxHjx6tuLg4jR079spNzOoMoAbuu+8+47bbbjO6d+9uTJkyxdm+Zs0a48d/jN566y0jLCzM8PLyMkJCQoznn3++RuNXVVUZPXv2NCIiIozKykq3/d9++61hGIaxYcMGQ5Lzs2EYRn5+viHJOHDggGEYhrFkyRLDbrcb77zzjtGjRw/D09PT2L9/vxESEmL88Y9/NO677z7Dz8/PGDVqlGEYhvHxxx8b8fHxho+Pj9G+fXtjwoQJxqlTp5zjh4SEGE8//bQxevRoo0WLFkZwcLDxyiuvOPdLctluuOGGn5zvjBkzjN69exu33XabMWvWLGf7xx9/bLRu3dr43e9+V+041c3/x4YNG2bExsYalZWVxsSJE41OnToZJ0+e/Ml6gEs1ZMgQo127di7nynnffvutceDAAUOSkZ+f79IuydiwYYNhGP/353ndunVGRESE0bRpU+Pvf/+7ccMNNxgPPvig8fDDDxutWrUyBg4caBiGYRQWFhpDhgwxfH19jTZt2hj33nuv8Z///Mc5/g033GBMmDDBmDJlinHttdcaAQEBxowZM5z7Q0JCXM7VkJCQn5zn+b9Pfv/73xtjx451th88eNDw9vY2nnjiiWrHqW7+P/bwww8bnTt3Nk6dOmXMnTvX8Pf3Nw4dOvST9aDusAKEGvP09NQzzzyjhQsXuq1aSFJeXp5Gjhypu+66S7t27dJTTz2ladOmKTMz8yfHLigoUGFhoR555BF5eLj/sfT397+kWk+fPq309HS9+uqrKiwsVJs2bSRJzz33nMLDw5WXl6dp06Zp165dGjx4sIYPH65PP/1UWVlZ2rJli37/+9+7jPfCCy8oMjJS+fn5Gj9+vH73u9/p888/lyS33/RWr15d4zofeOABl38/r7/+uu655x55eXld0nzPe/nll1VUVKR77rlHL774ojIzM9WiRYtajQVcyDfffKN169bpwQcflK+vr9v+Sz1fH3vsMaWnp2vPnj3q1auXJGnp0qVq0qSJPv74Y73yyityOBy64YYb1KdPH33yySdat26djhw5opEjR7qMtXTpUvn6+mr79u169tlnlZaW5rwEvGPHDknnVlnPr8jU1JgxY5SVlaXTp09LOndp7NZbb1VAQMAlzfW8Z555Rk2bNtW9996rP/zhD1q4cKHatWtXq7FQS2YnMDQO9913nzF06FDDMAwjOjraeOCBBwzDcF0Buvvuu41bbrnF5bgpU6YYYWFhPzl+VlaWIcn417/+ddF+NV0BkmQUFBS4HBsSEmIkJia6tCUnJxspKSkubZs3bzY8PDyMM2fOOI+79957nfurqqqMNm3aGIsXLzYM46d/06vO+RWgiooKo02bNsbGjRuNU6dOGddcc42xc+dOY+LEibVaATIMw3j55ZcNScbvfve7GtcDXIrt27cbkozVq1dfsM+lrABlZ2e7HHvDDTcYffr0cWmbNm2akZCQ4NJ28OBBQ5Kxd+9e53E///nPXfr079/fePzxx52fJRlr1qyp6VSdK0CGYRh9+vQxli5dalRVVRmdO3c23n77bWPu3Lm1WgEyDMNYt26dIckYMmRIjetB3WEFCJds9uzZWrp0qXbv3u3SvmfPHsXFxbm0xcXFqaioSJWVlRcd0/j/DySvqxuqvby8nL9J/lhkZKTL57y8POcqyflt8ODBqqqq0oEDB5z9fjyWzWZTYGCgjh49etl1nv8NcMmSJVq5cqVCQ0OrrbumKisrtXTpUjVv3lz/+Mc/qr2fArhcdX2+/vd5WV1bXl6eNmzY4HKudu/eXZL05ZdfOvv99/kTFBRUJ+eqdG7FdsmSJdq4caNOnTqlX/7yl5c13muvvabmzZtr165dKi0trZMaUXMEIFyygQMHavDgwfrDH/7g0m4YhttfiEYN37QSGhoq6VyIupjzl8d+PO7333/v1q9Zs2bV/uX838v1VVVV+u1vf6uCggLntnPnThUVFalz587Ofk2bNnU5zmazVXtjdW088MADWrlypV566SU98MADlzXW888/r6KiIu3YsUNff/21nnnmmTqpEfixrl27ymazXfR8rem5Krmfl9W1VVVV6fbbb3c5VwsKClRUVKSBAwc6+9XnuXrPPffoH//4h5566imNGjVKTZo0qfVYWVlZWrt2rbZs2SK73a6HH364TmpEzRGAUCt/+tOf9M4772jr1q3OtrCwMG3ZssWl39atWxUaGipPT8+LjtenTx+FhYXphRdeqPYvqxMnTkiSfvazn0mSHA6Hc9+lfP38v/Xr10+FhYXq0qWL21bT+3DO9/upVa4L6dmzp3r27KnPPvtMd999d63GkKTCwkLNmDFDixcvVlhYmF5++WXNmjVLn376aa3HBKrTsmVLDR48WC+99JK+++47t/0nTpyot3O1Y8eObudqdQHqQpo2bVrrc7Vly5a64447tHHjxsv6ZeXIkSN68MEHNWvWLPXt21eZmZn6y1/+ovfee6/WY+LSEYBQK9dff73uueceLVy40Nn2yCOPaP369frjH/+offv2aenSpXrxxRf16KOP/uR4NptNS5Ys0b59+zRw4EDl5ORo//79+vTTT/X0009r6NChkqQuXbooODhYTz31lPbt26d3331XL7zwQq3n8fjjj2vbtm168MEHnb9Nrl27VhMmTKjxGG3atFGzZs2cN2XWZin773//uxwOxyXfPHreDz/8oPvuu0/Dhg3TiBEjJEmJiYm68847df/993MpDHVu0aJFqqysVFRUlFatWqWioiLt2bNHCxYsUExMjJo1a6bo6Gj96U9/0u7du7Vp0yY9+eSTtf55Dz74oL755hv9+te/1j//+U/t379fH3zwgR544IFLCjQdO3bU+vXrVVJSUqtHemRmZurYsWPOy2+18dvf/lbdunVzPgsoMjJSjz32mFJSUrgUdgURgFBrf/zjH12Wt/v166c333xTb7zxhsLDwzV9+nSlpaXp/vvvr9F4UVFR+uSTT9S5c2f95je/UY8ePXTHHXeosLBQ8+bNk3Tut7cVK1bo888/V+/evTV79mzNmjWr1nPo1auXNm7cqKKiIsXHx6tv376aNm2agoKCajxGkyZNtGDBAr3yyitq27atM6xdCl9f31qHH+ncN0oOHz6sF1980aV94cKFcjgcXApDnevUqZP+9a9/6Re/+IUeeeQRhYeH65ZbbtH69eu1ePFiSee+1fj9998rMjJSEydOvKxztW3btvr4449VWVmpwYMHKzw8XBMnTpTdbq/2m6MX8sILLyg3N1fBwcHq27fvJdfRrFkztWrV6pKPO2/ZsmXKzc1VZmamS90zZsyQv78/l8KuIJtR05s0AAAArhKsAAEAAMshAOGK2Lx5s8vXV/97u9pcbK7nH38PwHw9e/a84Lm6fPlys8tDPeISGK6IM2fO6PDhwxfc36VLlytYTf374osvLrivXbt2atas2RWsBsCF/Pvf/77g1/MDAgJ0zTXXXOGKcKUQgAAAgOVwCQwAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQgAAFgOAQhAo/XUU0+pT58+ZpcBoBEiAAFAHbnQ82QANDwEIACmqqqq0uzZs9WlSxd5e3urQ4cOevrppyVJjz/+uEJDQ9W8eXNdd911mjZtmjNkZGZmaubMmdq5c6dsNptsNpsyMzMlSaWlpUpJSVGbNm3k5+enm266STt37nT5ubNmzVKbNm10zTXXaOzYsXriiSdcVpOqqqqUlpam9u3by9vbW3369NG6deuc+7/66ivZbDa9+eabuvHGG+Xj46OMjAz5+fnprbfecvlZ77zzjnx9fXXy5Ml6+DcIoDYIQABMlZqaqtmzZ2vatGnavXu3/va3vykgIECSdM011ygzM1O7d+/W/Pnz9ec//1lz586VJCUlJemRRx5Rz5495XA45HA4lJSUJMMw9Ktf/UolJSXKyclRXl6e+vXrp0GDBumbb76RJC1fvlxPP/20Zs+erby8PHXo0MH5BvPz5s+frxdeeEHPP/+8Pv30Uw0ePFh33HGHioqKXPo9/vjjeuihh7Rnzx4NGzZMd911l5YsWeLSZ8mSJRoxYgRPFQYaEgMATFJWVmZ4e3sbf/7zn2vU/9lnnzUiIiKcn2fMmGH07t3bpc/69esNPz8/4+zZsy7tnTt3Nl555RXDMAxjwIABxoMPPuiyPy4uzmWstm3bGk8//bRLn/79+xvjx483DMMwDhw4YEgy5s2b59Jn+/bthqenp3H48GHDMAzjP//5j9G0aVPjo48+qtEcAVwZrAABMM2ePXtUXl6uQYMGVbv/rbfe0s9//nMFBgaqRYsWmjZtmoqLiy86Zl5enk6dOqVWrVq5vNjywIED+vLLLyVJe/fuVVRUlMtxP/5cVlamr7/+WnFxcS594uLitGfPHpe2yMhIt3F69uypZcuWSZL+8pe/qEOHDho4cOBF6wZwZTUxuwAA1nWxl8L+4x//0F133aWZM2dq8ODBstvteuONN/TCCy9cdMyqqioFBQXpo48+ctvn7+/v/Gebzeayz6jmtYjV9fnvNl9fX7fjxo4dqxdffFFPPPGElixZotGjR7sdB8BcrAABME3Xrl3VrFkzrV+/3m3fxx9/rJCQEE2dOlWRkZHq2rWr/v3vf7v08fLyUmVlpUtbv379VFJSoiZNmqhLly4uW+vWrSVJ3bp10z//+U+X4z755BPnP/v5+alt27basmWLS5+tW7eqR48ePzmve++9V8XFxVqwYIEKCwt13333/eQxAK4sVoAAmMbHx0ePP/64HnvsMXl5eSkuLk7/+c9/VFhYqC5duqi4uFhvvPGG+vfvr3fffVdr1qxxOb5jx446cOCACgoK1L59e11zzTW6+eabFRMTo8TERM2ePVvdunXT119/rZycHCUmJioyMlITJkzQb37zG0VGRio2NlZZWVn69NNPdd111znHnjJlimbMmKHOnTurT58+WrJkiQoKCrR8+fKfnNe1116r4cOHa8qUKUpISFD79u3r/N8dgMtk9k1IAKytsrLSmDVrlhESEmI0bdrU6NChg/HMM88YhmEYU6ZMMVq1amW0aNHCSEpKMubOnWvY7XbnsWfPnjX+53/+x/D39zckGUuWLDEM49zN1RMmTDDatm1rNG3a1AgODjbuueceo7i42HlsWlqa0bp1a6NFixbGAw88YDz00ENGdHS0S10zZ8402rVrZzRt2tTo3bu38d577zn3n78JOj8/v9p5rV+/3pBkvPnmm3X3LwtAnbEZRjUXvgHAYm655RYFBgbqL3/5S52Mt3z5ck2cOFFff/21vLy86mRMAHWHS2AALOf06dN6+eWXNXjwYHl6emrFihX68MMPlZubWydjHzhwQOnp6frtb39L+AEaKG6CBmA5NptNOTk5io+PV0REhN555x2tWrVKN99882WP/eyzz6pPnz4KCAhQampqHVQLoD5wCQwAAFgOK0AAAMByCEAAAMByCEAAAMByCEAAAMByCEAAAMByCEAAAMByCEAAAMByCEAAAMBy/h8zhqBd9DYRLwAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# investigate whether MTX treated twins converge to unaffected twins\n",
    "\n",
    "# manually input metadata from Patient_data_KB.xls\n",
    "samp_to_MTX = {\n",
    "    '500A': 1,\n",
    "    '502A': 1,\n",
    "    '504A': 1,\n",
    "    '506A': 0, # other DMARDs\n",
    "    '508A': 0, # unclear? HCQ\n",
    "    '510A': 1,\n",
    "    '512A': 1,\n",
    "    '514A': 0#, # previosuly, not currently\n",
    "    #'516A.Fecal': 0, # in 2012, not currently,\n",
    "}\n",
    "# present MTX vs present biologic?\n",
    "# to treatment\n",
    "samp_to_tmt = {\n",
    "    '500A': 1, # MTX\n",
    "    '502A': 1, # MTX\n",
    "    '504A': 1, # MTX\n",
    "    '506A': 1, # other DMARDs\n",
    "    '508A': 1, # on HCQ\n",
    "    '510A': 1,\n",
    "    '512A': 1,\n",
    "    '514A': 0#, # previously, not currently\n",
    "    #'516A.Fecal': 0, # in 2012, not currently,\n",
    "}\n",
    "\n",
    "# get copy\n",
    "df_meta = df_map.copy()\n",
    "\n",
    "# replace with column dict per above\n",
    "df_meta['MTX'] = df_meta.index.map(samp_to_MTX)\n",
    "\n",
    "# replace nan with 0 i.e. untreated\n",
    "df_meta['MTX'] = df_meta['MTX'].fillna(0) \n",
    "\n",
    "# copy for future\n",
    "df_med = df_meta.copy()\n",
    "\n",
    "# create TwinPair to MTX mapping\n",
    "df = df_meta[df_meta['Diagnosis'] == 'RA'].set_index('TwinPair')\n",
    "TP_to_MTX = df['MTX'].to_dict()\n",
    "\n",
    "# Things to ask\n",
    "# 1. Do the twin pair distances decrease in treated vs untreated twins?\n",
    "df_paired_beta['MTX'] = df_paired_beta.index.map(TP_to_MTX)\n",
    "\n",
    "# compare distances\n",
    "inter_untreated = df_paired_beta[df_paired_beta['MTX'] == 0]['Bray_Curtis']\n",
    "inter_treated = df_paired_beta[df_paired_beta['MTX'] == 1]['Bray_Curtis']\n",
    "\n",
    "u, p = scipy.stats.mannwhitneyu(inter_untreated, inter_treated)\n",
    "print(u, p)\n",
    "\n",
    "t, p = scipy.stats.ttest_ind(inter_untreated, inter_treated)\n",
    "print(t, p)\n",
    "\n",
    "# sns.boxplot([inter_untreated, inter_treated])\n",
    "\n",
    "category = ['No_Current_MTX']*len(inter_untreated) + ['Current_MTX']*len(inter_treated)\n",
    "distances = list(inter_untreated) + list(inter_treated)\n",
    "df_dist = pd.DataFrame(data=np.array([category,distances]).T, columns=['category','distance'])\n",
    "df_dist['distance'] = df_dist['distance'].astype(float)\n",
    "\n",
    "# f test of variance\n",
    "\n",
    "F = np.var(inter_untreated) / np.var(inter_treated)\n",
    "df1 = len(inter_untreated) - 1\n",
    "df2 = len(inter_treated) - 1\n",
    "\n",
    "p_value = scipy.stats.f.cdf(F, df1, df2)\n",
    "\n",
    "print(F,p_value)\n",
    "\n",
    "# levene test non parametric\n",
    "l, p  = scipy.stats.levene(inter_untreated, inter_treated)\n",
    "print(l, p)\n",
    "\n",
    "sns.boxplot(data=df_dist, x='category', y='distance')\n",
    "sns.swarmplot(data=df_dist, x='category', y='distance', color='black')\n",
    "\n",
    "df_dist = df_dist.rename(columns={'category':'MTX', 'distance': 'Bray_Curtis'})\n",
    "# df_dist['MTX'] = df_dist['MTX'].map(lambda x: 'No_Current_MTX' if int(x) == 0 else 'Current_MTX')\n",
    "df_dist.to_csv(path + 'outputs/jobs02/beta_MTX.tsv',sep='\\t')\n",
    "\n",
    "sns.despine()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11.0 0.39285714285714285\n",
      "0.6711590447710831 0.5270980576728104\n",
      "0.8890306398926742 0.5207570909770318\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAkMAAAGwCAYAAACq12GxAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAufUlEQVR4nO3deXRUZYL+8acSkgogKYSQSphOWNQQZBsI3bI0QQVDAwOKnh66XZBNRMahSaTpQzsK6CjqQURHASNggtiacUDcsCGyBGQZDZB2aBOwbSQxZgECVSRCtqrfHwz1M5MA4aYqlcr9fs65p6m33lv11OlT+PDeW/da3G63WwAAACYV5O8AAAAA/kQZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApkYZugq32y2n0ykuxwQAQOtEGbqKc+fOyWaz6dy5c/6OAgAAfIAyBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATK2NvwMALUVeXp727NmjkydPqkuXLhoxYoTi4+P9HQsA4GOsDAG6WIQyMjL0ww8/qLq6Wj/88IMyMjKUl5fn72gAAB9jZQhec+HCBeXn5/s7hiGZmZkNjn/22WcKCgrcfzPExsYqLCzM3zEAoEWjDMFr8vPzNWvWLH/HMCQxMVHBwcH1xktKSgL2M0lSamqq4uLi/B0DAFo0i9vtdvs7REvmdDpls9nkcDgUHh7u7zgtWiCvDG3dulVlZWX1xjt37qykpCQ/JPIOVoYA4OpYGYLXhIWFBewqhMvlUkZGRr3x0aNHB+xnAgA0TuCeDAF4UXx8vCZPnqzOnTurpqZGnTt31uTJk/k1GQCYAGUI+F/x8fFKSkrSnj17lJSURBECAJOgDAEAAFOjDAEAAFOjDAEAAFOjDAEAAFOjDAEAAFOjDAEAAFOjDAEAAFOjDAEAAFPjdhwAgFavtrZWX331lcrKytSpUyf179+/wZszw5woQwCAVm337t1auXKliouLPWNRUVGaM2eOEhMT/ZgMLQVlCADQau3evVuLFi3S8OHDNWzYMJWXl+u6665TaWmpFi1apCVLllCIwDlDAIDWqba2VitXrtTw4cMVEhKiM2fOqLq6WmfOnFFISIiGDx+uVatWqba21t9R4WeUIQBAq/TVV1+puLhYkZGRDT4fGRmpoqIiffXVV82cDC0NZQgA0CqVlZVJksrLyxt8/tL4pXkwL8oQAKBV6tSpkyTpuuuua/D5S+OX5sG8KEMAgFapf//+ioqKUmlpaYPPl5aWKjo6Wv3792/mZGhpKEMAgFYpODhYc+bM0d69e1VdXa3rr79eISEhuv7661VdXa29e/fqkUce4XpD4Kf1AIDWKzExUUuWLNHKlSv1+eefe8ajo6P5WT08KEMAgFYtMTFRw4cP5wrUuCzKEACg1QsODtbAgQP9HQMtFOcMAQAAU6MMAQAAU6MMAQAAU6MMAQAAUwuoMrR7925NmDBBXbt2lcVi0ebNm684f9euXbJYLPW2vLy85gkMAABavID6NVlFRYUGDBigadOm6Z577mn0fkePHlV4eLjncZcuXXwRDwAABKCAKkNjx47V2LFjr3m/yMhIdezYsVFzKysrVVlZ6XnsdDqv+f0AAEDgCKjDZEYNHDhQ0dHRGjVqlHbu3HnFuUuXLpXNZvNsMTExzZQSAAD4Q6suQ9HR0UpNTdXGjRu1adMm9erVS6NGjdLu3bsvu8/ChQvlcDg8W0FBQTMmBgAAzS2gDpNdq169eqlXr16ex0OHDlVBQYGWLVt22fvRWK1WWa3W5ooIAAD8rFWvDDVkyJAh+uabb/wdAwAAtBCmK0OHDx9WdHS0v2MAAIAWIqAOk5WXl+tvf/ub5/Hx48eVk5OjTp06KTY2VgsXLlRhYaHWr18vSVqxYoW6d++uPn36qKqqShs2bNDGjRu1ceNGf30EAADQwgRUGcrOztZtt93meZySkiJJevDBB5WWlqaioiLl5+d7nq+qqtL8+fNVWFiotm3bqk+fPvrkk080bty4Zs8OAABaJovb7Xb7O0RL5nQ6ZbPZ5HA46ly4Ea3TsWPHNGvWLKWmpiouLs7fcQAAzcB05wwBAAD8FGUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYGmUIAACYWkCVod27d2vChAnq2rWrLBaLNm/efNV9srKylJCQoLCwMPXs2VOrV6/2fVAAABAwAqoMVVRUaMCAAXr11VcbNf/48eMaN26cRowYocOHD+uPf/yj5s6dq40bN/o4KQAACBRt/B3gWowdO1Zjx45t9PzVq1crNjZWK1askCT17t1b2dnZWrZsme655x4fpQQAAIEkoFaGrtX+/fuVlJRUZ2zMmDHKzs5WdXV1g/tUVlbK6XTW2QAAQOvVqstQcXGx7HZ7nTG73a6amhqdOnWqwX2WLl0qm83m2WJiYpojKgAA8JNWXYYkyWKx1HnsdrsbHL9k4cKFcjgcnq2goMDnGQEAgP8E1DlD1yoqKkrFxcV1xkpLS9WmTRt17ty5wX2sVqusVmtzxAMAAC1Aq14ZGjp0qDIzM+uMbdu2TYMHD1ZISIifUgEAgJYkoMpQeXm5cnJylJOTI+niT+dzcnKUn58v6eIhrilTpnjmz549WydOnFBKSopyc3O1bt06rV27VvPnz/dHfAAA0AIF1GGy7Oxs3XbbbZ7HKSkpkqQHH3xQaWlpKioq8hQjSerRo4e2bNmi5ORkvfbaa+ratateeeUVflYPAAA8AqoM3XrrrZ4ToBuSlpZWb2zkyJE6dOiQD1MBAIBAFlCHyQAAALyNMgQAAEyt0WWIKzEDAIDWqNFl6Prrr1dpaakk6fbbb9fZs2d9lQkAAKDZNLoMXXfddTp9+rQkadeuXZe9txcAAEAgafSvyUaPHq3bbrtNvXv3liRNmjRJoaGhDc7dsWOHd9IBAAD4WKPL0IYNG5Senq5vv/1WWVlZ6tOnj9q1a+fLbAAAAD7X6DJUXV2t2bNnS7p48cPnn39eHTt29FUuAACAZmHoBOrL3fEdAAAg0DR6ZejSCdSRkZHKysriBGovKykpkcPh8HcM0ztx4kSd/4V/2Ww22e12f8cA0MpZ3Fe6v8VP3HPPPdq7d6969+6trKwsDRs2zBQnUDudTtlsNjkcDoWHh/vkPUpKSnT/A1NUXVXpk9cHAlVIqFUb3lpPIQLgU5xA3QI4HA5VV1XqfM+RcoXZ/B0HaBGCLjikv2fJ4XBQhgD4VKPLUNu2bTmB2sdcYTa52kf4OwYAAKZi6K71O3fu9HYOAAAAv2h0GUpJSdHTTz+t9u3bKyUl5Ypzly9f3uRgAAAAzaHRZejw4cOeX5AdPnz4svP42T0AAAgkjS5DPz00xmEyAADQWjT6oosAAACtUaNXhu6+++5Gv+imTZsMhQEAAGhujV4Zstlsni08PFzbt29Xdna25/mDBw9q+/btstm4Tg4AAAgcjV4ZevPNNz1//sMf/qB//ud/1urVqxUcHCxJqq2t1Zw5c3x2lWYAAABfMHTO0Lp16zR//nxPEZKk4OBgpaSkaN26dV4LBwAA4GuGylBNTY1yc3Prjefm5srlcjU5FAAAQHMxdAXqadOmafr06frb3/6mIUOGSJIOHDig5557TtOmTfNqQAAAAF8yVIaWLVumqKgovfTSSyoqKpIkRUdHa8GCBXrssce8GhAAAMCXDJWhoKAgLViwQAsWLJDT6ZSkBk+c3rt3rwYPHiyr1dq0lAAAAD7S5IsuhoeHX/YXZGPHjlVhYWFT3wIAAMBnfHoFarfb7cuXBwAAaDJuxwEAAEyNMgQAAEyNMgQAAEzNp2XIYrH48uUBAACajBOoAQCAqRkqQ4sXL9aJEyeuOu/cuXPq2bOnkbcAAABoFobK0EcffaQbbrhBo0aN0p/+9CdduHDB27kAAACahaEydPDgQR06dEj9+/dXcnKyoqOj9cgjj+jLL7/0dj4AAACfMnzOUP/+/fXSSy+psLBQ69atU2FhoYYPH65+/frp5ZdflsPh8GZOAAAAn2jyCdQul0tVVVWqrKyU2+1Wp06dtGrVKsXExCgjI8MbGQEAAHzGcBk6ePCgHn30UUVHRys5OVkDBw5Ubm6usrKylJeXp0WLFmnu3LnezAoAAOB1hspQ//79NWTIEB0/flxr165VQUGBnnvuOd14442eOVOmTNHJkye9FhQAAMAX2hjZ6de//rWmT5+uf/iHf7jsnC5dusjlchkOBgAA0BwMlaEnnnjC8+dLF1bkatMAACAQGT5naO3aterbt6/CwsIUFhamvn37as2aNd7MBgAA4HOGytATTzyh3/3ud5owYYLee+89vffee5owYYKSk5P1b//2b97OWMfKlSvVo0cPhYWFKSEhQXv27Lns3F27dslisdTb8vLyfJoRAAAEDkOHyVatWqU33nhDv/3tbz1jEydOVP/+/fWv//qv+vd//3evBfypjIwMzZs3TytXrtTw4cP1+uuva+zYsfr6668VGxt72f2OHj2q8PBwz+MuXbr4JB8AAAg8hlaGamtrNXjw4HrjCQkJqqmpaXKoy1m+fLlmzJihmTNnqnfv3lqxYoViYmK0atWqK+4XGRmpqKgozxYcHOyzjAAAILAYKkP3339/gwUkNTVV9913X5NDNaSqqkoHDx5UUlJSnfGkpCTt27fvivsOHDhQ0dHRGjVqlHbu3HnFuZWVlXI6nXU2AADQehk6TCZdPIF627ZtGjJkiCTpwIEDKigo0JQpU5SSkuKZt3z58qanlHTq1CnV1tbKbrfXGbfb7SouLm5wn+joaKWmpiohIUGVlZV66623NGrUKO3atUuJiYkN7rN06VItWbLEK5kBAEDLZ6gMHTlyRIMGDZIkffvtt5IunofTpUsXHTlyxDPPFz+3/7+v6Xa7L/s+vXr1Uq9evTyPhw4dqoKCAi1btuyyZWjhwoV1ypzT6VRMTIwXkgMAgJbIUBm62qEmX4iIiFBwcHC9VaDS0tJ6q0VXMmTIEG3YsOGyz1utVlmtVsM5AQBAYGnyjVq///57FRYWeiPLFYWGhiohIUGZmZl1xjMzMzVs2LBGv87hw4cVHR3t7XgAACBAGSpDLpdLTz31lGw2m7p166bY2Fh17NhRTz/9tE9vwZGSkqI1a9Zo3bp1ys3NVXJysvLz8zV79mxJFw9xTZkyxTN/xYoV2rx5s7755hv99a9/1cKFC7Vx40Y9+uijPssIAAACi6HDZI8//rjWrl2r5557TsOHD5fb7dbevXu1ePFiXbhwQc8884y3c0qSJk+erNOnT+upp55SUVGR+vbtqy1btqhbt26SpKKiIuXn53vmV1VVaf78+SosLFTbtm3Vp08fffLJJxo3bpxP8gEAgMBjcV+6udg16Nq1q1avXq2JEyfWGf/ggw80Z86cZjls1lycTqdsNpscDkedCzd607FjxzRr1ixV3DxRrvYRPnkPINAEVZxS+68/VGpqquLi4vwdB0ArZugwWVlZmeLj4+uNx8fHq6ysrMmhAAAAmouhMjRgwAC9+uqr9cZfffVVDRgwoMmhAAAAmouhc4ZeeOEFjR8/Xp999pmGDh0qi8Wiffv2qaCgQFu2bPF2RgAAAJ8xtDI0cuRIHTt2TJMmTdLZs2dVVlamu+++W0ePHtWIESO8nREAAMBnrnllqLq6WklJSXr99dd99qsxAACA5nLNK0MhISE6cuSIT261AQAA0NwMHSabMmWK1q5d6+0sAAAAzc7QCdRVVVVas2aNMjMzNXjwYLVv377O8966Uz0AAICvNfmu9ceOHfNqIAAAgOYUMHetBwAA8AVD5wxNnz5d586dqzdeUVGh6dOnNzkUAABAczFUhtLT03X+/Pl64+fPn9f69eubHAoAAKC5XNNhMqfTKbfbLbfbrXPnziksLMzzXG1trbZs2aLIyEivhwQAAPCVaypDHTt2lMVikcViafAu0haLRUuWLPFaOAAAAF+7pjK0c+dOud1u3X777dq4caM6derkeS40NFTdunVT165dvR4SAADAV66pDI0cOVKSdPz4ccXExCgoyNApRwAAAC2GoZ/Wd+vWTWfPntUXX3yh0tJSuVyuOs9PmTLFK+EAAAB8zVAZ+uijj3TfffepoqJCHTp0qHOfMovFQhkCAAABw9Bxrscee8xzraGzZ8/qzJkznq2srMzbGQEAAHzGUBkqLCzU3Llz1a5dO2/nAQAAaFaGytCYMWOUnZ3t7SwAAADNztA5Q+PHj9fvf/97ff311+rXr59CQkLqPD9x4kSvhAMAAPA1Q2XooYcekiQ99dRT9Z6zWCyqra1tWioAALwoLy9Pe/bs0cmTJ9WlSxeNGDFC8fHx/o6FFsLQYTKXy3XZjSIEAGhJ8vLylJGRoR9++EHV1dX64YcflJGRoby8PH9HQwthaGXopy5cuFDnHmUAgLpKSkrkcDj8HaNJKisrVVxc7O8Yhnz11VcNjn/yyScqKCho5jTeExUVJavV6u8YhtlsNtntdn/HkGSwDNXW1urZZ5/V6tWrVVJSomPHjqlnz5564okn1L17d82YMcPbOQEgIJWUlOj+B6aouqrS31FMKzExUcHBwfXGz549q2eeecYPiSBJIaFWbXhrfYsoRIbK0DPPPKP09HS98MILnvOHJKlfv3566aWXKEMA8L8cDoeqqyp1vudIucJs/o5jnKtGQZXl/k5hyLmaIHWs34VUXhus8z0Smz+Ql7is10lBTT7A4xdBFxzS37PkcDgCtwytX79eqampGjVqlGbPnu0Z79+/P8dgAaABrjCbXO0j/B2jSVwd/J3AmGOuCv3cXaqf3CxBbrd0zGVXTUR7/wVDi2H4oos33nhjvXGXy6Xq6uomhwIAwFuKa9vrywuROlNrVY3bojO1Vn15IVLFtRQhXGRoZahPnz7as2ePunXrVmf8vffe08CBA70SDAAAbymuba/i85QfNMxQGVq0aJEeeOABFRYWyuVyadOmTTp69KjWr1+vjz/+2NsZAQAAfMbQYbIJEyYoIyNDW7ZskcVi0ZNPPqnc3Fx99NFHuuOOO7ydEQAAwGcMn4Y+ZswYjRkz5opz3nnnHU2cOFHt27M0CQAAWiZDK0ON9fDDD6ukpMSXbwEAANAkPi1Dbrfbly8PAADQZD4tQwAAAC0dZQgAAJgaZQgAAJgaZQgAAJiaT8tQt27dFBIS4su3AAAAaJIm3e62qqpKpaWlcrlcdcZjY2MlSUeOHGnKywMAAPicoTL0zTffaPr06dq3b1+dcbfbLYvFotraWq+EAwAA8DVDh8mmTp2qoKAgffzxxzp48KAOHTqkQ4cO6fDhwzp06JC3M9axcuVK9ejRQ2FhYUpISNCePXuuOD8rK0sJCQkKCwtTz549tXr1ap/mAwAAgcXQylBOTo4OHjyo+Ph4b+e5ooyMDM2bN08rV67U8OHD9frrr2vs2LH6+uuvPYfmfur48eMaN26cHnroIW3YsEF79+7VnDlz1KVLF91zzz3Nmh0AALRMhlaGbr75Zp06dcrbWa5q+fLlmjFjhmbOnKnevXtrxYoViomJ0apVqxqcv3r1asXGxmrFihXq3bu3Zs6cqenTp2vZsmXNnBwAALRUhsrQ888/rwULFmjXrl06ffq0nE5nnc0XqqqqdPDgQSUlJdUZT0pKqnfu0iX79++vN3/MmDHKzs5WdXV1g/tUVlY2y+cBAAAtg6HDZKNHj5YkjRo1qs64L0+gPnXqlGpra2W32+uM2+12FRcXN7hPcXFxg/Nramp06tQpRUdH19tn6dKlWrJkifeCAwCAFs1QGdq5c6e3czSaxWKp8/hSAbuW+Q2NX7Jw4UKlpKR4HjudTsXExBiNCwAAWjhDZWjkyJHeznFVERERCg4OrrcKVFpaWm/155KoqKgG57dp00adO3ducB+r1Sqr1eqd0AAAoMUzfNHFs2fPau3atcrNzZXFYtHNN9+s6dOny2azeTOfR2hoqBISEpSZmalJkyZ5xjMzM3XnnXc2uM/QoUP10Ucf1Rnbtm2bBg8ezJWxAQCAJIMnUGdnZ+uGG27QSy+9pLKyMp06dUrLly/XDTfc4NPrDKWkpGjNmjVat26dcnNzlZycrPz8fM2ePVvSxUNcU6ZM8cyfPXu2Tpw4oZSUFOXm5mrdunVau3at5s+f77OMAAAgsBhaGUpOTtbEiRP1xhtvqE2biy9RU1OjmTNnat68edq9e7dXQ14yefJknT59Wk899ZSKiorUt29fbdmyRd26dZMkFRUVKT8/3zO/R48e2rJli5KTk/Xaa6+pa9eueuWVV7jGEAAA8DBUhrKzs+sUIUlq06aNFixYoMGDB3stXEPmzJmjOXPmNPhcWlpavbGRI0f6/KrYAAAgcBk6TBYeHl5nBeaSgoICdejQocmhAAAAmouhMjR58mTNmDFDGRkZKigo0Pfff693331XM2fO1G9/+1tvZwQAAPAZQ4fJli1bJovFoilTpqimpkaSFBISokceeUTPPfecVwMCAAD4kqEyFBoaqpdffllLly7Vt99+K7fbrRtvvFHt2rXzdj7A66KCK3RT6Fl1CKrWOVeIvqnqqOLa9v6OBQDwE8PXGZKkdu3aqV+/ft7KAvhcVHCFftG21PP4+uAq/TysVF9eiKQQAYBJGSpDFRUVeu6557R9+3aVlpbK5XLVef7vf/+7V8KZTdD5s/6O0OrFdaqoN2axSDe1OaVS53k/JMLl8H0A0FwMlaGZM2cqKytLDzzwgKKjo694bzA0Xtvjvrk+E/6/DomJkoLrj1uq1P7r7c0fCADgd4bK0KeffqpPPvlEw4cP93YeU7vwD4PkDr3O3zFatXM1QepYvwupvDZY4X0SdYMtSNeFSOXV0rcOl0p+bP6MuMhSVa6wQq4RBsD3DJWh66+/Xp06dfJ2FtOy2WwKCbVK/MXvcwXnImTr27fOaqbb7ZajOF8J/3slc0nqaJUGdQnSkSNHdOrUKX9EhaSQUKvP7ncIAJcYKkNPP/20nnzySaWnp/MLMi+w2+3a8NZ6ORwOf0dpksrKShUXF/s7xlWdPn1ahYWF+vHHH9WuXTv97Gc/U4cOHVReXl5nnsVi0dChQ9W/f38/JW26qKgoWa1Wf8cwzGazyW63+zsGgFbOUBl68cUX9e2338put6t79+717gDP7S+und1uD/i/9I8dO6ZnnnnG3zEMSUxMVHBw/eNnZ8+eDdjPJEmpqamKi4vzdwzA77ikBq7EUBm66667vBwDrUFsbKxSU1P9HcOQrVu3qqysrN643W4P2M8kXfz/BDA7LqmBqzFUhhYtWuTtHGgFwsLCAnYVwuVyKSMjo9746NGjA/YzoWXhUgH+wyU1Wp6W9n1o0kUXq6qqGrzOEP8aRaCJj4/X5MmT9fnnn6u0tFSRkZH65S9/qfj4eH9HQyvBpTP8h0tq4GoMlaFjx45pxowZ2rdvX51xt9sti8Wi2tpar4QDmlN8fDzlBz5zvkeiXG07+juGKZ1zVahjsKv+uDtUFTdP9EMiBJ0/26L+gWCoDE2bNk1t2rTRxx9/zEUXAaARXG07ytU+wt8xTOlYTVv9vE2pfvqfKrdb+qYmQq72nDMEg2UoJydHBw8e5F/RAIAWr7i2vb68EKmbQh3qEFSlc65QfVNl4+RpeBgqQzfffDMXogMABIzi2vYqPk/5QcOCjOz0/PPPa8GCBdq1a5dOnz4tp9NZZwMAAAgUhlaGRo8eLUkaNWpUnXFOoAYAAIHGUBnauXOnt3MAAAD4haEyNHLkSG/nAAAA8IsmXXTxxx9/VH5+vqqqquqMB/KNLQEAgLkYKkMnT57UtGnT9Omnnzb4POcMAQCAQGHo12Tz5s3TmTNndODAAbVt21Z//vOflZ6erptuukkffvihtzMCAAD4jKGVoR07duiDDz7Qz3/+cwUFBalbt2664447FB4erqVLl2r8+PHezgkAAOAThlaGKioqFBkZKUnq1KmTTp48KUnq16+fDh065L10AAAAPmaoDPXq1UtHjx6VJP3jP/6jXn/9dRUWFmr16tWKjo72akAAAABfMnSYbN68eSoqKpIkLVq0SGPGjNHbb7+t0NBQpaWleTMfAACATxkqQ/fdd5/nzwMHDtR3332nvLw8xcbGKiKCuzIDAIDA0aTrDF3Srl07DRo0yBsvBQAA0KwMlaHa2lqlpaVp+/btKi0tlcvlqvP8jh07vBIOAADA1wyVod/97ndKS0vT+PHj1bdvX1ksFm/nAgAAaBaGytC7776r//zP/9S4ceO8nQcAAKBZGfppfWhoqG688UZvZwEAAGh2hsrQY489ppdffllut9vbeQAAAJpVow+T3X333XUe79ixQ59++qn69OmjkJCQOs9t2rTJO+kAAAB8rNFlyGaz1Xk8adIkr4cBAABobo0uQ2+++abnz+fPn5fL5VL79u0lSd999502b96s3r17a8yYMd5PCQAA4COGzhm688479dZbb0mSzp49qyFDhujFF1/UXXfdpVWrVnk1IAAAgC8ZKkOHDh3SiBEjJEn/9V//JbvdrhMnTmj9+vV65ZVXvBoQAICmigqu0Ii2hRrX/juNaFuoqOAKf0dCC2KoDP3444/q0KGDJGnbtm26++67FRQUpCFDhujEiRNeDQgAQFNEBVfoF21LdX1wldpY3Lo+uEo/DyulEMHD0EUXb7zxRm3evFmTJk3S1q1blZycLEkqLS1VeHi4VwMCQGsQdMHh7wimFdepfumxWKSb2pxSqfO8HxKhpX0fDJWhJ598Uvfee6+Sk5M1atQoDR06VNLFVaKBAwd6NeAlZ86c0dy5c/Xhhx9KkiZOnKj/+I//UMeOHS+7z9SpU5Wenl5n7JZbbtGBAwd8khEA/i+bzaaQUKv09yx/RzGtDomJkoLrj1uq1P7r7c0fCJKkkFBrvV+q+4vFbfDKicXFxSoqKtKAAQMUFHTxaNsXX3yh8PBwxcfHezWkJI0dO1bff/+9UlNTJUmzZs1S9+7d9dFHH112n6lTp6qkpKTOL+FCQ0PVqVOnRr+v0+mUzWaTw+Fg1QuAISUlJXI4Wta/hM1k69atKisrqzfeuXNnJSUl+SERpIv/ULDb7f6OIcngypAkRUVFKSoqqs7YL37xiyYHakhubq7+/Oc/68CBA7rlllskSW+88YaGDh2qo0ePqlevXpfd12q11ssJAM3Jbre3mL/0zcjlcikjI6Pe+OjRoxUXF+eHRGhpDJ1A3dz2798vm83mKUKSNGTIENlsNu3bt++K++7atUuRkZGKi4vTQw89pNLS0ivOr6yslNPprLMBAAJXfHy8Jk+erM6dO6umpkadO3fW5MmTfXIUA4EpIMpQcXGxIiMj641HRkaquLj4svuNHTtWb7/9tnbs2KEXX3xRX375pW6//XZVVlZedp+lS5fKZrN5tpiYGK98BgCA/8THxyspKUl79uxRUlISRQh1+LUMLV68WBaL5Ypbdna2JMlisdTb3+12Nzh+yeTJkzV+/Hj17dtXEyZM0Keffqpjx47pk08+uew+CxculMPh8GwFBQVN/6AAAKDFMnzOkDc8+uij+s1vfnPFOd27d9dXX32lkpKSes+dPHnymo7DR0dHq1u3bvrmm28uO8dqtcpqtTb6NQEAQGDzaxmKiIhQRETEVecNHTpUDodDX3zxheck7f/+7/+Ww+HQsGHDGv1+p0+fVkFBgaKjow1nBgAArUtAnDPUu3dv/epXv9JDDz2kAwcO6MCBA3rooYf0T//0T3V+SRYfH6/3339fklReXq758+dr//79+u6777Rr1y5NmDBBERERmjRpkr8+CgAAaGECogxJ0ttvv61+/fopKSlJSUlJ6t+/v+dmsZccPXrUcy2P4OBg/c///I/uvPNOxcXF6cEHH1RcXJz279/vuZUIAACAXw+TXYtOnTppw4YNV5zz0+tHtm3bVlu3bvV1LAAAEOACZmUIAADAFyhDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1ChDAADA1AKmDD3zzDMaNmyY2rVrp44dOzZqH7fbrcWLF6tr165q27atbr31Vv31r3/1bVAAABBQAqYMVVVV6de//rUeeeSRRu/zwgsvaPny5Xr11Vf15ZdfKioqSnfccYfOnTvnw6QAACCQtPF3gMZasmSJJCktLa1R891ut1asWKHHH39cd999tyQpPT1ddrtdf/rTn/Twww83uF9lZaUqKys9j51OZ9OCAwCAFi1gVoau1fHjx1VcXKykpCTPmNVq1ciRI7Vv377L7rd06VLZbDbPFhMT0xxxAQCAn7TaMlRcXCxJstvtdcbtdrvnuYYsXLhQDofDsxUUFPg0JwAA8C+/lqHFixfLYrFcccvOzm7Se1gsljqP3W53vbGfslqtCg8Pr7MBAIDWy6/nDD366KP6zW9+c8U53bt3N/TaUVFRki6uEEVHR3vGS0tL660WAQAA8/JrGYqIiFBERIRPXrtHjx6KiopSZmamBg4cKOniL9KysrL0/PPP++Q9AQBA4AmYc4by8/OVk5Oj/Px81dbWKicnRzk5OSovL/fMiY+P1/vvvy/p4uGxefPm6dlnn9X777+vI0eOaOrUqWrXrp3uvfdef30MAADQwgTMT+uffPJJpaenex5fWu3ZuXOnbr31VknS0aNH5XA4PHMWLFig8+fPa86cOTpz5oxuueUWbdu2TR06dGjW7AAAoOWyuN1ut79DtGROp1M2m00Oh4OTqQEggB07dkyzZs1Samqq4uLi/B0HLUjAHCYDAADwBcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAAAwNcoQAKDVy8vL09atW5WYmKitW7cqLy/P35HQglCGAACtWl5enjIyMlRWVqbg4GCVlZUpIyODQgSPNv4OAABo+S5cuKD8/Hx/xzAkMzOzwfHPPvtMQUGBuyYQGxursLAwf8doFShDAICrys/P16xZs/wdw5DExEQFBwfXGy8pKQnYzyRJqampiouL83eMVsHidrvd/g7RkjmdTtlsNjkcDoWHh/s7DgD4RSCvDG3dulVlZWX1xjt37qykpCQ/JPIOVoa8h5UhAMBVhYWFBewqhMvlUkZGRr3x0aNHB+xngnexMnQVrAwBQODLy8vT559/rtLSUkVGRuqXv/yl4uPj/R0LLQRl6CooQwAAtG6Bexo9AACAF1CGAACAqVGGAACAqVGGAACAqVGGAACAqVGGAACAqVGGAACAqVGGAACAqVGGAACAqVGGAACAqVGGAACAqXHX+qu4dOs2p9Pp5yQAAOBadejQQRaL5YpzKENXce7cOUlSTEyMn5MAAIBr1ZgbrXPX+qtwuVz64YcfGtUsEficTqdiYmJUUFBw1S8PgMDC99ucWBnygqCgIP3sZz/zdww0s/DwcP6yBFopvt/4vziBGgAAmBplCAAAmBplCPgJq9WqRYsWyWq1+jsKAC/j+43L4QRqAABgaqwMAQAAU6MMAQAAU6MMAQAAU6MMAQAAU6MMwXRWrlypHj16KCwsTAkJCdqzZ88V52dlZSkhIUFhYWHq2bOnVq9e3UxJAVyL3bt3a8KECeratassFos2b9581X34fkOiDMFkMjIyNG/ePD3++OM6fPiwRowYobFjxyo/P7/B+cePH9e4ceM0YsQIHT58WH/84x81d+5cbdy4sZmTA7iaiooKDRgwQK+++mqj5vP9xiX8tB6mcsstt2jQoEFatWqVZ6x379666667tHTp0nrz//CHP+jDDz9Ubm6uZ2z27Nn6y1/+ov379zdLZgDXzmKx6P3339ddd9112Tl8v3EJK0MwjaqqKh08eFBJSUl1xpOSkrRv374G99m/f3+9+WPGjFF2draqq6t9lhWA7/H9xiWUIZjGqVOnVFtbK7vdXmfcbreruLi4wX2Ki4sbnF9TU6NTp075LCsA3+P7jUsoQzAdi8VS57Hb7a43drX5DY0DCDx8vyFRhmAiERERCg4OrrcKVFpaWu9fh5dERUU1OL9Nmzbq3Lmzz7IC8D2+37iEMgTTCA0NVUJCgjIzM+uMZ2ZmatiwYQ3uM3To0Hrzt23bpsGDByskJMRnWQH4Ht9vXEIZgqmkpKRozZo1WrdunXJzc5WcnKz8/HzNnj1bkrRw4UJNmTLFM3/27Nk6ceKEUlJSlJubq3Xr1mnt2rWaP3++vz4CgMsoLy9XTk6OcnJyJF386XxOTo7n0hl8v3FZbsBkXnvtNXe3bt3coaGh7kGDBrmzsrI8zz344IPukSNH1pm/a9cu98CBA92hoaHu7t27u1etWtXMiQE0xs6dO92S6m0PPvig2+3m+43L4zpDAADA1DhMBgAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBAAATI0yBKBVmDp1qiwWi+c+cz81Z84cWSwWz5wrbVOnTtVf/vIXWa1Wffjhh3VeZ+PGjQoLC9ORI0ea62MBaAaUIQCtRkxMjN59912dP3/eM3bhwgW98847io2NlSQVFRV5thUrVig8PLzO2Msvv6wBAwboiSee0KxZs3T69GlJUmlpqWbPnq0lS5aob9++fvl8AHyDMgSg1Rg0aJBiY2O1adMmz9imTZsUExOjgQMHSpKioqI8m81mk8ViqTcmXbzDeWxsrP7lX/5FkvTwww/rpptu4o7mQCtEGQLQqkybNk1vvvmm5/G6des0ffr0a36d4OBgpaen64MPPtC9996rrVu3Ki0tTcHBwd6MC6AFoAwBaFUeeOABff755/ruu+904sQJ7d27V/fff7+h1+rdu7fmzZund955R4sXL1ZcXJyX0wJoCShDAFqViIgIjR8/Xunp6XrzzTc1fvx4RUREGHqt8vJyZWRkqF27dtqzZ4+XkwJoKShDAFqd6dOnKy0tTenp6YYOkV3y+9//XqGhodq3b5+2b9+u9evXezElgJaCMgSg1fnVr36lqqoqVVVVacyYMYZeIzMzU2vWrFFaWpoGDBigZ599VvPmzVNRUZGX0wLwN8oQgFYnODhYubm5ys3NNXTCs9Pp1IwZMzR//nwNGTJEkjR37lz16dNHs2bN8nZcAH5GGQLQKoWHhys8PNzQvvPmzZPNZtOSJUs8Y0FBQXrzzTe1Y8cODpcBrYzF7Xa7/R0CAADAX1gZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApkYZAgAApvb/ACaAN/ZNERh1AAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# compare distances\n",
    "df_paired_div = pd.concat([df_paired_alpha, df_paired_beta], axis=1)\n",
    "\n",
    "inter_untreated = df_paired_div[df_paired_div['MTX'] == 0][alpha_metric + '_diff']\n",
    "inter_treated = df_paired_div[df_paired_div['MTX'] == 1][alpha_metric + '_diff']\n",
    "\n",
    "u, p = scipy.stats.mannwhitneyu(inter_untreated, inter_treated)\n",
    "print(u, p)\n",
    "\n",
    "t, p = scipy.stats.ttest_ind(inter_untreated, inter_treated)\n",
    "print(t, p)\n",
    "\n",
    "# f test of variance\n",
    "\n",
    "F = np.var(inter_untreated) / np.var(inter_treated)\n",
    "df1 = len(inter_untreated) - 1\n",
    "df2 = len(inter_treated) - 1\n",
    "\n",
    "p_value = scipy.stats.f.cdf(F, df1, df2)\n",
    "\n",
    "print(F,p_value)\n",
    "\n",
    "#sns.boxplot([inter_untreated, inter_treated])\n",
    "ax = sns.boxplot(data=df_paired_div, x='MTX', y=alpha_metric + '_diff')\n",
    "sns.swarmplot(data=df_paired_div, x='MTX', y=alpha_metric + '_diff', color='grey')\n",
    "df_paired_div['MTX'] = df_paired_div['MTX'].map(lambda x: 'No_Current_MTX' if x == 0 else 'Current_MTX')\n",
    "df_paired_div.to_csv(path + 'outputs/jobs02/alpha_MTX.tsv',sep='\\t')\n",
    "\n",
    "sns.despine()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "##### SJC TJC #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bray_Curtis</th>\n",
       "      <th>MTX</th>\n",
       "      <th>SJC</th>\n",
       "      <th>TJC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T2</th>\n",
       "      <td>0.705551</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3</th>\n",
       "      <td>0.474034</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T4</th>\n",
       "      <td>0.685602</td>\n",
       "      <td>1.0</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T5</th>\n",
       "      <td>0.725669</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T6</th>\n",
       "      <td>0.757568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Bray_Curtis  MTX  SJC  TJC\n",
       "T2     0.705551  1.0    1    1\n",
       "T3     0.474034  1.0    7    7\n",
       "T4     0.685602  1.0   27   25\n",
       "T5     0.725669  0.0    3   21\n",
       "T6     0.757568  0.0    6   13"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add in metadata TJC and SJC\n",
    "samp_to_TJC = {\n",
    "    '501A': 0,\n",
    "    '503A': 2,\n",
    "    '505A': 8,\n",
    "    '507A': 0,\n",
    "    '509A': 0,\n",
    "    '511A': 0,\n",
    "    '513A': 0,\n",
    "    '515A': 0,\n",
    "    '500A': 1, \n",
    "    '502A': 7, \n",
    "    '504A': 25, \n",
    "    '506A': 21, \n",
    "    '508A': 13,\n",
    "    '510A': 0,\n",
    "    '512A': 4,\n",
    "    '514A': 12\n",
    "}\n",
    "\n",
    "samp_to_SJC = {\n",
    "    '501A': 0,\n",
    "    '503A': 0,\n",
    "    '505A': 2,\n",
    "    '507A': 0,\n",
    "    '509A': 0,\n",
    "    '511A': 0,\n",
    "    '513A': 0,\n",
    "    '515A': 0,\n",
    "    '500A': 1, \n",
    "    '502A': 7, \n",
    "    '504A': 27, \n",
    "    '506A': 3, \n",
    "    '508A': 6,\n",
    "    '510A': 0,\n",
    "    '512A': 1,\n",
    "    '514A': 5\n",
    "}\n",
    "\n",
    "# why is 510 so well controlled...? even w/o steroids\n",
    "# Actinomycetales?\n",
    "\n",
    "# correlate SJC/TJC with distance\n",
    "df_meta = df_map.copy()\n",
    "\n",
    "# replace with column dict per above\n",
    "df_meta['TJC'] = df_meta.index.map(samp_to_TJC)\n",
    "df_meta['SJC'] = df_meta.index.map(samp_to_SJC)\n",
    "\n",
    "# create TwinPair to var mapping\n",
    "df = df_meta[df_meta['Diagnosis'] == 'RA'].set_index('TwinPair')\n",
    "TP_to_TJC = df['TJC'].to_dict()\n",
    "TP_to_SJC = df['SJC'].to_dict()\n",
    "\n",
    "# create column in paired beta\n",
    "df_paired_beta['SJC'] = df_paired_beta.index.map(TP_to_SJC)\n",
    "df_paired_beta['TJC'] = df_paired_beta.index.map(TP_to_TJC)\n",
    "\n",
    "df_paired_beta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\nr, p = scipy.stats.pearsonr(x=df_paired_beta['Bray_Curtis'],y=df_paired_beta['SJC'])\\nsns.lmplot(data=df_paired_beta, x='Bray_Curtis', y='SJC')\\nprint(r, p)\\n\\nr, p = scipy.stats.pearsonr(x=df_paired_beta['Bray_Curtis'],y=df_paired_beta['TJC'])\\nsns.lmplot(data=df_paired_beta, x='Bray_Curtis', y='TJC')\\nprint(r, p)\\n\\nr, p = scipy.stats.spearmanr(a=df_paired_beta['Bray_Curtis'],b=df_paired_beta['SJC'])\\nprint(r, p)\\n\\nr, p = scipy.stats.spearmanr(a=df_paired_beta['Bray_Curtis'],b=df_paired_beta['TJC'])\\nprint(r, p)\\n\\n# alpha version of plot\\ndf_paired_div = pd.concat([df_paired_alpha, df_paired_beta], axis=1)\\n\\nr, p = scipy.stats.pearsonr(x=df_paired_div[alpha_metric + '_diff'],y=df_paired_div['TJC'])\\nsns.lmplot(data=df_paired_div, x=alpha_metric + '_diff', y='TJC')\\nprint(r, p)\\n\\nr, p = scipy.stats.pearsonr(x=df_paired_div[alpha_metric + '_diff'],y=df_paired_div['SJC'])\\nsns.lmplot(data=df_paired_div, x=alpha_metric + '_diff', y='SJC')\\nprint(r, p)\\n\\nr, p = scipy.stats.spearmanr(a=df_paired_div[alpha_metric + '_diff'],b=df_paired_div['TJC'])\\nprint(r, p)\\n\\nr, p = scipy.stats.spearmanr(a=df_paired_div[alpha_metric + '_diff'],b=df_paired_div['SJC'])\\nprint(r, p)\\n\""
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# conduct correlation\n",
    "'''\n",
    "r, p = scipy.stats.pearsonr(x=df_paired_beta['Bray_Curtis'],y=df_paired_beta['SJC'])\n",
    "sns.lmplot(data=df_paired_beta, x='Bray_Curtis', y='SJC')\n",
    "print(r, p)\n",
    "\n",
    "r, p = scipy.stats.pearsonr(x=df_paired_beta['Bray_Curtis'],y=df_paired_beta['TJC'])\n",
    "sns.lmplot(data=df_paired_beta, x='Bray_Curtis', y='TJC')\n",
    "print(r, p)\n",
    "\n",
    "r, p = scipy.stats.spearmanr(a=df_paired_beta['Bray_Curtis'],b=df_paired_beta['SJC'])\n",
    "print(r, p)\n",
    "\n",
    "r, p = scipy.stats.spearmanr(a=df_paired_beta['Bray_Curtis'],b=df_paired_beta['TJC'])\n",
    "print(r, p)\n",
    "\n",
    "# alpha version of plot\n",
    "df_paired_div = pd.concat([df_paired_alpha, df_paired_beta], axis=1)\n",
    "\n",
    "r, p = scipy.stats.pearsonr(x=df_paired_div[alpha_metric + '_diff'],y=df_paired_div['TJC'])\n",
    "sns.lmplot(data=df_paired_div, x=alpha_metric + '_diff', y='TJC')\n",
    "print(r, p)\n",
    "\n",
    "r, p = scipy.stats.pearsonr(x=df_paired_div[alpha_metric + '_diff'],y=df_paired_div['SJC'])\n",
    "sns.lmplot(data=df_paired_div, x=alpha_metric + '_diff', y='SJC')\n",
    "print(r, p)\n",
    "\n",
    "r, p = scipy.stats.spearmanr(a=df_paired_div[alpha_metric + '_diff'],b=df_paired_div['TJC'])\n",
    "print(r, p)\n",
    "\n",
    "r, p = scipy.stats.spearmanr(a=df_paired_div[alpha_metric + '_diff'],b=df_paired_div['SJC'])\n",
    "print(r, p)\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get alpha against TJC/SJC\n",
    "df_meta_new = df_meta.copy()\n",
    "df_meta_new.index = df_meta_new.index.map(lambda x: x.split('.')[0])\n",
    "df_meta_new = df_meta_new.drop(['Diagnosis','TwinPair','TJC','SJC','Diagnosis'],axis=1)\n",
    "df = pd.concat([df_meta_new,df_alpha], axis=1)\n",
    "df = df[df['Diagnosis'] == 'RA'].set_index('TwinPair')\n",
    "df = pd.concat([df,df_paired_beta],axis=1)\n",
    "df = df.drop('Diagnosis',axis=1)\n",
    "df.to_csv(path + 'outputs/jobs02/df_beta_meta.tsv',sep='\\t')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TwinPair</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>shannon_entropy</th>\n",
       "      <th>TJC</th>\n",
       "      <th>SJC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>T2</td>\n",
       "      <td>RA</td>\n",
       "      <td>3.860518</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>T2</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>5.193414</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>T3</td>\n",
       "      <td>RA</td>\n",
       "      <td>3.976850</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>T3</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>4.439629</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>T4</td>\n",
       "      <td>RA</td>\n",
       "      <td>5.171083</td>\n",
       "      <td>25</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     TwinPair   Diagnosis  shannon_entropy  TJC  SJC\n",
       "500A       T2          RA         3.860518    1    1\n",
       "501A       T2  Unaffected         5.193414    0    0\n",
       "502A       T3          RA         3.976850    7    7\n",
       "503A       T3  Unaffected         4.439629    2    0\n",
       "504A       T4          RA         5.171083   25   27"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get alpha against TJC/SJC\n",
    "df_meta_new = df_meta.copy()\n",
    "df_meta_new.index = df_meta_new.index.map(lambda x: x.split('.')[0])\n",
    "df_meta_new = df_meta_new.drop(['Diagnosis','TwinPair','TJC','SJC','Diagnosis'],axis=1)\n",
    "df = pd.concat([df_meta_new,df_alpha], axis=1)\n",
    "df['TJC'] = samp_to_TJC\n",
    "df['SJC'] = samp_to_SJC\n",
    "df.to_csv(path + 'inputs/df_alpha_metadata.tsv',sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TwinPair</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>TJC</th>\n",
       "      <th>SJC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>T2</td>\n",
       "      <td>RA</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>T2</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>T3</td>\n",
       "      <td>RA</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>T3</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>T4</td>\n",
       "      <td>RA</td>\n",
       "      <td>25</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          TwinPair   Diagnosis  TJC  SJC\n",
       "#SampleID                               \n",
       "500A            T2          RA    1    1\n",
       "501A            T2  Unaffected    0    0\n",
       "502A            T3          RA    7    7\n",
       "503A            T3  Unaffected    2    0\n",
       "504A            T4          RA   25   27"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df_meta: TwinPair, Diagnosis, TJC and SJC\n",
    "df_meta_edit = df_meta.copy()\n",
    "df_meta_edit.index = df_meta_edit.index.map(lambda x: x.split('.')[0])\n",
    "#df_meta_edit = df_meta_edit.drop(['516A','517A'])\n",
    "df_meta_edit.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "##### Pathway, OLINK, Metabolon, FA #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>metadata_KEGG_Pathways</th>\n",
       "      <th>Metabolism</th>\n",
       "      <th>Genetic Information Processing</th>\n",
       "      <th>Genetic Information Processing|Translation</th>\n",
       "      <th>Genetic Information Processing|Translation|Ribosome</th>\n",
       "      <th>Metabolism|Carbohydrate metabolism</th>\n",
       "      <th>Metabolism|Amino acid metabolism</th>\n",
       "      <th>Metabolism|Metabolism of cofactors and vitamins</th>\n",
       "      <th>Metabolism|Energy metabolism</th>\n",
       "      <th>Environmental Information Processing</th>\n",
       "      <th>Metabolism|Nucleotide metabolism</th>\n",
       "      <th>...</th>\n",
       "      <th>Metabolism|Biosynthesis of other secondary metabolites|Phenazine biosynthesis</th>\n",
       "      <th>Metabolism|Biosynthesis of other secondary metabolites|Tropane, piperidine and pyridine alkaloid biosynthesis</th>\n",
       "      <th>Metabolism|Glycan biosynthesis and metabolism|Various types of N-glycan biosynthesis</th>\n",
       "      <th>Metabolism|Glycan biosynthesis and metabolism|Glycosphingolipid biosynthesis - ganglio series</th>\n",
       "      <th>Cellular Processes|Cell growth and death|Apoptosis - fly</th>\n",
       "      <th>Organismal Systems|Endocrine system|PPAR signaling pathway</th>\n",
       "      <th>Human Diseases|Infectious disease: viral|Coronavirus disease - COVID-19</th>\n",
       "      <th>Metabolism|Xenobiotics biodegradation and metabolism|Chloroalkane and chloroalkene degradation</th>\n",
       "      <th>Metabolism|Metabolism of terpenoids and polyketides|Biosynthesis of vancomycin group antibiotics</th>\n",
       "      <th>Human Diseases|Cancer: overview|Viral carcinogenesis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>0.325986</td>\n",
       "      <td>0.294074</td>\n",
       "      <td>0.235304</td>\n",
       "      <td>0.206306</td>\n",
       "      <td>0.064753</td>\n",
       "      <td>0.065598</td>\n",
       "      <td>0.054545</td>\n",
       "      <td>0.038364</td>\n",
       "      <td>0.032569</td>\n",
       "      <td>0.035376</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000227</td>\n",
       "      <td>0.000225</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.000484</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>0.000100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>0.334549</td>\n",
       "      <td>0.233318</td>\n",
       "      <td>0.178332</td>\n",
       "      <td>0.152325</td>\n",
       "      <td>0.070002</td>\n",
       "      <td>0.062591</td>\n",
       "      <td>0.054963</td>\n",
       "      <td>0.041637</td>\n",
       "      <td>0.038675</td>\n",
       "      <td>0.036591</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.000246</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>0.001629</td>\n",
       "      <td>0.000490</td>\n",
       "      <td>0.000106</td>\n",
       "      <td>0.000222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>0.355141</td>\n",
       "      <td>0.244382</td>\n",
       "      <td>0.186823</td>\n",
       "      <td>0.160917</td>\n",
       "      <td>0.076275</td>\n",
       "      <td>0.064360</td>\n",
       "      <td>0.062004</td>\n",
       "      <td>0.042723</td>\n",
       "      <td>0.029320</td>\n",
       "      <td>0.038488</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000201</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000159</td>\n",
       "      <td>0.000141</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>0.000137</td>\n",
       "      <td>0.000153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>0.336738</td>\n",
       "      <td>0.228318</td>\n",
       "      <td>0.175192</td>\n",
       "      <td>0.151843</td>\n",
       "      <td>0.071781</td>\n",
       "      <td>0.056697</td>\n",
       "      <td>0.060269</td>\n",
       "      <td>0.041416</td>\n",
       "      <td>0.040608</td>\n",
       "      <td>0.036610</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000245</td>\n",
       "      <td>0.000168</td>\n",
       "      <td>0.000157</td>\n",
       "      <td>0.000157</td>\n",
       "      <td>0.000167</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>0.000147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>0.347150</td>\n",
       "      <td>0.239916</td>\n",
       "      <td>0.182507</td>\n",
       "      <td>0.156919</td>\n",
       "      <td>0.073963</td>\n",
       "      <td>0.065695</td>\n",
       "      <td>0.054216</td>\n",
       "      <td>0.041697</td>\n",
       "      <td>0.035611</td>\n",
       "      <td>0.038717</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000118</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>0.000133</td>\n",
       "      <td>0.000119</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 200 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "metadata_KEGG_Pathways  Metabolism  Genetic Information Processing  \\\n",
       "500A                      0.325986                        0.294074   \n",
       "501A                      0.334549                        0.233318   \n",
       "502A                      0.355141                        0.244382   \n",
       "503A                      0.336738                        0.228318   \n",
       "504A                      0.347150                        0.239916   \n",
       "\n",
       "metadata_KEGG_Pathways  Genetic Information Processing|Translation  \\\n",
       "500A                                                      0.235304   \n",
       "501A                                                      0.178332   \n",
       "502A                                                      0.186823   \n",
       "503A                                                      0.175192   \n",
       "504A                                                      0.182507   \n",
       "\n",
       "metadata_KEGG_Pathways  Genetic Information Processing|Translation|Ribosome  \\\n",
       "500A                                                             0.206306     \n",
       "501A                                                             0.152325     \n",
       "502A                                                             0.160917     \n",
       "503A                                                             0.151843     \n",
       "504A                                                             0.156919     \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Carbohydrate metabolism  \\\n",
       "500A                                              0.064753   \n",
       "501A                                              0.070002   \n",
       "502A                                              0.076275   \n",
       "503A                                              0.071781   \n",
       "504A                                              0.073963   \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Amino acid metabolism  \\\n",
       "500A                                            0.065598   \n",
       "501A                                            0.062591   \n",
       "502A                                            0.064360   \n",
       "503A                                            0.056697   \n",
       "504A                                            0.065695   \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Metabolism of cofactors and vitamins  \\\n",
       "500A                                                           0.054545   \n",
       "501A                                                           0.054963   \n",
       "502A                                                           0.062004   \n",
       "503A                                                           0.060269   \n",
       "504A                                                           0.054216   \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Energy metabolism  \\\n",
       "500A                                        0.038364   \n",
       "501A                                        0.041637   \n",
       "502A                                        0.042723   \n",
       "503A                                        0.041416   \n",
       "504A                                        0.041697   \n",
       "\n",
       "metadata_KEGG_Pathways  Environmental Information Processing  \\\n",
       "500A                                                0.032569   \n",
       "501A                                                0.038675   \n",
       "502A                                                0.029320   \n",
       "503A                                                0.040608   \n",
       "504A                                                0.035611   \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Nucleotide metabolism  ...  \\\n",
       "500A                                            0.035376  ...   \n",
       "501A                                            0.036591  ...   \n",
       "502A                                            0.038488  ...   \n",
       "503A                                            0.036610  ...   \n",
       "504A                                            0.038717  ...   \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Biosynthesis of other secondary metabolites|Phenazine biosynthesis  \\\n",
       "500A                                                             0.000227                               \n",
       "501A                                                             0.000154                               \n",
       "502A                                                             0.000267                               \n",
       "503A                                                             0.000245                               \n",
       "504A                                                             0.000161                               \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Biosynthesis of other secondary metabolites|Tropane, piperidine and pyridine alkaloid biosynthesis  \\\n",
       "500A                                                             0.000225                                                               \n",
       "501A                                                             0.000246                                                               \n",
       "502A                                                             0.000201                                                               \n",
       "503A                                                             0.000168                                                               \n",
       "504A                                                             0.000149                                                               \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Glycan biosynthesis and metabolism|Various types of N-glycan biosynthesis  \\\n",
       "500A                                                             0.000101                                      \n",
       "501A                                                             0.000161                                      \n",
       "502A                                                             0.000267                                      \n",
       "503A                                                             0.000157                                      \n",
       "504A                                                             0.000187                                      \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Glycan biosynthesis and metabolism|Glycosphingolipid biosynthesis - ganglio series  \\\n",
       "500A                                                             0.000101                                               \n",
       "501A                                                             0.000161                                               \n",
       "502A                                                             0.000267                                               \n",
       "503A                                                             0.000157                                               \n",
       "504A                                                             0.000187                                               \n",
       "\n",
       "metadata_KEGG_Pathways  Cellular Processes|Cell growth and death|Apoptosis - fly  \\\n",
       "500A                                                             0.000484          \n",
       "501A                                                             0.000088          \n",
       "502A                                                             0.000159          \n",
       "503A                                                             0.000167          \n",
       "504A                                                             0.000118          \n",
       "\n",
       "metadata_KEGG_Pathways  Organismal Systems|Endocrine system|PPAR signaling pathway  \\\n",
       "500A                                                             0.000174            \n",
       "501A                                                             0.000204            \n",
       "502A                                                             0.000141            \n",
       "503A                                                             0.000129            \n",
       "504A                                                             0.000109            \n",
       "\n",
       "metadata_KEGG_Pathways  Human Diseases|Infectious disease: viral|Coronavirus disease - COVID-19  \\\n",
       "500A                                                             0.000000                         \n",
       "501A                                                             0.001629                         \n",
       "502A                                                             0.000042                         \n",
       "503A                                                             0.000000                         \n",
       "504A                                                             0.000000                         \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Xenobiotics biodegradation and metabolism|Chloroalkane and chloroalkene degradation  \\\n",
       "500A                                                             0.000026                                                \n",
       "501A                                                             0.000490                                                \n",
       "502A                                                             0.000043                                                \n",
       "503A                                                             0.000014                                                \n",
       "504A                                                             0.000136                                                \n",
       "\n",
       "metadata_KEGG_Pathways  Metabolism|Metabolism of terpenoids and polyketides|Biosynthesis of vancomycin group antibiotics  \\\n",
       "500A                                                             0.000104                                                  \n",
       "501A                                                             0.000106                                                  \n",
       "502A                                                             0.000137                                                  \n",
       "503A                                                             0.000129                                                  \n",
       "504A                                                             0.000133                                                  \n",
       "\n",
       "metadata_KEGG_Pathways  Human Diseases|Cancer: overview|Viral carcinogenesis  \n",
       "500A                                                             0.000100     \n",
       "501A                                                             0.000222     \n",
       "502A                                                             0.000153     \n",
       "503A                                                             0.000147     \n",
       "504A                                                             0.000119     \n",
       "\n",
       "[5 rows x 200 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pathway data from humann3\n",
    "# df_pathway 8579 cols\n",
    "# df_path = pd.read_csv(path + 'outputs/ensemble_humann3/humann3_joined_pathabundance_no_misses_relabund.tsv', sep='\\t', index_col=0)\n",
    "df_path = pd.read_csv(path + 'outputs/jobs08/KOs_no_misses_relabund_kegg.tsv', sep='\\t', index_col=0)\n",
    "df_path = df_path.T\n",
    "\n",
    "# drop unintegrated, unmapped and normalize\n",
    "for x in list(df_path.columns.values):\n",
    "    if 'UNMAPPED' in x or 'UNINTEGRATED' in x:\n",
    "        df_path = df_path.drop(x, axis=1)\n",
    "\n",
    "# drop \n",
    "drops = ['EMPTY',\t'M',\t'E',\t'T',\t'P',\t'Y']\n",
    "df_path = df_path.drop(drops,axis=1)\n",
    "\n",
    "# normalize; first you obtain row sums (axis=1 specifies that weirdly) and then divide on a row basis\n",
    "# df_path = df_path.div(df_path.sum(axis=1),axis=0)\n",
    "\n",
    "# change special chars\n",
    "df_path.columns = df_path.columns.map(lambda x: x.replace('\\'', 'prime'))\n",
    "df_path.index = df_path.index.map(lambda x: x.split('_')[0])\n",
    "\n",
    "# grab top filt asv by mean rel abundance\n",
    "filt = 200\n",
    "sorted_means = df_path.mean(axis=0).sort_values(ascending=False)\n",
    "top_filt = sorted_means[:filt].index.values\n",
    "df_path_filt = df_path.loc[:,top_filt]\n",
    "# df_path_top100.to_csv(path + 'inputs/df_path_top100.tsv', sep='\\t')\n",
    "df_path_filt.to_csv(path + 'inputs/df_path_top100.tsv', sep='\\t')\n",
    "\n",
    "df_path.to_csv(path + 'inputs/df_path_mp4all.tsv', sep='\\t')\n",
    "\n",
    "# split on groups\n",
    "# create new df for asv\n",
    "df_pathmeta = pd.concat([df_meta_edit, df_path], axis=1) \n",
    "df_pathmeta = df_pathmeta.drop(['TwinPair','SJC','TJC'],axis=1)\n",
    "df_pathmeta.to_csv(path + 'inputs/df_pathmeta.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "# split df asv on disease groups\n",
    "df_path_RA = df_pathmeta[df_pathmeta['Diagnosis'] == 'RA']\n",
    "df_path_RA = df_path_RA.drop('Diagnosis',axis=1)\n",
    "df_path_RA.to_csv(path + 'inputs/df_path_RA.tsv', sep='\\t',index_label='SampleID')\n",
    "df_path_UA = df_pathmeta[df_pathmeta['Diagnosis'] == 'Unaffected']\n",
    "df_path_UA = df_path_UA.drop('Diagnosis',axis=1)\n",
    "df_path_UA.to_csv(path + 'inputs/df_path_UA.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "df_path_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IL8</th>\n",
       "      <th>VEGFA</th>\n",
       "      <th>CD8A</th>\n",
       "      <th>MCP.3</th>\n",
       "      <th>GDNF</th>\n",
       "      <th>CDCP1</th>\n",
       "      <th>CD244</th>\n",
       "      <th>IL7</th>\n",
       "      <th>OPG</th>\n",
       "      <th>LAP.TGF.beta.1</th>\n",
       "      <th>...</th>\n",
       "      <th>CX3CL1</th>\n",
       "      <th>TNFRSF9</th>\n",
       "      <th>NT.3</th>\n",
       "      <th>TWEAK</th>\n",
       "      <th>CCL20</th>\n",
       "      <th>ST1A1</th>\n",
       "      <th>STAMBP</th>\n",
       "      <th>ADA</th>\n",
       "      <th>TNFB</th>\n",
       "      <th>CSF.1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>4.460090</td>\n",
       "      <td>11.747740</td>\n",
       "      <td>10.112573</td>\n",
       "      <td>0.692187</td>\n",
       "      <td>1.39987</td>\n",
       "      <td>3.256127</td>\n",
       "      <td>5.67191</td>\n",
       "      <td>0.716623</td>\n",
       "      <td>9.629157</td>\n",
       "      <td>6.129923</td>\n",
       "      <td>...</td>\n",
       "      <td>4.15028</td>\n",
       "      <td>6.636600</td>\n",
       "      <td>1.995827</td>\n",
       "      <td>9.269563</td>\n",
       "      <td>5.610247</td>\n",
       "      <td>1.866237</td>\n",
       "      <td>5.395570</td>\n",
       "      <td>6.546317</td>\n",
       "      <td>3.851637</td>\n",
       "      <td>9.882993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>5.640333</td>\n",
       "      <td>11.811737</td>\n",
       "      <td>9.727983</td>\n",
       "      <td>1.801717</td>\n",
       "      <td>1.41795</td>\n",
       "      <td>3.843057</td>\n",
       "      <td>5.06545</td>\n",
       "      <td>0.807270</td>\n",
       "      <td>9.408657</td>\n",
       "      <td>6.907237</td>\n",
       "      <td>...</td>\n",
       "      <td>4.50843</td>\n",
       "      <td>6.256277</td>\n",
       "      <td>2.298063</td>\n",
       "      <td>9.381473</td>\n",
       "      <td>7.969883</td>\n",
       "      <td>2.666090</td>\n",
       "      <td>6.828657</td>\n",
       "      <td>6.373057</td>\n",
       "      <td>4.304750</td>\n",
       "      <td>9.600263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505A</th>\n",
       "      <td>5.183700</td>\n",
       "      <td>11.414580</td>\n",
       "      <td>8.847250</td>\n",
       "      <td>0.740210</td>\n",
       "      <td>1.03830</td>\n",
       "      <td>3.077490</td>\n",
       "      <td>5.54233</td>\n",
       "      <td>0.497640</td>\n",
       "      <td>9.908030</td>\n",
       "      <td>6.267330</td>\n",
       "      <td>...</td>\n",
       "      <td>4.46019</td>\n",
       "      <td>6.141750</td>\n",
       "      <td>2.900720</td>\n",
       "      <td>9.146430</td>\n",
       "      <td>5.922580</td>\n",
       "      <td>2.619130</td>\n",
       "      <td>5.220380</td>\n",
       "      <td>6.155300</td>\n",
       "      <td>4.564380</td>\n",
       "      <td>9.629940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>507A</th>\n",
       "      <td>4.888180</td>\n",
       "      <td>11.082720</td>\n",
       "      <td>9.029760</td>\n",
       "      <td>0.787420</td>\n",
       "      <td>0.93138</td>\n",
       "      <td>3.108910</td>\n",
       "      <td>5.12005</td>\n",
       "      <td>0.297860</td>\n",
       "      <td>9.616240</td>\n",
       "      <td>6.038080</td>\n",
       "      <td>...</td>\n",
       "      <td>3.95748</td>\n",
       "      <td>5.397950</td>\n",
       "      <td>0.909560</td>\n",
       "      <td>8.876210</td>\n",
       "      <td>6.204450</td>\n",
       "      <td>2.527400</td>\n",
       "      <td>4.836990</td>\n",
       "      <td>6.085700</td>\n",
       "      <td>3.006930</td>\n",
       "      <td>9.412340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509A</th>\n",
       "      <td>6.051210</td>\n",
       "      <td>12.279060</td>\n",
       "      <td>10.656030</td>\n",
       "      <td>1.387800</td>\n",
       "      <td>1.17672</td>\n",
       "      <td>4.046590</td>\n",
       "      <td>5.35261</td>\n",
       "      <td>2.330700</td>\n",
       "      <td>9.917410</td>\n",
       "      <td>7.123720</td>\n",
       "      <td>...</td>\n",
       "      <td>3.76442</td>\n",
       "      <td>6.645510</td>\n",
       "      <td>1.887630</td>\n",
       "      <td>9.608720</td>\n",
       "      <td>7.496650</td>\n",
       "      <td>4.781400</td>\n",
       "      <td>7.217610</td>\n",
       "      <td>6.351550</td>\n",
       "      <td>4.694640</td>\n",
       "      <td>9.953740</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                IL8      VEGFA       CD8A     MCP.3     GDNF     CDCP1  \\\n",
       "Sample ID                                                                \n",
       "501A       4.460090  11.747740  10.112573  0.692187  1.39987  3.256127   \n",
       "503A       5.640333  11.811737   9.727983  1.801717  1.41795  3.843057   \n",
       "505A       5.183700  11.414580   8.847250  0.740210  1.03830  3.077490   \n",
       "507A       4.888180  11.082720   9.029760  0.787420  0.93138  3.108910   \n",
       "509A       6.051210  12.279060  10.656030  1.387800  1.17672  4.046590   \n",
       "\n",
       "             CD244       IL7       OPG  LAP.TGF.beta.1  ...   CX3CL1  \\\n",
       "Sample ID                                               ...            \n",
       "501A       5.67191  0.716623  9.629157        6.129923  ...  4.15028   \n",
       "503A       5.06545  0.807270  9.408657        6.907237  ...  4.50843   \n",
       "505A       5.54233  0.497640  9.908030        6.267330  ...  4.46019   \n",
       "507A       5.12005  0.297860  9.616240        6.038080  ...  3.95748   \n",
       "509A       5.35261  2.330700  9.917410        7.123720  ...  3.76442   \n",
       "\n",
       "            TNFRSF9      NT.3     TWEAK     CCL20     ST1A1    STAMBP  \\\n",
       "Sample ID                                                               \n",
       "501A       6.636600  1.995827  9.269563  5.610247  1.866237  5.395570   \n",
       "503A       6.256277  2.298063  9.381473  7.969883  2.666090  6.828657   \n",
       "505A       6.141750  2.900720  9.146430  5.922580  2.619130  5.220380   \n",
       "507A       5.397950  0.909560  8.876210  6.204450  2.527400  4.836990   \n",
       "509A       6.645510  1.887630  9.608720  7.496650  4.781400  7.217610   \n",
       "\n",
       "                ADA      TNFB     CSF.1  \n",
       "Sample ID                                \n",
       "501A       6.546317  3.851637  9.882993  \n",
       "503A       6.373057  4.304750  9.600263  \n",
       "505A       6.155300  4.564380  9.629940  \n",
       "507A       6.085700  3.006930  9.412340  \n",
       "509A       6.351550  4.694640  9.953740  \n",
       "\n",
       "[5 rows x 86 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# OLINK cytokine data, 92 columns\n",
    "# drop OG rows 2-10 (1 indexed) and 20, the avg of 501 and 503 are used b/c they have repeat measures\n",
    "df_olink = pd.read_csv(path + 'inputs/RA_twins/Normalized Olink TWINRA_RBB/Norm_Olink_TWINRA_KB.csv', index_col='Sample ID')\n",
    "df_olink = df_olink.drop(['mtx Responder?','Twin pair','Timepoint','Group','Project','Dx', 'barcode'], axis=1)\n",
    "df_olink = df_olink.dropna()\n",
    "df_olink = df_olink.iloc[8:,:]\n",
    "\n",
    "# for cleaning purposes and consistency:\n",
    "# n this step, I removed columns (biomarkers) with 6 or more negative values. \n",
    "# There were 92 biomakers before the data cleaning, and I kept 86 of them. \n",
    "# get sum of number of negative values per column, hence axis = 0 which is default\n",
    "df_olink = df_olink.loc[:, ((df_olink < 0).sum(axis=0) <= 5)]\n",
    "\n",
    "# Also, I replaced all negative values in those 86 biomarkers with 0.\n",
    "df_olink[df_olink < 0] = df_olink\n",
    "\n",
    "# make index match other data\n",
    "df_olink.index = df_olink.index.map(lambda x: x.split('.')[1] + 'A')\n",
    "\n",
    "# drop 516, 517\n",
    "df_olink = df_olink.drop(['516A','517A'])\n",
    "\n",
    "# export for R\n",
    "df_olink_R = df_map.copy()\n",
    "df_olink_R['Subject'] = df_olink_R.index.map(lambda x: x.split('A')[0])\n",
    "df_olink_R = pd.concat([df_olink_R, df_olink], axis=1)\n",
    "df_olink_R['Diagnosis'] = df_olink_R['Diagnosis'].replace('Unaffected','1_Unaffected')\n",
    "df_olink_R['Diagnosis'] = df_olink_R['Diagnosis'].replace('RA','2_RA')\n",
    "df_olink_R = df_olink_R.rename(columns={'TwinPair':'Sibling_pair'})\n",
    "df_olink_R['Sibling_pair'] = df_olink_R['Sibling_pair'].map(lambda x: 'sib_0' + str(int(x.split('T')[1])-1))\n",
    "df_olink_R.to_csv(path + 'inputs/df_olink_R.tsv', sep='\\t', index=True)\n",
    "\n",
    "df_olink.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IL8</th>\n",
       "      <th>VEGFA</th>\n",
       "      <th>CD8A</th>\n",
       "      <th>MCP.3</th>\n",
       "      <th>GDNF</th>\n",
       "      <th>CDCP1</th>\n",
       "      <th>CD244</th>\n",
       "      <th>IL7</th>\n",
       "      <th>OPG</th>\n",
       "      <th>LAP.TGF.beta.1</th>\n",
       "      <th>...</th>\n",
       "      <th>CX3CL1</th>\n",
       "      <th>TNFRSF9</th>\n",
       "      <th>NT.3</th>\n",
       "      <th>TWEAK</th>\n",
       "      <th>CCL20</th>\n",
       "      <th>ST1A1</th>\n",
       "      <th>STAMBP</th>\n",
       "      <th>ADA</th>\n",
       "      <th>TNFB</th>\n",
       "      <th>CSF.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>4.830700</td>\n",
       "      <td>11.524570</td>\n",
       "      <td>9.998650</td>\n",
       "      <td>1.168630</td>\n",
       "      <td>1.14834</td>\n",
       "      <td>3.751960</td>\n",
       "      <td>5.58873</td>\n",
       "      <td>0.842700</td>\n",
       "      <td>9.718790</td>\n",
       "      <td>6.149300</td>\n",
       "      <td>...</td>\n",
       "      <td>4.08211</td>\n",
       "      <td>6.272690</td>\n",
       "      <td>1.138920</td>\n",
       "      <td>8.969190</td>\n",
       "      <td>6.591220</td>\n",
       "      <td>1.945380</td>\n",
       "      <td>4.995270</td>\n",
       "      <td>6.294770</td>\n",
       "      <td>3.681940</td>\n",
       "      <td>9.760340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>4.460090</td>\n",
       "      <td>11.747740</td>\n",
       "      <td>10.112573</td>\n",
       "      <td>0.692187</td>\n",
       "      <td>1.39987</td>\n",
       "      <td>3.256127</td>\n",
       "      <td>5.67191</td>\n",
       "      <td>0.716623</td>\n",
       "      <td>9.629157</td>\n",
       "      <td>6.129923</td>\n",
       "      <td>...</td>\n",
       "      <td>4.15028</td>\n",
       "      <td>6.636600</td>\n",
       "      <td>1.995827</td>\n",
       "      <td>9.269563</td>\n",
       "      <td>5.610247</td>\n",
       "      <td>1.866237</td>\n",
       "      <td>5.395570</td>\n",
       "      <td>6.546317</td>\n",
       "      <td>3.851637</td>\n",
       "      <td>9.882993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>4.750650</td>\n",
       "      <td>11.357440</td>\n",
       "      <td>9.202560</td>\n",
       "      <td>1.433600</td>\n",
       "      <td>1.48875</td>\n",
       "      <td>3.593450</td>\n",
       "      <td>5.01703</td>\n",
       "      <td>0.623210</td>\n",
       "      <td>9.351790</td>\n",
       "      <td>6.390890</td>\n",
       "      <td>...</td>\n",
       "      <td>4.48619</td>\n",
       "      <td>5.507010</td>\n",
       "      <td>1.715140</td>\n",
       "      <td>9.430550</td>\n",
       "      <td>5.855660</td>\n",
       "      <td>2.458220</td>\n",
       "      <td>5.611470</td>\n",
       "      <td>5.851970</td>\n",
       "      <td>4.351010</td>\n",
       "      <td>9.531600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>5.640333</td>\n",
       "      <td>11.811737</td>\n",
       "      <td>9.727983</td>\n",
       "      <td>1.801717</td>\n",
       "      <td>1.41795</td>\n",
       "      <td>3.843057</td>\n",
       "      <td>5.06545</td>\n",
       "      <td>0.807270</td>\n",
       "      <td>9.408657</td>\n",
       "      <td>6.907237</td>\n",
       "      <td>...</td>\n",
       "      <td>4.50843</td>\n",
       "      <td>6.256277</td>\n",
       "      <td>2.298063</td>\n",
       "      <td>9.381473</td>\n",
       "      <td>7.969883</td>\n",
       "      <td>2.666090</td>\n",
       "      <td>6.828657</td>\n",
       "      <td>6.373057</td>\n",
       "      <td>4.304750</td>\n",
       "      <td>9.600263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>6.671700</td>\n",
       "      <td>12.199100</td>\n",
       "      <td>9.201960</td>\n",
       "      <td>1.090900</td>\n",
       "      <td>1.68983</td>\n",
       "      <td>3.581590</td>\n",
       "      <td>5.51924</td>\n",
       "      <td>1.036510</td>\n",
       "      <td>10.235030</td>\n",
       "      <td>6.442820</td>\n",
       "      <td>...</td>\n",
       "      <td>4.24226</td>\n",
       "      <td>5.280160</td>\n",
       "      <td>2.191120</td>\n",
       "      <td>9.262260</td>\n",
       "      <td>6.745420</td>\n",
       "      <td>4.357420</td>\n",
       "      <td>6.330690</td>\n",
       "      <td>6.868090</td>\n",
       "      <td>4.360540</td>\n",
       "      <td>10.076210</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           IL8      VEGFA       CD8A     MCP.3     GDNF     CDCP1    CD244  \\\n",
       "500A  4.830700  11.524570   9.998650  1.168630  1.14834  3.751960  5.58873   \n",
       "501A  4.460090  11.747740  10.112573  0.692187  1.39987  3.256127  5.67191   \n",
       "502A  4.750650  11.357440   9.202560  1.433600  1.48875  3.593450  5.01703   \n",
       "503A  5.640333  11.811737   9.727983  1.801717  1.41795  3.843057  5.06545   \n",
       "504A  6.671700  12.199100   9.201960  1.090900  1.68983  3.581590  5.51924   \n",
       "\n",
       "           IL7        OPG  LAP.TGF.beta.1  ...   CX3CL1   TNFRSF9      NT.3  \\\n",
       "500A  0.842700   9.718790        6.149300  ...  4.08211  6.272690  1.138920   \n",
       "501A  0.716623   9.629157        6.129923  ...  4.15028  6.636600  1.995827   \n",
       "502A  0.623210   9.351790        6.390890  ...  4.48619  5.507010  1.715140   \n",
       "503A  0.807270   9.408657        6.907237  ...  4.50843  6.256277  2.298063   \n",
       "504A  1.036510  10.235030        6.442820  ...  4.24226  5.280160  2.191120   \n",
       "\n",
       "         TWEAK     CCL20     ST1A1    STAMBP       ADA      TNFB      CSF.1  \n",
       "500A  8.969190  6.591220  1.945380  4.995270  6.294770  3.681940   9.760340  \n",
       "501A  9.269563  5.610247  1.866237  5.395570  6.546317  3.851637   9.882993  \n",
       "502A  9.430550  5.855660  2.458220  5.611470  5.851970  4.351010   9.531600  \n",
       "503A  9.381473  7.969883  2.666090  6.828657  6.373057  4.304750   9.600263  \n",
       "504A  9.262260  6.745420  4.357420  6.330690  6.868090  4.360540  10.076210  \n",
       "\n",
       "[5 rows x 86 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# OLINK\n",
    "df_olink = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/12_olink/olink.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "# drop samples\n",
    "# df_mb = df_mb.drop(['516A','517A'])\n",
    "\n",
    "# drop cols\n",
    "df_olink = df_olink.drop(['Subject','Sibling_pair','Diagnosis'], axis=1)\n",
    "\n",
    "df_olink.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2-Methylbutyric acid</th>\n",
       "      <th>Acetic acid</th>\n",
       "      <th>Butyric acid</th>\n",
       "      <th>Hexanoic acid</th>\n",
       "      <th>Isobutyric acid</th>\n",
       "      <th>Isovaleric acid</th>\n",
       "      <th>Propionic acid</th>\n",
       "      <th>Valeric acid</th>\n",
       "      <th>Tot SCFA</th>\n",
       "      <th>Tot butyric acid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>215</td>\n",
       "      <td>1870</td>\n",
       "      <td>94.7</td>\n",
       "      <td>86.4</td>\n",
       "      <td>86.0</td>\n",
       "      <td>231</td>\n",
       "      <td>151</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2752.0</td>\n",
       "      <td>395.936284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>93.6</td>\n",
       "      <td>2708</td>\n",
       "      <td>16.4</td>\n",
       "      <td>83.8</td>\n",
       "      <td>44.9</td>\n",
       "      <td>50.5</td>\n",
       "      <td>94.2</td>\n",
       "      <td>10.7</td>\n",
       "      <td>3102.0</td>\n",
       "      <td>154.936779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505A</th>\n",
       "      <td>68.3</td>\n",
       "      <td>2217</td>\n",
       "      <td>0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>35.3</td>\n",
       "      <td>44.9</td>\n",
       "      <td>101</td>\n",
       "      <td>7.51</td>\n",
       "      <td>2530.0</td>\n",
       "      <td>103.613311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>507A</th>\n",
       "      <td>66.9</td>\n",
       "      <td>1495</td>\n",
       "      <td>4.88</td>\n",
       "      <td>75.7</td>\n",
       "      <td>43.6</td>\n",
       "      <td>40.9</td>\n",
       "      <td>80.3</td>\n",
       "      <td>5.32</td>\n",
       "      <td>1813.0</td>\n",
       "      <td>115.366995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509A</th>\n",
       "      <td>47.9</td>\n",
       "      <td>887</td>\n",
       "      <td>48.1</td>\n",
       "      <td>93.9</td>\n",
       "      <td>24.6</td>\n",
       "      <td>38.6</td>\n",
       "      <td>86.4</td>\n",
       "      <td>18.7</td>\n",
       "      <td>1245.0</td>\n",
       "      <td>120.617455</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         2-Methylbutyric acid Acetic acid Butyric acid Hexanoic acid  \\\n",
       "SampleID                                                               \n",
       "501A                      215        1870         94.7          86.4   \n",
       "503A                     93.6        2708         16.4          83.8   \n",
       "505A                     68.3        2217            0          56.0   \n",
       "507A                     66.9        1495         4.88          75.7   \n",
       "509A                     47.9         887         48.1          93.9   \n",
       "\n",
       "         Isobutyric acid Isovaleric acid Propionic acid Valeric acid  \\\n",
       "SampleID                                                               \n",
       "501A                86.0             231            151         18.0   \n",
       "503A                44.9            50.5           94.2         10.7   \n",
       "505A                35.3            44.9            101         7.51   \n",
       "507A                43.6            40.9           80.3         5.32   \n",
       "509A                24.6            38.6           86.4         18.7   \n",
       "\n",
       "          Tot SCFA  Tot butyric acid  \n",
       "SampleID                              \n",
       "501A        2752.0        395.936284  \n",
       "503A        3102.0        154.936779  \n",
       "505A        2530.0        103.613311  \n",
       "507A        1813.0        115.366995  \n",
       "509A        1245.0        120.617455  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# metabolon\n",
    "df_mb = pd.read_csv(path + 'inputs/Metabolon_SCFA_TWIN RA copy_KB_Sheet2.csv', skiprows=2)\n",
    "df_mb = df_mb.iloc[:-1,:-3] # drop unnamed trailing col and trailing row\n",
    "df_mb = df_mb.rename(columns={'Unnamed: 0':'Diagnosis', 'Unnamed: 1': 'SampleID'})#'Sibling_pair'})\n",
    "df_mb['SampleID'] = df_mb['SampleID'].astype(str)\n",
    "df_mb['SampleID'] = df_mb['SampleID'].map(lambda x: x.split('twinRA.')[1] + 'A')\n",
    "df_mb = df_mb.set_index('SampleID')\n",
    "df_mb = df_mb.drop('Diagnosis',axis=1)\n",
    "df_mb = df_mb.drop(['516A','517A'])\n",
    "\n",
    "# replace <0 with 0\n",
    "df_mb = df_mb.replace('< 0',0)\n",
    "\n",
    "# convert for R\n",
    "df_mb_R = pd.concat([df_map, df_mb],axis=1)\n",
    "df_mb_R['Diagnosis'] = df_mb_R['Diagnosis'].replace('Unaffected','1_Unaffected')\n",
    "df_mb_R['Diagnosis'] = df_mb_R['Diagnosis'].replace('RA','2_RA')\n",
    "df_mb_R = df_mb_R.rename(columns={'TwinPair':'Sibling_pair'})\n",
    "df_mb_R['Sibling_pair'] = df_mb_R['Sibling_pair'].map(lambda x: 'sib_0' + str(int(str(x).split('T')[1])-1))\n",
    "df_mb_R.to_csv(path + 'inputs/df_mb_R.tsv', sep='\\t', index=True)\n",
    "\n",
    "df_mb.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2-Methylbutyric acid</th>\n",
       "      <th>Acetic acid</th>\n",
       "      <th>Butyric acid</th>\n",
       "      <th>Hexanoic acid</th>\n",
       "      <th>Isobutyric acid</th>\n",
       "      <th>Isovaleric acid</th>\n",
       "      <th>Propionic acid</th>\n",
       "      <th>Valeric acid</th>\n",
       "      <th>Tot SCFA</th>\n",
       "      <th>Tot butyric acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>110.0</td>\n",
       "      <td>1491</td>\n",
       "      <td>26.5</td>\n",
       "      <td>83.9</td>\n",
       "      <td>43.9</td>\n",
       "      <td>133.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>45.00</td>\n",
       "      <td>2147.0</td>\n",
       "      <td>180.332978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>215.0</td>\n",
       "      <td>1870</td>\n",
       "      <td>94.7</td>\n",
       "      <td>86.4</td>\n",
       "      <td>86.0</td>\n",
       "      <td>231.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>18.00</td>\n",
       "      <td>2752.0</td>\n",
       "      <td>395.936284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>79.8</td>\n",
       "      <td>3380</td>\n",
       "      <td>62.2</td>\n",
       "      <td>94.7</td>\n",
       "      <td>35.2</td>\n",
       "      <td>48.7</td>\n",
       "      <td>152.0</td>\n",
       "      <td>22.30</td>\n",
       "      <td>3875.0</td>\n",
       "      <td>177.142365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>93.6</td>\n",
       "      <td>2708</td>\n",
       "      <td>16.4</td>\n",
       "      <td>83.8</td>\n",
       "      <td>44.9</td>\n",
       "      <td>50.5</td>\n",
       "      <td>94.2</td>\n",
       "      <td>10.70</td>\n",
       "      <td>3102.0</td>\n",
       "      <td>154.936779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>60.2</td>\n",
       "      <td>497</td>\n",
       "      <td>0.0</td>\n",
       "      <td>51.7</td>\n",
       "      <td>31.3</td>\n",
       "      <td>50.7</td>\n",
       "      <td>32.3</td>\n",
       "      <td>4.82</td>\n",
       "      <td>728.0</td>\n",
       "      <td>91.570762</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      2-Methylbutyric acid  Acetic acid  Butyric acid  Hexanoic acid  \\\n",
       "500A                 110.0         1491          26.5           83.9   \n",
       "501A                 215.0         1870          94.7           86.4   \n",
       "502A                  79.8         3380          62.2           94.7   \n",
       "503A                  93.6         2708          16.4           83.8   \n",
       "504A                  60.2          497           0.0           51.7   \n",
       "\n",
       "      Isobutyric acid  Isovaleric acid  Propionic acid  Valeric acid  \\\n",
       "500A             43.9            133.0           214.0         45.00   \n",
       "501A             86.0            231.0           151.0         18.00   \n",
       "502A             35.2             48.7           152.0         22.30   \n",
       "503A             44.9             50.5            94.2         10.70   \n",
       "504A             31.3             50.7            32.3          4.82   \n",
       "\n",
       "      Tot SCFA  Tot butyric acid  \n",
       "500A    2147.0        180.332978  \n",
       "501A    2752.0        395.936284  \n",
       "502A    3875.0        177.142365  \n",
       "503A    3102.0        154.936779  \n",
       "504A     728.0         91.570762  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# metabolon\n",
    "df_mb = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/13_metabolon/metabolon.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "# drop samples\n",
    "# df_mb = df_mb.drop(['516A','517A'])\n",
    "\n",
    "# drop cols\n",
    "df_mb = df_mb.drop(['Sibling_pair','Diagnosis'], axis=1)\n",
    "\n",
    "df_mb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Acetate</th>\n",
       "      <th>Propionate</th>\n",
       "      <th>Butyrate</th>\n",
       "      <th>Isovalerate</th>\n",
       "      <th>Valerate</th>\n",
       "      <th>Hexanoate</th>\n",
       "      <th>Heptanoate</th>\n",
       "      <th>Octanoate</th>\n",
       "      <th>Nonanoic_acid</th>\n",
       "      <th>Decanoic_acid</th>\n",
       "      <th>Dodecanoic_acid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>63.203350</td>\n",
       "      <td>0.018168</td>\n",
       "      <td>0.123664</td>\n",
       "      <td>0.071874</td>\n",
       "      <td>0.258681</td>\n",
       "      <td>0.067507</td>\n",
       "      <td>0.000151</td>\n",
       "      <td>0.009360</td>\n",
       "      <td>0.003035</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>18.961611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>144.772864</td>\n",
       "      <td>0.024557</td>\n",
       "      <td>0.257001</td>\n",
       "      <td>0.023236</td>\n",
       "      <td>0.018034</td>\n",
       "      <td>0.814973</td>\n",
       "      <td>0.012756</td>\n",
       "      <td>0.013322</td>\n",
       "      <td>0.003790</td>\n",
       "      <td>0.000128</td>\n",
       "      <td>23.364729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>24.348469</td>\n",
       "      <td>0.007449</td>\n",
       "      <td>0.103740</td>\n",
       "      <td>0.043493</td>\n",
       "      <td>0.186273</td>\n",
       "      <td>0.159585</td>\n",
       "      <td>0.000869</td>\n",
       "      <td>0.015359</td>\n",
       "      <td>0.002884</td>\n",
       "      <td>0.000077</td>\n",
       "      <td>6.222841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>72.685492</td>\n",
       "      <td>0.027111</td>\n",
       "      <td>0.183125</td>\n",
       "      <td>0.014804</td>\n",
       "      <td>0.036577</td>\n",
       "      <td>0.014324</td>\n",
       "      <td>0.000465</td>\n",
       "      <td>0.070759</td>\n",
       "      <td>0.004436</td>\n",
       "      <td>0.000069</td>\n",
       "      <td>11.350637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>27.178033</td>\n",
       "      <td>0.002476</td>\n",
       "      <td>0.022492</td>\n",
       "      <td>0.086892</td>\n",
       "      <td>0.271040</td>\n",
       "      <td>0.079170</td>\n",
       "      <td>0.010737</td>\n",
       "      <td>0.184629</td>\n",
       "      <td>0.012075</td>\n",
       "      <td>0.000072</td>\n",
       "      <td>26.324529</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Acetate   Propionate   Butyrate   Isovalerate   Valerate   \\\n",
       "SampleID                                                                \n",
       "500A       63.203350     0.018168   0.123664      0.071874   0.258681   \n",
       "501A      144.772864     0.024557   0.257001      0.023236   0.018034   \n",
       "502A       24.348469     0.007449   0.103740      0.043493   0.186273   \n",
       "503A       72.685492     0.027111   0.183125      0.014804   0.036577   \n",
       "504A       27.178033     0.002476   0.022492      0.086892   0.271040   \n",
       "\n",
       "          Hexanoate   Heptanoate   Octanoate   Nonanoic_acid  Decanoic_acid  \\\n",
       "SampleID                                                                      \n",
       "500A        0.067507     0.000151    0.009360       0.003035       0.000109   \n",
       "501A        0.814973     0.012756    0.013322       0.003790       0.000128   \n",
       "502A        0.159585     0.000869    0.015359       0.002884       0.000077   \n",
       "503A        0.014324     0.000465    0.070759       0.004436       0.000069   \n",
       "504A        0.079170     0.010737    0.184629       0.012075       0.000072   \n",
       "\n",
       "          Dodecanoic_acid  \n",
       "SampleID                   \n",
       "500A            18.961611  \n",
       "501A            23.364729  \n",
       "502A             6.222841  \n",
       "503A            11.350637  \n",
       "504A            26.324529  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# metabolites, fatty acids; 11 collumns\n",
    "df_fa = pd.read_csv(path + 'inputs/RA_twins/Metabolites/metabolites_KB.csv', index_col=0)\n",
    "df_fa = df_fa.drop(['Subject','Sibling_pair'], axis=1)\n",
    "df_fa = df_fa.drop(['182A','183A', 'Missing 507A', '517A','516A'])\n",
    "df_fa = df_fa.dropna()\n",
    "df_fa.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "##### ACPAs #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biglycan 247-266 sm-1 cyclic _IgG_Fecal</th>\n",
       "      <th>Clusterin 231-250 sm-1 cyclic _IgG_Fecal</th>\n",
       "      <th>Mix B RBB _IgA_Fecal</th>\n",
       "      <th>ApoE 277-296 Cit2 sm2 cyclic _IgG_Fecal</th>\n",
       "      <th>H2B/a 62-81 cyclic _IgG_Fecal</th>\n",
       "      <th>ApoA1 Cit _IgG_Fecal</th>\n",
       "      <th>Mix C RBB _IgA_Fecal</th>\n",
       "      <th>FibrinogenB 36-52 cit _IgG_Fecal</th>\n",
       "      <th>FibrinogenB 246-267 cit _IgG_Fecal</th>\n",
       "      <th>Vimentin 58-77 cit3 cyclic small-1 _IgG_Fecal</th>\n",
       "      <th>...</th>\n",
       "      <th>Vimentin _IgG_Fecal</th>\n",
       "      <th>Fibrinogen _IgG_Fecal</th>\n",
       "      <th>Fibrinogen cit _IgG_Fecal</th>\n",
       "      <th>Filaggrin 48-65 arg2 v1 cyclic _IgG_Fecal</th>\n",
       "      <th>Fibronectin cit _IgA_Fecal</th>\n",
       "      <th>FibrinogenA 556-575 sm cyclic _IgG_Fecal</th>\n",
       "      <th>Histones2B cit _IgG_Fecal</th>\n",
       "      <th>Fibronectin cit 1029-1042 cit2 _IgG_Fecal</th>\n",
       "      <th>FibrinogenA 211-230 cit small cyclic _IgG_Fecal</th>\n",
       "      <th>Mix B RBB _IgG_Fecal</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>16</td>\n",
       "      <td>18</td>\n",
       "      <td>48</td>\n",
       "      <td>23</td>\n",
       "      <td>21</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>136</td>\n",
       "      <td>109</td>\n",
       "      <td>17</td>\n",
       "      <td>131</td>\n",
       "      <td>18</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>48</td>\n",
       "      <td>45</td>\n",
       "      <td>16</td>\n",
       "      <td>19</td>\n",
       "      <td>48</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>143</td>\n",
       "      <td>113</td>\n",
       "      <td>15</td>\n",
       "      <td>132</td>\n",
       "      <td>17</td>\n",
       "      <td>37</td>\n",
       "      <td>16</td>\n",
       "      <td>30</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>18</td>\n",
       "      <td>16</td>\n",
       "      <td>50</td>\n",
       "      <td>46</td>\n",
       "      <td>15</td>\n",
       "      <td>19</td>\n",
       "      <td>45</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>26</td>\n",
       "      <td>130</td>\n",
       "      <td>109</td>\n",
       "      <td>16</td>\n",
       "      <td>135</td>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>33</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>506A</th>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>48</td>\n",
       "      <td>23</td>\n",
       "      <td>19</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>27</td>\n",
       "      <td>129</td>\n",
       "      <td>109</td>\n",
       "      <td>15</td>\n",
       "      <td>138</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>32</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>508A</th>\n",
       "      <td>33</td>\n",
       "      <td>22</td>\n",
       "      <td>78</td>\n",
       "      <td>50</td>\n",
       "      <td>18</td>\n",
       "      <td>21</td>\n",
       "      <td>62</td>\n",
       "      <td>23</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>130</td>\n",
       "      <td>108</td>\n",
       "      <td>20</td>\n",
       "      <td>141</td>\n",
       "      <td>22</td>\n",
       "      <td>20</td>\n",
       "      <td>17</td>\n",
       "      <td>33</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 116 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Biglycan 247-266 sm-1 cyclic _IgG_Fecal  \\\n",
       "500A                                      18   \n",
       "502A                                      20   \n",
       "504A                                      18   \n",
       "506A                                      19   \n",
       "508A                                      33   \n",
       "\n",
       "     Clusterin 231-250 sm-1 cyclic _IgG_Fecal Mix B RBB _IgA_Fecal  \\\n",
       "500A                                       15                   47   \n",
       "502A                                       15                   48   \n",
       "504A                                       16                   50   \n",
       "506A                                       15                   46   \n",
       "508A                                       22                   78   \n",
       "\n",
       "     ApoE 277-296 Cit2 sm2 cyclic _IgG_Fecal H2B/a 62-81 cyclic _IgG_Fecal  \\\n",
       "500A                                      47                            16   \n",
       "502A                                      45                            16   \n",
       "504A                                      46                            15   \n",
       "506A                                      45                            16   \n",
       "508A                                      50                            18   \n",
       "\n",
       "     ApoA1 Cit _IgG_Fecal Mix C RBB _IgA_Fecal  \\\n",
       "500A                   18                   48   \n",
       "502A                   19                   48   \n",
       "504A                   19                   45   \n",
       "506A                   22                   48   \n",
       "508A                   21                   62   \n",
       "\n",
       "     FibrinogenB 36-52 cit _IgG_Fecal FibrinogenB 246-267 cit _IgG_Fecal  \\\n",
       "500A                               23                                 21   \n",
       "502A                               21                                 20   \n",
       "504A                               23                                 20   \n",
       "506A                               23                                 19   \n",
       "508A                               23                                 19   \n",
       "\n",
       "     Vimentin 58-77 cit3 cyclic small-1 _IgG_Fecal  ... Vimentin _IgG_Fecal  \\\n",
       "500A                                            17  ...                  31   \n",
       "502A                                            17  ...                  29   \n",
       "504A                                            15  ...                  26   \n",
       "506A                                            17  ...                  27   \n",
       "508A                                            15  ...                  28   \n",
       "\n",
       "     Fibrinogen _IgG_Fecal Fibrinogen cit _IgG_Fecal  \\\n",
       "500A                   136                       109   \n",
       "502A                   143                       113   \n",
       "504A                   130                       109   \n",
       "506A                   129                       109   \n",
       "508A                   130                       108   \n",
       "\n",
       "     Filaggrin 48-65 arg2 v1 cyclic _IgG_Fecal Fibronectin cit _IgA_Fecal  \\\n",
       "500A                                        17                        131   \n",
       "502A                                        15                        132   \n",
       "504A                                        16                        135   \n",
       "506A                                        15                        138   \n",
       "508A                                        20                        141   \n",
       "\n",
       "     FibrinogenA 556-575 sm cyclic _IgG_Fecal Histones2B cit _IgG_Fecal  \\\n",
       "500A                                       18                        17   \n",
       "502A                                       17                        37   \n",
       "504A                                       17                        18   \n",
       "506A                                       14                        15   \n",
       "508A                                       22                        20   \n",
       "\n",
       "     Fibronectin cit 1029-1042 cit2 _IgG_Fecal  \\\n",
       "500A                                        17   \n",
       "502A                                        16   \n",
       "504A                                        15   \n",
       "506A                                        17   \n",
       "508A                                        17   \n",
       "\n",
       "     FibrinogenA 211-230 cit small cyclic _IgG_Fecal Mix B RBB _IgG_Fecal  \n",
       "500A                                              32                   43  \n",
       "502A                                              30                   45  \n",
       "504A                                              33                   44  \n",
       "506A                                              32                   47  \n",
       "508A                                              33                   49  \n",
       "\n",
       "[5 rows x 116 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# acpas fecal\n",
    "df_acpa_fecal = pd.read_csv(path + 'inputs/RA_twins/ACPA/CCP_Fecal_Analytes_COMPARISON_KB.csv', index_col=0)\n",
    "\n",
    "# concatenate index\n",
    "df_acpa_fecal.index = df_acpa_fecal.index + '_' + df_acpa_fecal['Secondary'] + '_Fecal'\n",
    "\n",
    "# transpose\n",
    "df_acpa_fecal = df_acpa_fecal.T\n",
    "\n",
    "# drop useless rows\n",
    "df_acpa_fecal = df_acpa_fecal.drop(['Secondary','Units', 'AVG1','TTESTP','AVG2','182A_F','183A_F', '516A_F','517A_F'])\n",
    "\n",
    "# convert to normal\n",
    "df_acpa_fecal.index = df_acpa_fecal.index.map(lambda x: x.split('_')[0])\n",
    "df_acpa_fecal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biglycan 247-266 sm-1 cyclic IgG_Fecal</th>\n",
       "      <th>Clusterin 231-250 sm-1 cyclic IgG_Fecal</th>\n",
       "      <th>Mix B RBB IgA_Fecal</th>\n",
       "      <th>ApoE 277-296 Cit2 sm2 cyclic IgG_Fecal</th>\n",
       "      <th>H2B-a 62-81 cyclic IgG_Fecal</th>\n",
       "      <th>ApoA1 Cit IgG_Fecal</th>\n",
       "      <th>Mix C RBB IgA_Fecal</th>\n",
       "      <th>FibrinogenB 36-52 cit IgG_Fecal</th>\n",
       "      <th>FibrinogenB 246-267 cit IgG_Fecal</th>\n",
       "      <th>Vimentin 58-77 cit3 cyclic small-1 IgG_Fecal</th>\n",
       "      <th>...</th>\n",
       "      <th>Vimentin IgG_Fecal</th>\n",
       "      <th>Fibrinogen IgG_Fecal</th>\n",
       "      <th>Fibrinogen cit IgG_Fecal</th>\n",
       "      <th>Filaggrin 48-65 arg2 v1 cyclic IgG_Fecal</th>\n",
       "      <th>Fibronectin cit IgA_Fecal</th>\n",
       "      <th>FibrinogenA 556-575 sm cyclic IgG_Fecal</th>\n",
       "      <th>Histones2B cit IgG_Fecal</th>\n",
       "      <th>Fibronectin cit 1029-1042 cit2 IgG_Fecal</th>\n",
       "      <th>FibrinogenA 211-230 cit small cyclic IgG_Fecal</th>\n",
       "      <th>Mix B RBB IgG_Fecal</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>18</td>\n",
       "      <td>15.0</td>\n",
       "      <td>47</td>\n",
       "      <td>47.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>23</td>\n",
       "      <td>21.0</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>30.5</td>\n",
       "      <td>135.5</td>\n",
       "      <td>109.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>130.5</td>\n",
       "      <td>18.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>17</td>\n",
       "      <td>16.0</td>\n",
       "      <td>46</td>\n",
       "      <td>44.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>48.5</td>\n",
       "      <td>23</td>\n",
       "      <td>18.0</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>138.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>15.5</td>\n",
       "      <td>122.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>46.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>20</td>\n",
       "      <td>14.5</td>\n",
       "      <td>48</td>\n",
       "      <td>45.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>21</td>\n",
       "      <td>20.0</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>29.0</td>\n",
       "      <td>142.5</td>\n",
       "      <td>113.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>132.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>45.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>17</td>\n",
       "      <td>15.0</td>\n",
       "      <td>71</td>\n",
       "      <td>47.0</td>\n",
       "      <td>14.5</td>\n",
       "      <td>21.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>24</td>\n",
       "      <td>20.0</td>\n",
       "      <td>18</td>\n",
       "      <td>...</td>\n",
       "      <td>27.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>106.5</td>\n",
       "      <td>18.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>18</td>\n",
       "      <td>16.0</td>\n",
       "      <td>50</td>\n",
       "      <td>46.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>23</td>\n",
       "      <td>19.5</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>26.0</td>\n",
       "      <td>129.5</td>\n",
       "      <td>109.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 116 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Biglycan 247-266 sm-1 cyclic IgG_Fecal  \\\n",
       "SampleID                                           \n",
       "500A                                          18   \n",
       "501A                                          17   \n",
       "502A                                          20   \n",
       "503A                                          17   \n",
       "504A                                          18   \n",
       "\n",
       "          Clusterin 231-250 sm-1 cyclic IgG_Fecal  Mix B RBB IgA_Fecal  \\\n",
       "SampleID                                                                 \n",
       "500A                                         15.0                   47   \n",
       "501A                                         16.0                   46   \n",
       "502A                                         14.5                   48   \n",
       "503A                                         15.0                   71   \n",
       "504A                                         16.0                   50   \n",
       "\n",
       "          ApoE 277-296 Cit2 sm2 cyclic IgG_Fecal  \\\n",
       "SampleID                                           \n",
       "500A                                        47.0   \n",
       "501A                                        44.0   \n",
       "502A                                        45.0   \n",
       "503A                                        47.0   \n",
       "504A                                        46.0   \n",
       "\n",
       "          H2B-a 62-81 cyclic IgG_Fecal  ApoA1 Cit IgG_Fecal  \\\n",
       "SampleID                                                      \n",
       "500A                              16.0                 18.0   \n",
       "501A                              15.0                 19.0   \n",
       "502A                              16.0                 19.0   \n",
       "503A                              14.5                 21.0   \n",
       "504A                              15.0                 19.0   \n",
       "\n",
       "          Mix C RBB IgA_Fecal  FibrinogenB 36-52 cit IgG_Fecal  \\\n",
       "SampleID                                                         \n",
       "500A                     48.0                               23   \n",
       "501A                     48.5                               23   \n",
       "502A                     48.0                               21   \n",
       "503A                     61.0                               24   \n",
       "504A                     45.0                               23   \n",
       "\n",
       "          FibrinogenB 246-267 cit IgG_Fecal  \\\n",
       "SampleID                                      \n",
       "500A                                   21.0   \n",
       "501A                                   18.0   \n",
       "502A                                   20.0   \n",
       "503A                                   20.0   \n",
       "504A                                   19.5   \n",
       "\n",
       "          Vimentin 58-77 cit3 cyclic small-1 IgG_Fecal  ...  \\\n",
       "SampleID                                                ...   \n",
       "500A                                                17  ...   \n",
       "501A                                                16  ...   \n",
       "502A                                                17  ...   \n",
       "503A                                                18  ...   \n",
       "504A                                                15  ...   \n",
       "\n",
       "          Vimentin IgG_Fecal  Fibrinogen IgG_Fecal  Fibrinogen cit IgG_Fecal  \\\n",
       "SampleID                                                                       \n",
       "500A                    30.5                 135.5                     109.0   \n",
       "501A                    25.0                 138.0                     111.0   \n",
       "502A                    29.0                 142.5                     113.0   \n",
       "503A                    27.0                 125.0                     106.5   \n",
       "504A                    26.0                 129.5                     109.0   \n",
       "\n",
       "          Filaggrin 48-65 arg2 v1 cyclic IgG_Fecal  Fibronectin cit IgA_Fecal  \\\n",
       "SampleID                                                                        \n",
       "500A                                          17.0                      130.5   \n",
       "501A                                          15.5                      122.0   \n",
       "502A                                          15.0                      132.0   \n",
       "503A                                          18.0                      178.0   \n",
       "504A                                          16.0                      135.0   \n",
       "\n",
       "          FibrinogenA 556-575 sm cyclic IgG_Fecal  Histones2B cit IgG_Fecal  \\\n",
       "SampleID                                                                      \n",
       "500A                                         18.0                      17.0   \n",
       "501A                                         16.0                      16.0   \n",
       "502A                                         17.0                      37.0   \n",
       "503A                                         19.0                      17.0   \n",
       "504A                                         17.0                      18.0   \n",
       "\n",
       "          Fibronectin cit 1029-1042 cit2 IgG_Fecal  \\\n",
       "SampleID                                             \n",
       "500A                                          17.0   \n",
       "501A                                          16.0   \n",
       "502A                                          16.0   \n",
       "503A                                          16.0   \n",
       "504A                                          15.0   \n",
       "\n",
       "          FibrinogenA 211-230 cit small cyclic IgG_Fecal  Mix B RBB IgG_Fecal  \n",
       "SampleID                                                                       \n",
       "500A                                                32.0                 43.0  \n",
       "501A                                                33.0                 46.0  \n",
       "502A                                                30.0                 45.0  \n",
       "503A                                                31.0                 44.0  \n",
       "504A                                                33.0                 44.0  \n",
       "\n",
       "[5 rows x 116 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# acpas fecal\n",
    "df_acpa_fecal = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/10_acpa_fecal/acpa_fecal.txt', sep='\\t', index_col=0)\n",
    "\n",
    "# drop samples\n",
    "df_acpa_fecal = df_acpa_fecal.drop(['182A','183A','516A','517A'])\n",
    "\n",
    "# drop cols\n",
    "df_acpa_fecal = df_acpa_fecal.drop(['Subject','Sibling_pair','Diagnosis'], axis=1)\n",
    "\n",
    "# add suffix to columns\n",
    "df_acpa_fecal.columns = df_acpa_fecal.columns + '_Fecal'\n",
    "\n",
    "df_acpa_fecal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CCP 3.1_ccpnan_Plasma</th>\n",
       "      <th>FibrinogenA 41-60 cit3 cyclic _IgA_Plasma</th>\n",
       "      <th>FibrinogenA 27-43 cit _IgA_Plasma</th>\n",
       "      <th>Biglycan 247-266 cit sm-1 cyclic _IgA_Plasma</th>\n",
       "      <th>FibrinogenA 41-60 cyclic _IgA_Plasma</th>\n",
       "      <th>H2B/a 62-81 cyclic _IgA_Plasma</th>\n",
       "      <th>ApoE 277-296 Cit2 sm2 cyclic _IgG_Plasma</th>\n",
       "      <th>Enolase-1A 5-21 cit _IgA_Plasma</th>\n",
       "      <th>FibrinogenA 556-575 sm cyclic _IgA_Plasma</th>\n",
       "      <th>FibrinogenB 54-72 _IgA_Plasma</th>\n",
       "      <th>...</th>\n",
       "      <th>ApoA1 Cit _IgA_Plasma</th>\n",
       "      <th>H2B/a 62-81 cit cyclic _IgG_Plasma</th>\n",
       "      <th>Histones2B _IgG_Plasma</th>\n",
       "      <th>ApoA1 _IgA_Plasma</th>\n",
       "      <th>Mix C SVB _IgA_Plasma</th>\n",
       "      <th>Histones2A cit _IgG_Plasma</th>\n",
       "      <th>H2A/a 1-20 cit sm-2 cyclic _IgA_Plasma</th>\n",
       "      <th>Histones2A cit _IgA_Plasma</th>\n",
       "      <th>FibrinogenA 556-575 cit sm cyclic _IgA_Plasma</th>\n",
       "      <th>Vimentin 58-77 cit3 cyclic small-1 _IgA_Plasma</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>44</td>\n",
       "      <td>32</td>\n",
       "      <td>33</td>\n",
       "      <td>38</td>\n",
       "      <td>64</td>\n",
       "      <td>67</td>\n",
       "      <td>36</td>\n",
       "      <td>43</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>35</td>\n",
       "      <td>46</td>\n",
       "      <td>20</td>\n",
       "      <td>45</td>\n",
       "      <td>34</td>\n",
       "      <td>47</td>\n",
       "      <td>34</td>\n",
       "      <td>29</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>94384.5</td>\n",
       "      <td>46.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>437.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>...</td>\n",
       "      <td>35.0</td>\n",
       "      <td>233.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>722.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>154.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>228.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>4406.13</td>\n",
       "      <td>22.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>437.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>506A</th>\n",
       "      <td>1010</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>36</td>\n",
       "      <td>43</td>\n",
       "      <td>42</td>\n",
       "      <td>113</td>\n",
       "      <td>64</td>\n",
       "      <td>43</td>\n",
       "      <td>53</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>79</td>\n",
       "      <td>88</td>\n",
       "      <td>29</td>\n",
       "      <td>77</td>\n",
       "      <td>31</td>\n",
       "      <td>34</td>\n",
       "      <td>125</td>\n",
       "      <td>32</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>508A</th>\n",
       "      <td>47444.8</td>\n",
       "      <td>51.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>347.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1101.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>247.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 117 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CCP 3.1_ccpnan_Plasma FibrinogenA 41-60 cit3 cyclic _IgA_Plasma  \\\n",
       "500A                     0                                        21   \n",
       "502A               94384.5                                      46.0   \n",
       "504A               4406.13                                      22.0   \n",
       "506A                  1010                                        26   \n",
       "508A               47444.8                                      51.0   \n",
       "\n",
       "     FibrinogenA 27-43 cit _IgA_Plasma  \\\n",
       "500A                                44   \n",
       "502A                              36.0   \n",
       "504A                              25.0   \n",
       "506A                                26   \n",
       "508A                              33.0   \n",
       "\n",
       "     Biglycan 247-266 cit sm-1 cyclic _IgA_Plasma  \\\n",
       "500A                                           32   \n",
       "502A                                         53.0   \n",
       "504A                                         26.0   \n",
       "506A                                           36   \n",
       "508A                                         36.0   \n",
       "\n",
       "     FibrinogenA 41-60 cyclic _IgA_Plasma H2B/a 62-81 cyclic _IgA_Plasma  \\\n",
       "500A                                   33                             38   \n",
       "502A                                 65.0                           74.0   \n",
       "504A                                 27.0                           32.0   \n",
       "506A                                   43                             42   \n",
       "508A                                 41.0                           48.0   \n",
       "\n",
       "     ApoE 277-296 Cit2 sm2 cyclic _IgG_Plasma Enolase-1A 5-21 cit _IgA_Plasma  \\\n",
       "500A                                       64                              67   \n",
       "502A                                     75.0                           437.0   \n",
       "504A                                     64.0                           137.0   \n",
       "506A                                      113                              64   \n",
       "508A                                     83.0                           347.0   \n",
       "\n",
       "     FibrinogenA 556-575 sm cyclic _IgA_Plasma FibrinogenB 54-72 _IgA_Plasma  \\\n",
       "500A                                        36                            43   \n",
       "502A                                      79.0                          74.0   \n",
       "504A                                      29.0                          37.0   \n",
       "506A                                        43                            53   \n",
       "508A                                      52.0                          58.0   \n",
       "\n",
       "      ... ApoA1 Cit _IgA_Plasma H2B/a 62-81 cit cyclic _IgG_Plasma  \\\n",
       "500A  ...                    19                                 35   \n",
       "502A  ...                  35.0                              233.0   \n",
       "504A  ...                  21.0                               34.0   \n",
       "506A  ...                    24                                 79   \n",
       "508A  ...                  23.0                             1101.0   \n",
       "\n",
       "     Histones2B _IgG_Plasma ApoA1 _IgA_Plasma Mix C SVB _IgA_Plasma  \\\n",
       "500A                     46                20                    45   \n",
       "502A                   47.0              26.0                  89.0   \n",
       "504A                   45.0              19.0                  49.0   \n",
       "506A                     88                29                    77   \n",
       "508A                   56.0              26.0                  76.0   \n",
       "\n",
       "     Histones2A cit _IgG_Plasma H2A/a 1-20 cit sm-2 cyclic _IgA_Plasma  \\\n",
       "500A                         34                                     47   \n",
       "502A                      722.0                                  166.0   \n",
       "504A                       55.0                                   29.0   \n",
       "506A                         31                                     34   \n",
       "508A                      247.0                                   34.0   \n",
       "\n",
       "     Histones2A cit _IgA_Plasma FibrinogenA 556-575 cit sm cyclic _IgA_Plasma  \\\n",
       "500A                         34                                            29   \n",
       "502A                      154.0                                          66.0   \n",
       "504A                      437.0                                          26.0   \n",
       "506A                        125                                            32   \n",
       "508A                       83.0                                          61.0   \n",
       "\n",
       "     Vimentin 58-77 cit3 cyclic small-1 _IgA_Plasma  \n",
       "500A                                             51  \n",
       "502A                                          228.0  \n",
       "504A                                           25.0  \n",
       "506A                                             30  \n",
       "508A                                           49.0  \n",
       "\n",
       "[5 rows x 117 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# acpas plasma\n",
    "df_acpa_plasma = pd.read_csv(path + 'inputs/RA_twins/ACPA/CCP_Plasma_Analytes_COMPARISON_KB.csv', index_col=0)\n",
    "\n",
    "# convert first nan to ccp\n",
    "df_acpa_plasma.iloc[0,0] = 'ccpnan'\n",
    "\n",
    "# concatenate index\n",
    "df_acpa_plasma.index = df_acpa_plasma.index + '_' + df_acpa_plasma['Secondary'] + '_Plasma'\n",
    "\n",
    "# transpose\n",
    "df_acpa_plasma = df_acpa_plasma.T\n",
    "\n",
    "# drop useless rows\n",
    "df_acpa_plasma = df_acpa_plasma.drop(['Secondary','Units', 'AVG1','TTESTP','AVG2','182A_P','183A_P', '516A_P', '517A_P'])\n",
    "\n",
    "# convert to normal\n",
    "df_acpa_plasma.index = df_acpa_plasma.index.map(lambda x: x.split('_')[0])\n",
    "df_acpa_plasma.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CCP 3.1_Plasma</th>\n",
       "      <th>FibrinogenA 41-60 cit3 cyclic IgA_Plasma</th>\n",
       "      <th>FibrinogenA 27-43 cit IgA_Plasma</th>\n",
       "      <th>Biglycan 247-266 cit sm-1 cyclic IgA_Plasma</th>\n",
       "      <th>FibrinogenA 41-60 cyclic IgA_Plasma</th>\n",
       "      <th>H2B-a 62-81 cyclic IgA_Plasma</th>\n",
       "      <th>ApoE 277-296 Cit2 sm2 cyclic IgG_Plasma</th>\n",
       "      <th>Enolase-1A 5-21 cit IgA_Plasma</th>\n",
       "      <th>FibrinogenA 556-575 sm cyclic IgA_Plasma</th>\n",
       "      <th>FibrinogenB 54-72 IgA_Plasma</th>\n",
       "      <th>...</th>\n",
       "      <th>ApoA1 Cit IgA_Plasma</th>\n",
       "      <th>H2B-a 62-81 cit cyclic IgG_Plasma</th>\n",
       "      <th>Histones2B IgG_Plasma</th>\n",
       "      <th>ApoA1 IgA_Plasma</th>\n",
       "      <th>Mix C SVB IgA_Plasma</th>\n",
       "      <th>Histones2A cit IgG_Plasma</th>\n",
       "      <th>H2A-a 1-20 cit sm-2 cyclic IgA_Plasma</th>\n",
       "      <th>Histones2A cit IgA_Plasma</th>\n",
       "      <th>FibrinogenA 556-575 cit sm cyclic IgA_Plasma</th>\n",
       "      <th>Vimentin 58-77 cit3 cyclic small-1 IgA_Plasma</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>0.00</td>\n",
       "      <td>21.0</td>\n",
       "      <td>44</td>\n",
       "      <td>32</td>\n",
       "      <td>33.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>...</td>\n",
       "      <td>19.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>51.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>0.00</td>\n",
       "      <td>15.0</td>\n",
       "      <td>18</td>\n",
       "      <td>16</td>\n",
       "      <td>20.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>38.5</td>\n",
       "      <td>28.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>94384.50</td>\n",
       "      <td>46.0</td>\n",
       "      <td>36</td>\n",
       "      <td>53</td>\n",
       "      <td>64.5</td>\n",
       "      <td>74.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>437.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>...</td>\n",
       "      <td>34.5</td>\n",
       "      <td>233.0</td>\n",
       "      <td>46.5</td>\n",
       "      <td>26.0</td>\n",
       "      <td>88.5</td>\n",
       "      <td>722.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>154.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>228.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>104484.50</td>\n",
       "      <td>45.0</td>\n",
       "      <td>36</td>\n",
       "      <td>38</td>\n",
       "      <td>55.5</td>\n",
       "      <td>55.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>60.5</td>\n",
       "      <td>64.0</td>\n",
       "      <td>...</td>\n",
       "      <td>178.0</td>\n",
       "      <td>3646.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>238.0</td>\n",
       "      <td>2797.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>607.0</td>\n",
       "      <td>264.5</td>\n",
       "      <td>799.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>4406.13</td>\n",
       "      <td>22.0</td>\n",
       "      <td>25</td>\n",
       "      <td>26</td>\n",
       "      <td>27.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>63.5</td>\n",
       "      <td>137.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>437.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 117 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CCP 3.1_Plasma  FibrinogenA 41-60 cit3 cyclic IgA_Plasma  \\\n",
       "SampleID                                                             \n",
       "500A                0.00                                      21.0   \n",
       "501A                0.00                                      15.0   \n",
       "502A            94384.50                                      46.0   \n",
       "503A           104484.50                                      45.0   \n",
       "504A             4406.13                                      22.0   \n",
       "\n",
       "          FibrinogenA 27-43 cit IgA_Plasma  \\\n",
       "SampleID                                     \n",
       "500A                                    44   \n",
       "501A                                    18   \n",
       "502A                                    36   \n",
       "503A                                    36   \n",
       "504A                                    25   \n",
       "\n",
       "          Biglycan 247-266 cit sm-1 cyclic IgA_Plasma  \\\n",
       "SampleID                                                \n",
       "500A                                               32   \n",
       "501A                                               16   \n",
       "502A                                               53   \n",
       "503A                                               38   \n",
       "504A                                               26   \n",
       "\n",
       "          FibrinogenA 41-60 cyclic IgA_Plasma  H2B-a 62-81 cyclic IgA_Plasma  \\\n",
       "SampleID                                                                       \n",
       "500A                                     33.0                           38.0   \n",
       "501A                                     20.0                           23.0   \n",
       "502A                                     64.5                           74.0   \n",
       "503A                                     55.5                           55.0   \n",
       "504A                                     27.0                           32.0   \n",
       "\n",
       "          ApoE 277-296 Cit2 sm2 cyclic IgG_Plasma  \\\n",
       "SampleID                                            \n",
       "500A                                         64.0   \n",
       "501A                                         57.0   \n",
       "502A                                         75.0   \n",
       "503A                                         69.0   \n",
       "504A                                         63.5   \n",
       "\n",
       "          Enolase-1A 5-21 cit IgA_Plasma  \\\n",
       "SampleID                                   \n",
       "500A                                67.0   \n",
       "501A                                51.0   \n",
       "502A                               437.0   \n",
       "503A                                96.0   \n",
       "504A                               137.0   \n",
       "\n",
       "          FibrinogenA 556-575 sm cyclic IgA_Plasma  \\\n",
       "SampleID                                             \n",
       "500A                                          36.0   \n",
       "501A                                          22.0   \n",
       "502A                                          79.0   \n",
       "503A                                          60.5   \n",
       "504A                                          29.0   \n",
       "\n",
       "          FibrinogenB 54-72 IgA_Plasma  ...  ApoA1 Cit IgA_Plasma  \\\n",
       "SampleID                                ...                         \n",
       "500A                              43.0  ...                  19.0   \n",
       "501A                              28.0  ...                  18.0   \n",
       "502A                              74.0  ...                  34.5   \n",
       "503A                              64.0  ...                 178.0   \n",
       "504A                              37.0  ...                  21.0   \n",
       "\n",
       "          H2B-a 62-81 cit cyclic IgG_Plasma  Histones2B IgG_Plasma  \\\n",
       "SampleID                                                             \n",
       "500A                                   35.0                   46.0   \n",
       "501A                                   30.0                   53.0   \n",
       "502A                                  233.0                   46.5   \n",
       "503A                                 3646.0                   46.0   \n",
       "504A                                   34.0                   45.0   \n",
       "\n",
       "          ApoA1 IgA_Plasma  Mix C SVB IgA_Plasma  Histones2A cit IgG_Plasma  \\\n",
       "SampleID                                                                      \n",
       "500A                  20.0                  45.0                       34.0   \n",
       "501A                  21.0                  45.0                       38.5   \n",
       "502A                  26.0                  88.5                      722.0   \n",
       "503A                  28.0                 238.0                     2797.0   \n",
       "504A                  19.0                  49.0                       55.0   \n",
       "\n",
       "          H2A-a 1-20 cit sm-2 cyclic IgA_Plasma  Histones2A cit IgA_Plasma  \\\n",
       "SampleID                                                                     \n",
       "500A                                       47.0                       34.0   \n",
       "501A                                       28.0                       27.0   \n",
       "502A                                      166.0                      154.0   \n",
       "503A                                       88.0                      607.0   \n",
       "504A                                       29.0                      437.0   \n",
       "\n",
       "          FibrinogenA 556-575 cit sm cyclic IgA_Plasma  \\\n",
       "SampleID                                                 \n",
       "500A                                              29.0   \n",
       "501A                                              25.0   \n",
       "502A                                              66.0   \n",
       "503A                                             264.5   \n",
       "504A                                              26.0   \n",
       "\n",
       "          Vimentin 58-77 cit3 cyclic small-1 IgA_Plasma  \n",
       "SampleID                                                 \n",
       "500A                                               51.0  \n",
       "501A                                               19.0  \n",
       "502A                                              228.0  \n",
       "503A                                              799.5  \n",
       "504A                                               25.0  \n",
       "\n",
       "[5 rows x 117 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# acpas plasma\n",
    "df_acpa_plasma = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/11_acpa_plasma/acpa_plasma.txt', sep='\\t', index_col=0)\n",
    "\n",
    "# drop samples\n",
    "df_acpa_plasma = df_acpa_plasma.drop(['182A','183A','516A','517A'])\n",
    "\n",
    "# drop cols\n",
    "df_acpa_plasma = df_acpa_plasma.drop(['Subject','Sibling_pair','Diagnosis'], axis=1)\n",
    "\n",
    "# add suffix to columns\n",
    "df_acpa_plasma.columns = df_acpa_plasma.columns + '_Plasma'\n",
    "\n",
    "df_acpa_plasma.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### run up to after this #####"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "##### Combined Analyses #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Phocaeicola|s  Phocaeicola vulgatus</th>\n",
       "      <th>k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Bacteroides|s  Bacteroides uniformis</th>\n",
       "      <th>k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Bacteroides|s  Bacteroides ovatus</th>\n",
       "      <th>k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Bacteroides|s  Bacteroides stercoris</th>\n",
       "      <th>k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Phocaeicola|s  Phocaeicola dorei</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Lachnospiraceae unclassified|s  Eubacterium rectale</th>\n",
       "      <th>k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Rikenellaceae|g  Alistipes|s  Alistipes onderdonkii</th>\n",
       "      <th>k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Phocaeicola|s  Phocaeicola plebeius</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  CFGB3002|o  OFGB3002|f  FGB3002|g  GGB9463|s  GGB9463 SGB14855</th>\n",
       "      <th>k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Rikenellaceae|g  Alistipes|s  Alistipes putredinis</th>\n",
       "      <th>...</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Coriobacteriia|o  Coriobacteriales|f  Coriobacteriaceae|g  Enorma|s  Enorma massiliensis</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Dorea|s  Dorea phocaeensis</th>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Coriobacteriia|o  Coriobacteriales|f  Coriobacteriaceae|g  Collinsella|s  Collinsella tanakaei</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  CFGB1359|o  OFGB1359|f  FGB1359|g  GGB3321|s  GGB3321 SGB4394</th>\n",
       "      <th>k  Bacteria|p  Proteobacteria|c  Betaproteobacteria|o  Burkholderiales|f  Sutterellaceae|g  Parasutterella|s  Parasutterella SGB9260</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  CFGB1538|o  OFGB1538|f  FGB1538|g  GGB3892|s  GGB3892 SGB5290</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB51269|s  GGB51269 SGB5062</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Oscillospiraceae|g  Agathobaculum|s  Agathobaculum butyriciproducens</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Eubacteriales Family XIII Incertae Sedis|g  GGB32387|s  GGB32387 SGB47381</th>\n",
       "      <th>k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Eubacteriales Family XIII Incertae Sedis|g  Eubacteriales Family XIII Incertae Sedis unclassified|s  Clostridiales Family XIII bacterium BX16</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>0.063512</td>\n",
       "      <td>0.260751</td>\n",
       "      <td>0.002257</td>\n",
       "      <td>0.017321</td>\n",
       "      <td>0.000083</td>\n",
       "      <td>0.004326</td>\n",
       "      <td>0.000645</td>\n",
       "      <td>0.023989</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.007745</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>0.135520</td>\n",
       "      <td>0.074287</td>\n",
       "      <td>0.029319</td>\n",
       "      <td>0.039050</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.026713</td>\n",
       "      <td>0.027435</td>\n",
       "      <td>0.064394</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000338</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>0.213437</td>\n",
       "      <td>0.130766</td>\n",
       "      <td>0.017082</td>\n",
       "      <td>0.238545</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.038197</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.021635</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000841</td>\n",
       "      <td>0.000113</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>0.151226</td>\n",
       "      <td>0.170454</td>\n",
       "      <td>0.026042</td>\n",
       "      <td>0.083876</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.054987</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.012060</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007189</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>0.102240</td>\n",
       "      <td>0.151471</td>\n",
       "      <td>0.001853</td>\n",
       "      <td>0.077741</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001610</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.028988</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005232</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000564</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 200 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Phocaeicola|s  Phocaeicola vulgatus  \\\n",
       "500A                                           0.063512                                                                       \n",
       "501A                                           0.135520                                                                       \n",
       "502A                                           0.213437                                                                       \n",
       "503A                                           0.151226                                                                       \n",
       "504A                                           0.102240                                                                       \n",
       "\n",
       "      k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Bacteroides|s  Bacteroides uniformis  \\\n",
       "500A                                           0.260751                                                                        \n",
       "501A                                           0.074287                                                                        \n",
       "502A                                           0.130766                                                                        \n",
       "503A                                           0.170454                                                                        \n",
       "504A                                           0.151471                                                                        \n",
       "\n",
       "      k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Bacteroides|s  Bacteroides ovatus  \\\n",
       "500A                                           0.002257                                                                     \n",
       "501A                                           0.029319                                                                     \n",
       "502A                                           0.017082                                                                     \n",
       "503A                                           0.026042                                                                     \n",
       "504A                                           0.001853                                                                     \n",
       "\n",
       "      k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Bacteroides|s  Bacteroides stercoris  \\\n",
       "500A                                           0.017321                                                                        \n",
       "501A                                           0.039050                                                                        \n",
       "502A                                           0.238545                                                                        \n",
       "503A                                           0.083876                                                                        \n",
       "504A                                           0.077741                                                                        \n",
       "\n",
       "      k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Phocaeicola|s  Phocaeicola dorei  \\\n",
       "500A                                           0.000083                                                                    \n",
       "501A                                           0.000000                                                                    \n",
       "502A                                           0.000000                                                                    \n",
       "503A                                           0.000000                                                                    \n",
       "504A                                           0.000000                                                                    \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Lachnospiraceae unclassified|s  Eubacterium rectale  \\\n",
       "500A                                           0.004326                                                                                    \n",
       "501A                                           0.026713                                                                                    \n",
       "502A                                           0.038197                                                                                    \n",
       "503A                                           0.054987                                                                                    \n",
       "504A                                           0.000000                                                                                    \n",
       "\n",
       "      k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Rikenellaceae|g  Alistipes|s  Alistipes onderdonkii  \\\n",
       "500A                                           0.000645                                                                     \n",
       "501A                                           0.027435                                                                     \n",
       "502A                                           0.000000                                                                     \n",
       "503A                                           0.000000                                                                     \n",
       "504A                                           0.001610                                                                     \n",
       "\n",
       "      k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Bacteroidaceae|g  Phocaeicola|s  Phocaeicola plebeius  \\\n",
       "500A                                           0.023989                                                                       \n",
       "501A                                           0.064394                                                                       \n",
       "502A                                           0.000000                                                                       \n",
       "503A                                           0.000000                                                                       \n",
       "504A                                           0.000000                                                                       \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  CFGB3002|o  OFGB3002|f  FGB3002|g  GGB9463|s  GGB9463 SGB14855  \\\n",
       "500A                                                0.0                                             \n",
       "501A                                                0.0                                             \n",
       "502A                                                0.0                                             \n",
       "503A                                                0.0                                             \n",
       "504A                                                0.0                                             \n",
       "\n",
       "      k  Bacteria|p  Bacteroidetes|c  Bacteroidia|o  Bacteroidales|f  Rikenellaceae|g  Alistipes|s  Alistipes putredinis  \\\n",
       "500A                                           0.007745                                                                    \n",
       "501A                                           0.000000                                                                    \n",
       "502A                                           0.021635                                                                    \n",
       "503A                                           0.012060                                                                    \n",
       "504A                                           0.028988                                                                    \n",
       "\n",
       "      ...  \\\n",
       "500A  ...   \n",
       "501A  ...   \n",
       "502A  ...   \n",
       "503A  ...   \n",
       "504A  ...   \n",
       "\n",
       "      k  Bacteria|p  Actinobacteria|c  Coriobacteriia|o  Coriobacteriales|f  Coriobacteriaceae|g  Enorma|s  Enorma massiliensis  \\\n",
       "500A                                                0.0                                                                           \n",
       "501A                                                0.0                                                                           \n",
       "502A                                                0.0                                                                           \n",
       "503A                                                0.0                                                                           \n",
       "504A                                                0.0                                                                           \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Dorea|s  Dorea phocaeensis  \\\n",
       "500A                                           0.000071                                                           \n",
       "501A                                           0.000000                                                           \n",
       "502A                                           0.000000                                                           \n",
       "503A                                           0.000000                                                           \n",
       "504A                                           0.000000                                                           \n",
       "\n",
       "      k  Bacteria|p  Actinobacteria|c  Coriobacteriia|o  Coriobacteriales|f  Coriobacteriaceae|g  Collinsella|s  Collinsella tanakaei  \\\n",
       "500A                                           0.000000                                                                                 \n",
       "501A                                           0.000000                                                                                 \n",
       "502A                                           0.000000                                                                                 \n",
       "503A                                           0.007189                                                                                 \n",
       "504A                                           0.000000                                                                                 \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  CFGB1359|o  OFGB1359|f  FGB1359|g  GGB3321|s  GGB3321 SGB4394  \\\n",
       "500A                                                0.0                                            \n",
       "501A                                                0.0                                            \n",
       "502A                                                0.0                                            \n",
       "503A                                                0.0                                            \n",
       "504A                                                0.0                                            \n",
       "\n",
       "      k  Bacteria|p  Proteobacteria|c  Betaproteobacteria|o  Burkholderiales|f  Sutterellaceae|g  Parasutterella|s  Parasutterella SGB9260  \\\n",
       "500A                                           0.000000                                                                                      \n",
       "501A                                           0.000000                                                                                      \n",
       "502A                                           0.000000                                                                                      \n",
       "503A                                           0.000000                                                                                      \n",
       "504A                                           0.005232                                                                                      \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  CFGB1538|o  OFGB1538|f  FGB1538|g  GGB3892|s  GGB3892 SGB5290  \\\n",
       "500A                                                0.0                                            \n",
       "501A                                                0.0                                            \n",
       "502A                                                0.0                                            \n",
       "503A                                                0.0                                            \n",
       "504A                                                0.0                                            \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB51269|s  GGB51269 SGB5062  \\\n",
       "500A                                                0.0                                                             \n",
       "501A                                                0.0                                                             \n",
       "502A                                                0.0                                                             \n",
       "503A                                                0.0                                                             \n",
       "504A                                                0.0                                                             \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Oscillospiraceae|g  Agathobaculum|s  Agathobaculum butyriciproducens  \\\n",
       "500A                                           0.000073                                                                                  \n",
       "501A                                           0.000338                                                                                  \n",
       "502A                                           0.000841                                                                                  \n",
       "503A                                           0.000000                                                                                  \n",
       "504A                                           0.000000                                                                                  \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Eubacteriales Family XIII Incertae Sedis|g  GGB32387|s  GGB32387 SGB47381  \\\n",
       "500A                                           0.000000                                                                                       \n",
       "501A                                           0.000000                                                                                       \n",
       "502A                                           0.000113                                                                                       \n",
       "503A                                           0.000000                                                                                       \n",
       "504A                                           0.000564                                                                                       \n",
       "\n",
       "      k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Eubacteriales Family XIII Incertae Sedis|g  Eubacteriales Family XIII Incertae Sedis unclassified|s  Clostridiales Family XIII bacterium BX16  \n",
       "500A                                           0.000000                                                                                                                                                          \n",
       "501A                                           0.000129                                                                                                                                                          \n",
       "502A                                           0.000000                                                                                                                                                          \n",
       "503A                                           0.000000                                                                                                                                                          \n",
       "504A                                           0.000000                                                                                                                                                          \n",
       "\n",
       "[5 rows x 200 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combine them all except OTU and path\n",
    "df_quant = pd.concat([df_meta_edit, df_olink, df_fa, df_acpa_fecal, df_acpa_plasma, df_mb], axis=1)\n",
    "df_quant = df_quant.drop(['TwinPair'],axis=1)\n",
    "df_quant.columns = df_quant.columns.map(lambda x: x.replace('\\'','prime'))\n",
    "df_quant.columns = df_quant.columns.map(lambda x: x.strip())\n",
    "df_quant.to_csv(path + 'inputs/df_quant.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "# create new df for asv\n",
    "df_asv = pd.concat([df_meta_edit, df_otu], axis=1) \n",
    "df_asv = df_asv.drop(['TwinPair','SJC','TJC'],axis=1)\n",
    "df_asv.to_csv(path + 'inputs/df_asv.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "# split df asv on disease groups\n",
    "df_asv_RA = df_asv[df_asv['Diagnosis'] == 'RA']\n",
    "df_asv_RA = df_asv_RA.drop('Diagnosis',axis=1)\n",
    "df_asv_RA.to_csv(path + 'inputs/df_asv_RA.tsv', sep='\\t',index_label='SampleID')\n",
    "df_asv_UA = df_asv[df_asv['Diagnosis'] == 'Unaffected']\n",
    "df_asv_UA = df_asv_UA.drop('Diagnosis',axis=1)\n",
    "df_asv_UA.to_csv(path + 'inputs/df_asv_UA.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "# get top x df asv\n",
    "filt = 200\n",
    "df_dx = df_asv['Diagnosis']\n",
    "df_asv_filt = df_asv.drop(['Diagnosis'],axis=1).T\n",
    "df_asv_filt['sum'] = df_asv_filt.sum(axis=1)\n",
    "df_asv_filt = df_asv_filt.sort_values(by = 'sum', ascending=False)\n",
    "df_asv_filt = df_asv_filt.iloc[0:filt,:].T\n",
    "sums = df_asv_filt.loc['sum',:]\n",
    "df_asv_filt = df_asv_filt.drop('sum')\n",
    "df_asv_filt_pre=df_asv_filt.copy()\n",
    "df_asv_filt.to_csv(path + 'inputs/df_asv200.tsv', sep='\\t',index_label='SampleID')\n",
    "# df_asv100.to_csv(path + 'inputs/df_asv100.tsv', sep='\\t',index_label='SampleID')\n",
    "df_asv_filt_dx = pd.concat([df_asv_filt, df_dx],axis=1)\n",
    "\n",
    "# for plotting\n",
    "#taxa = ['k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia',\n",
    "#        'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis']\n",
    "\n",
    "# df = pd.concat([df_meta_edit,df_asv.drop('Diagnosis',axis=1)],axis=1)\n",
    "# df = df[['TwinPair','Diagnosis'] + taxa]\n",
    "# df = df.rename(columns={'TwinPair':'Sibling_pair', taxa[0]:'Bilophila wadsworthia', taxa[1]: 'Blautia faecis'})\n",
    "df = pd.concat([df_meta_edit.loc[:,['TwinPair','Diagnosis']],df_asv.drop('Diagnosis',axis=1)],axis=1)\n",
    "df = df.rename(columns={'TwinPair':'Sibling_pair'})#, taxa[0]:'Bilophila wadsworthia', taxa[1]: 'Blautia faecis'})\n",
    "df = df.rename(columns=lambda x: x.split('|s  ')[-1] if '|s' in x else x) #, inplace=True)\n",
    "df['Sibling_pair'] = df['Sibling_pair'].apply(lambda x: 'sib_0' + str(int(x.split('T')[1]) - 1))\n",
    "df['Diagnosis'] = df['Diagnosis'].apply(lambda x: '2_' + str(x) if x == 'RA' else '1_' + str(x))\n",
    "df.to_csv(path + 'inputs/RA_twins/16S/jobs/15_taxa/taxa.tsv', sep='\\t')\n",
    "\n",
    "\n",
    "# for the pathways# create BRITE pathways for R\n",
    "df_path = pd.read_csv(path + 'inputs/df_path_mp4all.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "# merge with df meta with twin data\n",
    "df_path = pd.concat([df_meta_edit.loc[:,['TwinPair','Diagnosis']],df_path],axis=1)\n",
    "df = df_path.rename(columns={'TwinPair':'Sibling_pair'})#, taxa[0]:'Bilophila wadsworthia', taxa[1]: 'Blautia faecis'})\n",
    "df['Sibling_pair'] = df['Sibling_pair'].apply(lambda x: 'sib_0' + str(int(x.split('T')[1]) - 1))\n",
    "df['Diagnosis'] = df['Diagnosis'].apply(lambda x: '2_' + str(x) if x == 'RA' else '1_' + str(x))\n",
    "# replace | and \\ special chars\n",
    "df = df.rename(columns=lambda x: x.replace('|', '_').replace('/', '__').replace('\\\\', '__'))\n",
    "df = df.rename(columns=lambda x: x.split('_')[-1])\n",
    "df = df.rename(columns={'pair':'Sibling_pair'})\n",
    "df.to_csv(path + 'inputs/RA_twins/16S/jobs/16_path/path.tsv', sep='\\t')\n",
    "\n",
    "# subset for CUTIE into sep groups\n",
    "df_asv_RA = df_asv_filt_dx[df_asv_filt_dx['Diagnosis'] == 'RA']\n",
    "df_asv_RA = df_asv_RA.drop('Diagnosis',axis=1)\n",
    "df_asv_RA.to_csv(path + 'inputs/df_asv200_RA.tsv', sep='\\t',index_label='SampleID')\n",
    "df_asv_UA = df_asv_filt_dx[df_asv_filt_dx['Diagnosis'] == 'Unaffected']\n",
    "df_asv_UA = df_asv_UA.drop('Diagnosis',axis=1)\n",
    "df_asv_UA.to_csv(path + 'inputs/df_asv200_UA.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "df_quant_RA = df_quant[df_quant['Diagnosis'] == 'RA']\n",
    "df_quant_RA = df_quant_RA.drop('Diagnosis',axis=1)\n",
    "df_quant_RA.to_csv(path + 'inputs/df_quant_RA.tsv', sep='\\t',index_label='SampleID')\n",
    "df_quant_UA = df_quant[df_quant['Diagnosis'] == 'Unaffected']\n",
    "df_quant_UA = df_quant_UA.drop('Diagnosis',axis=1)\n",
    "df_quant_UA.to_csv(path + 'inputs/df_quant_UA.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "# combine them all\n",
    "df_quant = pd.concat([df_meta_edit, df_olink, df_fa, df_acpa_fecal, df_acpa_plasma, df_mb, df_asv_filt, df_path_filt], axis=1)\n",
    "df_quant = pd.concat([df_meta_edit, df_olink, df_fa, df_acpa_fecal, df_acpa_plasma, df_mb, df_asv.drop(['Diagnosis'],axis=1), df_path.drop(['TwinPair','Diagnosis'],axis=1)], axis=1)\n",
    "df_quant = df_quant.drop(['TwinPair'],axis=1)\n",
    "df_quant.columns = df_quant.columns.map(lambda x: x.replace('\\'','prime'))\n",
    "df_quant.columns = df_quant.columns.map(lambda x: x.strip())\n",
    "df_quant.to_csv(path + 'inputs/df_quant_R.tsv', sep='\\t',index_label='SampleID')\n",
    "df_quant.to_csv(path + 'inputs/df_quant_R_meta.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "df_asv_filt_pre.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metabolism</th>\n",
       "      <th>Metabolism|Carbohydrate metabolism</th>\n",
       "      <th>Metabolism|Carbohydrate metabolism|Glycolysis / Gluconeogenesis</th>\n",
       "      <th>Metabolism|Carbohydrate metabolism|Pyruvate metabolism</th>\n",
       "      <th>Metabolism|Lipid metabolism</th>\n",
       "      <th>Metabolism|Lipid metabolism|Fatty acid degradation</th>\n",
       "      <th>Metabolism|Amino acid metabolism</th>\n",
       "      <th>Metabolism|Amino acid metabolism|Tyrosine metabolism</th>\n",
       "      <th>Metabolism|Metabolism of cofactors and vitamins</th>\n",
       "      <th>Metabolism|Metabolism of cofactors and vitamins|Retinol metabolism</th>\n",
       "      <th>...</th>\n",
       "      <th>Metabolism|Glycan biosynthesis and metabolism|Glycosphingolipid biosynthesis - globo and isoglobo series</th>\n",
       "      <th>Metabolism|Lipid metabolism|Biosynthesis of unsaturated fatty acids</th>\n",
       "      <th>Metabolism|Lipid metabolism|Steroid hormone biosynthesis</th>\n",
       "      <th>Metabolism|Glycan biosynthesis and metabolism|Various types of N-glycan biosynthesis</th>\n",
       "      <th>Metabolism|Glycan biosynthesis and metabolism|Glycosphingolipid biosynthesis - ganglio series</th>\n",
       "      <th>Human Diseases|Infectious disease: bacterial|Yersinia infection</th>\n",
       "      <th>Human Diseases|Infectious disease: bacterial|Bacterial invasion of epithelial cells</th>\n",
       "      <th>Metabolism|Biosynthesis of other secondary metabolites|Betalain biosynthesis</th>\n",
       "      <th>Metabolism|Metabolism of terpenoids and polyketides|Tetracycline biosynthesis</th>\n",
       "      <th>Genetic Information Processing|Information processing in viruses|Virion - Bacteriophage lambda</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>0.325986</td>\n",
       "      <td>0.064753</td>\n",
       "      <td>0.005362</td>\n",
       "      <td>0.004788</td>\n",
       "      <td>0.009472</td>\n",
       "      <td>0.000137</td>\n",
       "      <td>0.065598</td>\n",
       "      <td>0.000231</td>\n",
       "      <td>0.054545</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000258</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001356</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>0.334549</td>\n",
       "      <td>0.070002</td>\n",
       "      <td>0.005384</td>\n",
       "      <td>0.004861</td>\n",
       "      <td>0.015821</td>\n",
       "      <td>0.000146</td>\n",
       "      <td>0.062591</td>\n",
       "      <td>0.000323</td>\n",
       "      <td>0.054963</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000381</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>0.355141</td>\n",
       "      <td>0.076275</td>\n",
       "      <td>0.006307</td>\n",
       "      <td>0.005539</td>\n",
       "      <td>0.012117</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.064360</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>0.062004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000505</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>0.336738</td>\n",
       "      <td>0.071781</td>\n",
       "      <td>0.006146</td>\n",
       "      <td>0.005629</td>\n",
       "      <td>0.012767</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>0.056697</td>\n",
       "      <td>0.000188</td>\n",
       "      <td>0.060269</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000379</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000157</td>\n",
       "      <td>0.000157</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.001366</td>\n",
       "      <td>0.000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>0.347150</td>\n",
       "      <td>0.073963</td>\n",
       "      <td>0.005880</td>\n",
       "      <td>0.005605</td>\n",
       "      <td>0.013336</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>0.065695</td>\n",
       "      <td>0.000167</td>\n",
       "      <td>0.054216</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000443</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.000083</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002128</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 327 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Metabolism  Metabolism|Carbohydrate metabolism  \\\n",
       "500A    0.325986                            0.064753   \n",
       "501A    0.334549                            0.070002   \n",
       "502A    0.355141                            0.076275   \n",
       "503A    0.336738                            0.071781   \n",
       "504A    0.347150                            0.073963   \n",
       "\n",
       "      Metabolism|Carbohydrate metabolism|Glycolysis / Gluconeogenesis  \\\n",
       "500A                                           0.005362                 \n",
       "501A                                           0.005384                 \n",
       "502A                                           0.006307                 \n",
       "503A                                           0.006146                 \n",
       "504A                                           0.005880                 \n",
       "\n",
       "      Metabolism|Carbohydrate metabolism|Pyruvate metabolism  \\\n",
       "500A                                           0.004788        \n",
       "501A                                           0.004861        \n",
       "502A                                           0.005539        \n",
       "503A                                           0.005629        \n",
       "504A                                           0.005605        \n",
       "\n",
       "      Metabolism|Lipid metabolism  \\\n",
       "500A                     0.009472   \n",
       "501A                     0.015821   \n",
       "502A                     0.012117   \n",
       "503A                     0.012767   \n",
       "504A                     0.013336   \n",
       "\n",
       "      Metabolism|Lipid metabolism|Fatty acid degradation  \\\n",
       "500A                                           0.000137    \n",
       "501A                                           0.000146    \n",
       "502A                                           0.000060    \n",
       "503A                                           0.000105    \n",
       "504A                                           0.000109    \n",
       "\n",
       "      Metabolism|Amino acid metabolism  \\\n",
       "500A                          0.065598   \n",
       "501A                          0.062591   \n",
       "502A                          0.064360   \n",
       "503A                          0.056697   \n",
       "504A                          0.065695   \n",
       "\n",
       "      Metabolism|Amino acid metabolism|Tyrosine metabolism  \\\n",
       "500A                                           0.000231      \n",
       "501A                                           0.000323      \n",
       "502A                                           0.000204      \n",
       "503A                                           0.000188      \n",
       "504A                                           0.000167      \n",
       "\n",
       "      Metabolism|Metabolism of cofactors and vitamins  \\\n",
       "500A                                         0.054545   \n",
       "501A                                         0.054963   \n",
       "502A                                         0.062004   \n",
       "503A                                         0.060269   \n",
       "504A                                         0.054216   \n",
       "\n",
       "      Metabolism|Metabolism of cofactors and vitamins|Retinol metabolism  ...  \\\n",
       "500A                                           0.000000                   ...   \n",
       "501A                                           0.000013                   ...   \n",
       "502A                                           0.000000                   ...   \n",
       "503A                                           0.000006                   ...   \n",
       "504A                                           0.000049                   ...   \n",
       "\n",
       "      Metabolism|Glycan biosynthesis and metabolism|Glycosphingolipid biosynthesis - globo and isoglobo series  \\\n",
       "500A                                           0.000258                                                          \n",
       "501A                                           0.000381                                                          \n",
       "502A                                           0.000505                                                          \n",
       "503A                                           0.000379                                                          \n",
       "504A                                           0.000443                                                          \n",
       "\n",
       "      Metabolism|Lipid metabolism|Biosynthesis of unsaturated fatty acids  \\\n",
       "500A                                           0.000000                     \n",
       "501A                                           0.000023                     \n",
       "502A                                           0.000000                     \n",
       "503A                                           0.000143                     \n",
       "504A                                           0.000024                     \n",
       "\n",
       "      Metabolism|Lipid metabolism|Steroid hormone biosynthesis  \\\n",
       "500A                                           0.000011          \n",
       "501A                                           0.000006          \n",
       "502A                                           0.000029          \n",
       "503A                                           0.000000          \n",
       "504A                                           0.000083          \n",
       "\n",
       "      Metabolism|Glycan biosynthesis and metabolism|Various types of N-glycan biosynthesis  \\\n",
       "500A                                           0.000101                                      \n",
       "501A                                           0.000161                                      \n",
       "502A                                           0.000267                                      \n",
       "503A                                           0.000157                                      \n",
       "504A                                           0.000187                                      \n",
       "\n",
       "      Metabolism|Glycan biosynthesis and metabolism|Glycosphingolipid biosynthesis - ganglio series  \\\n",
       "500A                                           0.000101                                               \n",
       "501A                                           0.000161                                               \n",
       "502A                                           0.000267                                               \n",
       "503A                                           0.000157                                               \n",
       "504A                                           0.000187                                               \n",
       "\n",
       "      Human Diseases|Infectious disease: bacterial|Yersinia infection  \\\n",
       "500A                                           0.000000                 \n",
       "501A                                           0.000000                 \n",
       "502A                                           0.000000                 \n",
       "503A                                           0.000017                 \n",
       "504A                                           0.000000                 \n",
       "\n",
       "      Human Diseases|Infectious disease: bacterial|Bacterial invasion of epithelial cells  \\\n",
       "500A                                           0.000000                                     \n",
       "501A                                           0.000000                                     \n",
       "502A                                           0.000000                                     \n",
       "503A                                           0.000017                                     \n",
       "504A                                           0.000000                                     \n",
       "\n",
       "      Metabolism|Biosynthesis of other secondary metabolites|Betalain biosynthesis  \\\n",
       "500A                                           0.000000                              \n",
       "501A                                           0.000028                              \n",
       "502A                                           0.000000                              \n",
       "503A                                           0.000031                              \n",
       "504A                                           0.000000                              \n",
       "\n",
       "      Metabolism|Metabolism of terpenoids and polyketides|Tetracycline biosynthesis  \\\n",
       "500A                                           0.001356                               \n",
       "501A                                           0.000000                               \n",
       "502A                                           0.000937                               \n",
       "503A                                           0.001366                               \n",
       "504A                                           0.002128                               \n",
       "\n",
       "      Genetic Information Processing|Information processing in viruses|Virion - Bacteriophage lambda  \n",
       "500A                                           0.000000                                               \n",
       "501A                                           0.000000                                               \n",
       "502A                                           0.000000                                               \n",
       "503A                                           0.000005                                               \n",
       "504A                                           0.000000                                               \n",
       "\n",
       "[5 rows x 327 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# subset for dgca subplots\n",
    "name_to_df = {\n",
    "    'olink':df_olink,\n",
    "    'fa': df_fa,\n",
    "    'acpa_fecal': df_acpa_fecal,\n",
    "    'acpa_plasma': df_acpa_plasma,\n",
    "    'mb': df_mb,\n",
    "    'asv_filt': df_asv_filt,\n",
    "    'path_filt': df_path_filt,\n",
    "    'asv': df_asv.drop('Diagnosis',axis=1),\n",
    "    'path': df_path.drop(['TwinPair','Diagnosis'],axis=1)\n",
    "}\n",
    "for x in name_to_df:\n",
    "    df = pd.concat([df_meta_edit, name_to_df[x]], axis=1)\n",
    "    # for R to handle strings properly with ' character\n",
    "    df.columns = df.columns.map(lambda x: x.replace('\\'','prime'))\n",
    "    # eliminate trailing spaces\n",
    "    df.columns = df.columns.map(lambda x: x.strip())\n",
    "    \n",
    "    # keep meta col\n",
    "    df = df.drop(['TwinPair'],axis=1)\n",
    "    df.to_csv(path + 'inputs/df_' + x + '_meta.tsv', sep='\\t',index_label='SampleID')\n",
    "\n",
    "    # drop meta col\n",
    "    df = df.drop(['Diagnosis', 'TJC', 'SJC'],axis=1)\n",
    "    df.to_csv(path + 'inputs/df_' + x + '.tsv', sep='\\t',index_label='SampleID')\n",
    "    \n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>var_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IL8</td>\n",
       "      <td>Plasma_Cytokines</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VEGFA</td>\n",
       "      <td>Plasma_Cytokines</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>Plasma_Cytokines</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCP.3</td>\n",
       "      <td>Plasma_Cytokines</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDNF</td>\n",
       "      <td>Plasma_Cytokines</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index          var_type\n",
       "0    IL8  Plasma_Cytokines\n",
       "1  VEGFA  Plasma_Cytokines\n",
       "2   CD8A  Plasma_Cytokines\n",
       "3  MCP.3  Plasma_Cytokines\n",
       "4   GDNF  Plasma_Cytokines"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# for R, annotate quant vars\n",
    "# df_quant_R = pd.concat([df_quant, df_asv100, df_path_top100], axis=1)\n",
    "df_quant_R = df_quant.copy() #([df_quant, df_asv100, df_path_top100], axis=1)\n",
    "df_quant_R = df_quant_R.drop(['Diagnosis','SJC','TJC'],axis=1).T\n",
    "# this is in the order they were entered i.e. \n",
    "# [df_olink, df_fa, df_acpa_fecal, df_acpa_plasma, df_plasma, df_rbfa, df_mb]\n",
    " # ['rbfa']*len(df_rbfa.columns) + \\  \n",
    "df_quant_R['var_type'] = ['Plasma_Cytokines']*len(df_olink.columns) + \\\n",
    "                         ['Stool_Fatty_Acids']*len(df_fa.columns) + \\\n",
    "                         ['Fecal_Autoantibodies']*len(df_acpa_fecal.columns) + \\\n",
    "                         ['Plasma_Autoantibodies']*len(df_acpa_plasma.columns) + \\\n",
    "                         ['Serum_Fatty_Acids']*len(df_mb.columns) + \\\n",
    "                         ['Metagenomic_Taxa']*len(df_asv.drop('Diagnosis',axis=1).columns) + \\\n",
    "                         ['Metagenomic_Pathways']*len(df_path.drop(['TwinPair','Diagnosis'],axis=1).columns)\n",
    "df_quant_R = df_quant_R['var_type'].reset_index()\n",
    "df_quant_R.to_csv(path + 'inputs/df_quant_R_labels.tsv', sep='\\t')\n",
    "df_quant_R.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ab</th>\n",
       "      <th>UA_means</th>\n",
       "      <th>UA_stds</th>\n",
       "      <th>RA_means</th>\n",
       "      <th>RA_stds</th>\n",
       "      <th>Pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clusterin 231-250 sm-1 cyclic IgG_Fecal</td>\n",
       "      <td>15.0000</td>\n",
       "      <td>2.121320</td>\n",
       "      <td>16.5625</td>\n",
       "      <td>2.228193</td>\n",
       "      <td>0.040457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FibrinogenA 556-575 sm cyclic IgA_Plasma</td>\n",
       "      <td>34.8125</td>\n",
       "      <td>13.374270</td>\n",
       "      <td>46.5625</td>\n",
       "      <td>16.802971</td>\n",
       "      <td>0.019108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Biglycan 247-266 cit sm-1 cyclic IgA_Plasma</td>\n",
       "      <td>26.1250</td>\n",
       "      <td>9.346624</td>\n",
       "      <td>37.2500</td>\n",
       "      <td>11.110243</td>\n",
       "      <td>0.020316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H2B-a 62-81 cyclic IgA_Plasma</td>\n",
       "      <td>33.2500</td>\n",
       "      <td>11.065148</td>\n",
       "      <td>45.8125</td>\n",
       "      <td>16.669466</td>\n",
       "      <td>0.021194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FibrinogenA 41-60 cyclic IgA_Plasma</td>\n",
       "      <td>29.1875</td>\n",
       "      <td>11.494394</td>\n",
       "      <td>41.4375</td>\n",
       "      <td>14.952921</td>\n",
       "      <td>0.023381</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Ab  UA_means    UA_stds  RA_means  \\\n",
       "0      Clusterin 231-250 sm-1 cyclic IgG_Fecal   15.0000   2.121320   16.5625   \n",
       "1     FibrinogenA 556-575 sm cyclic IgA_Plasma   34.8125  13.374270   46.5625   \n",
       "2  Biglycan 247-266 cit sm-1 cyclic IgA_Plasma   26.1250   9.346624   37.2500   \n",
       "3                H2B-a 62-81 cyclic IgA_Plasma   33.2500  11.065148   45.8125   \n",
       "4          FibrinogenA 41-60 cyclic IgA_Plasma   29.1875  11.494394   41.4375   \n",
       "\n",
       "     RA_stds    Pvalue  \n",
       "0   2.228193  0.040457  \n",
       "1  16.802971  0.019108  \n",
       "2  11.110243  0.020316  \n",
       "3  16.669466  0.021194  \n",
       "4  14.952921  0.023381  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# table for manuscript\n",
    "df_acpas = pd.concat([df_dx, df_acpa_fecal, df_acpa_plasma], axis=1)\n",
    "\n",
    "# subset on relevant variables\n",
    "fecal = ['Clusterin 231-250 sm-1 cyclic IgG']\n",
    "ab = [f + '_Fecal' for f in fecal]\n",
    "plasma = ['FibrinogenA 556-575 sm cyclic IgA', 'Biglycan 247-266 cit sm-1 cyclic IgA',\n",
    "          'H2B-a 62-81 cyclic IgA', 'FibrinogenA 41-60 cyclic IgA', 'FibrinogenA 27-43 cit IgA',\n",
    "          'FibrinogenA 41-60 cit3 cyclic IgA', 'FibrinogenA 582-599 IgA', 'ApoE 277-296 Cit2 sm2 cyclic IgG',\n",
    "          'Vimentin 1-16 cit IgA', 'Enolase-1A 5-21 cit IgA', 'Enolase-1A 5-21 cit IgG',\n",
    "          'ApoA1 231-248 IgA', 'CCP IgA', 'FibrinogenB 36-52 cit IgA']\n",
    "plasma_ab = [p + '_Plasma' for p in plasma]\n",
    "ab.extend(plasma_ab)\n",
    "df_rel = df_acpas.loc[:,['Diagnosis'] + ab]\n",
    "\n",
    "# grab means, std dev, and pvalues for relevant variables\n",
    "UA_means = []\n",
    "UA_stds = []\n",
    "RA_means = []\n",
    "RA_stds = []\n",
    "pvals = []\n",
    "\n",
    "for x in ab:\n",
    "    UA_vals = df_rel[df_rel['Diagnosis'] == 'Unaffected'][x].values\n",
    "    UA_means.append(np.mean(UA_vals))\n",
    "    UA_stds.append(np.std(UA_vals))\n",
    "    RA_vals = df_rel[df_rel['Diagnosis'] == 'RA'][x].values\n",
    "    RA_means.append(np.mean(RA_vals))\n",
    "    RA_stds.append(np.std(RA_vals))\n",
    "    pvals.append(scipy.stats.ttest_rel(RA_vals, UA_vals)[1])\n",
    "\n",
    "df_results = pd.DataFrame({'Ab': ab, 'UA_means': UA_means, 'UA_stds': UA_stds, 'RA_means': RA_means, 'RA_stds': RA_stds, 'Pvalue': pvals})\n",
    "df_results.to_csv(path + 'outputs/jobs02/Table2.tsv', sep='\\t')\n",
    "df_results.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>var_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>2-methylbutyric acid</td>\n",
       "      <td>Serum_Fatty_Acids</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>acetate</td>\n",
       "      <td>Stool_Fatty_Acids</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>acetic acid</td>\n",
       "      <td>Serum_Fatty_Acids</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>ada</td>\n",
       "      <td>Plasma_Cytokines</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>apoa1 231-248 cit1 iga_fecal</td>\n",
       "      <td>Fecal_Autoantibodies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index              var_type\n",
       "332          2-methylbutyric acid     Serum_Fatty_Acids\n",
       "88                        acetate     Stool_Fatty_Acids\n",
       "333                   acetic acid     Serum_Fatty_Acids\n",
       "85                            ada      Plasma_Cytokines\n",
       "160  apoa1 231-248 cit1 iga_fecal  Fecal_Autoantibodies"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# for corr heatmaps \n",
    "df_quant_R = df_quant.copy() #([df_quant, df_asv100, df_path_top100], axis=1)\n",
    "df_quant_R = df_quant_R.drop(['Diagnosis'],axis=1).T\n",
    "# this is in the order they were entered i.e. \n",
    "# [df_olink, df_fa, df_acpa_fecal, df_acpa_plasma, df_plasma, df_rbfa, df_mb]\n",
    " # ['rbfa']*len(df_rbfa.columns) + \\  \n",
    "df_quant_R['var_type'] = ['Clinical']*2 + \\\n",
    "                         ['Plasma_Cytokines']*len(df_olink.columns) + \\\n",
    "                         ['Stool_Fatty_Acids']*len(df_fa.columns) + \\\n",
    "                         ['Fecal_Autoantibodies']*len(df_acpa_fecal.columns) + \\\n",
    "                         ['Plasma_Autoantibodies']*len(df_acpa_plasma.columns) + \\\n",
    "                         ['Serum_Fatty_Acids']*len(df_mb.columns) + \\\n",
    "                         ['Metagenomic_ASVs']*len(df_asv.drop('Diagnosis',axis=1).columns) + \\\n",
    "                         ['Metagenomic_Pathways']*len(df_path.drop(['TwinPair','Diagnosis'],axis=1).columns)\n",
    "df_quant_R = df_quant_R[~df_quant_R['var_type'].isin(['Metagenomic_ASVs','Metagenomic_Pathways'])]\n",
    "df_quant_R = df_quant_R['var_type'].reset_index()\n",
    "df_quant_R['index'] = df_quant_R['index'].str.lower()\n",
    "df_quant_R = df_quant_R.sort_values(by='index',ascending=True)\n",
    "df_quant_R.to_csv(path + 'inputs/df_quant_R_corr_labels.tsv', sep='\\t')\n",
    "df_quant_R.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of diff inflam markers\n",
      "23\n",
      "Number of TP RA corrs involving b faecis\n",
      "0\n",
      "Number of TP RA corrs involving b faecis and diff markers\n",
      "0\n",
      "Number of TP UA corrs involving b faecis\n",
      "30\n",
      "Number of TP UA corrs involving b faecis and diff markers\n",
      "0\n",
      "set()\n"
     ]
    }
   ],
   "source": [
    "# asv analysis\n",
    "# grab differential variables from the t-tests\n",
    "int_var = []\n",
    "for p in ['13_metabolon', '12_olink', '11_acpa_plasma', '10_acpa_fecal', '8_metabolites']:\n",
    "    df = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/' + p + '/' + p + '_stats_clean.csv')\n",
    "    df = df[df['ttest'] < 0.05]\n",
    "    int_var.extend(list(df['feature'].values))\n",
    "print('Number of diff inflam markers')\n",
    "print(len(int_var))\n",
    "# grab CUTIE results in UA and RA groups with these pathways \n",
    "# grab jobs18 RA and jobs17 UA results\n",
    "df_RA = pd.read_csv(path + 'outputs/jobs18/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "df_RA = df_RA[df_RA['class'] == 'TP']\n",
    "df_RA = df_RA[df_RA['var1'] == 'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis']\n",
    "print('Number of TP RA corrs involving b faecis')\n",
    "print(len(df_RA))\n",
    "#  [i + j for i, j in zip(test_list1, test_list2)]\n",
    "RA_corrs = [i + '_corr_with_' + j for i,j in zip(list(df_RA['var1']),list(df_RA['var2']))]\n",
    "print('Number of TP RA corrs involving b faecis and diff markers')\n",
    "print(len(df_RA[df_RA['var2'].isin(int_var)]))\n",
    "      \n",
    "df_UA = pd.read_csv(path + 'outputs/jobs17/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "df_UA = df_UA[df_UA['class'] == 'TP']\n",
    "df_UA = df_UA[df_UA['var1'] == 'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis']\n",
    "print('Number of TP UA corrs involving b faecis')\n",
    "print(len(df_UA))\n",
    "UA_corrs = [i + '_corr_with_' + j for i,j in zip(list(df_UA['var1']),list(df_UA['var2']))]\n",
    "print('Number of TP UA corrs involving b faecis and diff markers')\n",
    "print(len(df_UA[df_UA['var2'].isin(int_var)]))\n",
    "      \n",
    "print(set(RA_corrs).intersection(set(UA_corrs)))\n",
    "# path_int\n",
    "# print(len(df_quant_R))\n",
    "# df_RA[df_RA['var2'].isin(int_var)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>var1</th>\n",
       "      <th>var2</th>\n",
       "      <th>pvalues</th>\n",
       "      <th>correlations</th>\n",
       "      <th>r2vals</th>\n",
       "      <th>TP_rev_indicators</th>\n",
       "      <th>FP_rev_indicators</th>\n",
       "      <th>extreme_p</th>\n",
       "      <th>extreme_r</th>\n",
       "      <th>p_ratio</th>\n",
       "      <th>r2_ratio</th>\n",
       "      <th>class</th>\n",
       "      <th>reverse</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>141960</th>\n",
       "      <td>k  Bacteria|p  Firmicutes|c  Clostridia|o  Eub...</td>\n",
       "      <td>SCF</td>\n",
       "      <td>0.002008</td>\n",
       "      <td>0.904762</td>\n",
       "      <td>0.818594</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.013697</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>6.820442</td>\n",
       "      <td>0.897507</td>\n",
       "      <td>TP</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     var1 var2   pvalues  \\\n",
       "141960  k  Bacteria|p  Firmicutes|c  Clostridia|o  Eub...  SCF  0.002008   \n",
       "\n",
       "        correlations    r2vals  TP_rev_indicators  FP_rev_indicators  \\\n",
       "141960      0.904762  0.818594               -1.0               -1.0   \n",
       "\n",
       "        extreme_p  extreme_r   p_ratio  r2_ratio class reverse  \n",
       "141960   0.013697   0.857143  6.820442  0.897507    TP      No  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "df_UA = pd.read_csv(path + 'outputs/jobs17/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "df_UA = df_UA[df_UA['var1'] == 'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis']\n",
    "df_UA = df_UA[df_UA['var2'] == 'SCF']\n",
    "df_UA\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of diff inflam markers\n",
      "23\n",
      "Number of TP RA corrs involving int paths\n",
      "34\n",
      "Number of TP RA corrs involving int paths and int vars\n",
      "2\n",
      "Number of TP UA corrs involving int paths\n",
      "48\n",
      "Number of TP UA corrs involving int paths and int vars\n",
      "0\n",
      "set()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>var1</th>\n",
       "      <th>var2</th>\n",
       "      <th>pvalues</th>\n",
       "      <th>correlations</th>\n",
       "      <th>r2vals</th>\n",
       "      <th>TP_rev_indicators</th>\n",
       "      <th>FP_rev_indicators</th>\n",
       "      <th>extreme_p</th>\n",
       "      <th>extreme_r</th>\n",
       "      <th>p_ratio</th>\n",
       "      <th>r2_ratio</th>\n",
       "      <th>class</th>\n",
       "      <th>reverse</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30806</th>\n",
       "      <td>Metabolism|Biosynthesis of other secondary met...</td>\n",
       "      <td>CXCL1</td>\n",
       "      <td>0.00653</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>0.734694</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.036238</td>\n",
       "      <td>0.785714</td>\n",
       "      <td>5.549520</td>\n",
       "      <td>0.840278</td>\n",
       "      <td>TP</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47222</th>\n",
       "      <td>Metabolism|Metabolism of other amino acids|Cya...</td>\n",
       "      <td>CXCL1</td>\n",
       "      <td>0.00385</td>\n",
       "      <td>0.880952</td>\n",
       "      <td>0.776077</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.023449</td>\n",
       "      <td>0.821429</td>\n",
       "      <td>6.090093</td>\n",
       "      <td>0.869430</td>\n",
       "      <td>TP</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    var1   var2  pvalues  \\\n",
       "30806  Metabolism|Biosynthesis of other secondary met...  CXCL1  0.00653   \n",
       "47222  Metabolism|Metabolism of other amino acids|Cya...  CXCL1  0.00385   \n",
       "\n",
       "       correlations    r2vals  TP_rev_indicators  FP_rev_indicators  \\\n",
       "30806      0.857143  0.734694               -1.0               -1.0   \n",
       "47222      0.880952  0.776077               -1.0               -1.0   \n",
       "\n",
       "       extreme_p  extreme_r   p_ratio  r2_ratio class reverse  \n",
       "30806   0.036238   0.785714  5.549520  0.840278    TP      No  \n",
       "47222   0.023449   0.821429  6.090093  0.869430    TP      No  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pathway analysis of sig pathways\n",
    "path_int = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/16_path/16_path_stats_clean.csv')\n",
    "path_int = path_int[path_int['wilcoxon'] < 0.05]\n",
    "path_int = list(path_int['feature'].values)\n",
    "\n",
    "# jk we do this\n",
    "path_int = ['Metabolism|Xenobiotics biodegradation and metabolism|Atrazine degradation',\n",
    "'Cellular Processes',\n",
    "'Human Diseases|Cardiovascular disease|Fluid shear stress and atherosclerosis',\n",
    "'Metabolism|Amino acid metabolism|Lysine biosynthesis',\n",
    "'Metabolism|Biosynthesis of other secondary metabolites|Biosynthesis of various plant secondary metabolites',\n",
    "'Metabolism|Biosynthesis of other secondary metabolites|Monobactam biosynthesis',\n",
    "'Metabolism|Metabolism of other amino acids|Cyanoamino acid metabolism']\n",
    "\n",
    "# grab differential variables from the t-tests\n",
    "int_var = []\n",
    "for p in ['13_metabolon', '12_olink', '11_acpa_plasma', '10_acpa_fecal', '8_metabolites']:\n",
    "    df = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/' + p + '/' + p + '_stats_clean.csv')\n",
    "    df = df[df['ttest'] < 0.05]\n",
    "    int_var.extend(list(df['feature'].values))\n",
    "\n",
    "print('Number of diff inflam markers')\n",
    "print(len(int_var))\n",
    "# grab CUTIE results in UA and RA groups with these pathways \n",
    "# grab jobs20 RA and jobs21 UA results\n",
    "df_RA = pd.read_csv(path + 'outputs/jobs20/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "df_RA = df_RA[df_RA['class'] == 'TP']\n",
    "df_RA = df_RA[df_RA['var1'].isin(path_int)]\n",
    "print('Number of TP RA corrs involving int paths')\n",
    "print(len(df_RA))\n",
    "#  [i + j for i, j in zip(test_list1, test_list2)]\n",
    "RA_corrs = [i + '_corr_with_' + j for i,j in zip(list(df_RA['var1']),list(df_RA['var2']))]\n",
    "print('Number of TP RA corrs involving int paths and int vars')\n",
    "print(len(df_RA[df_RA['var2'].isin(int_var)]))\n",
    "      \n",
    "df_UA = pd.read_csv(path + 'outputs/jobs21/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "df_UA = df_UA[df_UA['class'] == 'TP']\n",
    "df_UA = df_UA[df_UA['var1'].isin(path_int)]\n",
    "print('Number of TP UA corrs involving int paths')\n",
    "print(len(df_UA))\n",
    "UA_corrs = [i + '_corr_with_' + j for i,j in zip(list(df_UA['var1']),list(df_UA['var2']))]\n",
    "print('Number of TP UA corrs involving int paths and int vars')\n",
    "print(len(df_UA[df_UA['var2'].isin(int_var)]))\n",
    "      \n",
    "print(set(RA_corrs).intersection(set(UA_corrs)))\n",
    "# path_int\n",
    "# print(len(df_quant_R))\n",
    "df_RA[df_RA['var2'].isin(int_var)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "56\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>var1</th>\n",
       "      <th>var2</th>\n",
       "      <th>pvalues</th>\n",
       "      <th>correlations</th>\n",
       "      <th>r2vals</th>\n",
       "      <th>TP_rev_indicators</th>\n",
       "      <th>FP_rev_indicators</th>\n",
       "      <th>extreme_p</th>\n",
       "      <th>extreme_r</th>\n",
       "      <th>p_ratio</th>\n",
       "      <th>r2_ratio</th>\n",
       "      <th>class</th>\n",
       "      <th>reverse</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8232</th>\n",
       "      <td>Metabolism|Amino acid metabolism|Lysine biosyn...</td>\n",
       "      <td>IL.1.alpha</td>\n",
       "      <td>0.160026</td>\n",
       "      <td>0.547619</td>\n",
       "      <td>0.299887</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.248998</td>\n",
       "      <td>0.0</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8234</th>\n",
       "      <td>Metabolism|Amino acid metabolism|Lysine biosyn...</td>\n",
       "      <td>CXCL1</td>\n",
       "      <td>0.455645</td>\n",
       "      <td>0.309524</td>\n",
       "      <td>0.095805</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.194692</td>\n",
       "      <td>0.0</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8263</th>\n",
       "      <td>Metabolism|Amino acid metabolism|Lysine biosyn...</td>\n",
       "      <td>MMP.10</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8266</th>\n",
       "      <td>Metabolism|Amino acid metabolism|Lysine biosyn...</td>\n",
       "      <td>CCL23</td>\n",
       "      <td>0.651401</td>\n",
       "      <td>-0.190476</td>\n",
       "      <td>0.036281</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.535152</td>\n",
       "      <td>0.0</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8285</th>\n",
       "      <td>Metabolism|Amino acid metabolism|Lysine biosyn...</td>\n",
       "      <td>CCL25</td>\n",
       "      <td>0.492726</td>\n",
       "      <td>-0.285714</td>\n",
       "      <td>0.081633</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.029525</td>\n",
       "      <td>0.0</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   var1        var2   pvalues  \\\n",
       "8232  Metabolism|Amino acid metabolism|Lysine biosyn...  IL.1.alpha  0.160026   \n",
       "8234  Metabolism|Amino acid metabolism|Lysine biosyn...       CXCL1  0.455645   \n",
       "8263  Metabolism|Amino acid metabolism|Lysine biosyn...      MMP.10  1.000000   \n",
       "8266  Metabolism|Amino acid metabolism|Lysine biosyn...       CCL23  0.651401   \n",
       "8285  Metabolism|Amino acid metabolism|Lysine biosyn...       CCL25  0.492726   \n",
       "\n",
       "      correlations    r2vals  TP_rev_indicators  FP_rev_indicators  extreme_p  \\\n",
       "8232      0.547619  0.299887                0.0                0.0        1.0   \n",
       "8234      0.309524  0.095805                0.0                0.0        1.0   \n",
       "8263      0.000000  0.000000                0.0                0.0        1.0   \n",
       "8266     -0.190476  0.036281                0.0                0.0        1.0   \n",
       "8285     -0.285714  0.081633                0.0                0.0        1.0   \n",
       "\n",
       "      extreme_r   p_ratio  r2_ratio class reverse  \n",
       "8232        0.0  6.248998       0.0     N     NaN  \n",
       "8234        0.0  2.194692       0.0     N     NaN  \n",
       "8263        0.0  1.000000       NaN     N     NaN  \n",
       "8266        0.0  1.535152       0.0     N     NaN  \n",
       "8285        0.0  2.029525       0.0     N     NaN  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_RA = pd.read_csv(path + 'outputs/jobs20/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "df_RA = df_RA[df_RA['var1'].isin(path_int)]\n",
    "df_RA = df_RA[df_RA['var2'].isin(int_var)]\n",
    "print(len(df_RA))\n",
    "df_RA.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "incomplete input (798214678.py, line 6)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn[38], line 6\u001b[0;36m\u001b[0m\n\u001b[0;31m    print(len(\u001b[0m\n\u001b[0m              ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m incomplete input\n"
     ]
    }
   ],
   "source": [
    "df_UA = pd.read_csv(path + 'outputs/jobs21/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "df_UA = df_UA[df_UA['var1'].isin(['Metabolism|Biosynthesis of other secondary metabolites|Monobactam biosynthesis',\n",
    "                                 'Metabolism|Metabolism of other amino acids|Cyanoamino acid metabolism'])]\n",
    "df_UA = df_UA[df_UA['var2'] == 'CXCL1']\n",
    "df_UA\n",
    "print(len())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Scratch #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TtestResult(statistic=1.0306243614492072, pvalue=0.34245913778236703, df=6.0)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Axes: xlabel='MTX', ylabel='shannon_entropy'>"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjcAAAGwCAYAAABVdURTAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAmnUlEQVR4nO3dfXRU9YH/8c+EPAGHDCsPeWgz4UkEwoMh6ZrIWk4bS7ZQWavt2bpoFFSa4hYhTeOJHAUqIdpDK3JwoYFAiLTA7oZaVFyhSoIrsMtjj9jwsIJJjAkxKhlEkkAyvz88pMwvCSR3JtzJN+/XOXNgvvO9w2c4Z8iHe7/3XofH4/EIAADAEEF2BwAAAPAnyg0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFGC7Q5gh5aWFn3yyScaMGCAHA6H3XEAAEAneDweXbhwQTExMQoK6nj/TK8sN5988oliY2PtjgEAACyorKzUN7/5zQ5f75XlZsCAAZK+/suJiIiwOQ0AAOgMt9ut2NjY1p/jHemV5ebqoaiIiAjKDQAAPcyNlpSwoBgAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGIVyAwAAjEK5AQAARqHcAAAAo1BuAACAUXrlXcHROQ0NDaqoqLA7Bq7hcrkUHh5udwwACGiUG3SooqJCc+fOtTsGrpGfn6/Ro0fbHQMAAhrlBh1yuVzKz8+3O4bPysvLlZubq0WLFikuLs7uOD5xuVx2RwCAgEe5QYfCw8ON2ksQFxdn1OcBALSPBcUAAMAolBsAAGAU28tNVVWVHnzwQQ0aNEj9+vXT7bffrsOHD193m9LSUiUmJio8PFwjRozQ2rVrb1JaAAAQ6Gxdc/PFF19oypQp+s53vqM333xTQ4cO1YcffqiBAwd2uM3Zs2c1ffp0Pf7449q8ebPee+89zZs3T0OGDNH9999/88IDAICAZGu5eeGFFxQbG6uNGze2jg0bNuy626xdu1Yul0srV66UJI0dO1aHDh3SihUrOiw3jY2NamxsbH3udrt9zg4AAAKTrYelduzYoaSkJP34xz/W0KFDlZCQoHXr1l13m/3792vatGleY2lpaTp06JAuX77c7jZ5eXlyOp2tj9jYWL99BgAAEFhsLTdnzpzRmjVrdOutt+qtt95SRkaG5s+fr6Kiog63qampUWRkpNdYZGSkrly5orq6una3ycnJUX19feujsrLSr58DAAAEDlsPS7W0tCgpKUnLly+XJCUkJOiDDz7QmjVrlJ6e3uF2DofD67nH42l3/KqwsDCFhYX5KTUAAAhktu65iY6O1rhx47zGxo4de937GUVFRammpsZrrLa2VsHBwRo0aFC35AQAAD2HreVmypQpOnnypNfYqVOnrnuJ/JSUFO3evdtrbNeuXUpKSlJISEi35AQAAD2HreVm4cKFOnDggJYvX67/+7//0x/+8Afl5+friSeeaJ2Tk5PjdYgqIyND5eXlyszMVFlZmTZs2KCCggJlZWXZ8REAAECAsbXcfOtb39If//hHbdmyRePHj9dzzz2nlStXatasWa1zqqurvQ5TDR8+XDt37lRJSYluv/12Pffcc1q1ahXXuAEAAJIC4MaZP/jBD/SDH/ygw9cLCwvbjE2dOlVHjhzpxlQAAKCnsv32CwAAAP5EuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGIVyAwAAjEK5AQAARqHcAAAAo1BuAACAUSg3AADAKJQbAABgFMoNAAAwCuUGAAAYhXIDAACMQrkBAABGodwAAACjUG4AAIBRKDcAAMAolBsAAGAUyg0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGIVyAwAAjEK5AQAARqHcAAAAo1BuAACAUSg3AADAKJQbAABgFMoNAAAwCuUGAAAYhXIDAACMQrkBAABGodwAAACjUG4AAIBRKDcAAMAolBsAAGAUyg0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGIVyAwAAjEK5AQAARqHcAAAAo1BuAACAUSg3AADAKMF2BzDVuXPnVF9fb3cMSCovL/f6FfZyOp2KjIy0OwYAgzk8Ho/H7hA3m9vtltPpVH19vSIiIvz+/ufOndODD6XrclOj398b6OlCQsO0+ZUiCg6ALuvsz2/23HSD+vp6XW5q1KURU9US7rQ7DhAwghrqpTOlqq+vp9wA6DaUm27UEu5US//BdscAAKBXsXVB8ZIlS+RwOLweUVFRHc4vKSlpM9/hcOjEiRM3MTUAAAhktu+5iY+P15///OfW53369LnhNidPnvQ61jZkyJBuyQYAAHoe28tNcHDwdffWtGfo0KEaOHBg9wQCAAA9mu3XuTl9+rRiYmI0fPhw/eQnP9GZM2duuE1CQoKio6OVmpqqPXv23HB+Y2Oj3G631wMAAJjJ1nJzxx13qKioSG+99ZbWrVunmpoa3Xnnnfrss8/anR8dHa38/HwVFxdr+/btuu2225Samqq9e/de98/Jy8uT0+lsfcTGxnbHxwEAAAEgoK5zc/HiRY0cOVLZ2dnKzMzs1Db33HOPHA6HduzY0eGcxsZGNTb+7ZozbrdbsbGx3Xadm1OnTmnu3Lm6OG4mZ0sB1wi6WKf+f92h/Px8jR492u44AHqYzl7nxvbDUtfq37+/JkyYoNOnT3d6m+Tk5BvODwsLU0REhNcDAACYKaDKTWNjo8rKyhQdHd3pbY4ePdql+QAAwGy2ni2VlZWle+65Ry6XS7W1tVq2bJncbrcefvhhSVJOTo6qqqpUVFQkSVq5cqWGDRum+Ph4NTU1afPmzSouLlZxcbGdHwMAAAQQW8vNxx9/rAceeEB1dXUaMmSIkpOTdeDAAcXFxUmSqqurVVFR0Tq/qalJWVlZqqqqUt++fRUfH6833nhD06dPt+sjAACAAGNrudm6det1Xy8sLPR6np2drezs7G5MBAAAerqAWnMDAADgK8oNAAAwCuUGAAAYhXIDAACMQrkBAABGodwAAACjUG4AAIBRKDcAAMAolBsAAGAUyg0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGMVSuVmyZInKy8v9nQUAAMBnlsrNa6+9ppEjRyo1NVV/+MMf1NDQ4O9cAAAAllgqN4cPH9aRI0c0ceJELVy4UNHR0frZz36mgwcP+jsfAABAl1heczNx4kS9+OKLqqqq0oYNG1RVVaUpU6ZowoQJeumll1RfX+/PnAAAAJ3i84LilpYWNTU1qbGxUR6PR7fccovWrFmj2NhYbdu2zR8ZAQAAOs1yuTl8+LD+9V//VdHR0Vq4cKESEhJUVlam0tJSnThxQosXL9b8+fP9mRUAAOCGLJWbiRMnKjk5WWfPnlVBQYEqKyv1/PPPa9SoUa1z0tPT9emnn/otKAAAQGcEW9noxz/+sebMmaNvfOMbHc4ZMmSIWlpaLAcDAACwwlK5eeaZZ1p/7/F4JEkOh8M/iQAAAHxgec1NQUGBxo8fr/DwcIWHh2v8+PFav369P7MBAAB0meU9Ny+++KJ+/vOfKyUlRZK0f/9+LVy4UB999JGWLVvm15AAAACdZancrFmzRuvWrdMDDzzQOjZz5kxNnDhRP//5zyk3AADANpYOSzU3NyspKanNeGJioq5cueJzKAAAAKsslZsHH3xQa9asaTOen5+vWbNm+RwKAADAKkuHpaSvFxTv2rVLycnJkqQDBw6osrJS6enpyszMbJ3329/+1veUAAAAnWSp3Bw/flyTJ0+WJH344YeSvr6uzZAhQ3T8+PHWeZweDgAAbjZL5WbPnj3+zgEAAOAXPt848+OPP1ZVVZU/sgAAAPjMUrlpaWnRr371KzmdTsXFxcnlcmngwIF67rnnuOUCAACwlaXDUosWLVJBQYGef/55TZkyRR6PR++9956WLFmihoYG5ebm+jsnAABAp1gqN5s2bdL69es1c+bM1rFJkybpG9/4hubNm0e5AQAAtrF0WOrzzz/XmDFj2oyPGTNGn3/+uc+hAAAArLJUbiZNmqTVq1e3GV+9erUmTZrkcygAAACrLB2W+vWvf60ZM2boz3/+s1JSUuRwOLRv3z5VVlZq586d/s4IAPCzhoYGVVRU2B0D13C5XAoPD7c7hhEslZupU6fq1KlTevnll3XixAl5PB7dd999mjdvnmJiYvydEQDgZxUVFZo7d67dMXCN/Px8jR492u4YRuhyubl8+bKmTZum3/3udywcBoAeyuVyKT8/3+4YPisvL1dubq4WLVqkuLg4u+P4xOVy2R3BGF0uNyEhITp+/Di3VgCAHiw8PNyovQRxcXFGfR74xtKC4vT0dBUUFPg7CwAAgM8srblpamrS+vXrtXv3biUlJal///5er3MncAAAYBef7wp+6tQpvwYCAADwBXcFBwAARrG05mbOnDm6cOFCm/GLFy9qzpw5PocCAACwylK52bRpky5dutRm/NKlSyoqKvI5FAAAgFVdOizldrvl8Xjk8Xh04cIFryspNjc3a+fOnRo6dKjfQwIAAHRWl8rNwIED5XA45HA42r2egMPh0NKlS/0WDgAAoKu6VG727Nkjj8ej7373uyouLtYtt9zS+lpoaKji4uK4/QIAALBVl8rN1KlTJUlnz55VbGysgoIsLdkBAADoNpZOBY+Li9P58+f1v//7v6qtrVVLS4vX6+np6X4JBwAA0FWWys1rr72mWbNm6eLFixowYIDXfaYcDgflBgAA2MbScaVf/OIXrde6OX/+vL744ovWx+eff+7vjAAAAJ1mac9NVVWV5s+fr379+vk7DwAEvHPnzqm+vt7uGJBUXl7u9Svs5XQ6FRkZaXcMa+UmLS1Nhw4d0ogRI/ydBwAC2rlz5/TgQ+m63NRodxRcIzc31+4IkBQSGqbNrxTZXnAslZsZM2bol7/8pf76179qwoQJCgkJ8Xp95syZfgkHAIGmvr5el5sadWnEVLWEO+2OAwSMoIZ66Uyp6uvre2a5efzxxyVJv/rVr9q85nA41Nzc7FsqAAhwLeFOtfQfbHcMAO2wVG7+/1O/0b6gS+ftjgAEFL4TAG4GS+XmWg0NDV73mMLf9D271+4IAAD0OpbKTXNzs5YvX661a9fq3LlzOnXqlEaMGKFnnnlGw4YN06OPPurvnD3SpeHfVkvfgXbHAAJG0KXzlH4A3c5SucnNzdWmTZv061//unX9jSRNmDBBL774YqfLzZIlS9rcaDMyMlI1NTUdblNaWqrMzEx98MEHiomJUXZ2tjIyMqx8jG7X0ncgx+QBALjJLF3Er6ioSPn5+Zo1a5b69OnTOj5x4kSdOHGiS+8VHx+v6urq1sf777/f4dyzZ89q+vTpuuuuu3T06FE9/fTTmj9/voqLi618DAAAYCDLF/EbNWpUm/GWlhZdvny5awGCgxUVFdWpuWvXrpXL5dLKlSslSWPHjtWhQ4e0YsUK3X///R1u19jYqMbGv12Twu12dykjAADoOSztuYmPj9e7777bZvw//uM/lJCQ0KX3On36tGJiYjR8+HD95Cc/0ZkzZzqcu3//fk2bNs1r7OoFBa9XqvLy8uR0OlsfsbGxXcoIAAB6Dkt7bhYvXqyHHnpIVVVVamlp0fbt23Xy5EkVFRXp9ddf7/T73HHHHSoqKtLo0aN17tw5LVu2THfeeac++OADDRo0qM38mpqaNhcGioyM1JUrV1RXV6fo6Oh2/5ycnBxlZma2Pne73RQcAAAMZanc3HPPPdq2bZuWL18uh8OhZ599VpMnT9Zrr72m733ve51+n+9///utv58wYYJSUlI0cuRIbdq0yauMXOvaO5BLksfjaXf8WmFhYQoLC+t0LgAA0HNZvs5NWlqa0tLSrjtny5Ytmjlzpvr379+p9+zfv78mTJig06dPt/t6VFRUmzOpamtrFRwc3O6eHgAA0PtYWnPTWT/96U917ty5Ts9vbGxUWVlZh4eXUlJStHv3bq+xXbt2KSkpqc39rQAAQO/UreXm6iGjjmRlZam0tFRnz57V//zP/+hHP/qR3G63Hn74YUlfr5VJT09vnZ+RkaHy8nJlZmaqrKxMGzZsUEFBgbKysrrzYwAAgB7E59sv+OLjjz/WAw88oLq6Og0ZMkTJyck6cOCA4uLiJEnV1dWqqKhonT98+HDt3LlTCxcu1Msvv6yYmBitWrXquqeBAwCA3sXWcrN169brvl5YWNhmbOrUqTpy5Eg3JQIAAD1dtx6WAgAAuNkoNwAAwCjdWm7i4uI4iwkAANxUPq25aWpqUm1trVpaWrzGXS6XJOn48eO+vD0AAECXWSo3p0+f1pw5c7Rv3z6vcY/HI4fDoebmZr+EAwAA6CpL5eaRRx5RcHCwXn/9dUVHR1/31gcAAAA3k6Vyc+zYMR0+fFhjxozxdx4AAACfWFpQPG7cONXV1fk7CwAAgM8slZsXXnhB2dnZKikp0WeffSa32+31AAAAsIulw1J33323JCk1NdVrnAXFAADAbpbKzZ49e/ydAwAAwC8slZupU6f6OwcAAIBfWL6I3/nz51VQUKCysjI5HA6NGzdOc+bMkdPp9Gc+AACALrG0oPjQoUMaOXKkXnzxRX3++eeqq6vTb3/7W40cOZI7dgMAAFtZ2nOzcOFCzZw5U+vWrVNw8NdvceXKFT322GNasGCB9u7d69eQAAAAnWWp3Bw6dMir2EhScHCwsrOzlZSU5LdwAAAAXWXpsFRERIQqKirajFdWVmrAgAE+hwIAALDKUrn553/+Zz366KPatm2bKisr9fHHH2vr1q167LHH9MADD/g7IwAAQKdZOiy1YsUKORwOpaen68qVK5KkkJAQ/exnP9Pzzz/v14AAAABdYanchIaG6qWXXlJeXp4+/PBDeTwejRo1Sv369fN3PgAAgC6xfJ0bSerXr58mTJjgrywAAAA+s1RuLl68qOeff15vv/22amtr1dLS4vX6mTNn/BIOAACgqyyVm8cee0ylpaV66KGHFB0dLYfD4e9cAAAAllgqN2+++abeeOMNTZkyxd95AAAAfGLpVPC/+7u/0y233OLvLAAAAD6zVG6ee+45Pfvss/rqq6/8nQcAAMAnlg5L/eY3v9GHH36oyMhIDRs2TCEhIV6vc/NMAABgF0vl5t577/VzDAAAAP+wVG4WL17s7xwAAAB+4dNF/Jqamtq9zo3L5fIpFAAAgFWWys2pU6f06KOPat++fV7jHo9HDodDzc3NfgkHAADQVZbKzezZsxUcHKzXX3+di/gBAICAYqncHDt2TIcPH9aYMWP8nQcAAMAnlq5zM27cONXV1fk7CwAAgM8slZsXXnhB2dnZKikp0WeffSa32+31AAAAsIulw1J33323JCk1NdVrnAXFAADAbpbKzZ49e/ydAwAAwC8slZupU6f6OwcAAIBf+HQRv6+++koVFRVqamryGp84caJPoQAAAKyyVG4+/fRTzZ49W2+++Wa7r7PmBgAA2MXS2VILFizQF198oQMHDqhv3776r//6L23atEm33nqrduzY4e+MAAAAnWZpz80777yjP/3pT/rWt76loKAgxcXF6Xvf+54iIiKUl5enGTNm+DsnAABAp1jac3Px4kUNHTpUknTLLbfo008/lSRNmDBBR44c8V86AACALrJUbm677TadPHlSknT77bfrd7/7naqqqrR27VpFR0f7NSAAAEBXWDostWDBAlVXV0uSFi9erLS0NP3+979XaGioCgsL/ZkPAACgSyyVm1mzZrX+PiEhQR999JFOnDghl8ulwYMH+y0cAABAV/l0nZur+vXrp8mTJ/vjrQAAAHxiqdw0NzersLBQb7/9tmpra9XS0uL1+jvvvOOXcAAAAF1lqdw8+eSTKiws1IwZMzR+/Hg5HA5/5wIAALDEUrnZunWr/v3f/13Tp0/3dx4AAACfWDoVPDQ0VKNGjfJ3FgAAAJ9ZKje/+MUv9NJLL8nj8fg7DwAAgE86fVjqvvvu83r+zjvv6M0331R8fLxCQkK8Xtu+fbt/0gEAAHRRp8uN0+n0ev7DH/7Q72EAAAB81elys3HjxtbfX7p0SS0tLerfv78k6aOPPtKrr76qsWPHKi0tzf8pAQAAOsnSmpt/+qd/0iuvvCJJOn/+vJKTk/Wb3/xG9957r9asWePXgAAAAF1hqdwcOXJEd911lyTpP//zPxUZGany8nIVFRVp1apVfg0IAADQFZbKzVdffaUBAwZIknbt2qX77rtPQUFBSk5OVnl5uV8DAgAAdIWlcjNq1Ci9+uqrqqys1FtvvaVp06ZJkmpraxUREeHXgAAAAF1hqdw8++yzysrK0rBhw3THHXcoJSVF0td7cRISEvwaEAAAoCss3X7hRz/6kf7hH/5B1dXVmjRpUut4amoqp4gDAABbWSo3khQVFaWoqCivsb//+7/3ORAAAIAvLB2WAgAACFSUGwAAYBTLh6VwY0EN9XZHAAIK3wkAN0PAlJu8vDw9/fTTevLJJ7Vy5cp255SUlOg73/lOm/GysjKNGTOmmxN2ntPpVEhomHSm1O4oQMAJCQ1rc686APCngCg3Bw8eVH5+viZOnNip+SdPnvS6ns6QIUO6K5olkZGR2vxKkerr+V9qICgvL1dubq4WLVqkuLg4u+P0ek6nU5GRkXbHAGAw28vNl19+qVmzZmndunVatmxZp7YZOnSoBg4c2L3BfBQZGck/4AEmLi5Oo0ePtjsGAKCb2b6g+IknntCMGTN09913d3qbhIQERUdHKzU1VXv27Lnh/MbGRrndbq8HAAAwk617brZu3aojR47o4MGDnZofHR2t/Px8JSYmqrGxUa+88opSU1NVUlKib3/72x1ul5eXp6VLl/orNgAACGC2lZvKyko9+eST2rVrl8LDwzu1zW233abbbrut9XlKSooqKyu1YsWK65abnJwcZWZmtj53u92KjY21Hh4AAAQs2w5LHT58WLW1tUpMTFRwcLCCg4NVWlqqVatWKTg4WM3NzZ16n+TkZJ0+ffq6c8LCwhQREeH1AAAAZrJtz01qaqref/99r7HZs2drzJgxeuqpp9SnT59Ovc/Ro0cVHR3dHREBAEAPZFu5GTBggMaPH+811r9/fw0aNKh1PCcnR1VVVSoqKpIkrVy5UsOGDVN8fLyampq0efNmFRcXq7i4+KbnBwAAgcn2U8Gvp7q6WhUVFa3Pm5qalJWVpaqqKvXt21fx8fF64403NH36dBtTAgCAQBJQ5aakpMTreWFhodfz7OxsZWdn37xAAACgx7H9OjcAAAD+RLkBAABGodwAAACjUG4AAIBRKDcAAMAolBsAAGAUyg0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGIVyAwAAjEK5AQAARqHcAAAAo1BuAACAUSg3AADAKJQbAABgFMoNAAAwCuUGAAAYhXIDAACMQrkBAABGodwAAACjUG4AAIBRKDcAAMAolBsAAGAUyg0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGCXY7gAA0BMFXTpvdwQgoATSd4JyAwAW9D271+4IADpAuQEACy4N/7Za+g60OwYQMIIunQ+Y0k+5AQALWvoOVEv/wXbHANAOFhQDAACjUG4AAIBRKDcAAMAolBsAAGAUyg0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGIVyAwAAjEK5AQAARqHcAAAAo1BuAACAUSg3AADAKJQbAABgFMoNAAAwCuUGAAAYJWDKTV5enhwOhxYsWHDdeaWlpUpMTFR4eLhGjBihtWvX3pyAAACgRwi2O4AkHTx4UPn5+Zo4ceJ15509e1bTp0/X448/rs2bN+u9997TvHnzNGTIEN1///03KW3v0dDQoIqKCrtj+Ky8vNzr157M5XIpPDzc7hgAENBsLzdffvmlZs2apXXr1mnZsmXXnbt27Vq5XC6tXLlSkjR27FgdOnRIK1asoNx0g4qKCs2dO9fuGH6Tm5trdwSf5efna/To0XbHAICAZnu5eeKJJzRjxgzdfffdNyw3+/fv17Rp07zG0tLSVFBQoMuXLyskJKTd7RobG9XY2Nj63O12+x68F3C5XMrPz7c7Bq7hcrnsjgAAAc/WcrN161YdOXJEBw8e7NT8mpoaRUZGeo1FRkbqypUrqqurU3R0dLvb5eXlaenSpT7n7W3Cw8PZSwAA6HFsW1BcWVmpJ598Ups3b+7SGgKHw+H13OPxtDt+rZycHNXX17c+KisrrYUGAAABz7Y9N4cPH1Ztba0SExNbx5qbm7V3716tXr1ajY2N6tOnj9c2UVFRqqmp8Rqrra1VcHCwBg0a1OGfFRYWprCwMP9+AAC9WlBDvd0RgIASSN8J28pNamqq3n//fa+x2bNna8yYMXrqqafaFBtJSklJ0WuvveY1tmvXLiUlJXW43gYA/MnpdCokNEw6U2p3FCDghISGyel02h3DvnIzYMAAjR8/3musf//+GjRoUOt4Tk6OqqqqVFRUJEnKyMjQ6tWrlZmZqccff1z79+9XQUGBtmzZctPzA+idIiMjtfmVItXXB87/Unuz8vJy5ebmatGiRYqLi7M7Tq/ndDrbrI21g+1nS11PdXW113VWhg8frp07d2rhwoV6+eWXFRMTo1WrVnEaOICbKjIyMiD+AcffxMXFcQIEWgVUuSkpKfF6XlhY2GbO1KlTdeTIkZsTCAAA9DgBc/sFAAAAf6DcAAAAo1BuAACAUSg3AADAKJQbAABgFMoNAAAwCuUGAAAYhXIDAACMQrkBAABGodwAAACjUG4AAIBRKDcAAMAolBsAAGCUgLorOADg5mhoaFBFRYXdMXxWXl7u9WtP5nK5FB4ebncMI1BuAKAXqqio0Ny5c+2O4Te5ubl2R/BZfn6+Ro8ebXcMI1BuAKAXcrlcys/PtzsGruFyueyOYAzKDQD0QuHh4ewlgLFYUAwAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGIVyAwAAjEK5AQAARqHcAAAAo1BuAACAUSg3AADAKL3yruAej0eS5Ha7bU4CAAA66+rP7as/xzvSK8vNhQsXJEmxsbE2JwEAAF114cIFOZ3ODl93eG5UfwzU0tKiTz75RAMGDJDD4bA7DrqZ2+1WbGysKisrFRERYXccAH7E97t38Xg8unDhgmJiYhQU1PHKml655yYoKEjf/OY37Y6BmywiIoJ//ABD8f3uPa63x+YqFhQDAACjUG4AAIBRKDcwXlhYmBYvXqywsDC7owDwM77faE+vXFAMAADMxZ4bAABgFMoNAAAwCuUGAAAYhXIDAACMQrlBj/dv//ZvGj58uMLDw5WYmKh33333uvNLS0uVmJio8PBwjRgxQmvXrr1JSQF0xd69e3XPPfcoJiZGDodDr7766g234fsNiXKDHm7btm1asGCBFi1apKNHj+quu+7S97//fVVUVLQ7/+zZs5o+fbruuusuHT16VE8//bTmz5+v4uLim5wcwI1cvHhRkyZN0urVqzs1n+83ruJUcPRod9xxhyZPnqw1a9a0jo0dO1b33nuv8vLy2sx/6qmntGPHDpWVlbWOZWRk6C9/+Yv2799/UzID6DqHw6E//vGPuvfeezucw/cbV7HnBj1WU1OTDh8+rGnTpnmNT5s2Tfv27Wt3m/3797eZn5aWpkOHDuny5cvdlhVA9+P7jasoN+ix6urq1NzcrMjISK/xyMhI1dTUtLtNTU1Nu/OvXLmiurq6bssKoPvx/cZVlBv0eA6Hw+u5x+NpM3aj+e2NA+h5+H5DotygBxs8eLD69OnTZi9NbW1tm/+9XRUVFdXu/ODgYA0aNKjbsgLofny/cRXlBj1WaGioEhMTtXv3bq/x3bt3684772x3m5SUlDbzd+3apaSkJIWEhHRbVgDdj+83rqLcoEfLzMzU+vXrtWHDBpWVlWnhwoWqqKhQRkaGJCknJ0fp6emt8zMyMlReXq7MzEyVlZVpw4YNKigoUFZWll0fAUAHvvzySx07dkzHjh2T9PWp3seOHWu91APfb3TIA/RwL7/8sicuLs4TGhrqmTx5sqe0tLT1tYcfftgzdepUr/klJSWehIQET2hoqGfYsGGeNWvW3OTEADpjz549HkltHg8//LDH4+H7jY5xnRsAAGAUDksBAACjUG4AAIBRKDcAAMAolBsAAGAUyg0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAErEceeUQOh6P1XmHXmjdvnhwOR+uc6z0eeeQR/eUvf1FYWJh27Njh9T7FxcUKDw/X8ePHb9bHAtDNKDcAAlpsbKy2bt2qS5cutY41NDRoy5YtcrlckqTq6urWx8qVKxUREeE19tJLL2nSpEl65plnNHfuXH322WeSpNraWmVkZGjp0qUaP368LZ8PgP9RbgAEtMmTJ8vlcmn79u2tY9u3b1dsbKwSEhIkSVFRUa0Pp9Mph8PRZkz6+i7SLpdLTzzxhCTppz/9qW699VbuGg0YhnIDIODNnj1bGzdubH2+YcMGzZkzp8vv06dPH23atEl/+tOf9C//8i966623VFhYqD59+vgzLgCbUW4ABLyHHnpI//3f/62PPvpI5eXleu+99/Tggw9aeq+xY8dqwYIF2rJli5YsWaLRo0f7OS0Au1FuAAS8wYMHa8aMGdq0aZM2btyoGTNmaPDgwZbe68svv9S2bdvUr18/vfvuu35OCiAQUG4A9Ahz5sxRYWGhNm3aZOmQ1FW//OUvFRoaqn379untt99WUVGRH1MCCASUGwA9wj/+4z+qqalJTU1NSktLs/Qeu3fv1vr161VYWKhJkyZp+fLlWrBggaqrq/2cFoCdKDcAeoQ+ffqorKxMZWVllhYAu91uPfroo8rKylJycrIkaf78+YqPj9fcuXP9HReAjSg3AHqMiIgIRUREWNp2wYIFcjqdWrp0aetYUFCQNm7cqHfeeYfDU4BBHB6Px2N3CAAAAH9hzw0AADAK5QYAABiFcgMAAIxCuQEAAEah3AAAAKNQbgAAgFEoNwAAwCiUGwAAYBTKDQAAMArlBgAAGIVyAwAAjPL/AK7UsJ18+rrHAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df = pd.concat([df_meta,df_med['MTX'],df_alpha['shannon_entropy']],axis=1)\n",
    "df.to_csv(path + 'inputs/df_lmer.csv')#, sep='\\t')\n",
    "df = df[df['Diagnosis'] == 'RA']\n",
    "# df\n",
    "print(scipy.stats.ttest_ind(df[df['MTX']==0]['shannon_entropy'],df[df['MTX']==1]['shannon_entropy']))\n",
    "sns.boxplot(y='shannon_entropy',x='MTX',data=df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['TwinPair', 'Diagnosis', 'TJC', 'SJC', 'MTX', 'shannon_entropy',\n",
       "       'k__Bacteria.p__Actinobacteria.c__Coriobacteriia',\n",
       "       'k__Bacteria.p__Firmicutes.c__Bacilli',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridia_unclassified.f__Clostridia_unclassified.g__Clostridia_unclassified.s__Clostridia_bacterium',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Bacteroidaceae.g__Bacteroides.s__Bacteroides_caccae',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Tannerellaceae.g__Parabacteroides',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Tannerellaceae.g__Parabacteroides.s__Parabacteroides_distasonis',\n",
       "       'k__Bacteria.p__Proteobacteria.c__Gammaproteobacteria.o__Enterobacterales.f__Enterobacteriaceae.g__Escherichia',\n",
       "       'k__Bacteria.p__Proteobacteria.c__Gammaproteobacteria.o__Enterobacterales.f__Enterobacteriaceae.g__Escherichia.s__Escherichia_coli'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge lefses\n",
    "df_lefse_taxa_uk = pd.read_csv(path + 'inputs/lefse_uk.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "df_lefse_taxa_nyu = pd.read_csv(path + 'outputs/jobs06/lefse_results.tsv', sep='\\t', header=None)\n",
    "df_lefse_taxa_nyu.columns = ['Taxa','ESorMean_NYU','Group_NYU','LDA_NYU','p_value_NYU']\n",
    "df_lefse_taxa_nyu = df_lefse_taxa_nyu.set_index('Taxa')\n",
    "\n",
    "df_lefse_taxa_merge = pd.concat([df_lefse_taxa_uk,df_lefse_taxa_nyu],axis=1)\n",
    "df_lefse_taxa_merge = df_lefse_taxa_merge.dropna(subset='p_value')\n",
    "\n",
    "# otu l7\n",
    "df_otu = pd.read_csv(path + 'outputs/jobs06/lefse_table.tsv', sep='\\t', skiprows=2, index_col=0).T\n",
    "df_otu.columns = df_otu.columns.map(lambda x: x.replace('|','.'))\n",
    "df_otu.index = df_otu.index.map(lambda x: x + 'A')\n",
    "\n",
    "# normalize\n",
    "df_otu = df_otu.div(df_otu.sum(axis=1),axis=0)\n",
    "\n",
    "# df combined\n",
    "df_merge = pd.concat([df_meta,df_med['MTX'],df_alpha['shannon_entropy']],axis=1)\n",
    "\n",
    "# for each taxa\n",
    "for taxa in df_lefse_taxa_merge.index:\n",
    "    cols = []\n",
    "    for col in df_otu.columns:\n",
    "        if taxa in col:\n",
    "            cols.append(col)\n",
    "\n",
    "    if len(cols) > 0:\n",
    "        # print(taxa, cols)\n",
    "        df_merge[taxa] = df_otu[cols].sum(axis=1)\n",
    "        # break\n",
    "    # sum all columns with those values\n",
    "    \n",
    "\n",
    "#df_lefse_taxa_merge['wsr_p']\n",
    "\n",
    "#df_lefse_taxa_merge\n",
    "# df_otu\n",
    "df_merge.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "      <th>LDA</th>\n",
       "      <th>p_value</th>\n",
       "      <th>t</th>\n",
       "      <th>t_pvalue</th>\n",
       "      <th>direction</th>\n",
       "      <th>wsr_pvalue</th>\n",
       "      <th>wrs_pvalue</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Taxa</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>k__Bacteria.p__Actinobacteria.c__Actinobacteria.o__Actinomycetales.f__Actinomycetaceae.g__Actinomyces.s__Actinomyces_sp_ICM58</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.48</td>\n",
       "      <td>0.012</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Bacteria.p__Actinobacteria.c__Actinobacteria.o__Actinomycetales.f__Actinomycetaceae.g__Actinomyces.s__Actinomyces_sp_ICM58.t__SGB17169</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.48</td>\n",
       "      <td>0.012</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Bacteria.p__Actinobacteria.c__Coriobacteriia</th>\n",
       "      <td>Affected</td>\n",
       "      <td>4.27</td>\n",
       "      <td>0.048</td>\n",
       "      <td>-0.004</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.945</td>\n",
       "      <td>0.721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Bacteria.p__Firmicutes.c__Bacilli</th>\n",
       "      <td>Affected</td>\n",
       "      <td>4.07</td>\n",
       "      <td>0.048</td>\n",
       "      <td>-0.516</td>\n",
       "      <td>0.622</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridia_unclassified.f__Clostridia_unclassified.g__Clostridia_unclassified.s__Clostridia_bacterium</th>\n",
       "      <td>Affected</td>\n",
       "      <td>4.04</td>\n",
       "      <td>0.035</td>\n",
       "      <td>0.623</td>\n",
       "      <td>0.553</td>\n",
       "      <td>Affected</td>\n",
       "      <td>0.499</td>\n",
       "      <td>0.488</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       group   LDA  p_value  \\\n",
       "Taxa                                                                          \n",
       "k__Bacteria.p__Actinobacteria.c__Actinobacteria...  Affected  2.48    0.012   \n",
       "k__Bacteria.p__Actinobacteria.c__Actinobacteria...  Affected  2.48    0.012   \n",
       "k__Bacteria.p__Actinobacteria.c__Coriobacteriia     Affected  4.27    0.048   \n",
       "k__Bacteria.p__Firmicutes.c__Bacilli                Affected  4.07    0.048   \n",
       "k__Bacteria.p__Firmicutes.c__Clostridia.o__Clos...  Affected  4.04    0.035   \n",
       "\n",
       "                                                        t t_pvalue  \\\n",
       "Taxa                                                                 \n",
       "k__Bacteria.p__Actinobacteria.c__Actinobacteria...     NA       NA   \n",
       "k__Bacteria.p__Actinobacteria.c__Actinobacteria...     NA       NA   \n",
       "k__Bacteria.p__Actinobacteria.c__Coriobacteriia    -0.004    0.997   \n",
       "k__Bacteria.p__Firmicutes.c__Bacilli               -0.516    0.622   \n",
       "k__Bacteria.p__Firmicutes.c__Clostridia.o__Clos...  0.623    0.553   \n",
       "\n",
       "                                                     direction wsr_pvalue  \\\n",
       "Taxa                                                                        \n",
       "k__Bacteria.p__Actinobacteria.c__Actinobacteria...          NA         NA   \n",
       "k__Bacteria.p__Actinobacteria.c__Actinobacteria...          NA         NA   \n",
       "k__Bacteria.p__Actinobacteria.c__Coriobacteriia     Unaffected      0.945   \n",
       "k__Bacteria.p__Firmicutes.c__Bacilli                Unaffected      0.844   \n",
       "k__Bacteria.p__Firmicutes.c__Clostridia.o__Clos...    Affected      0.499   \n",
       "\n",
       "                                                   wrs_pvalue  \n",
       "Taxa                                                           \n",
       "k__Bacteria.p__Actinobacteria.c__Actinobacteria...         NA  \n",
       "k__Bacteria.p__Actinobacteria.c__Actinobacteria...         NA  \n",
       "k__Bacteria.p__Actinobacteria.c__Coriobacteriia         0.721  \n",
       "k__Bacteria.p__Firmicutes.c__Bacilli                    0.959  \n",
       "k__Bacteria.p__Firmicutes.c__Clostridia.o__Clos...      0.488  "
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_RA = df_merge[df_merge['Diagnosis']=='RA']\n",
    "df_UA = df_merge[df_merge['Diagnosis']=='Unaffected']\n",
    "\n",
    "\n",
    "ts = []\n",
    "tps = []\n",
    "wsrps = []\n",
    "wrsps = []\n",
    "dirs = []\n",
    "\n",
    "for taxa in df_lefse_taxa_uk.index:\n",
    "    try:\n",
    "        ra = df_RA[taxa].values\n",
    "        ua = df_UA[taxa].values\n",
    "    \n",
    "        t,p = scipy.stats.ttest_1samp(ra-ua,popmean=0) # RA-UA order\n",
    "        if t > 0:\n",
    "            dirs.append('Affected')\n",
    "        elif t < 0:\n",
    "            dirs.append('Unaffected')\n",
    "        wsr,wsrp = scipy.stats.wilcoxon(ra,ua)\n",
    "        wrs,wrsp = scipy.stats.mannwhitneyu(ra,ua)\n",
    "        ts.append(np.round(t,3))\n",
    "        tps.append(np.round(p,3))\n",
    "        wsrps.append(np.round(wsrp,3))\n",
    "        wrsps.append(np.round(wrsp,3))\n",
    "    except:\n",
    "        dirs.append('NA')\n",
    "        ts.append('NA')\n",
    "        tps.append('NA')\n",
    "        wsrps.append('NA')\n",
    "        wrsps.append('NA')\n",
    "    \n",
    "df_lefse_taxa_uk['t'] = ts\n",
    "df_lefse_taxa_uk['t_pvalue'] = tps\n",
    "df_lefse_taxa_uk['direction'] = dirs\n",
    "df_lefse_taxa_uk['wsr_pvalue'] = wsrps\n",
    "df_lefse_taxa_uk['wrs_pvalue'] = wrsps\n",
    "\n",
    "df_lefse_taxa_uk.to_csv(path + 'inputs/df_lefse_taxa_uk_nyu.tsv', sep='\\t')\n",
    "df_lefse_taxa_uk.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['k__Bacteria.p__Actinobacteria.c__Actinobacteria.o__Actinomycetales.f__Actinomycetaceae.g__Actinomyces.s__Actinomyces_sp_ICM58',\n",
       "       'k__Bacteria.p__Actinobacteria.c__Actinobacteria.o__Actinomycetales.f__Actinomycetaceae.g__Actinomyces.s__Actinomyces_sp_ICM58.t__SGB17169',\n",
       "       'k__Bacteria.p__Actinobacteria.c__Coriobacteriia',\n",
       "       'k__Bacteria.p__Firmicutes.c__Bacilli',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridia_unclassified.f__Clostridia_unclassified.g__Clostridia_unclassified.s__Clostridia_bacterium',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridia_unclassified.f__Clostridia_unclassified.g__Clostridia_unclassified.s__Clostridia_bacterium.t__SGB14854',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Bacteroidaceae.g__Bacteroides.s__Bacteroides_caccae',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Bacteroidaceae.g__Bacteroides.s__Bacteroides_caccae.t__SGB1877',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Tannerellaceae.g__Parabacteroides',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Tannerellaceae.g__Parabacteroides.s__Parabacteroides_distasonis',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Tannerellaceae.g__Parabacteroides.s__Parabacteroides_distasonis.t__SGB1934',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Faecalicatena',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Faecalicatena.s__Faecalicatena_fissicatena',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Faecalicatena.s__Faecalicatena_fissicatena.t__SGB4871_group',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Lachnospira.s__Lachnospira_pectinoschiza',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Lachnospira.s__Lachnospira_pectinoschiza.t__SGB5075_group',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Oscillospiraceae.g__Dysosmobacter',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Ruminococcaceae.g__Gemmiger',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Ruminococcaceae.g__Gemmiger.s__Gemmiger_formicilis',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Ruminococcaceae.g__Gemmiger.s__Gemmiger_formicilis.t__SGB15300',\n",
       "       'k__Bacteria.p__Proteobacteria.c__Gammaproteobacteria.o__Enterobacterales.f__Enterobacteriaceae.g__Escherichia',\n",
       "       'k__Bacteria.p__Proteobacteria.c__Gammaproteobacteria.o__Enterobacterales.f__Enterobacteriaceae.g__Escherichia.s__Escherichia_coli',\n",
       "       'k__Bacteria.p__Proteobacteria.c__Gammaproteobacteria.o__Enterobacterales.f__Enterobacteriaceae.g__Escherichia.s__Escherichia_coli.t__SGB10068'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_lefse_taxa_uk.index.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SampleID</th>\n",
       "      <th>500A</th>\n",
       "      <th>501A</th>\n",
       "      <th>502A</th>\n",
       "      <th>503A</th>\n",
       "      <th>504A</th>\n",
       "      <th>505A</th>\n",
       "      <th>506A</th>\n",
       "      <th>507A</th>\n",
       "      <th>508A</th>\n",
       "      <th>509A</th>\n",
       "      <th>510A</th>\n",
       "      <th>511A</th>\n",
       "      <th>512A</th>\n",
       "      <th>513A</th>\n",
       "      <th>514A</th>\n",
       "      <th>515A</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Taxon</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>k  Archaea|p  Candidatus Thermoplasmatota|c  Thermoplasmata|o  Methanomassiliicoccales|f  Methanomassiliicoccaceae|g  Methanomassiliicoccaceae unclassified|s  Methanomassiliicoccaceae unclassified SGB380</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000239</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Archaea|p  Euryarchaeota|c  Methanobacteria|o  Methanobacteriales|f  Methanobacteriaceae|g  Methanobrevibacter|s  Methanobrevibacter smithii</th>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.011409</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000123</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004630</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001290</td>\n",
       "      <td>0.000179</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000487</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000094</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000984</td>\n",
       "      <td>0.000395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Actinomyces|s  Actinomyces SGB17154</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Bacteria|p  Actinobacteria|c  Actinomycetia|o  Actinomycetales|f  Actinomycetaceae|g  Fannyhessea|s  Fannyhessea vaginae</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Bacteria|p  Synergistetes|c  Synergistia|o  Synergistales|f  Synergistaceae|g  GGB9469|s  GGB9469 SGB14863</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001091</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1787|f  FGB1787|g  GGB4745|s  GGB4745 SGB6570</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.015351</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia muciniphila</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.009421</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010926</td>\n",
       "      <td>0.010352</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.044276</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.015001</td>\n",
       "      <td>0.023885</td>\n",
       "      <td>0.005809</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp BIOML A40</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014649</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k  Bacteria|p  Verrucomicrobia|c  Verrucomicrobiae|o  Verrucomicrobiales|f  Akkermansiaceae|g  Akkermansia|s  Akkermansia sp KLE1798</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.041014</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>690 rows Ã— 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SampleID                                               500A      501A  502A  \\\n",
       "Taxon                                                                         \n",
       "k  Archaea|p  Candidatus Thermoplasmatota|c  Th...  0.00000  0.000239   0.0   \n",
       "k  Archaea|p  Euryarchaeota|c  Methanobacteria|...  0.00003  0.011409   0.0   \n",
       "k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota ...  0.00000  0.000000   0.0   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.00000  0.000000   0.0   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.00000  0.000000   0.0   \n",
       "...                                                     ...       ...   ...   \n",
       "k  Bacteria|p  Synergistetes|c  Synergistia|o  ...  0.00000  0.000000   0.0   \n",
       "k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1...  0.00000  0.000000   0.0   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.00000  0.009421   0.0   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.00000  0.000000   0.0   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.00000  0.000000   0.0   \n",
       "\n",
       "SampleID                                                503A      504A  \\\n",
       "Taxon                                                                    \n",
       "k  Archaea|p  Candidatus Thermoplasmatota|c  Th...  0.000000  0.000000   \n",
       "k  Archaea|p  Euryarchaeota|c  Methanobacteria|...  0.000000  0.000000   \n",
       "k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota ...  0.000000  0.001290   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000  0.000000   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000075  0.000000   \n",
       "...                                                      ...       ...   \n",
       "k  Bacteria|p  Synergistetes|c  Synergistia|o  ...  0.000000  0.000000   \n",
       "k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1...  0.000000  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.014649   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000   \n",
       "\n",
       "SampleID                                                505A      506A  \\\n",
       "Taxon                                                                    \n",
       "k  Archaea|p  Candidatus Thermoplasmatota|c  Th...  0.000000  0.000000   \n",
       "k  Archaea|p  Euryarchaeota|c  Methanobacteria|...  0.000000  0.000000   \n",
       "k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota ...  0.000179  0.000029   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000  0.000000   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000  0.000000   \n",
       "...                                                      ...       ...   \n",
       "k  Bacteria|p  Synergistetes|c  Synergistia|o  ...  0.000000  0.001091   \n",
       "k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1...  0.000000  0.015351   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.010926  0.010352   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000   \n",
       "\n",
       "SampleID                                                507A  508A      509A  \\\n",
       "Taxon                                                                          \n",
       "k  Archaea|p  Candidatus Thermoplasmatota|c  Th...  0.000000   0.0  0.000000   \n",
       "k  Archaea|p  Euryarchaeota|c  Methanobacteria|...  0.000000   0.0  0.000123   \n",
       "k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota ...  0.000000   0.0  0.000000   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000   0.0  0.000000   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000   0.0  0.000000   \n",
       "...                                                      ...   ...       ...   \n",
       "k  Bacteria|p  Synergistetes|c  Synergistia|o  ...  0.000000   0.0  0.000000   \n",
       "k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1...  0.000000   0.0  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000601   0.0  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000   0.0  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000   0.0  0.000000   \n",
       "\n",
       "SampleID                                                510A      511A  \\\n",
       "Taxon                                                                    \n",
       "k  Archaea|p  Candidatus Thermoplasmatota|c  Th...  0.000000  0.000000   \n",
       "k  Archaea|p  Euryarchaeota|c  Methanobacteria|...  0.000000  0.000000   \n",
       "k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota ...  0.000487  0.000000   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000  0.000000   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000  0.000000   \n",
       "...                                                      ...       ...   \n",
       "k  Bacteria|p  Synergistetes|c  Synergistia|o  ...  0.000000  0.000105   \n",
       "k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1...  0.000000  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.044276  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.041014   \n",
       "\n",
       "SampleID                                                512A      513A  \\\n",
       "Taxon                                                                    \n",
       "k  Archaea|p  Candidatus Thermoplasmatota|c  Th...  0.000000  0.000000   \n",
       "k  Archaea|p  Euryarchaeota|c  Methanobacteria|...  0.000000  0.000000   \n",
       "k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota ...  0.000094  0.000000   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000020  0.000000   \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000  0.000000   \n",
       "...                                                      ...       ...   \n",
       "k  Bacteria|p  Synergistetes|c  Synergistia|o  ...  0.000000  0.000000   \n",
       "k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1...  0.000000  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.015001  0.023885   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000   \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000   \n",
       "\n",
       "SampleID                                                514A      515A  \n",
       "Taxon                                                                   \n",
       "k  Archaea|p  Candidatus Thermoplasmatota|c  Th...  0.000000  0.000000  \n",
       "k  Archaea|p  Euryarchaeota|c  Methanobacteria|...  0.004630  0.000000  \n",
       "k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota ...  0.000984  0.000395  \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000  0.000000  \n",
       "k  Bacteria|p  Actinobacteria|c  Actinomycetia|...  0.000000  0.000000  \n",
       "...                                                      ...       ...  \n",
       "k  Bacteria|p  Synergistetes|c  Synergistia|o  ...  0.000000  0.000000  \n",
       "k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB1...  0.000000  0.000000  \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.005809  0.000000  \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000  \n",
       "k  Bacteria|p  Verrucomicrobia|c  Verrucomicrob...  0.000000  0.000000  \n",
       "\n",
       "[690 rows x 16 columns]"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# taxa barplot\n",
    "L_to_start_str = {\n",
    "    'L2': '.c__', # for phylum you want to split on class\n",
    "    'L3': '.o__',\n",
    "    'L4': '.f__',\n",
    "    'L5': '.g__',\n",
    "    'L6': '.s__',\n",
    "    'L7': 'NOTHING',\n",
    "}\n",
    "L_to_end_str = {\n",
    "    'L2': '.p__', \n",
    "    'L3': '.c__',\n",
    "    'L4': '.o__',\n",
    "    'L5': '.f__',\n",
    "    'L6': '.g__',\n",
    "    'L7': '.s__'\n",
    "}\n",
    "\n",
    "df_otu2 = pd.read_csv(path + '/inputs/df_asv.tsv', sep='\\t',skiprows=0,index_col=0)#, header=True)\n",
    "\n",
    "l_to_taxa = {}\n",
    "l_to_df = {}\n",
    "for l in L_to_start_str:\n",
    "\n",
    "    # get string to split on\n",
    "    ss,es = L_to_start_str[l], L_to_end_str[l]\n",
    "\n",
    "\n",
    "    # get taxa data\n",
    "    df = df_otu2.copy() # .loc[keep,:]\n",
    "\n",
    "    #if l != 'L7':\n",
    "    #    df.columns = df.columns.map(lambda x: x.split(ss)[0])\n",
    "\n",
    "    # normally this should happen AFTER the groupby but there's silly inconsistences\n",
    "    # like bacillota === firmicutes but it's not consistent in the mapping\n",
    "    \n",
    "    # split string on ending '|'\n",
    "    # df.columns = df.columns.map(lambda x: x.split(es)[-1])\n",
    "\n",
    "    # this collapses taxa and sums up ones with same names\n",
    "    df = df.groupby(df.columns, axis=1).sum()\n",
    "\n",
    "    # taxa_tables.append(df)\n",
    "\n",
    "    # transpose\n",
    "    df = df.T\n",
    "\n",
    "    # set name of index\n",
    "    df.index.name = 'Taxon'\n",
    "\n",
    "    # check that no dup rows\n",
    "    if np.max(df.index.value_counts()) > 1:\n",
    "        print('dup row')\n",
    "    \n",
    "    # save output\n",
    "    # this is untidy format\n",
    "    df.to_csv(path + 'inputs/df_taxa_' + l + '.tsv', sep='\\t')\n",
    "    l_to_df[l] = df\n",
    "\n",
    "    \n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>t</th>\n",
       "      <th>t_pvalue</th>\n",
       "      <th>direction</th>\n",
       "      <th>wsr_pvalue</th>\n",
       "      <th>wrs_pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>k  Archaea|p  Candidatus Thermoplasmatota|c  T...</td>\n",
       "      <td>-1.000</td>\n",
       "      <td>0.351</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.317</td>\n",
       "      <td>0.382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>k  Archaea|p  Euryarchaeota|c  Methanobacteria...</td>\n",
       "      <td>-0.534</td>\n",
       "      <td>0.610</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.593</td>\n",
       "      <td>0.945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota...</td>\n",
       "      <td>2.005</td>\n",
       "      <td>0.085</td>\n",
       "      <td>Affected</td>\n",
       "      <td>0.043</td>\n",
       "      <td>0.148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>k  Bacteria|p  Actinobacteria|c  Actinomycetia...</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.351</td>\n",
       "      <td>Affected</td>\n",
       "      <td>0.317</td>\n",
       "      <td>0.382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>k  Bacteria|p  Actinobacteria|c  Actinomycetia...</td>\n",
       "      <td>-1.000</td>\n",
       "      <td>0.351</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.317</td>\n",
       "      <td>0.382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>685</th>\n",
       "      <td>k  Bacteria|p  Synergistetes|c  Synergistia|o ...</td>\n",
       "      <td>0.888</td>\n",
       "      <td>0.404</td>\n",
       "      <td>Affected</td>\n",
       "      <td>0.655</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>686</th>\n",
       "      <td>k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB...</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.351</td>\n",
       "      <td>Affected</td>\n",
       "      <td>0.317</td>\n",
       "      <td>0.382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>687</th>\n",
       "      <td>k  Bacteria|p  Verrucomicrobia|c  Verrucomicro...</td>\n",
       "      <td>0.602</td>\n",
       "      <td>0.566</td>\n",
       "      <td>Affected</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0.866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>k  Bacteria|p  Verrucomicrobia|c  Verrucomicro...</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.351</td>\n",
       "      <td>Affected</td>\n",
       "      <td>0.317</td>\n",
       "      <td>0.382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>k  Bacteria|p  Verrucomicrobia|c  Verrucomicro...</td>\n",
       "      <td>-1.000</td>\n",
       "      <td>0.351</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.317</td>\n",
       "      <td>0.382</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4140 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  taxa      t  t_pvalue  \\\n",
       "0    k  Archaea|p  Candidatus Thermoplasmatota|c  T... -1.000     0.351   \n",
       "1    k  Archaea|p  Euryarchaeota|c  Methanobacteria... -0.534     0.610   \n",
       "2    k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota...  2.005     0.085   \n",
       "3    k  Bacteria|p  Actinobacteria|c  Actinomycetia...  1.000     0.351   \n",
       "4    k  Bacteria|p  Actinobacteria|c  Actinomycetia... -1.000     0.351   \n",
       "..                                                 ...    ...       ...   \n",
       "685  k  Bacteria|p  Synergistetes|c  Synergistia|o ...  0.888     0.404   \n",
       "686  k  Bacteria|p  Tenericutes|c  CFGB1787|o  OFGB...  1.000     0.351   \n",
       "687  k  Bacteria|p  Verrucomicrobia|c  Verrucomicro...  0.602     0.566   \n",
       "688  k  Bacteria|p  Verrucomicrobia|c  Verrucomicro...  1.000     0.351   \n",
       "689  k  Bacteria|p  Verrucomicrobia|c  Verrucomicro... -1.000     0.351   \n",
       "\n",
       "      direction  wsr_pvalue  wrs_pvalue  \n",
       "0    Unaffected       0.317       0.382  \n",
       "1    Unaffected       0.593       0.945  \n",
       "2      Affected       0.043       0.148  \n",
       "3      Affected       0.317       0.382  \n",
       "4    Unaffected       0.317       0.382  \n",
       "..          ...         ...         ...  \n",
       "685    Affected       0.655       1.000  \n",
       "686    Affected       0.317       0.382  \n",
       "687    Affected       0.917       0.866  \n",
       "688    Affected       0.317       0.382  \n",
       "689  Unaffected       0.317       0.382  \n",
       "\n",
       "[4140 rows x 6 columns]"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_taxa = []\n",
    "\n",
    "taxa_tables = []\n",
    "\n",
    "for l in L_to_start_str:\n",
    "    df_otu = l_to_df[l].copy().T\n",
    "    df_otu.index.name = 'SampleID'\n",
    "    df_merge = pd.concat([df_meta['Diagnosis'],df_otu],axis=1)\n",
    "    \n",
    "    df_RA = df_merge[df_merge['Diagnosis']=='RA']\n",
    "    df_UA = df_merge[df_merge['Diagnosis']=='Unaffected']\n",
    "\n",
    "    taxas = []\n",
    "    ts = []\n",
    "    tps = []\n",
    "    wsrps = []\n",
    "    wrsps = []\n",
    "    dirs = []\n",
    "    \n",
    "    for taxa in df_otu.columns.values:\n",
    "        taxas.append(taxa)\n",
    "        try:\n",
    "            ra = df_RA[taxa].values\n",
    "            ua = df_UA[taxa].values\n",
    "        \n",
    "            t,p = scipy.stats.ttest_1samp(ra-ua,popmean=0) # RA-UA order\n",
    "            if t > 0:\n",
    "                dirs.append('Affected')\n",
    "            elif t < 0:\n",
    "                dirs.append('Unaffected')\n",
    "            wsr,wsrp = scipy.stats.wilcoxon(ra,ua)\n",
    "            wrs,wrsp = scipy.stats.mannwhitneyu(ra,ua)\n",
    "            ts.append(np.round(t,3))\n",
    "            tps.append(np.round(p,3))\n",
    "            wsrps.append(np.round(wsrp,3))\n",
    "            wrsps.append(np.round(wrsp,3))\n",
    "        except:\n",
    "            dirs.append('NA')\n",
    "            ts.append('NA')\n",
    "            tps.append('NA')\n",
    "            wsrps.append('NA')\n",
    "            wrsps.append('NA')\n",
    "\n",
    "    df_res = pd.DataFrame({'taxa': taxas, \n",
    "                          't': ts,\n",
    "                          't_pvalue': tps,\n",
    "                          'direction': dirs,\n",
    "                          'wsr_pvalue': wsrps,\n",
    "                          'wrs_pvalue': wrsps})\n",
    "    taxa_tables.append(df_res)\n",
    "    \n",
    "df_final = pd.concat(taxa_tables,axis=0)\n",
    "df_final\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1011|o  OFGB1011|f  FGB1011|g  GGB2653|s  GGB2653 SGB3574',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1227|o  OFGB1227|f  FGB1227|g  GGB3005|s  GGB3005 SGB3996',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB3004|o  OFGB3004|f  FGB3004|g  GGB58158|s  GGB58158 SGB79798',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB3746|s  GGB3746 SGB5089',\n",
       "       'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia',\n",
       "       'k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1011|o  OFGB1011|f  FGB1011|g  GGB2653|s  GGB2653 SGB3574',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1227|o  OFGB1227|f  FGB1227|g  GGB3005|s  GGB3005 SGB3996',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB3004|o  OFGB3004|f  FGB3004|g  GGB58158|s  GGB58158 SGB79798',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB3746|s  GGB3746 SGB5089',\n",
       "       'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia',\n",
       "       'k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1011|o  OFGB1011|f  FGB1011|g  GGB2653|s  GGB2653 SGB3574',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1227|o  OFGB1227|f  FGB1227|g  GGB3005|s  GGB3005 SGB3996',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB3004|o  OFGB3004|f  FGB3004|g  GGB58158|s  GGB58158 SGB79798',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB3746|s  GGB3746 SGB5089',\n",
       "       'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia',\n",
       "       'k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1011|o  OFGB1011|f  FGB1011|g  GGB2653|s  GGB2653 SGB3574',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1227|o  OFGB1227|f  FGB1227|g  GGB3005|s  GGB3005 SGB3996',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB3004|o  OFGB3004|f  FGB3004|g  GGB58158|s  GGB58158 SGB79798',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB3746|s  GGB3746 SGB5089',\n",
       "       'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia',\n",
       "       'k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1011|o  OFGB1011|f  FGB1011|g  GGB2653|s  GGB2653 SGB3574',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1227|o  OFGB1227|f  FGB1227|g  GGB3005|s  GGB3005 SGB3996',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB3004|o  OFGB3004|f  FGB3004|g  GGB58158|s  GGB58158 SGB79798',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB3746|s  GGB3746 SGB5089',\n",
       "       'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia',\n",
       "       'k  Archaea|p  Thaumarchaeota|c  Thaumarchaeota unclassified|o  Nitrosopumilales|f  Nitrosopumilaceae|g  Nitrosopumilus|s  Nitrosopumilus SGB14899',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1011|o  OFGB1011|f  FGB1011|g  GGB2653|s  GGB2653 SGB3574',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB1227|o  OFGB1227|f  FGB1227|g  GGB3005|s  GGB3005 SGB3996',\n",
       "       'k  Bacteria|p  Firmicutes|c  CFGB3004|o  OFGB3004|f  FGB3004|g  GGB58158|s  GGB58158 SGB79798',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis',\n",
       "       'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB3746|s  GGB3746 SGB5089',\n",
       "       'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df_final[df_final['wsr_pvalue'] < 0.05]\n",
    "df.taxa.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "      <th>LDA</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathway</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>COMPLETE_ARO_PWY_superpathwayofaromaticaminoacidbiosynthesis</th>\n",
       "      <td>Affected</td>\n",
       "      <td>3.02</td>\n",
       "      <td>0.035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P125_PWY_superpathwayof_R_R__butanediolbiosynthesis</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.06</td>\n",
       "      <td>0.047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_4722_creatininedegradationII</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.07</td>\n",
       "      <td>0.048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_6471_peptidoglycanbiosynthesisIV_Enterococcusfaecium_</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.97</td>\n",
       "      <td>0.048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AST_PWY_L_argininedegradationII_ASTpathway_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.79</td>\n",
       "      <td>0.048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CITRULBIO_PWY_L_citrullinebiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.84</td>\n",
       "      <td>0.025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnosedegradation</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.67</td>\n",
       "      <td>0.018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLYCOLYSIS_TCA_GLYOX_BYPASS_superpathwayofglycolysis_pyruvatedehydrogenase_TCA_andglyoxylatebypass</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.80</td>\n",
       "      <td>0.017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLYOXYLATE_BYPASS_glyoxylatecycle</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.66</td>\n",
       "      <td>0.017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HEXITOLDEGSUPER_PWY_superpathwayofhexitoldegradation_bacteria_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.82</td>\n",
       "      <td>0.025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAGLIPASYN_PWY_lipidIVAbiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.70</td>\n",
       "      <td>0.025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P105_PWY_TCAcycleIV_2_oxoglutaratedecarboxylase_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.66</td>\n",
       "      <td>0.025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_4702_phytatedegradationI</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.48</td>\n",
       "      <td>0.047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_5083_NAD_NADHphosphorylationanddephosphorylation</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.62</td>\n",
       "      <td>0.025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_5675_nitratereductionV_assimilatory_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.59</td>\n",
       "      <td>0.007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_5705_allantoindegradationtoglyoxylateIII</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.08</td>\n",
       "      <td>0.025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_6282_palmitoleatebiosynthesisI_from_5Z__dodec_5_enoate_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.81</td>\n",
       "      <td>0.018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_6531_mannitolcycle</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.61</td>\n",
       "      <td>0.024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_7323_superpathwayofGDP_mannose_derivedO_antigenbuildingblocksbiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.62</td>\n",
       "      <td>0.048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_7664_oleatebiosynthesisIV_anaerobic_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.85</td>\n",
       "      <td>0.025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY0_1261_anhydromuropeptidesrecycling</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.91</td>\n",
       "      <td>0.048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY0_862__5Z__dodec_5_enoatebiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.84</td>\n",
       "      <td>0.018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWYG_321_mycolatebiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.88</td>\n",
       "      <td>0.018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypassandTCA</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.76</td>\n",
       "      <td>0.017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                         group   LDA  p_value\n",
       "Pathway                                                                      \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci...    Affected  3.02    0.035\n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...    Affected  2.06    0.047\n",
       "PWY_4722_creatininedegradationII                      Affected  2.07    0.048\n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...    Affected  2.97    0.048\n",
       "AST_PWY_L_argininedegradationII_ASTpathway_         Unaffected  2.17    0.017\n",
       "BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI         Unaffected  2.79    0.048\n",
       "CITRULBIO_PWY_L_citrullinebiosynthesis              Unaffected  2.84    0.025\n",
       "FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnose...  Unaffected  2.67    0.018\n",
       "GLYCOLYSIS_TCA_GLYOX_BYPASS_superpathwayofglyco...  Unaffected  2.80    0.017\n",
       "GLYOXYLATE_BYPASS_glyoxylatecycle                   Unaffected  2.66    0.017\n",
       "HEXITOLDEGSUPER_PWY_superpathwayofhexitoldegrad...  Unaffected  2.82    0.025\n",
       "NAGLIPASYN_PWY_lipidIVAbiosynthesis                 Unaffected  2.70    0.025\n",
       "P105_PWY_TCAcycleIV_2_oxoglutaratedecarboxylase_    Unaffected  2.66    0.025\n",
       "PWY_4702_phytatedegradationI                        Unaffected  2.48    0.047\n",
       "PWY_5083_NAD_NADHphosphorylationanddephosphoryl...  Unaffected  2.62    0.025\n",
       "PWY_5675_nitratereductionV_assimilatory_            Unaffected  2.59    0.007\n",
       "PWY_5705_allantoindegradationtoglyoxylateIII        Unaffected  2.08    0.025\n",
       "PWY_6282_palmitoleatebiosynthesisI_from_5Z__dod...  Unaffected  2.81    0.018\n",
       "PWY_6531_mannitolcycle                              Unaffected  2.61    0.024\n",
       "PWY_7323_superpathwayofGDP_mannose_derivedO_ant...  Unaffected  2.62    0.048\n",
       "PWY_7664_oleatebiosynthesisIV_anaerobic_            Unaffected  2.85    0.025\n",
       "PWY0_1261_anhydromuropeptidesrecycling              Unaffected  2.91    0.048\n",
       "PWY0_862__5Z__dodec_5_enoatebiosynthesis            Unaffected  2.84    0.018\n",
       "PWYG_321_mycolatebiosynthesis                       Unaffected  2.88    0.018\n",
       "TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypass...  Unaffected  2.76    0.017"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mc = pd.read_csv(path + 'outputs/jobs08/humann3_joined_pathabundance.tsv', sep='\\t', index_col=0).T\n",
    "df_mc.index = df_mc.index.map(lambda x: x.split('_')[0])\n",
    "df_mc.index.name = 'SampleID'\n",
    "df_mc = df_mc.drop(['UNMAPPED','UNINTEGRATED'],axis=1)\n",
    "\n",
    "# normalize\n",
    "df_mc = df_mc.div(df_mc.sum(axis=1),axis=0)\n",
    "# df.sum(axis=1)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#df = df_lefse_taxa_uk.copy()\n",
    "#sns.scatterplot(x='t_\n",
    "df_lefse_path_uk = pd.read_csv(path + 'inputs/metacyc_twinsuk.txt', sep='\\t', index_col=0)\n",
    "#for x in df_lefse_path_uk.index:\n",
    "#    y = x.split('PWY')\n",
    "df_merge = df_lefse_path_uk.copy()\n",
    "\n",
    "for fxn in df_lefse_path_uk.index:\n",
    "    cols = []\n",
    "    for col in df_mc.columns:\n",
    "        if fxn in col:\n",
    "            cols.append(col)\n",
    "\n",
    "    if len(cols) > 0:\n",
    "        print(fxn, cols)\n",
    "        df_merge[fxn] = df_mc[cols].sum(axis=1)\n",
    "\n",
    "df_merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI BIOTIN-BIOSYNTHESIS-PWY: biotin biosynthesis I\n",
      "CITRULBIO_PWY_L_citrullinebiosynthesis CITRULBIO_PWY_L_citrullinebiosynthesis CITRULBIO-PWY: L-citrulline biosynthesis\n",
      "COMPLETE_ARO_PWY_superpathwayofaromaticaminoacidbiosynthesis COMPLETE_ARO_PWY_superpathwayofaromaticaminoacidbiosynthesis COMPLETE-ARO-PWY: superpathway of aromatic amino acid biosynthesis\n",
      "FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnosedegradation FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnosedegradation FUC-RHAMCAT-PWY: superpathway of fucose and rhamnose degradation\n",
      "GLYOXYLATE_BYPASS_glyoxylatecycle GLYOXYLATE_BYPASS_glyoxylatecycle GLYOXYLATE-BYPASS: glyoxylate cycle\n",
      "PWY_5705_allantoindegradationtoglyoxylateIII PWY_5705_allantoindegradationtoglyoxylateIII PWY-5705: allantoin degradation to glyoxylate III\n",
      "PWY_6531_mannitolcycle PWY_6531_mannitolcycle PWY-6531: mannitol cycle\n",
      "PWY_7323_superpathwayofGDP_mannose_derivedO_antigenbuildingblocksbiosynthesis PWY_7323_superpathwayofGDP_mannose_derivedO_antigenbuildingblocksbiosynthesis PWY-7323: superpathway of GDP-mannose-derived O-antigen building blocks biosynthesis\n",
      "TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypassandTCA TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypassandTCA TCA-GLYOX-BYPASS: superpathway of glyoxylate bypass and TCA\n"
     ]
    }
   ],
   "source": [
    "df_mc_merge = df_mc.copy()\n",
    "df_mc_merge.columns = df_mc_merge.columns.map(lambda x: x.split('|')[0])\n",
    "df_mc_merge = df_mc_merge.groupby(lambda x:x, axis=1).sum()\n",
    "\n",
    "queries = []\n",
    "q_to_entry = {}\n",
    "for x in df_mc_merge.columns.values:\n",
    "    q = x.replace(': ','_')\n",
    "    q = q.replace(' ','')\n",
    "    q = q.replace('-','_')\n",
    "    q_to_entry[q] = x\n",
    "    queries.append(q)\n",
    "    \n",
    "for q in queries:\n",
    "    for t in df_merge.index:\n",
    "        if q in t:\n",
    "            print(q,t,q_to_entry[q])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "      <th>LDA</th>\n",
       "      <th>p_value</th>\n",
       "      <th>t</th>\n",
       "      <th>t_pvalue</th>\n",
       "      <th>direction</th>\n",
       "      <th>wsr_pvalue</th>\n",
       "      <th>wrs_pvalue</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathway</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>COMPLETE_ARO_PWY_superpathwayofaromaticaminoacidbiosynthesis</th>\n",
       "      <td>Affected</td>\n",
       "      <td>3.02</td>\n",
       "      <td>0.035</td>\n",
       "      <td>-1.203</td>\n",
       "      <td>0.268</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.312</td>\n",
       "      <td>0.442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P125_PWY_superpathwayof_R_R__butanediolbiosynthesis</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.06</td>\n",
       "      <td>0.047</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_4722_creatininedegradationII</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.07</td>\n",
       "      <td>0.048</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_6471_peptidoglycanbiosynthesisIV_Enterococcusfaecium_</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.97</td>\n",
       "      <td>0.048</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AST_PWY_L_argininedegradationII_ASTpathway_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.017</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                         group   LDA  p_value  \\\n",
       "Pathway                                                                         \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci...    Affected  3.02    0.035   \n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...    Affected  2.06    0.047   \n",
       "PWY_4722_creatininedegradationII                      Affected  2.07    0.048   \n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...    Affected  2.97    0.048   \n",
       "AST_PWY_L_argininedegradationII_ASTpathway_         Unaffected  2.17    0.017   \n",
       "\n",
       "                                                        t t_pvalue  \\\n",
       "Pathway                                                              \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci... -1.203    0.268   \n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...     NA       NA   \n",
       "PWY_4722_creatininedegradationII                       NA       NA   \n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...     NA       NA   \n",
       "AST_PWY_L_argininedegradationII_ASTpathway_            NA       NA   \n",
       "\n",
       "                                                     direction wsr_pvalue  \\\n",
       "Pathway                                                                     \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci...  Unaffected      0.312   \n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...          NA         NA   \n",
       "PWY_4722_creatininedegradationII                            NA         NA   \n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...          NA         NA   \n",
       "AST_PWY_L_argininedegradationII_ASTpathway_                 NA         NA   \n",
       "\n",
       "                                                   wrs_pvalue  \n",
       "Pathway                                                        \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci...      0.442  \n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...         NA  \n",
       "PWY_4722_creatininedegradationII                           NA  \n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...         NA  \n",
       "AST_PWY_L_argininedegradationII_ASTpathway_                NA  "
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat([df_meta,df_med['MTX'],df_alpha['shannon_entropy']],axis=1)\n",
    "\n",
    "df2 = df_mc_merge.copy()\n",
    "\n",
    "df2.columns = df2.columns.map(lambda x: x.replace(': ','_').replace(' ','').replace('-','_'))\n",
    "\n",
    "\n",
    "df_merge = pd.concat([df2,df],axis=1)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "df_RA = df_merge[df_merge['Diagnosis']=='RA']\n",
    "df_UA = df_merge[df_merge['Diagnosis']=='Unaffected']\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "ts = []\n",
    "tps = []\n",
    "wsrps = []\n",
    "wrsps = []\n",
    "dirs = []\n",
    "\n",
    "for fxn in df_lefse_path_uk.index:\n",
    "    try:\n",
    "        ra = df_RA[fxn].values\n",
    "        ua = df_UA[fxn].values\n",
    "    \n",
    "        t,p = scipy.stats.ttest_1samp(ra-ua,popmean=0) # RA-UA order\n",
    "        if t > 0:\n",
    "            dirs.append('Affected')\n",
    "        elif t < 0:\n",
    "            dirs.append('Unaffected')\n",
    "        wsr,wsrp = scipy.stats.wilcoxon(ra,ua)\n",
    "        wrs,wrsp = scipy.stats.mannwhitneyu(ra,ua)\n",
    "        ts.append(np.round(t,3))\n",
    "        tps.append(np.round(p,3))\n",
    "        wsrps.append(np.round(wsrp,3))\n",
    "        wrsps.append(np.round(wrsp,3))\n",
    "    except:\n",
    "        dirs.append('NA')\n",
    "        ts.append('NA')\n",
    "        tps.append('NA')\n",
    "        wsrps.append('NA')\n",
    "        wrsps.append('NA')\n",
    "    \n",
    "df_lefse_path_uk['t'] = ts\n",
    "df_lefse_path_uk['t_pvalue'] = tps\n",
    "df_lefse_path_uk['direction'] = dirs\n",
    "df_lefse_path_uk['wsr_pvalue'] = wsrps\n",
    "df_lefse_path_uk['wrs_pvalue'] = wrsps\n",
    "\n",
    "df_lefse_path_uk.to_csv(path + 'inputs/df_lefse_fxn_uk_nyu.tsv', sep='\\t')\n",
    "df_lefse_path_uk.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "      <th>LDA</th>\n",
       "      <th>p_value</th>\n",
       "      <th>t</th>\n",
       "      <th>t_pvalue</th>\n",
       "      <th>direction</th>\n",
       "      <th>wsr_pvalue</th>\n",
       "      <th>wrs_pvalue</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathway</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>COMPLETE_ARO_PWY_superpathwayofaromaticaminoacidbiosynthesis</th>\n",
       "      <td>Affected</td>\n",
       "      <td>3.02</td>\n",
       "      <td>0.035</td>\n",
       "      <td>-1.203</td>\n",
       "      <td>0.268</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.312</td>\n",
       "      <td>0.442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P125_PWY_superpathwayof_R_R__butanediolbiosynthesis</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.06</td>\n",
       "      <td>0.047</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_4722_creatininedegradationII</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.07</td>\n",
       "      <td>0.048</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_6471_peptidoglycanbiosynthesisIV_Enterococcusfaecium_</th>\n",
       "      <td>Affected</td>\n",
       "      <td>2.97</td>\n",
       "      <td>0.048</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AST_PWY_L_argininedegradationII_ASTpathway_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.017</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.79</td>\n",
       "      <td>0.048</td>\n",
       "      <td>-1.036</td>\n",
       "      <td>0.335</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.461</td>\n",
       "      <td>0.574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CITRULBIO_PWY_L_citrullinebiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.84</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.472</td>\n",
       "      <td>0.651</td>\n",
       "      <td>Affected</td>\n",
       "      <td>0.742</td>\n",
       "      <td>0.959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnosedegradation</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.67</td>\n",
       "      <td>0.018</td>\n",
       "      <td>-0.393</td>\n",
       "      <td>0.706</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLYCOLYSIS_TCA_GLYOX_BYPASS_superpathwayofglycolysis_pyruvatedehydrogenase_TCA_andglyoxylatebypass</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.80</td>\n",
       "      <td>0.017</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLYOXYLATE_BYPASS_glyoxylatecycle</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.66</td>\n",
       "      <td>0.017</td>\n",
       "      <td>-1.213</td>\n",
       "      <td>0.265</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.225</td>\n",
       "      <td>0.407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HEXITOLDEGSUPER_PWY_superpathwayofhexitoldegradation_bacteria_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.82</td>\n",
       "      <td>0.025</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAGLIPASYN_PWY_lipidIVAbiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.70</td>\n",
       "      <td>0.025</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P105_PWY_TCAcycleIV_2_oxoglutaratedecarboxylase_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.66</td>\n",
       "      <td>0.025</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_4702_phytatedegradationI</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.48</td>\n",
       "      <td>0.047</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_5083_NAD_NADHphosphorylationanddephosphorylation</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.62</td>\n",
       "      <td>0.025</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_5675_nitratereductionV_assimilatory_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.59</td>\n",
       "      <td>0.007</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_5705_allantoindegradationtoglyoxylateIII</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.08</td>\n",
       "      <td>0.025</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.351</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.317</td>\n",
       "      <td>0.382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_6282_palmitoleatebiosynthesisI_from_5Z__dodec_5_enoate_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.81</td>\n",
       "      <td>0.018</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_6531_mannitolcycle</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.61</td>\n",
       "      <td>0.024</td>\n",
       "      <td>-0.48</td>\n",
       "      <td>0.646</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.735</td>\n",
       "      <td>0.873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_7323_superpathwayofGDP_mannose_derivedO_antigenbuildingblocksbiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.62</td>\n",
       "      <td>0.048</td>\n",
       "      <td>-1.281</td>\n",
       "      <td>0.241</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.312</td>\n",
       "      <td>0.195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY_7664_oleatebiosynthesisIV_anaerobic_</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.85</td>\n",
       "      <td>0.025</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY0_1261_anhydromuropeptidesrecycling</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.91</td>\n",
       "      <td>0.048</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWY0_862__5Z__dodec_5_enoatebiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.84</td>\n",
       "      <td>0.018</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PWYG_321_mycolatebiosynthesis</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.88</td>\n",
       "      <td>0.018</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypassandTCA</th>\n",
       "      <td>Unaffected</td>\n",
       "      <td>2.76</td>\n",
       "      <td>0.017</td>\n",
       "      <td>-1.323</td>\n",
       "      <td>0.227</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>0.109</td>\n",
       "      <td>0.076</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                         group   LDA  p_value  \\\n",
       "Pathway                                                                         \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci...    Affected  3.02    0.035   \n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...    Affected  2.06    0.047   \n",
       "PWY_4722_creatininedegradationII                      Affected  2.07    0.048   \n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...    Affected  2.97    0.048   \n",
       "AST_PWY_L_argininedegradationII_ASTpathway_         Unaffected  2.17    0.017   \n",
       "BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI         Unaffected  2.79    0.048   \n",
       "CITRULBIO_PWY_L_citrullinebiosynthesis              Unaffected  2.84    0.025   \n",
       "FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnose...  Unaffected  2.67    0.018   \n",
       "GLYCOLYSIS_TCA_GLYOX_BYPASS_superpathwayofglyco...  Unaffected  2.80    0.017   \n",
       "GLYOXYLATE_BYPASS_glyoxylatecycle                   Unaffected  2.66    0.017   \n",
       "HEXITOLDEGSUPER_PWY_superpathwayofhexitoldegrad...  Unaffected  2.82    0.025   \n",
       "NAGLIPASYN_PWY_lipidIVAbiosynthesis                 Unaffected  2.70    0.025   \n",
       "P105_PWY_TCAcycleIV_2_oxoglutaratedecarboxylase_    Unaffected  2.66    0.025   \n",
       "PWY_4702_phytatedegradationI                        Unaffected  2.48    0.047   \n",
       "PWY_5083_NAD_NADHphosphorylationanddephosphoryl...  Unaffected  2.62    0.025   \n",
       "PWY_5675_nitratereductionV_assimilatory_            Unaffected  2.59    0.007   \n",
       "PWY_5705_allantoindegradationtoglyoxylateIII        Unaffected  2.08    0.025   \n",
       "PWY_6282_palmitoleatebiosynthesisI_from_5Z__dod...  Unaffected  2.81    0.018   \n",
       "PWY_6531_mannitolcycle                              Unaffected  2.61    0.024   \n",
       "PWY_7323_superpathwayofGDP_mannose_derivedO_ant...  Unaffected  2.62    0.048   \n",
       "PWY_7664_oleatebiosynthesisIV_anaerobic_            Unaffected  2.85    0.025   \n",
       "PWY0_1261_anhydromuropeptidesrecycling              Unaffected  2.91    0.048   \n",
       "PWY0_862__5Z__dodec_5_enoatebiosynthesis            Unaffected  2.84    0.018   \n",
       "PWYG_321_mycolatebiosynthesis                       Unaffected  2.88    0.018   \n",
       "TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypass...  Unaffected  2.76    0.017   \n",
       "\n",
       "                                                        t t_pvalue  \\\n",
       "Pathway                                                              \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci... -1.203    0.268   \n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...     NA       NA   \n",
       "PWY_4722_creatininedegradationII                       NA       NA   \n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...     NA       NA   \n",
       "AST_PWY_L_argininedegradationII_ASTpathway_            NA       NA   \n",
       "BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI        -1.036    0.335   \n",
       "CITRULBIO_PWY_L_citrullinebiosynthesis              0.472    0.651   \n",
       "FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnose... -0.393    0.706   \n",
       "GLYCOLYSIS_TCA_GLYOX_BYPASS_superpathwayofglyco...     NA       NA   \n",
       "GLYOXYLATE_BYPASS_glyoxylatecycle                  -1.213    0.265   \n",
       "HEXITOLDEGSUPER_PWY_superpathwayofhexitoldegrad...     NA       NA   \n",
       "NAGLIPASYN_PWY_lipidIVAbiosynthesis                    NA       NA   \n",
       "P105_PWY_TCAcycleIV_2_oxoglutaratedecarboxylase_       NA       NA   \n",
       "PWY_4702_phytatedegradationI                           NA       NA   \n",
       "PWY_5083_NAD_NADHphosphorylationanddephosphoryl...     NA       NA   \n",
       "PWY_5675_nitratereductionV_assimilatory_               NA       NA   \n",
       "PWY_5705_allantoindegradationtoglyoxylateIII         -1.0    0.351   \n",
       "PWY_6282_palmitoleatebiosynthesisI_from_5Z__dod...     NA       NA   \n",
       "PWY_6531_mannitolcycle                              -0.48    0.646   \n",
       "PWY_7323_superpathwayofGDP_mannose_derivedO_ant... -1.281    0.241   \n",
       "PWY_7664_oleatebiosynthesisIV_anaerobic_               NA       NA   \n",
       "PWY0_1261_anhydromuropeptidesrecycling                 NA       NA   \n",
       "PWY0_862__5Z__dodec_5_enoatebiosynthesis               NA       NA   \n",
       "PWYG_321_mycolatebiosynthesis                          NA       NA   \n",
       "TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypass... -1.323    0.227   \n",
       "\n",
       "                                                     direction wsr_pvalue  \\\n",
       "Pathway                                                                     \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci...  Unaffected      0.312   \n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...          NA         NA   \n",
       "PWY_4722_creatininedegradationII                            NA         NA   \n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...          NA         NA   \n",
       "AST_PWY_L_argininedegradationII_ASTpathway_                 NA         NA   \n",
       "BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI         Unaffected      0.461   \n",
       "CITRULBIO_PWY_L_citrullinebiosynthesis                Affected      0.742   \n",
       "FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnose...  Unaffected      0.844   \n",
       "GLYCOLYSIS_TCA_GLYOX_BYPASS_superpathwayofglyco...          NA         NA   \n",
       "GLYOXYLATE_BYPASS_glyoxylatecycle                   Unaffected      0.225   \n",
       "HEXITOLDEGSUPER_PWY_superpathwayofhexitoldegrad...          NA         NA   \n",
       "NAGLIPASYN_PWY_lipidIVAbiosynthesis                         NA         NA   \n",
       "P105_PWY_TCAcycleIV_2_oxoglutaratedecarboxylase_            NA         NA   \n",
       "PWY_4702_phytatedegradationI                                NA         NA   \n",
       "PWY_5083_NAD_NADHphosphorylationanddephosphoryl...          NA         NA   \n",
       "PWY_5675_nitratereductionV_assimilatory_                    NA         NA   \n",
       "PWY_5705_allantoindegradationtoglyoxylateIII        Unaffected      0.317   \n",
       "PWY_6282_palmitoleatebiosynthesisI_from_5Z__dod...          NA         NA   \n",
       "PWY_6531_mannitolcycle                              Unaffected      0.735   \n",
       "PWY_7323_superpathwayofGDP_mannose_derivedO_ant...  Unaffected      0.312   \n",
       "PWY_7664_oleatebiosynthesisIV_anaerobic_                    NA         NA   \n",
       "PWY0_1261_anhydromuropeptidesrecycling                      NA         NA   \n",
       "PWY0_862__5Z__dodec_5_enoatebiosynthesis                    NA         NA   \n",
       "PWYG_321_mycolatebiosynthesis                               NA         NA   \n",
       "TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypass...  Unaffected      0.109   \n",
       "\n",
       "                                                   wrs_pvalue  \n",
       "Pathway                                                        \n",
       "COMPLETE_ARO_PWY_superpathwayofaromaticaminoaci...      0.442  \n",
       "P125_PWY_superpathwayof_R_R__butanediolbiosynth...         NA  \n",
       "PWY_4722_creatininedegradationII                           NA  \n",
       "PWY_6471_peptidoglycanbiosynthesisIV_Enterococc...         NA  \n",
       "AST_PWY_L_argininedegradationII_ASTpathway_                NA  \n",
       "BIOTIN_BIOSYNTHESIS_PWY_biotinbiosynthesisI             0.574  \n",
       "CITRULBIO_PWY_L_citrullinebiosynthesis                  0.959  \n",
       "FUC_RHAMCAT_PWY_superpathwayoffucoseandrhamnose...      0.798  \n",
       "GLYCOLYSIS_TCA_GLYOX_BYPASS_superpathwayofglyco...         NA  \n",
       "GLYOXYLATE_BYPASS_glyoxylatecycle                       0.407  \n",
       "HEXITOLDEGSUPER_PWY_superpathwayofhexitoldegrad...         NA  \n",
       "NAGLIPASYN_PWY_lipidIVAbiosynthesis                        NA  \n",
       "P105_PWY_TCAcycleIV_2_oxoglutaratedecarboxylase_           NA  \n",
       "PWY_4702_phytatedegradationI                               NA  \n",
       "PWY_5083_NAD_NADHphosphorylationanddephosphoryl...         NA  \n",
       "PWY_5675_nitratereductionV_assimilatory_                   NA  \n",
       "PWY_5705_allantoindegradationtoglyoxylateIII            0.382  \n",
       "PWY_6282_palmitoleatebiosynthesisI_from_5Z__dod...         NA  \n",
       "PWY_6531_mannitolcycle                                  0.873  \n",
       "PWY_7323_superpathwayofGDP_mannose_derivedO_ant...      0.195  \n",
       "PWY_7664_oleatebiosynthesisIV_anaerobic_                   NA  \n",
       "PWY0_1261_anhydromuropeptidesrecycling                     NA  \n",
       "PWY0_862__5Z__dodec_5_enoatebiosynthesis                   NA  \n",
       "PWYG_321_mycolatebiosynthesis                              NA  \n",
       "TCA_GLYOX_BYPASS_superpathwayofglyoxylatebypass...      0.076  "
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_lefse_path_uk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Nitrosopumilus SGB14899', 'GGB2653 SGB3574', 'GGB3005 SGB3996',\n",
       "       'GGB58158 SGB79798', 'Candidatus Cibionibacter quicibialis',\n",
       "       'Eubacterium ramulus', 'Blautia faecis', 'GGB3746 SGB5089',\n",
       "       'Dysosmobacter welbionis', 'Bilophila wadsworthia'], dtype=object)"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = pd.read_csv('/Users/KevinBu/Desktop/clemente_lab/Projects/twinsra/inputs/RA_twins/16S/jobs/15_taxa/15_taxa_stats_clean.csv')\n",
    "df1 = df1[df1['wilcoxon'] <= 0.1]\n",
    "df1.feature.values\n",
    "# df1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['k__Bacteria.p__Actinobacteria.c__Actinobacteria.o__Actinomycetales.f__Actinomycetaceae.g__Actinomyces.s__Actinomyces_sp_ICM58',\n",
       "       'k__Bacteria.p__Actinobacteria.c__Actinobacteria.o__Actinomycetales.f__Actinomycetaceae.g__Actinomyces.s__Actinomyces_sp_ICM58.t__SGB17169',\n",
       "       'k__Bacteria.p__Actinobacteria.c__Coriobacteriia',\n",
       "       'k__Bacteria.p__Firmicutes.c__Bacilli',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridia_unclassified.f__Clostridia_unclassified.g__Clostridia_unclassified.s__Clostridia_bacterium',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridia_unclassified.f__Clostridia_unclassified.g__Clostridia_unclassified.s__Clostridia_bacterium.t__SGB14854',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Bacteroidaceae.g__Bacteroides.s__Bacteroides_caccae',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Bacteroidaceae.g__Bacteroides.s__Bacteroides_caccae.t__SGB1877',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Tannerellaceae.g__Parabacteroides',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Tannerellaceae.g__Parabacteroides.s__Parabacteroides_distasonis',\n",
       "       'k__Bacteria.p__Bacteroidetes.c__Bacteroidia.o__Bacteroidales.f__Tannerellaceae.g__Parabacteroides.s__Parabacteroides_distasonis.t__SGB1934',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Faecalicatena',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Faecalicatena.s__Faecalicatena_fissicatena',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Faecalicatena.s__Faecalicatena_fissicatena.t__SGB4871_group',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Lachnospira.s__Lachnospira_pectinoschiza',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Lachnospiraceae.g__Lachnospira.s__Lachnospira_pectinoschiza.t__SGB5075_group',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Oscillospiraceae.g__Dysosmobacter',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Ruminococcaceae.g__Gemmiger',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Ruminococcaceae.g__Gemmiger.s__Gemmiger_formicilis',\n",
       "       'k__Bacteria.p__Firmicutes.c__Clostridia.o__Clostridiales.f__Ruminococcaceae.g__Gemmiger.s__Gemmiger_formicilis.t__SGB15300',\n",
       "       'k__Bacteria.p__Proteobacteria.c__Gammaproteobacteria.o__Enterobacterales.f__Enterobacteriaceae.g__Escherichia',\n",
       "       'k__Bacteria.p__Proteobacteria.c__Gammaproteobacteria.o__Enterobacterales.f__Enterobacteriaceae.g__Escherichia.s__Escherichia_coli',\n",
       "       'k__Bacteria.p__Proteobacteria.c__Gammaproteobacteria.o__Enterobacterales.f__Enterobacteriaceae.g__Escherichia.s__Escherichia_coli.t__SGB10068'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_lefse_taxa_uk.index.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>n</th>\n",
       "      <th>r</th>\n",
       "      <th>CI95%</th>\n",
       "      <th>p-val</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>pearson</th>\n",
       "      <td>16</td>\n",
       "      <td>0.298158</td>\n",
       "      <td>[-0.25, 0.7]</td>\n",
       "      <td>0.280422</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          n         r         CI95%     p-val\n",
       "pearson  16  0.298158  [-0.25, 0.7]  0.280422"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pingouin import partial_corr\n",
    "df = pd.concat([df_meta,df_med['MTX'],df_alpha['shannon_entropy']],axis=1)\n",
    "df = pd.get_dummies(data=df,columns=['Diagnosis','TwinPair'],dtype=int)\n",
    "partial_corr(data=df, x='Diagnosis_RA', y='shannon_entropy', covar=['MTX','TwinPair'], method='pearson')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TJC</th>\n",
       "      <th>SJC</th>\n",
       "      <th>MTX</th>\n",
       "      <th>shannon_entropy</th>\n",
       "      <th>Diagnosis_RA</th>\n",
       "      <th>Diagnosis_Unaffected</th>\n",
       "      <th>TwinPair_T2</th>\n",
       "      <th>TwinPair_T3</th>\n",
       "      <th>TwinPair_T4</th>\n",
       "      <th>TwinPair_T5</th>\n",
       "      <th>TwinPair_T6</th>\n",
       "      <th>TwinPair_T7</th>\n",
       "      <th>TwinPair_T8</th>\n",
       "      <th>TwinPair_T9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.860518</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.193414</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.976850</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.439629</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>25</td>\n",
       "      <td>27</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.171083</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505A</th>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.541273</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>506A</th>\n",
       "      <td>21</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.038667</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>507A</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.159244</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>508A</th>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.013039</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509A</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.171476</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510A</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.062185</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>511A</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.245597</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>512A</th>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.183647</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>513A</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.985378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514A</th>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.055377</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>515A</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.252750</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      TJC  SJC  MTX  shannon_entropy  Diagnosis_RA  Diagnosis_Unaffected  \\\n",
       "500A    1    1  1.0         3.860518             1                     0   \n",
       "501A    0    0  0.0         5.193414             0                     1   \n",
       "502A    7    7  1.0         3.976850             1                     0   \n",
       "503A    2    0  0.0         4.439629             0                     1   \n",
       "504A   25   27  1.0         5.171083             1                     0   \n",
       "505A    8    2  0.0         3.541273             0                     1   \n",
       "506A   21    3  0.0         5.038667             1                     0   \n",
       "507A    0    0  0.0         5.159244             0                     1   \n",
       "508A   13    6  0.0         4.013039             1                     0   \n",
       "509A    0    0  0.0         4.171476             0                     1   \n",
       "510A    0    0  1.0         5.062185             1                     0   \n",
       "511A    0    0  0.0         5.245597             0                     1   \n",
       "512A    4    1  1.0         4.183647             1                     0   \n",
       "513A    0    0  0.0         3.985378             0                     1   \n",
       "514A   12    5  0.0         6.055377             1                     0   \n",
       "515A    0    0  0.0         4.252750             0                     1   \n",
       "\n",
       "      TwinPair_T2  TwinPair_T3  TwinPair_T4  TwinPair_T5  TwinPair_T6  \\\n",
       "500A            1            0            0            0            0   \n",
       "501A            1            0            0            0            0   \n",
       "502A            0            1            0            0            0   \n",
       "503A            0            1            0            0            0   \n",
       "504A            0            0            1            0            0   \n",
       "505A            0            0            1            0            0   \n",
       "506A            0            0            0            1            0   \n",
       "507A            0            0            0            1            0   \n",
       "508A            0            0            0            0            1   \n",
       "509A            0            0            0            0            1   \n",
       "510A            0            0            0            0            0   \n",
       "511A            0            0            0            0            0   \n",
       "512A            0            0            0            0            0   \n",
       "513A            0            0            0            0            0   \n",
       "514A            0            0            0            0            0   \n",
       "515A            0            0            0            0            0   \n",
       "\n",
       "      TwinPair_T7  TwinPair_T8  TwinPair_T9  \n",
       "500A            0            0            0  \n",
       "501A            0            0            0  \n",
       "502A            0            0            0  \n",
       "503A            0            0            0  \n",
       "504A            0            0            0  \n",
       "505A            0            0            0  \n",
       "506A            0            0            0  \n",
       "507A            0            0            0  \n",
       "508A            0            0            0  \n",
       "509A            0            0            0  \n",
       "510A            1            0            0  \n",
       "511A            1            0            0  \n",
       "512A            0            1            0  \n",
       "513A            0            1            0  \n",
       "514A            0            0            1  \n",
       "515A            0            0            1  "
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Scratch? #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([-1.96053116e-04, -3.99226564e-02,  0.00000000e+00, -1.07387222e-03,\n",
       "        3.54714627e-05,  0.00000000e+00,  0.00000000e+00,  0.00000000e+00])"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# scipy.stats.ttest_1samp(ra-ua)\n",
    "ra-ua"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Abx\n",
      "Abx           n  y\n",
      "Diagnosis         \n",
      "1_Unaffected  4  4\n",
      "2_RA          5  3\n",
      "Chi-Square statistic: 0.0\n",
      "P-value: 1.0\n",
      "smoking previously\n",
      "smoking previously  n  y\n",
      "Diagnosis               \n",
      "1_Unaffected        4  4\n",
      "2_RA                3  5\n",
      "Chi-Square statistic: 0.0\n",
      "P-value: 1.0\n",
      "Alcohol current\n",
      "Alcohol current  n  y\n",
      "Diagnosis            \n",
      "1_Unaffected     4  4\n",
      "2_RA             4  4\n",
      "Chi-Square statistic: 0.0\n",
      "P-value: 1.0\n",
      "NSAIDS current\n",
      "NSAIDS current  n  y\n",
      "Diagnosis           \n",
      "1_Unaffected    4  4\n",
      "2_RA            3  5\n",
      "Chi-Square statistic: 0.0\n",
      "P-value: 1.0\n",
      "NSAIDS last 3 mo\n",
      "NSAIDS last 3 mo  n  y\n",
      "Diagnosis             \n",
      "1_Unaffected      0  8\n",
      "2_RA              2  6\n",
      "Chi-Square statistic: 0.5714285714285714\n",
      "P-value: 0.4496917979688908\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sibling_pair</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>Abx</th>\n",
       "      <th>smoking previously</th>\n",
       "      <th>Alcohol current</th>\n",
       "      <th>NSAIDS current</th>\n",
       "      <th>NSAIDS last 3 mo</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Subject</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>sib_01</td>\n",
       "      <td>1_Unaffected</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>sib_02</td>\n",
       "      <td>1_Unaffected</td>\n",
       "      <td>y</td>\n",
       "      <td>y</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505</th>\n",
       "      <td>sib_03</td>\n",
       "      <td>1_Unaffected</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>507</th>\n",
       "      <td>sib_04</td>\n",
       "      <td>1_Unaffected</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>y</td>\n",
       "      <td>y</td>\n",
       "      <td>y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509</th>\n",
       "      <td>sib_05</td>\n",
       "      <td>1_Unaffected</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Sibling_pair     Diagnosis Abx smoking previously Alcohol current  \\\n",
       "Subject                                                                     \n",
       "501           sib_01  1_Unaffected   y                  n               n   \n",
       "503           sib_02  1_Unaffected   y                  y               y   \n",
       "505           sib_03  1_Unaffected   n                  n               n   \n",
       "507           sib_04  1_Unaffected   n                  y               y   \n",
       "509           sib_05  1_Unaffected   n                  y               n   \n",
       "\n",
       "        NSAIDS current NSAIDS last 3 mo  \n",
       "Subject                                  \n",
       "501                  n                y  \n",
       "503                  n                y  \n",
       "505                  n                y  \n",
       "507                  y                y  \n",
       "509                  n                y  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load in medication data\n",
    "df_med = pd.read_csv(path + 'inputs/patient_data_meds.csv', skiprows=1)\n",
    "\n",
    "# drop na rows\n",
    "df_med = df_med.iloc[:20,:]\n",
    "\n",
    "# convert subject to index and drop\n",
    "df_med = df_med.set_index('Subject')\n",
    "df_med.index = df_med.index.astype(int).astype(str)\n",
    "df_med = df_med.drop(['183','182','516','517'])\n",
    "\n",
    "# metadata subset\n",
    "mcols = ['Abx', 'smoking previously', 'Alcohol current', 'NSAIDS current', 'NSAIDS last 3 mo']\n",
    "df_med = df_med.loc[:,['Sibling_pair', 'Diagnosis'] + mcols]\n",
    "\n",
    "# specific sub for 509\n",
    "df_med.loc['509','smoking previously'] = 'y'\n",
    "\n",
    "# convert 'na' to np.nan and run chi sq\n",
    "for m in mcols:\n",
    "    print(m)\n",
    "    df_med[m] = df_med[m].apply(lambda x: np.nan if x == 'na' else x)\n",
    "\n",
    "    # binarize column\n",
    "    df_med[m] = df_med[m].apply(lambda x: 'n' if x in ['No abx', 'n'] else 'y')\n",
    " \n",
    "    # get contingency matrix\n",
    "    contingency_table = pd.crosstab(df_med['Diagnosis'], df_med[m])\n",
    "    print(contingency_table)\n",
    "\n",
    "    # compute chi2\n",
    "    chi2, p, dof, expected = scipy.stats.chi2_contingency(contingency_table)\n",
    "    \n",
    "    print(\"Chi-Square statistic:\", chi2)\n",
    "    print(\"P-value:\", p)\n",
    "\n",
    "df_med.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Abx_diff</th>\n",
       "      <th>smoking previously_diff</th>\n",
       "      <th>Alcohol current_diff</th>\n",
       "      <th>NSAIDS current_diff</th>\n",
       "      <th>NSAIDS last 3 mo_diff</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sibling_pair</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T2</th>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T4</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T5</th>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T6</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T7</th>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T8</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T9</th>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Abx_diff  smoking previously_diff  Alcohol current_diff  \\\n",
       "Sibling_pair                                                            \n",
       "T2                  -1                        0                     0   \n",
       "T3                   0                        0                     0   \n",
       "T4                   0                        1                     1   \n",
       "T5                   1                       -1                     0   \n",
       "T6                   0                        0                     0   \n",
       "T7                  -1                        0                    -1   \n",
       "T8                   1                        0                     0   \n",
       "T9                  -1                        1                     0   \n",
       "\n",
       "              NSAIDS current_diff  NSAIDS last 3 mo_diff  \n",
       "Sibling_pair                                              \n",
       "T2                              1                      0  \n",
       "T3                              1                      0  \n",
       "T4                              0                     -1  \n",
       "T5                              0                      0  \n",
       "T6                              1                      0  \n",
       "T7                             -1                     -1  \n",
       "T8                             -1                      0  \n",
       "T9                              0                      0  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check for diversity as a function of metadata difference\n",
    "df = df_med.copy()\n",
    "\n",
    "pairs = df['Sibling_pair'].unique()\n",
    "\n",
    "df_dict = {\n",
    "    'Sibling_pair': pairs\n",
    "}\n",
    "# collect paired values\n",
    "for m in mcols:\n",
    "    df_UA = df[df['Diagnosis'] == '1_Unaffected'].set_index('Sibling_pair')\n",
    "    df_RA = df[df['Diagnosis'] == '2_RA'].set_index('Sibling_pair')\n",
    "    df_dict[m + '_diff'] = df_RA[m].replace(['y','n'],[1,0]) - df_UA[m].replace(['y','n'],[1,0]) \n",
    "\n",
    "df = pd.DataFrame(df_dict)\n",
    "df.index = df.index.map(lambda x: 'T' + str(int(x.split('_0')[-1]) + 1))\n",
    "df = df.drop('Sibling_pair', axis=1)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smoking previously shannon_entropy_diff\n",
      "0.04608584897606692\n",
      "Alcohol current SJC\n",
      "0.011992044791843633\n",
      "NSAIDS current TJC\n",
      "0.76955186851755\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Abx_diff</th>\n",
       "      <th>smoking previously_diff</th>\n",
       "      <th>Alcohol current_diff</th>\n",
       "      <th>NSAIDS current_diff</th>\n",
       "      <th>NSAIDS last 3 mo_diff</th>\n",
       "      <th>shannon_entropy_diff</th>\n",
       "      <th>Bray_Curtis</th>\n",
       "      <th>MTX</th>\n",
       "      <th>SJC</th>\n",
       "      <th>TJC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T2</th>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1.332895</td>\n",
       "      <td>0.705551</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.462779</td>\n",
       "      <td>0.474034</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T4</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.629810</td>\n",
       "      <td>0.685602</td>\n",
       "      <td>1.0</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T5</th>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.120577</td>\n",
       "      <td>0.725669</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T6</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.158437</td>\n",
       "      <td>0.757568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Abx_diff  smoking previously_diff  Alcohol current_diff  \\\n",
       "T2        -1                        0                     0   \n",
       "T3         0                        0                     0   \n",
       "T4         0                        1                     1   \n",
       "T5         1                       -1                     0   \n",
       "T6         0                        0                     0   \n",
       "\n",
       "    NSAIDS current_diff  NSAIDS last 3 mo_diff  shannon_entropy_diff  \\\n",
       "T2                    1                      0             -1.332895   \n",
       "T3                    1                      0             -0.462779   \n",
       "T4                    0                     -1              1.629810   \n",
       "T5                    0                      0             -0.120577   \n",
       "T6                    1                      0             -0.158437   \n",
       "\n",
       "    Bray_Curtis  MTX  SJC  TJC  \n",
       "T2     0.705551  1.0    1    1  \n",
       "T3     0.474034  1.0    7    7  \n",
       "T4     0.685602  1.0   27   25  \n",
       "T5     0.725669  0.0    3   21  \n",
       "T6     0.757568  0.0    6   13  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test corr \n",
    "df_merge = pd.concat([df, df_paired_alpha, df_paired_beta], axis=1)\n",
    "\n",
    "# for each metadata difference, test each q dff\n",
    "qcols = ['shannon_entropy_diff', 'Bray_Curtis', 'MTX', 'SJC', 'TJC']\n",
    "for m in mcols:\n",
    "    for q in qcols:\n",
    "        f, p = scipy.stats.f_oneway(*[gp[q] for g, gp in df_merge.groupby(m + '_diff')])\n",
    "        if p < 0.05:\n",
    "            r, p = scipy.stats.pearsonr(df_merge[m + '_diff'], df_merge[q])\n",
    "            print(m,q)\n",
    "            print(p)\n",
    "\n",
    "        plt.figure(figsize=(4,3))\n",
    "        sns.scatterplot(data=df_merge, x=m + '_diff', y=q)\n",
    "        sns.despine()\n",
    "        plt.tight_layout()\n",
    "        plt.savefig(path + 'outputs/jobs22/' + m + '_' + q + '.pdf')\n",
    "        plt.close()\n",
    "\n",
    "df_merge.to_csv(path + 'inputs/df_glm_R.csv')\n",
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bray_Curtis</th>\n",
       "      <th>MTX</th>\n",
       "      <th>SJC</th>\n",
       "      <th>TJC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T2</th>\n",
       "      <td>0.705551</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3</th>\n",
       "      <td>0.474034</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T4</th>\n",
       "      <td>0.685602</td>\n",
       "      <td>1.0</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T5</th>\n",
       "      <td>0.725669</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T6</th>\n",
       "      <td>0.757568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T7</th>\n",
       "      <td>0.839525</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T8</th>\n",
       "      <td>0.452763</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T9</th>\n",
       "      <td>0.648924</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Bray_Curtis  MTX  SJC  TJC\n",
       "T2     0.705551  1.0    1    1\n",
       "T3     0.474034  1.0    7    7\n",
       "T4     0.685602  1.0   27   25\n",
       "T5     0.725669  0.0    3   21\n",
       "T6     0.757568  0.0    6   13\n",
       "T7     0.839525  1.0    0    0\n",
       "T8     0.452763  1.0    1    4\n",
       "T9     0.648924  0.0    5   12"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# intra_pair beta, lowest for T3 and T8\n",
    "df_paired_beta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SignificanceResult(statistic=0.2735294117647059, pvalue=0.3053231563074052)\n",
      "SignificanceResult(statistic=0.38095238095238104, pvalue=0.35181255311756476)\n",
      "SignificanceResult(statistic=0.28571428571428575, pvalue=0.4927262450169573)\n",
      "SignificanceResult(statistic=-0.7142857142857144, pvalue=0.04652823228416724)\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAnYAAAHWCAYAAAD6oMSKAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAAA6VElEQVR4nO3de3RU1cH+8WcITCYhZAJMSYLEkBcjBCMIiZBLibe+AbxUWt4Sb6FYkKYXJKX6CguthdoGXH29QCVKi0YtRarRyloFJb5eEk3grWkQK4qpt3CZSEMlQ36YDITz+8OVWQ4zCbmRkzn5ftY6azl777Nnb85CH/c5Z4/NMAxDAAAACHmDzB4AAAAAegfBDgAAwCIIdgAAABZBsAMAALAIgh0AAIBFEOwAAAAsgmAHAABgEQQ7AAAAiyDYBWEYhjwej9i7GQAAhBKCXRDHjx+X0+nU8ePHzR4KAABApxHsAAAALIJgBwAAYBEEOwAAAIvoF8Fuw4YNSkpKksPhUFpamioqKjpsv3nzZk2ePFmRkZGKj4/XrbfeqqNHj/rqS0pKZLPZAo7m5uZzPRUAAADTmB7stm7dqsLCQq1cuVI1NTWaMWOGZs+erbq6uqDt33zzTc2fP18LFy7Ue++9p2effVZ/+9vftGjRIr920dHRcrvdfofD4eiLKQEAAJjC9GD3wAMPaOHChVq0aJFSUlL00EMPKSEhQcXFxUHb79q1S2PHjtXtt9+upKQkffOb39QPf/hDvf32237tbDab4uLi/A4AAAArMzXYeb1eVVdXKzc31688NzdXlZWVQc/JysrSwYMHtX37dhmGoc8//1zPPfecrrnmGr92TU1NSkxM1JgxY3Tttdeqpqam3XG0tLTI4/H4HQAAAKHG1GDX0NCg1tZWxcbG+pXHxsaqvr4+6DlZWVnavHmz8vLyZLfbFRcXp5iYGK1fv97XZsKECSopKdG2bdu0ZcsWORwOZWdnq7a2NmifRUVFcjqdviMhIaH3JgkAANBHTL8VK3112/TrDMMIKGuzb98+3X777frFL36h6upqvfTSS/rkk09UUFDga5ORkaFbbrlFkydP1owZM/TnP/9ZF154oV/4+7oVK1aosbHRdxw4cKD3JgcAANBHBpv55S6XS2FhYQGrc0eOHAlYxWtTVFSk7Oxs3XnnnZKkSZMmaejQoZoxY4buu+8+xcfHB5wzaNAgXXrppe2u2IWHhys8PLyHswEAADCXqSt2drtdaWlpKisr8ysvKytTVlZW0HNOnDihQYP8hx0WFiZJ7f62q2EY2rNnT9DQBwAAYBWmrthJ0rJly5Sfn6/09HRlZmZq48aNqqur891aXbFihQ4dOqSnnnpKknTdddfptttuU3FxsWbOnCm3263CwkJNmzZNo0ePliStWrVKGRkZSk5Olsfj0bp167Rnzx498sgjps0TAADgXDM92OXl5eno0aNavXq13G63UlNTtX37diUmJkqS3G633552CxYs0PHjx/W73/1OP//5zxUTE6Mrr7xSa9eu9bU5duyYFi9erPr6ejmdTk2ZMkXl5eWaNm1an88PAACgr9iM9u5fDmAej0dOp1ONjY2Kjo7u9f4bT3jV0OSVp/mkoiOGyDXULmekvde/BwAADCymr9gNNIePfam7SveqorbBV5aT7NKauZM0OibCxJEBAIBQ1y+2OxkoGk94A0KdJJXXNmh56V41nvCaNDIAAGAFBLs+1NDkDQh1bcprG9TQRLADAADdR7DrQ57mkx3WHz9LPQAAQEcIdn0o2jGkw/phZ6kHAADoCMGuD7mi7MpJdgWty0l2yRXFm7EAAKD7CHZ9yBlp15q5kwLCXU6yS2vnTmLLEwAA0CPsYxdEX+1jd7z5pIY5hsgVxT52AACg59jHzgTOSIIcAADofdyKBQAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGAR/SLYbdiwQUlJSXI4HEpLS1NFRUWH7Tdv3qzJkycrMjJS8fHxuvXWW3X06FG/NqWlpZo4caLCw8M1ceJEvfDCC+dyCgAAAKYzPdht3bpVhYWFWrlypWpqajRjxgzNnj1bdXV1Qdu/+eabmj9/vhYuXKj33ntPzz77rP72t79p0aJFvjZVVVXKy8tTfn6+3nnnHeXn52vevHnavXt3X00LAACgz9kMwzDMHMD06dM1depUFRcX+8pSUlI0Z84cFRUVBbT/7W9/q+LiYn300Ue+svXr1+v+++/XgQMHJEl5eXnyeDzasWOHr82sWbM0fPhwbdmy5axj8ng8cjqdamxsVHR0dE+mBwAA0GdMXbHzer2qrq5Wbm6uX3lubq4qKyuDnpOVlaWDBw9q+/btMgxDn3/+uZ577jldc801vjZVVVUBfc6cObPdPltaWuTxePwOAACAUGNqsGtoaFBra6tiY2P9ymNjY1VfXx/0nKysLG3evFl5eXmy2+2Ki4tTTEyM1q9f72tTX1/fpT6LiorkdDp9R0JCQg9nBgAA0PdMf8ZOkmw2m99nwzACytrs27dPt99+u37xi1+ourpaL730kj755BMVFBR0u88VK1aosbHRd7Td0gUAAAglg838cpfLpbCwsICVtCNHjgSsuLUpKipSdna27rzzTknSpEmTNHToUM2YMUP33Xef4uPjFRcX16U+w8PDFR4e3gszAgAAMI+pK3Z2u11paWkqKyvzKy8rK1NWVlbQc06cOKFBg/yHHRYWJumrVTlJyszMDOhz586d7fYJAABgBaau2EnSsmXLlJ+fr/T0dGVmZmrjxo2qq6vz3VpdsWKFDh06pKeeekqSdN111+m2225TcXGxZs6cKbfbrcLCQk2bNk2jR4+WJC1dulQ5OTlau3atrr/+er344ot65ZVX9Oabb5o2TwAAgHPN9GCXl5eno0ePavXq1XK73UpNTdX27duVmJgoSXK73X572i1YsEDHjx/X7373O/385z9XTEyMrrzySq1du9bXJisrS88884zuvvtu3XPPPRo3bpy2bt2q6dOn9/n8AAAA+orp+9j1R+xjBwAAQlG/eCsWAAAAPUewAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLGGz2AAAAOFPjCa8amrzyNJ9UdMQQuYba5Yy0mz0soN8j2AEA+pXDx77UXaV7VVHb4CvLSXZpzdxJGh0TYeLIgP6PW7EAgH6j8YQ3INRJUnltg5aX7lXjCa9JIwNCA8EOANBvNDR5A0Jdm/LaBjU0EeyAjhDsAAD9hqf5ZIf1x89SDwx0BDsAQL8R7RjSYf2ws9QDAx3BDgDQb7ii7MpJdgWty0l2yRXFm7FAR/pFsNuwYYOSkpLkcDiUlpamioqKdtsuWLBANpst4Ljooot8bUpKSoK2aW5u7ovpAAC6yRlp15q5kwLCXU6yS2vnTmLLE+AsTN/uZOvWrSosLNSGDRuUnZ2txx57TLNnz9a+fft0/vnnB7R/+OGHtWbNGt/nU6dOafLkyfre977n1y46Olr79+/3K3M4HOdmEgCAXjM6JkLrb5yihiavjjef1DDHELmi2McO6AybYRiGmQOYPn26pk6dquLiYl9ZSkqK5syZo6KiorOe/5e//EXf/e539cknnygxMVHSVyt2hYWFOnbsWLfG5PF45HQ61djYqOjo6G71AQAA0NdMvRXr9XpVXV2t3Nxcv/Lc3FxVVlZ2qo9NmzbpW9/6li/UtWlqalJiYqLGjBmja6+9VjU1Ne320dLSIo/H43cAAACEGlODXUNDg1pbWxUbG+tXHhsbq/r6+rOe73a7tWPHDi1atMivfMKECSopKdG2bdu0ZcsWORwOZWdnq7a2Nmg/RUVFcjqdviMhIaH7kwIAADBJv3h5wmaz+X02DCOgLJiSkhLFxMRozpw5fuUZGRm65ZZbNHnyZM2YMUN//vOfdeGFF2r9+vVB+1mxYoUaGxt9x4EDB7o9FwAAALOY+vKEy+VSWFhYwOrckSNHAlbxzmQYhh5//HHl5+fLbu/4gdpBgwbp0ksvbXfFLjw8XOHh4V0bPAAAQD9j6oqd3W5XWlqaysrK/MrLysqUlZXV4blvvPGG/vnPf2rhwoVn/R7DMLRnzx7Fx8f3aLwAAAD9menbnSxbtkz5+flKT09XZmamNm7cqLq6OhUUFEj66jbpoUOH9NRTT/mdt2nTJk2fPl2pqakBfa5atUoZGRlKTk6Wx+PRunXrtGfPHj3yyCN9MicAAAAzmB7s8vLydPToUa1evVput1upqanavn277y1Xt9uturo6v3MaGxtVWlqqhx9+OGifx44d0+LFi1VfXy+n06kpU6aovLxc06ZNO+fzAQAAMIvp+9j1R+xjBwAAQlG/eCsWAAAAPUewAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACxisNkDAACYo/GEVw1NXnmaTyo6YohcQ+1yRtrNHhaAHiDYAcAAdPjYl7qrdK8qaht8ZTnJLq2ZO0mjYyJMHBmAnuBWLAAMMI0nvAGhTpLKaxu0vHSvGk94TRoZgJ4i2AHAANPQ5A0IdW3KaxvU0ESwA0IVwQ4ABhhP88kO64+fpR5A/0WwA4ABJtoxpMP6YWepB9B/EewAYIBxRdmVk+wKWpeT7JIrijdjgVBFsAOAAcYZadeauZMCwl1Osktr505iyxMghNkMwzDMHkR/4/F45HQ61djYqOjoaLOHAwDnRNs+dsebT2qYY4hcUexjB4Q69rEDgAHKGUmQA6yGW7EAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCH5SDDBB2290eppPKjpiiFxD+WknAEDPEeyAPnb42Je6q3SvKmobfGU5yS6tmTtJo2MiTBwZACDUcSsW6EONJ7wBoU6SymsbtLx0rxpPeE0aGQDACgh2QB9qaPIGhLo25bUNamgi2AEAuq9fBLsNGzYoKSlJDodDaWlpqqioaLftggULZLPZAo6LLrrIr11paakmTpyo8PBwTZw4US+88MK5ngZwVp7mkx3WHz9LPQAAHTE92G3dulWFhYVauXKlampqNGPGDM2ePVt1dXVB2z/88MNyu92+48CBAxoxYoS+973v+dpUVVUpLy9P+fn5euedd5Sfn6958+Zp9+7dfTUtIKhox5AO64edpR4AgI7YDMMwzBzA9OnTNXXqVBUXF/vKUlJSNGfOHBUVFZ31/L/85S/67ne/q08++USJiYmSpLy8PHk8Hu3YscPXbtasWRo+fLi2bNly1j49Ho+cTqcaGxsVHR3djVkBwTWe8GrJlhqVB7kdm5Ps0vobp/B2LACg20xdsfN6vaqurlZubq5feW5uriorKzvVx6ZNm/Stb33LF+qkr1bszuxz5syZne4TOFeckXatmTtJOckuv/KcZJfWzp1EqAMA9Iip2500NDSotbVVsbGxfuWxsbGqr68/6/lut1s7duzQn/70J7/y+vr6LvXZ0tKilpYW32ePx9PZKQBdNjomQutvnKKGJq+ON5/UMMcQuaLYxw4A0HP9Yh87m83m99kwjICyYEpKShQTE6M5c+b0qM+ioiKtWrWq8wMGesgZSZADAPQ+U2/FulwuhYWFBaykHTlyJGDF7UyGYejxxx9Xfn6+7Hb//0DGxcV1qc8VK1aosbHRdxw4cKAbswEAADCXqcHObrcrLS1NZWVlfuVlZWXKysrq8Nw33nhD//znP7Vw4cKAuszMzIA+d+7c2W6f4eHhio6O9jsAAABCjem3YpctW6b8/Hylp6crMzNTGzduVF1dnQoKCiR9tZp26NAhPfXUU37nbdq0SdOnT1dqampAn0uXLlVOTo7Wrl2r66+/Xi+++KJeeeUVvfnmm30yJwAAADOYHuzy8vJ09OhRrV69Wm63W6mpqdq+fbvvLVe32x2wp11jY6NKS0v18MMPB+0zKytLzzzzjO6++27dc889GjdunLZu3arp06ef8/kAAACYxfR97Poj9rEDAAChyPRfngAAAEDvINgBAABYBMEOAADAIgh2AAAAFmH6W7EA0BsaT3jV0OSVp/mkoiOGyDWUX/cAMPAQ7ACEvMPHvtRdpXtVUdvgK8tJdmnN3EkaHRNh4sgAoG9xKxZASGs84Q0IdZJUXtug5aV71XjCa9LIAKDvEewAhLSGJm9AqGtTXtughiaCHYCBg2AHIKR5mk92WH/8LPUAYCUEOwAhLdoxpMP6YWepBwArIdgBCGmuKLtykl1B63KSXXJF8WYsgIGDYAcgpDkj7Vozd1JAuMtJdmnt3ElseQJgQLEZhmGYPYj+xuPxyOl0qrGxUdHR0WYPB0AntO1jd7z5pIY5hsgVxT52AAYe9rEDYAnOSIIcAHArFgAAwCIIdgAAABZBsAMAALCITge7devWqbm5WZJUV1cn3rkAAADoXzr9VuzgwYN1+PBhjRo1SmFhYXK73Ro1atS5Hp8peCsWAACEok6/FTt69GiVlpbq6quvlmEYOnjwoG8F70znn39+rw0QAAAAndPpFbuNGzdqyZIlOnXqVLttDMOQzWZTa2trrw3QDKzYAQCAUNSlDYqPHz+uzz77TJMmTdIrr7yikSNHBm03efLkXhugGQh2AAAgFHX6Vuy6deu0ePFipaam6oknnlBmZqYiIiLO5dgAAADQBbw8EQQrdgAAIBTx8gQAAIBF8PJEEKzYAQCAUMTLE0EQ7AAAQCjq9K1YSRo2bJjv5Yns7GyFh4efq3EBAACgi7q0YjdQsGIHAABCUadX7EaMGKEPP/xQLpdLw4cPl81ma7ftv//9714ZHAAAADqv08HuwQcf1LBhw3z/3FGwAwAAQN/jVmwQ3IoFAAChqNMrdh6Pp9OdEoYAAAD6XqeDXUxMTKdvv4b6PnYAAAChqNPB7rXXXvP986effqrly5drwYIFyszMlCRVVVXpySefVFFRUe+PEgAAAGfVrWfsrrrqKi1atEg33nijX/mf/vQnbdy4Ua+//npvjc8UPGMHAABC0aDunFRVVaX09PSA8vT0dP3f//1fjwcFAACArutWsEtISNCjjz4aUP7YY48pISGhy/1t2LBBSUlJcjgcSktLU0VFRYftW1patHLlSiUmJio8PFzjxo3T448/7qsvKSmRzWYLOJqbm7s8NgAAgFDRpZ8Ua/Pggw9q7ty5evnll5WRkSFJ2rVrlz766COVlpZ2qa+tW7eqsLBQGzZsUHZ2th577DHNnj1b+/bt0/nnnx/0nHnz5unzzz/Xpk2bdMEFF+jIkSM6deqUX5vo6Gjt37/fr8zhcHRpbAAAAKGk2/vYHTx4UMXFxXr//fdlGIYmTpyogoKCLq/YTZ8+XVOnTlVxcbGvLCUlRXPmzAn6IsZLL72kG264QR9//LFGjBgRtM+SkhIVFhbq2LFjXRpLG56xAwAgtDSe8KqhyStP80lFRwyRa6hdzki72cPqc91asZOkMWPG6Ne//nWHbX784x9r9erVcrlcQeu9Xq+qq6u1fPlyv/Lc3FxVVlYGPWfbtm1KT0/X/fffr6efflpDhw7Vt7/9bf3qV79SRESEr11TU5MSExPV2tqqSy65RL/61a80ZcqULs4SAAD0d4ePfam7SveqorbBV5aT7NKauZM0OiaigzOtp1vP2HXWH//4xw43Nm5oaFBra6tiY2P9ymNjY1VfXx/0nI8//lhvvvmm/vGPf+iFF17QQw89pOeee04/+clPfG0mTJigkpISbdu2TVu2bJHD4VB2drZqa2uD9tnS0iKPx+N3AACA/q/xhDcg1ElSeW2DlpfuVeMJr0kjM0e3V+w6o7N3ec/c+NgwjHY3Qz59+rRsNps2b94sp9MpSXrggQf0X//1X3rkkUcUERGhjIwM37N/kpSdna2pU6dq/fr1WrduXUCfRUVFWrVqVWenBQAA+omGJm9AqGtTXtughibvgLole05X7M7G5XIpLCwsYHXuyJEjAat4beLj43Xeeef5Qp301TN5hmHo4MGDQc8ZNGiQLr300nZX7FasWKHGxkbfceDAgW7OCAAA9CVP88kO64+fpd5qTA12drtdaWlpKisr8ysvKytTVlZW0HOys7N1+PBhNTU1+co+/PBDDRo0SGPGjAl6jmEY2rNnj+Lj44PWh4eHKzo62u8AAAD9X7RjSIf1w85SbzWmBjtJWrZsmf7whz/o8ccf1/vvv6+f/exnqqurU0FBgaSvVtPmz5/va3/TTTdp5MiRuvXWW7Vv3z6Vl5frzjvv1A9+8APfyxOrVq3Syy+/rI8//lh79uzRwoULtWfPHl+fAADAGlxRduUkB39JMyfZJVfUwLkNK53jZ+w6Iy8vT0ePHtXq1avldruVmpqq7du3KzExUZLkdrtVV1fnax8VFaWysjItWbJE6enpGjlypObNm6f77rvP1+bYsWNavHix6uvr5XQ6NWXKFJWXl2vatGl9Pj8AAHDuOCPtWjN3kpaX7lX5GW/Frp07aUA9Xyf1YB+7zvjRj36kX/3qV+1ud9JfsY8dAAChpW0fu+PNJzXMMUSuqIG5j12Pgt2JEydUV1cnr9f/VeJJkyb1eGBmItgBAIBQ1K1bsf/617906623aseOHUHrW1tbezQoAAAAdF23Xp4oLCzUF198oV27dikiIkIvvfSSnnzySSUnJ2vbtm29PUYAAAB0QrdW7F599VW9+OKLuvTSSzVo0CAlJibqP//zPxUdHa2ioiJdc801vT1OAAAAnEW3Vuz+3//7fxo1apQkacSIEfrXv/4lSbr44ov197//vfdGBwAAgE7rVrAbP3689u/fL0m65JJL9Nhjj+nQoUN69NFH290EGAAAAOdWt27FFhYWyu12S5LuvfdezZw5U5s3b5bdbldJSUlvjg8AAACd1Cv72J04cUIffPCBzj///JDbsy4YtjsBAAChqFu3YlevXq0TJ074PkdGRmrq1KkaOnSoVq9e3WuDAwAAQOd1a8UuLCxMbrfb9wJFm6NHj2rUqFEhv48dK3YAACAUdWvFzjAM2Wy2gPJ33nlHI0aM6PGgAAAA0HVdenli+PDhstlsstlsuvDCC/3CXWtrq5qamlRQUNDrgwQAAMDZdSnYPfTQQzIMQz/4wQ+0atUqOZ1OX53dbtfYsWOVmZnZ64MEAADA2XXrGbs33nhD2dnZGjy4W7ul9Hs8YwcAAEJRt56xu+yyy/TZZ5/p7rvv1o033qgjR45Ikl566SW99957vTpAAAAAdE63gt0bb7yhiy++WLt379bzzz+vpqYmSdLevXt177339uoAAQAA0DndCnbLly/Xfffdp7KyMtntdl/5FVdcoaqqql4bHAAAADqvW8Hu3Xff1Xe+852A8m984xs6evRojwcFAACArutWsIuJifH9VuzX1dTU6LzzzuvxoAAAANB13Qp2N910k+666y7V19fLZrPp9OnTeuutt3THHXdo/vz5vT1GAAAAdEK3tjs5efKkFixYoGeeeUaGYWjw4ME6deqUbr75ZpWUlCgsLOxcjLXPsN0JAAAIRd0Kdm0+/vhjvf3227LZbJoyZYouuOCC3hybaQh2AAAgFHV7h+FNmzbpwQcfVG1trSQpOTlZhYWFWrRoUa8NDgAAAJ3XrWB3zz336MEHH9SSJUt8PyFWVVWln/3sZ/r0009133339eogAQAAcHbduhXrcrm0fv163XjjjX7lW7Zs0ZIlS9TQ0NBrAzQDt2IBAEAo6tZbsa2trUpPTw8oT0tL06lTp3o8KAAAAHRdt4LdLbfcouLi4oDyjRs36uabb+7xoAAAANB1nX7GbtmyZb5/ttls+sMf/qCdO3cqIyNDkrRr1y4dOHCAfewAAABM0uln7K644orOdWiz6dVXX+3RoMzGM3YAACAU9WgfO6si2AEAgFDUrWfsAAAA0P8Q7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFtEvgt2GDRuUlJQkh8OhtLQ0VVRUdNi+paVFK1euVGJiosLDwzVu3Dg9/vjjfm1KS0s1ceJEhYeHa+LEiXrhhRfO5RQAAABMZ3qw27p1qwoLC7Vy5UrV1NRoxowZmj17turq6to9Z968efrf//1fbdq0Sfv379eWLVs0YcIEX31VVZXy8vKUn5+vd955R/n5+Zo3b552797dF1MCAAAwhem/PDF9+nRNnTpVxcXFvrKUlBTNmTNHRUVFAe1feukl3XDDDfr44481YsSIoH3m5eXJ4/Fox44dvrJZs2Zp+PDh2rJly1nHxC9PAACAUGTqip3X61V1dbVyc3P9ynNzc1VZWRn0nG3btik9PV3333+/zjvvPF144YW644479OWXX/raVFVVBfQ5c+bMdvtsaWmRx+PxOwAAAELNYDO/vKGhQa2trYqNjfUrj42NVX19fdBzPv74Y7355ptyOBx64YUX1NDQoB//+Mf697//7XvOrr6+vkt9FhUVadWqVb0wIwAAAPOY/oydJNlsNr/PhmEElLU5ffq0bDabNm/erGnTpunqq6/WAw88oJKSEr9Vu670uWLFCjU2NvqOAwcO9HBGAAAAfc/UFTuXy6WwsLCAlbQjR44ErLi1iY+P13nnnSen0+krS0lJkWEYOnjwoJKTkxUXF9elPsPDwxUeHt7D2QAAAJjL1BU7u92utLQ0lZWV+ZWXlZUpKysr6DnZ2dk6fPiwmpqafGUffvihBg0apDFjxkiSMjMzA/rcuXNnu30CAABYgem3YpctW6Y//OEPevzxx/X+++/rZz/7merq6lRQUCDpq9uk8+fP97W/6aabNHLkSN16663at2+fysvLdeedd+oHP/iBIiIiJElLly7Vzp07tXbtWn3wwQdau3atXnnlFRUWFpoxRQAAgD5h6q1Y6autSY4eParVq1fL7XYrNTVV27dvV2JioiTJ7Xb77WkXFRWlsrIyLVmyROnp6Ro5cqTmzZun++67z9cmKytLzzzzjO6++27dc889GjdunLZu3arp06f3+fwAAAD6iun72PVH7GMHAABCkem3YgEAANA7CHYAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCIIdAACARRDsAAAALKJfBLsNGzYoKSlJDodDaWlpqqioaLft66+/LpvNFnB88MEHvjYlJSVB2zQ3N/fFdAAAAEwx2OwBbN26VYWFhdqwYYOys7P12GOPafbs2dq3b5/OP//8ds/bv3+/oqOjfZ+/8Y1v+NVHR0dr//79fmUOh6N3Bw8AANCPmB7sHnjgAS1cuFCLFi2SJD300EN6+eWXVVxcrKKionbPGzVqlGJiYtqtt9lsiouL6+3hAgAA9Fum3or1er2qrq5Wbm6uX3lubq4qKys7PHfKlCmKj4/XVVddpddeey2gvqmpSYmJiRozZoyuvfZa1dTUtNtXS0uLPB6P3wEAABBqTA12DQ0Nam1tVWxsrF95bGys6uvrg54THx+vjRs3qrS0VM8//7zGjx+vq666SuXl5b42EyZMUElJibZt26YtW7bI4XAoOztbtbW1QfssKiqS0+n0HQkJCb03SQAAgD5iMwzDMOvLDx8+rPPOO0+VlZXKzMz0lf/617/W008/7fdCREeuu+462Ww2bdu2LWj96dOnNXXqVOXk5GjdunUB9S0tLWppafF99ng8SkhIUGNjo99zfAAAAP2ZqSt2LpdLYWFhAatzR44cCVjF60hGRka7q3GSNGjQIF166aXttgkPD1d0dLTfAQAAEGpMDXZ2u11paWkqKyvzKy8rK1NWVlan+6mpqVF8fHy79YZhaM+ePR22AQAACHWmvxW7bNky5efnKz09XZmZmdq4caPq6upUUFAgSVqxYoUOHTqkp556StJXb82OHTtWF110kbxer/74xz+qtLRUpaWlvj5XrVqljIwMJScny+PxaN26ddqzZ48eeeQRU+YIAADQF0wPdnl5eTp69KhWr14tt9ut1NRUbd++XYmJiZIkt9uturo6X3uv16s77rhDhw4dUkREhC666CL99a9/1dVXX+1rc+zYMS1evFj19fVyOp2aMmWKysvLNW3atD6fHwAAQF8x9eWJ/srj8cjpdPLyBAAACCn94ifFAAAA0HMEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAiCHYAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAi+kWw27Bhg5KSkuRwOJSWlqaKiop2277++uuy2WwBxwcffODXrrS0VBMnTlR4eLgmTpyoF1544VxPAwAAwFSmB7utW7eqsLBQK1euVE1NjWbMmKHZs2errq6uw/P2798vt9vtO5KTk311VVVVysvLU35+vt555x3l5+dr3rx52r1797meDgAAgGlshmEYZg5g+vTpmjp1qoqLi31lKSkpmjNnjoqKigLav/7667riiiv0xRdfKCYmJmifeXl58ng82rFjh69s1qxZGj58uLZs2XLWMXk8HjmdTjU2Nio6OrrrkwIAADCBqSt2Xq9X1dXVys3N9SvPzc1VZWVlh+dOmTJF8fHxuuqqq/Taa6/51VVVVQX0OXPmzLP2CQAAEMoGm/nlDQ0Nam1tVWxsrF95bGys6uvrg54THx+vjRs3Ki0tTS0tLXr66ad11VVX6fXXX1dOTo4kqb6+vkt9trS0qKWlxffZ4/H0ZFoAAACmMDXYtbHZbH6fDcMIKGszfvx4jR8/3vc5MzNTBw4c0G9/+1tfsOtqn0VFRVq1alV3hw8AANAvmHor1uVyKSwsLGAl7ciRIwErbh3JyMhQbW2t73NcXFyX+lyxYoUaGxt9x4EDB7owCwAAgP7B1GBnt9uVlpamsrIyv/KysjJlZWV1up+amhrFx8f7PmdmZgb0uXPnznb7DA8PV3R0tN8BAAAQaky/Fbts2TLl5+crPT1dmZmZ2rhxo+rq6lRQUCDpq9W0Q4cO6amnnpIkPfTQQxo7dqwuuugieb1e/fGPf1RpaalKS0t9fS5dulQ5OTlau3atrr/+er344ot65ZVX9Oabb5oyRwAAgL5gerDLy8vT0aNHtXr1arndbqWmpmr79u1KTEyUJLndbr897bxer+644w4dOnRIERERuuiii/TXv/5VV199ta9NVlaWnnnmGd1999265557NG7cOG3dulXTp0/v8/kBAAD0FdP3seuP2McOAACEItN/eQIAAAC9g2AHAABgEQQ7AAAAiyDYAQAAWATBDgAAwCIIdgAAABZBsAMAALAIgh0AAIBFEOwAAAAsgmAHAABgEQQ7AAAAiyDYAQAAWATBDgAAwCIIdgAAABZBsAMAALCIwWYPAAAAIBQ1nvCqockrT/NJRUcMkWuoXc5Iu6ljItgBAAB00eFjX+qu0r2qqG3wleUku7Rm7iSNjokwbVzcigUAAOiCxhPegFAnSeW1DVpeuleNJ7wmjYxgBwAA0CUNTd6AUNemvLZBDU0EOwAAgJDgaT7ZYf3xs9SfSwQ7AACALoh2DOmwfthZ6s8lgh0AAEAXuKLsykl2Ba3LSXbJFWXem7EEOwAAgC5wRtq1Zu6kgHCXk+zS2rmTTN3yxGYYhmHat/dTHo9HTqdTjY2Nio6ONns4AACgH2rbx+5480kNcwyRK4p97AAAAEKSM9L8IHcmbsUCAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIvhJMQAAztD2G6Ce5pOKjhgi19D+99NRQDAEOwAAvubwsS91V+leVdQ2+Mpykl1aM3eSRsdEmDgy4Oz6xa3YDRs2KCkpSQ6HQ2lpaaqoqOjUeW+99ZYGDx6sSy65xK+8pKRENpst4Ghubj4HowcAWEXjCW9AqJOk8toGLS/dq8YTXpNGBnSO6cFu69atKiws1MqVK1VTU6MZM2Zo9uzZqqur6/C8xsZGzZ8/X1dddVXQ+ujoaLndbr/D4XCciykAACyiockbEOralNc2qKGJYIf+zfRg98ADD2jhwoVatGiRUlJS9NBDDykhIUHFxcUdnvfDH/5QN910kzIzM4PW22w2xcXF+R0AAHTE03yyw/rjZ6kHzGZqsPN6vaqurlZubq5feW5uriorK9s974knntBHH32ke++9t902TU1NSkxM1JgxY3Tttdeqpqam18YNALCmaMeQDuuHnaUeMJupwa6hoUGtra2KjY31K4+NjVV9fX3Qc2pra7V8+XJt3rxZgwcHf/djwoQJKikp0bZt27RlyxY5HA5lZ2ertrY2aPuWlhZ5PB6/AwAw8Lii7MpJdgWty0l2yRXFm7Ho30y/FSt9ddv06wzDCCiTpNbWVt10001atWqVLrzwwnb7y8jI0C233KLJkydrxowZ+vOf/6wLL7xQ69evD9q+qKhITqfTdyQkJPRsQgCAkOSMtGvN3EkB4S4n2aW1cyex5Qn6PZthGIZZX+71ehUZGalnn31W3/nOd3zlS5cu1Z49e/TGG2/4tT927JiGDx+usLAwX9np06dlGIbCwsK0c+dOXXnllUG/67bbbtPBgwe1Y8eOgLqWlha1tLT4Pns8HiUkJKixsVHR0dE9nSYAIMS07WN3vPmkhjmGyBXFPnYIDabuY2e325WWlqaysjK/YFdWVqbrr78+oH10dLTeffddv7INGzbo1Vdf1XPPPaekpKSg32MYhvbs2aOLL744aH14eLjCw8N7MBMAgJU4IwlyCE2mb1C8bNky5efnKz09XZmZmdq4caPq6upUUFAgSVqxYoUOHTqkp556SoMGDVJqaqrf+aNGjZLD4fArX7VqlTIyMpScnCyPx6N169Zpz549euSRR/p0bgAAAH3J9GCXl5eno0ePavXq1XK73UpNTdX27duVmJgoSXK73Wfd0+5Mx44d0+LFi1VfXy+n06kpU6aovLxc06ZNOxdTAAAA6BdMfcauv/J4PHI6nTxjBwAAQkq/eCsWAAAAPUewAwAAsAiCHQAAgEUQ7AAAACyCYAcAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFmH6T4r1R20/xuHxeEweCQAAwFeGDRsmm83WYRuCXRDHjx+XJCUkJJg8EgAAgK905qdO+a3YIE6fPq3Dhw93KhkPNB6PRwkJCTpw4AC/o9sPcD36F65H/8L16F+4Hj3Hil03DRo0SGPGjDF7GP1adHQ0fzH7Ea5H/8L16F+4Hv0L1+Pc4uUJAAAAiyDYAQAAWATBDl0SHh6ue++9V+Hh4WYPBeJ69Ddcj/6F69G/cD36Bi9PAAAAWAQrdgAAABZBsAMAALAIgh0AAIBFEOwGsC+++EL5+flyOp1yOp3Kz8/XsWPHOjzHMAz98pe/1OjRoxUREaHLL79c7733nl+blpYWLVmyRC6XS0OHDtW3v/1tHTx4MGh/LS0tuuSSS2Sz2bRnz55emlloMvN6fPvb39b5558vh8Oh+Ph45efn6/Dhw709xZBj1jX59NNPtXDhQiUlJSkiIkLjxo3TvffeK6/Xey6mGTLM/Dvy61//WllZWYqMjFRMTEwvzyw0bNiwQUlJSXI4HEpLS1NFRUWH7d944w2lpaXJ4XDoP/7jP/Too48GtCktLdXEiRMVHh6uiRMn6oUXXujx9w54BgasWbNmGampqUZlZaVRWVlppKamGtdee22H56xZs8YYNmyYUVpaarz77rtGXl6eER8fb3g8Hl+bgoIC47zzzjPKysqMv//978YVV1xhTJ482Th16lRAf7fffrsxe/ZsQ5JRU1PT21MMKWZejwceeMCoqqoyPv30U+Ott94yMjMzjczMzHM211Bh1jXZsWOHsWDBAuPll182PvroI+PFF180Ro0aZfz85z8/p/Pt78z8O/KLX/zCeOCBB4xly5YZTqfzXE2x33rmmWeMIUOGGL///e+Nffv2GUuXLjWGDh1qfPbZZ0Hbf/zxx0ZkZKSxdOlSY9++fcbvf/97Y8iQIcZzzz3na1NZWWmEhYUZv/nNb4z333/f+M1vfmMMHjzY2LVrV7e/F4ZBsBug9u3bZ0jy+wtUVVVlSDI++OCDoOecPn3aiIuLM9asWeMra25uNpxOp/Hoo48ahmEYx44dM4YMGWI888wzvjaHDh0yBg0aZLz00kt+/W3fvt2YMGGC8d577w34YNcfrsfXvfjii4bNZjO8Xm9Ppxay+ts1uf/++42kpKSeTitk9Zfr8cQTTwzIYDdt2jSjoKDAr2zChAnG8uXLg7b/7//+b2PChAl+ZT/84Q+NjIwM3+d58+YZs2bN8mszc+ZM44Ybbuj298IwuBU7QFVVVcnpdGr69Om+soyMDDmdTlVWVgY955NPPlF9fb1yc3N9ZeHh4brssst851RXV+vkyZN+bUaPHq3U1FS/fj///HPddtttevrppxUZGdnb0ws5Zl+Pr/v3v/+tzZs3KysrS0OGDOmN6YWk/nRNpK9+/HvEiBE9nVbI6m/XYyDxer2qrq72+zOSpNzc3Hb/jKqqqgLaz5w5U2+//bZOnjzZYZu2PrvzveAZuwGrvr5eo0aNCigfNWqU6uvr2z1HkmJjY/3KY2NjfXX19fWy2+0aPnx4u20Mw9CCBQtUUFCg9PT0Hs/FCsy8Hm3uuusuDR06VCNHjlRdXZ1efPHFbs/HCvrDNWnz0Ucfaf369SooKOjyPKyiP12PgaahoUGtra0d/jmeqb6+Pmj7U6dOqaGhocM2bX1253tBsLOcX/7yl7LZbB0eb7/9tiTJZrMFnG8YRtDyrzuzvjPnfL3N+vXr5fF4tGLFiq5MLSSFwvVoc+edd6qmpkY7d+5UWFiY5s+fL8OC+5eH0jWRpMOHD2vWrFn63ve+p0WLFp1teiEn1K7HQNbVP8dg7c8s70yf3bl+A9lgsweA3vXTn/5UN9xwQ4dtxo4dq7179+rzzz8PqPvXv/4V8H9HbeLi4iR99X9Z8fHxvvIjR474zomLi5PX69UXX3zh93/AR44cUVZWliTp1Vdf1a5duwJ+ViY9PV0333yznnzyyU7MNDSEwvVo43K55HK5dOGFFyolJUUJCQnatWuXMjMzOzfZEBFK1+Tw4cO64oorlJmZqY0bN3ZugiEmlK7HQOVyuRQWFhawSvb1P8czxcXFBW0/ePBgjRw5ssM2bX1253sh3oodqNoeRN69e7evbNeuXZ16EHnt2rW+spaWlqAPIm/dutXX5vDhw34PIn/22WfGu+++6ztefvllQ5Lx3HPPGQcOHDgX0+33zLwewdTV1RmSjNdee62HMwtdZl+TgwcPGsnJycYNN9wQ9I3ygcbs69FmIL888aMf/civLCUlpcOXJ1JSUvzKCgoKAl6emD17tl+bWbNmBbw80ZXvBW/FDmizZs0yJk2aZFRVVRlVVVXGxRdfHLB1wPjx443nn3/e93nNmjWG0+k0nn/+eePdd981brzxxqBbB4wZM8Z45ZVXjL///e/GlVde2e52J4ZhGJ988smAfyvWMMy7Hrt37zbWr19v1NTUGJ9++qnx6quvGt/85jeNcePGGc3NzX0z+X7KrGty6NAh44ILLjCuvPJK4+DBg4bb7fYdA5mZ/8767LPPjJqaGmPVqlVGVFSUUVNTY9TU1BjHjx8/9xPvB9q2Hdm0aZOxb98+o7Cw0Bg6dKjx6aefGoZhGMuXLzfy8/N97du2O/nZz35m7Nu3z9i0aVPAdidvvfWWERYWZqxZs8Z4//33jTVr1rS73Ul734tABLsB7OjRo8bNN99sDBs2zBg2bJhx8803G1988YVfG0nGE0884ft8+vRp49577zXi4uKM8PBwIycnx3j33Xf9zvnyyy+Nn/70p8aIESOMiIgI49prrzXq6uraHQfB7itmXY+9e/caV1xxhTFixAgjPDzcGDt2rFFQUGAcPHjwXE43JJh1TZ544glDUtBjIDPz31nf//73g16PgbSq/cgjjxiJiYmG3W43pk6darzxxhu+uu9///vGZZdd5tf+9ddfN6ZMmWLY7XZj7NixRnFxcUCfzz77rDF+/HhjyJAhxoQJE4zS0tIufS8C2QzDgk9HAwAADEC8FQsAAGARBDsAAACLINgBAABYBMEOAADAIgh2AAAAFkGwAwAAsAiCHQAAgEUQ7AAAACyCYAcAX3P55ZersLCw3foTJ05o7ty5io6Ols1m07Fjx7r8HWPHjtVDDz3k+2yz2fSXv/zF9/mDDz5QRkaGHA6HLrnkknbLAOBMg80eAACEkieffFIVFRWqrKyUy+WS0+nscZ9ut1vDhw/3fb733ns1dOhQ7d+/X1FRUe2WAcCZCHYA0AUfffSRUlJSlJqa2mt9xsXFBXzHNddco8TExA7LAOBM3IoFgDOcOnVKP/3pTxUTE6ORI0fq7rvvlmEYuvzyy/U///M/Ki8vl81m0+WXX37Wvo4cOaLrrrtOERERSkpK0ubNmwPafP1WrM1mU3V1tVavXi2bzaZf/vKXQcsAIBhW7ADgDE8++aQWLlyo3bt36+2339bixYuVmJio559/XsuXL9c//vEPPf/887Lb7Wfta8GCBTpw4IBeffVV2e123X777Tpy5Ei77d1ut771rW9p1qxZuuOOOxQVFaWCgoKAMgAIhmAHAGdISEjQgw8+KJvNpvHjx+vdd9/Vgw8+qNtuu02RkZGy2+0Bt0+D+fDDD7Vjxw7t2rVL06dPlyRt2rRJKSkp7Z4TFxenwYMHKyoqyvcdUVFRAWUAEAy3YgHgDBkZGbLZbL7PmZmZqq2tVWtra5f6ef/99zV48GClp6f7yiZMmKCYmJjeGioA+CHYAcA5YhiGJPmFRAA4lwh2AHCGXbt2BXxOTk5WWFhYl/pJSUnRqVOn9Pbbb/vK9u/f36297wCgMwh2AHCGAwcOaNmyZdq/f7+2bNmi9evXa+nSpV3uZ/z48Zo1a5Zuu+027d69W9XV1Vq0aJEiIiLOwagBgGAHAAHmz5+vL7/8UtOmTdNPfvITLVmyRIsXL+5WX0888YQSEhJ02WWX6bvf/a4WL16sUaNG9fKIAeArNqPtIRAAAACENFbsAAAALIJgBwDdVFFRoaioqHYPAOhr3IoFgG768ssvdejQoXbrL7jggj4cDQAQ7AAAACyDW7EAAAAWQbADAACwCIIdAACARRDsAAAALIJgBwAAYBEEOwAAAIsg2AEAAFgEwQ4AAMAi/j8tWpIp28uEagAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# testing associatoin of diversity \n",
    "div = 'shannon_entropy'\n",
    "bf = 'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis'\n",
    "df_merge = pd.concat([df_alpha, df_otu[bf]],axis=1)\n",
    "# df_merge = df_merge.drop(['502A', '503A'])\n",
    "\n",
    "print(scipy.stats.spearmanr(df_merge[div], df_merge[bf]))\n",
    "\n",
    "df_RA = df_merge[df_merge['Diagnosis'] == 'RA']\n",
    "df_UA = df_merge[df_merge['Diagnosis'] == 'Unaffected']\n",
    "\n",
    "print(scipy.stats.spearmanr(df_RA[div], df_RA[bf]))\n",
    "\n",
    "print(scipy.stats.spearmanr(df_RA[div].values - df_UA[div].values, df_RA[bf].values - df_UA[bf].values))\n",
    "\n",
    "# beta\n",
    "print(scipy.stats.spearmanr(df_paired_beta['Bray_Curtis'], df_RA[bf].values - df_UA[bf].values))\n",
    "# print(scipy.stats.spearmanr(df_paired_beta.drop('T3')['Bray_Curtis'], df_RA[bf].values - df_UA[bf].values))\n",
    "\n",
    "# \n",
    "df_merge = pd.DataFrame({'beta_diff': df_paired_beta['Bray_Curtis'], 'bf_diff': df_RA[bf].values - df_UA[bf].values})\n",
    "\n",
    "sns.scatterplot(data=df_merge, x='bf_diff', y='beta_diff')\n",
    "sns.despine()\n",
    "plt.tight_layout()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[-0.00316979  0.00053966 -0.00232258 -0.00155014 -0.00116292 -0.00445142\n",
      " -0.00083639  0.00024919]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bray_Curtis</th>\n",
       "      <th>MTX</th>\n",
       "      <th>SJC</th>\n",
       "      <th>TJC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T2</th>\n",
       "      <td>0.705551</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3</th>\n",
       "      <td>0.474034</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T4</th>\n",
       "      <td>0.685602</td>\n",
       "      <td>1.0</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T5</th>\n",
       "      <td>0.725669</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T6</th>\n",
       "      <td>0.757568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T7</th>\n",
       "      <td>0.839525</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T8</th>\n",
       "      <td>0.452763</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T9</th>\n",
       "      <td>0.648924</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Bray_Curtis  MTX  SJC  TJC\n",
       "T2     0.705551  1.0    1    1\n",
       "T3     0.474034  1.0    7    7\n",
       "T4     0.685602  1.0   27   25\n",
       "T5     0.725669  0.0    3   21\n",
       "T6     0.757568  0.0    6   13\n",
       "T7     0.839525  1.0    0    0\n",
       "T8     0.452763  1.0    1    4\n",
       "T9     0.648924  0.0    5   12"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(df_RA[bf].values - df_UA[bf].values)\n",
    "df_paired_beta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Acetate \n",
      "SignificanceResult(statistic=0.45, pvalue=0.09235737494812253)\n",
      "PearsonRResult(statistic=0.3745870399424124, pvalue=0.16893963710591794)\n",
      "SignificanceResult(statistic=0.3571428571428572, pvalue=0.38512064355625636)\n",
      "PearsonRResult(statistic=-0.09353532514884613, pvalue=0.825641489546712)\n",
      "Propionate \n",
      "SignificanceResult(statistic=0.28214285714285714, pvalue=0.3082844568200554)\n",
      "PearsonRResult(statistic=0.09141021464076574, pvalue=0.7459413071183899)\n",
      "SignificanceResult(statistic=0.30952380952380953, pvalue=0.4556448907375822)\n",
      "PearsonRResult(statistic=-0.1536364661994486, pvalue=0.7164326014219166)\n",
      "Butyrate \n",
      "SignificanceResult(statistic=0.22857142857142856, pvalue=0.41256617049636035)\n",
      "PearsonRResult(statistic=0.19543739622845313, pvalue=0.4851541034122076)\n",
      "SignificanceResult(statistic=0.30952380952380953, pvalue=0.4556448907375822)\n",
      "PearsonRResult(statistic=0.31802134547591104, pvalue=0.4426949949127731)\n",
      "Isovalerate \n",
      "SignificanceResult(statistic=0.010714285714285714, pvalue=0.9697698761545686)\n",
      "PearsonRResult(statistic=0.04745095841163002, pvalue=0.8666410687983006)\n",
      "SignificanceResult(statistic=-0.14285714285714288, pvalue=0.735764859879812)\n",
      "PearsonRResult(statistic=-0.29551549822688217, pvalue=0.47732228571125723)\n",
      "Valerate \n",
      "SignificanceResult(statistic=-0.35357142857142854, pvalue=0.19607167843419876)\n",
      "PearsonRResult(statistic=-0.32554024765150175, pvalue=0.23640120541184761)\n",
      "SignificanceResult(statistic=-0.4285714285714286, pvalue=0.2894032248467901)\n",
      "PearsonRResult(statistic=-0.3033622108182784, pvalue=0.46512989484288486)\n",
      "Hexanoate \n",
      "SignificanceResult(statistic=-0.0035714285714285713, pvalue=0.9899214068809781)\n",
      "PearsonRResult(statistic=0.2994966796740263, pvalue=0.2781631037989611)\n",
      "SignificanceResult(statistic=0.2380952380952381, pvalue=0.5701563208157683)\n",
      "PearsonRResult(statistic=0.2593223313188229, pvalue=0.535129511810669)\n",
      "Heptanoate \n",
      "SignificanceResult(statistic=0.02857142857142857, pvalue=0.9194899083821857)\n",
      "PearsonRResult(statistic=-0.17505970883614563, pvalue=0.5326061489243894)\n",
      "SignificanceResult(statistic=-0.19047619047619052, pvalue=0.6514014957024814)\n",
      "PearsonRResult(statistic=-0.37539134324618867, pvalue=0.35947033283522123)\n",
      "Octanoate \n",
      "SignificanceResult(statistic=0.11428571428571428, pvalue=0.6850657626568972)\n",
      "PearsonRResult(statistic=0.10360986645779144, pvalue=0.7132774057168408)\n",
      "SignificanceResult(statistic=-0.4285714285714286, pvalue=0.2894032248467901)\n",
      "PearsonRResult(statistic=-0.3482228136302642, pvalue=0.39794364646943814)\n",
      "Nonanoic_acid\n",
      "SignificanceResult(statistic=0.09285714285714286, pvalue=0.742045009907367)\n",
      "PearsonRResult(statistic=-0.03170173226008541, pvalue=0.9107000419821716)\n",
      "SignificanceResult(statistic=-0.09523809523809526, pvalue=0.8225054302036114)\n",
      "PearsonRResult(statistic=-0.22006975821732758, pvalue=0.6004983006806149)\n",
      "Decanoic_acid\n",
      "SignificanceResult(statistic=0.5535714285714286, pvalue=0.03228710377012356)\n",
      "PearsonRResult(statistic=0.133206015218875, pvalue=0.6360231119883923)\n",
      "SignificanceResult(statistic=0.30952380952380953, pvalue=0.4556448907375822)\n",
      "PearsonRResult(statistic=0.3629919786491989, pvalue=0.37681297356301063)\n",
      "Dodecanoic_acid\n",
      "SignificanceResult(statistic=0.5357142857142857, pvalue=0.03956697222954944)\n",
      "PearsonRResult(statistic=0.07823253556765286, pvalue=0.7816790436190095)\n",
      "SignificanceResult(statistic=0.5000000000000001, pvalue=0.20703124999999997)\n",
      "PearsonRResult(statistic=0.7034493425463277, pvalue=0.051557452859172556)\n"
     ]
    }
   ],
   "source": [
    "# df_fa\n",
    "bf = 'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis'\n",
    "df_merge = pd.concat([df_fa, df_otu[bf], df_dx],axis=1)\n",
    "# df_merge = df_merge.drop(['502A', '503A'])\n",
    "# df_merge = df_merge.dropna(how='any')\n",
    "\n",
    "for fa in df_fa.columns:\n",
    "    print(fa)\n",
    "    print(scipy.stats.spearmanr(df_merge[fa], df_merge[bf], nan_policy='omit'))\n",
    "    df_plot = pd.concat([df_merge[fa], df_merge[bf]], axis=1)\n",
    "    df_plot = df_plot.dropna()\n",
    "    print(scipy.stats.pearsonr(df_plot[fa], df_plot[bf]))\n",
    "    \n",
    "    df_RA = df_merge[df_merge['Diagnosis'] == 'RA']\n",
    "    df_UA = df_merge[df_merge['Diagnosis'] == 'Unaffected']\n",
    "\n",
    "    plt.figure(figsize=(6,4))\n",
    "    sns.scatterplot(data=df_merge, x=bf, y=fa, hue='Diagnosis')\n",
    "    sns.despine()\n",
    "    plt.tight_layout()\n",
    "    plt.savefig(path + 'outputs/jobs22/bf_fa_corrs/bf_' + fa + '.pdf') \n",
    "    plt.close()\n",
    "    \n",
    "    dfa = df_RA[fa] #- df_UA[fa].values\n",
    "    dbf = df_RA[bf] #- df_UA[bf].values\n",
    "    print(scipy.stats.spearmanr(dfa.values, dbf.values, nan_policy='omit'))\n",
    "    df_plot = pd.concat([dfa, dbf], axis=1)\n",
    "    df_plot = df_plot.dropna()    \n",
    "    print(scipy.stats.pearsonr(df_plot[fa], df_plot[bf]))\n",
    "\n",
    "    df_merged = pd.DataFrame({'diff_bf': dbf, 'diff_' + fa: dfa})\n",
    "    plt.figure(figsize=(6,4))\n",
    "    sns.scatterplot(data=df_merged, x='diff_bf', y='diff_' + fa)\n",
    "    sns.despine()\n",
    "    plt.tight_layout()\n",
    "    plt.savefig(path + 'outputs/jobs22/bf_fa_corrs/bf_' + fa + '_diff.pdf') \n",
    "    plt.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "500A      18.961611\n",
       "501A      23.364729\n",
       "502A       6.222841\n",
       "503A      11.350637\n",
       "504A      26.324529\n",
       "505A      40.646318\n",
       "506A       9.933258\n",
       "508A       5.415508\n",
       "509A     333.730602\n",
       "510A       3.006532\n",
       "511A      25.632757\n",
       "512A     666.873182\n",
       "513A     132.694466\n",
       "514A     801.416089\n",
       "515A    2976.820507\n",
       "507A            NaN\n",
       "Name: Dodecanoic_acid, dtype: float64"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge[fa]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cyanoamino_acid_metabolism</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>CXCL1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500A</th>\n",
       "      <td>0.001914</td>\n",
       "      <td>RA</td>\n",
       "      <td>7.062380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501A</th>\n",
       "      <td>0.001631</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>6.989813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502A</th>\n",
       "      <td>0.002139</td>\n",
       "      <td>RA</td>\n",
       "      <td>8.102450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503A</th>\n",
       "      <td>0.001869</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>7.471897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504A</th>\n",
       "      <td>0.002183</td>\n",
       "      <td>RA</td>\n",
       "      <td>9.101940</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Cyanoamino_acid_metabolism   Diagnosis     CXCL1\n",
       "500A                    0.001914          RA  7.062380\n",
       "501A                    0.001631  Unaffected  6.989813\n",
       "502A                    0.002139          RA  8.102450\n",
       "503A                    0.001869  Unaffected  7.471897\n",
       "504A                    0.002183          RA  9.101940"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# CUTIE pathway plots\n",
    "x =  'Metabolism|Biosynthesis of other secondary metabolites|Monobactam biosynthesis'\n",
    "df = pd.concat([df_path[x], df_dx, df_quant.loc[:,['CXCL1']]],axis=1)\n",
    "df.columns = ['Monobactam_biosynthesis','Diagnosis','CXCL1']\n",
    "df.to_csv(path + 'outputs/jobs02/df_mb.tsv', sep='\\t')\n",
    "df.head()\n",
    "\n",
    "x = 'Metabolism|Metabolism of other amino acids|Cyanoamino acid metabolism'\n",
    "df = pd.concat([df_path[x], df_dx, df_quant.loc[:,['CXCL1']]],axis=1)\n",
    "df.columns = ['Cyanoamino_acid_metabolism','Diagnosis','CXCL1']\n",
    "df.to_csv(path + 'outputs/jobs02/df_ca.tsv', sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%capture\n",
    "# pathway analysis\n",
    "# collect individual pathway dfs\n",
    "#pathway_fps = pathlib.Path(path + 'outputs/ensemble_humann3/').glob('*A_genefamilies.tsv')\n",
    "# pathway_fps = pathlib.Path(path + 'outputs/ensemble_humann3/').glob('*A_pathabundance.tsv')\n",
    "\n",
    "# collect and parse dfs\n",
    "#dfs = []\n",
    "#for fp in pathway_fps:\n",
    "#    df = pd.read_csv(path + 'outputs/ensemble_humann3/' + fp.name, sep='\\t', index_col=0)\n",
    "#    df = df/df.sum()\n",
    "#    dfs.append(df)\n",
    "\n",
    "# merge dataframes\n",
    "#df_pathway = pd.concat(dfs, axis=1).T\n",
    "df_pathway = pd.read_csv(path + 'outputs/jobs08/humann3_joined_pathabundance.tsv', sep='\\t',index_col=0)\n",
    "\n",
    "# drop UNINTEGRATED, UNMAPPED\n",
    "df_pathway = df_pathway.drop(['UNINTEGRATED','UNMAPPED'])\n",
    "\n",
    "# replace nan with 0\n",
    "df_pathway = df_pathway.replace(np.nan,0)\n",
    "\n",
    "# transpose and normalize\n",
    "df_pathway = df_pathway.T\n",
    "df_pathway = df_pathway.div(df_pathway.sum(axis=1),axis=0)\n",
    "\n",
    "# test first x pathways\n",
    "# df_pathway = df_pathway.iloc[:,:100]\n",
    "\n",
    "# conduct KW test\n",
    "RA_ids = ['512A_Abundance',#-RPKs', \n",
    "          '508A_Abundance',#-RPKs', \n",
    "          '514A_Abundance',#-RPKs',\n",
    "          '502A_Abundance',#-RPKs',\n",
    "          '500A_Abundance',#-RPKs',\n",
    "          '510A_Abundance',#-RPKs',\n",
    "          '504A_Abundance']#-RPKs',\n",
    "\n",
    "UA_ids = ['513A_Abundance',#-RPKs', \n",
    "          '509A_Abundance',#-RPKs', \n",
    "          '515A_Abundance',#-RPKs',\n",
    "          '503A_Abundance',#-RPKs', \n",
    "          '501A_Abundance',#-RPKs',\n",
    "          '511A_Abundance',#-RPKs',\n",
    "          '505A_Abundance']#-RPKs']\n",
    "\n",
    "pathways = list(df_pathway.columns.values)\n",
    "\n",
    "# SR ONLY\n",
    "names = []\n",
    "tvals = []\n",
    "pvals = []\n",
    "test = []\n",
    "direction = []\n",
    "\n",
    "for x in pathways:\n",
    "    plot = False\n",
    "    RA_vals = df_pathway.loc[RA_ids,x].values\n",
    "    UA_vals = df_pathway.loc[UA_ids,x].values\n",
    "    try:\n",
    "        #h,p = scipy.stats.kruskal(RA_vals, UA_vals)\n",
    "        if np.median(RA_vals) > np.median(UA_vals):\n",
    "            d = 'RA'\n",
    "        else:\n",
    "            d = 'UA'\n",
    "        #names.append(x)\n",
    "        #tvals.append(h)\n",
    "        #pvals.append(p)\n",
    "        #test.append('KW')\n",
    "        #direction.append(d)\n",
    "        #if p<0.05:\n",
    "        #    plot = True\n",
    "        diff = RA_vals - UA_vals\n",
    "        if sum(np.isnan(diff)) < 2:\n",
    "            w,p = scipy.stats.wilcoxon(RA_vals, UA_vals)\n",
    "            names.append(x)\n",
    "            tvals.append(w)\n",
    "            pvals.append(p)\n",
    "            test.append('SR')\n",
    "            direction.append(d)\n",
    "            if p < 0.05:\n",
    "                plot = True\n",
    "    except:\n",
    "        plot = False\n",
    "    if plot:\n",
    "        #sns.stripplot([RA_vals, UA_vals])\n",
    "        sns.boxplot([RA_vals, UA_vals])\n",
    "        plt.title(x)\n",
    "        plt.savefig(path + 'outputs/jobs08/sig_metacyc/' + x + '.pdf',  bbox_inches=\"tight\")\n",
    "        plt.close()\n",
    "\n",
    "df_sigpath = pd.DataFrame.from_dict({\n",
    "    'pathway': names,\n",
    "    'test_stat': tvals,\n",
    "    'pvalue': pvals,\n",
    "    'test': test,\n",
    "    'direction': direction\n",
    "})\n",
    "df_sigpath = df_sigpath.set_index('pathway')\n",
    "df_sigpath['FDR_bh'] = scipy.stats.false_discovery_control(df_sigpath['pvalue'].values)\n",
    "df_sigpath_FDR = df_sigpath[df_sigpath['FDR_bh'] < 0.05]\n",
    "df_sigpath = df_sigpath[df_sigpath['pvalue'] < 0.05] \n",
    "df_sigpath.to_csv(path + 'outputs/jobs08/df_sigpath.tsv', sep='\\t')\n",
    "df_sigpath_FDR.to_csv(path + 'outputs/jobs08/df_sigpath_FDR.tsv', sep='\\t')\n",
    "\n",
    "print(len(df_sigpath)) # should be 27\n",
    "df_sigpath.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%capture\n",
    "# pathway analysis for KO\n",
    "# collect individual pathway dfs\n",
    "#pathway_fps = pathlib.Path(path + 'outputs/ensemble_humann3/').glob('*A_genefamilies.tsv')\n",
    "# pathway_fps = pathlib.Path(path + 'outputs/ensemble_humann3/').glob('*A_pathabundance.tsv')\n",
    "# df_pathway = pd.read_csv(path + 'outputs/ensemble_humann3/humann3_joined_KO_no_misses_relabund_nobar_brite.tsv', sep='\\t')\n",
    "df_pathway = pd.read_csv(path + 'outputs/jobs08/humann3_joined_KO_no_misses_relabund_stratified_brite.tsv', sep='\\t')#,index_col=0)\n",
    "\n",
    "# set index and transpose\n",
    "df_pathway = df_pathway.set_index('pathways')\n",
    "df_pathway = df_pathway.T\n",
    "\n",
    "# rename samples\n",
    "df_pathway.index = df_pathway.index.map(lambda x: x.split('_')[0] + '_Abundance')\n",
    "\n",
    "# test first x pathways\n",
    "# df_pathway = df_pathway.iloc[:,:100]\n",
    "\n",
    "# conduct KW test\n",
    "RA_ids = ['512A_Abundance',#-RPKs', \n",
    "          '516A_Abundance',#-RPKs',\n",
    "          '508A_Abundance',#-RPKs', \n",
    "          '514A_Abundance',#-RPKs',\n",
    "          '502A_Abundance',#-RPKs',\n",
    "          '500A_Abundance',#-RPKs',\n",
    "          '510A_Abundance',#-RPKs',\n",
    "          '504A_Abundance']#-RPKs',\n",
    "\n",
    "UA_ids = ['513A_Abundance',#-RPKs', \n",
    "          '517A_Abundance',#-RPKs', \n",
    "          '509A_Abundance',#-RPKs', \n",
    "          '515A_Abundance',#-RPKs',\n",
    "          '503A_Abundance',#-RPKs', \n",
    "          '501A_Abundance',#-RPKs',\n",
    "          '511A_Abundance',#-RPKs',\n",
    "          '505A_Abundance']#-RPKs']\n",
    "\n",
    "pathways = list(df_pathway.columns.values)\n",
    "\n",
    "names = []\n",
    "tvals = []\n",
    "pvals = []\n",
    "test = []\n",
    "\n",
    "for x in pathways:\n",
    "    plot = False\n",
    "    RA_vals = df_pathway.loc[RA_ids,x].values\n",
    "    UA_vals = df_pathway.loc[UA_ids,x].values\n",
    "    try:\n",
    "        #h,p = scipy.stats.kruskal(RA_vals, UA_vals)\n",
    "        #if p < 0.05:\n",
    "        #    names.append(x)\n",
    "        #    tvals.append(h)\n",
    "        #    pvals.append(p)\n",
    "        #    test.append('KW')\n",
    "        #    plot = True\n",
    "        diff = RA_vals - UA_vals\n",
    "        if sum(np.isnan(diff)) < 2:\n",
    "            w,p = scipy.stats.wilcoxon(RA_vals, UA_vals)\n",
    "            names.append(x)\n",
    "            tvals.append(w)\n",
    "            pvals.append(p)\n",
    "            test.append('SR')\n",
    "            if p < 0.05:\n",
    "                plot = True\n",
    "    except:\n",
    "        plot = False\n",
    "    if plot:\n",
    "        #sns.stripplot([RA_vals, UA_vals])\n",
    "        sns.boxplot([RA_vals, UA_vals])\n",
    "        plt.title(x)\n",
    "        plt.savefig(path + 'outputs/ensemble_humann3/sig_ko/' + x + '.pdf',  bbox_inches=\"tight\")\n",
    "        plt.close()\n",
    "\n",
    "df_sigpath = pd.DataFrame.from_dict({\n",
    "    'pathway': names,\n",
    "    'test_stat': tvals,\n",
    "    'pvalue': pvals,\n",
    "    'test': test\n",
    "})\n",
    "df_sigpath = df_sigpath.set_index('pathway')\n",
    "df_sigpath['FDR_bh'] = scipy.stats.false_discovery_control(df_sigpath['pvalue'].values)\n",
    "df_sigpath_FDR = df_sigpath[df_sigpath['FDR_bh'] < 0.05]\n",
    "df_sigpath = df_sigpath[df_sigpath['pvalue'] < 0.05] \n",
    "df_sigpath.to_csv(path + 'outputs/jobs08/df_sigko.tsv', sep='\\t')\n",
    "df_sigpath_FDR.to_csv(path + 'outputs/jobs08/df_sigko_FDR.tsv', sep='\\t')\n",
    "\n",
    "print(len(df_sigpath_FDR)) # should be 0, before or after FDR\n",
    "df_sigpath_FDR.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# identify specific FA producers\n",
    "df_pathway = pd.read_csv(path + 'outputs/jobs08/humann3_joined_pathabundance.tsv', sep='\\t',index_col=0)\n",
    "\n",
    "# drop UNINTEGRATED, UNMAPPED\n",
    "df_pathway = df_pathway.drop(['UNINTEGRATED','UNMAPPED'])\n",
    "\n",
    "# replace nan with 0\n",
    "df_pathway = df_pathway.replace(np.nan,0)\n",
    "\n",
    "# transpose and normalize\n",
    "df_pathway = df_pathway.T\n",
    "df_pathway = df_pathway.div(df_pathway.sum(axis=1),axis=0)\n",
    "\n",
    "pathways = list(df_pathway.columns.values)\n",
    "\n",
    "# specific queries\n",
    "queries = ['butanoate','octanoate']\n",
    "\n",
    "# IDs of samples; dropped twin pair that didn't have FA measurement?\n",
    "RA_ids = ['512A_Abundance',#-RPKs', \n",
    "          '508A_Abundance',#-RPKs', \n",
    "          '514A_Abundance',#-RPKs',\n",
    "          '502A_Abundance',#-RPKs',\n",
    "          '500A_Abundance',#-RPKs',\n",
    "          '510A_Abundance',#-RPKs',\n",
    "          '504A_Abundance']#-RPKs',\n",
    "\n",
    "UA_ids = ['513A_Abundance',#-RPKs', \n",
    "          '509A_Abundance',#-RPKs', \n",
    "          '515A_Abundance',#-RPKs',\n",
    "          '503A_Abundance',#-RPKs', \n",
    "          '501A_Abundance',#-RPKs',\n",
    "          '511A_Abundance',#-RPKs',\n",
    "          '505A_Abundance']#-RPKs']\n",
    "\n",
    "for q in queries:\n",
    "    for x in pathways:\n",
    "        pathway = x.lower()\n",
    "        if q in pathway:\n",
    "            RA_vals = df_pathway.loc[RA_ids,x].values\n",
    "            UA_vals = df_pathway.loc[UA_ids,x].values\n",
    "            h,p = scipy.stats.kruskal(RA_vals, UA_vals)\n",
    "            if p < 0.05:\n",
    "                print(x)\n",
    "                print('KW')\n",
    "                print(p)\n",
    "            diff = RA_vals - UA_vals\n",
    "            if sum(np.isnan(diff)) < 2:\n",
    "                w,p = scipy.stats.wilcoxon(RA_vals, UA_vals)\n",
    "                if p<0.05:\n",
    "                    print(x)\n",
    "                    print('RS')\n",
    "                    print(p)\n",
    "            sns.boxplot([RA_vals, UA_vals])\n",
    "            plt.title(x)\n",
    "            plt.savefig(path + 'outputs/ensemble_humann3/' + q +'/' + x + '.pdf',  bbox_inches=\"tight\")\n",
    "            plt.close()        \n",
    "\n",
    "        \n",
    "    #if 'fatty acid' in x.lower():\n",
    "    #    print(x)\n",
    "    # if 'fatty acid' in pathway:\n",
    "    #    if 'mitochondria' not in pathway:\n",
    "    #        print(x)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# pseudo perm test chi square \n",
    "# iterate and grab each df post dropping near zero var\n",
    "import scipy\n",
    "import sklearn.metrics\n",
    "import itertools\n",
    "import random\n",
    "\n",
    "random.seed(0)\n",
    "\n",
    "name_to_df = {\n",
    "    'olink':df_olink,\n",
    "    'fa': df_fa,\n",
    "    'acpa_fecal': df_acpa_fecal,\n",
    "    'acpa_plasma': df_acpa_plasma,\n",
    "    'mb': df_mb,\n",
    "    'asv_filt': df_asv_filt,\n",
    "    'path_filt': df_path_filt,\n",
    "    'asv': df_asv.drop(['Diagnosis'],axis=1),\n",
    "    'path': df_path.drop(['Diagnosis','TwinPair'],axis=1),\n",
    "    'quant_R': pd.concat([df_quant, df_asv.drop(['Diagnosis'],axis=1), df_path.drop(['Diagnosis','TwinPair'],axis=1)], axis=1)\n",
    "}\n",
    "# each row will be data type, num clusters, type of test/association, and p value\n",
    "df_arr = []\n",
    "k_arr = []\n",
    "stat_arr = []\n",
    "p_arr = []\n",
    "\n",
    "# max cluster iter\n",
    "k_max = 6\n",
    "\n",
    "for x in name_to_df:\n",
    "    print(x)\n",
    "    df = pd.read_csv(path + 'outputs/jobs19/mat_' + x + '_meta.tsv',sep='\\t') # written from R code\n",
    "    if x == 'fa':\n",
    "        df.index = df_meta_edit.index.drop('507A')\n",
    "        df = pd.concat([df_meta_edit.drop('507A'), df], axis=1)\n",
    "    else:\n",
    "        df.index = df_meta_edit.index\n",
    "        df = pd.concat([df_meta_edit, df], axis=1)\n",
    "\n",
    "    if x == 'quant_R':\n",
    "        df = df.drop('507A')\n",
    "    \n",
    "    # get distance matrix\n",
    "    mdata = df.loc[:,['TwinPair','Diagnosis','TJC','SJC']]\n",
    "    data = df.drop(['TwinPair','Diagnosis','TJC','SJC'],axis=1)\n",
    "    if x == 'quant_R':\n",
    "        data = data.values.astype(float)\n",
    "    condensed_dist = scipy.spatial.distance.pdist(data, metric='euclidean')\n",
    "    # noncond_dist = distance.squareform(condensed_dist)\n",
    "    \n",
    "    # get linkages from dist matrix\n",
    "    linkages = scipy.cluster.hierarchy.linkage(condensed_dist, method='average', metric='euclidian')\n",
    "    \n",
    "    for k in range(2,k_max+1):\n",
    "        # [1 1 1 1 1 1 1 1 1 1 1 2 1 1 1 1]\n",
    "        cut = scipy.cluster.hierarchy.fcluster(linkages, k, criterion='maxclust')\n",
    "    \n",
    "        # we want completeness of twinpair per cluster (all twins from same pair in same cluster)\n",
    "        # and we want homogeneity of disease state per cluster (clusters should be as unfiorm as possible)\n",
    "        # we can construct a statistical test via non parametric permutation testing of shuffling of labels (either TP or Dx)\n",
    "        # given the same tree structure/cluster partitioning, so we keep cut the same and shuttle the labels\n",
    "    \n",
    "        # for the twins\n",
    "        true=mdata['TwinPair'].values\n",
    "        perm_cs = []\n",
    "        # do arbitrarily a subset of 1000 perms \n",
    "        n_perm = 100\n",
    "        for i in range(n_perm):\n",
    "            perm = random.sample(list(true), k=len(true))\n",
    "            perm_cs.append(sklearn.metrics.completeness_score(labels_true=perm, labels_pred=cut))\n",
    "    \n",
    "        true_cs = sklearn.metrics.completeness_score(labels_true=true, labels_pred=cut)\n",
    "    \n",
    "        percentile = scipy.stats.percentileofscore(perm_cs, true_cs)\n",
    "        df_arr.append(x)\n",
    "        k_arr.append(k)\n",
    "        stat_arr.append('TwinPair_completeness')\n",
    "        p_arr.append((100-percentile)/100) \n",
    "        \n",
    "        # for the dx\n",
    "        true=mdata['Diagnosis'].values\n",
    "        perm_cs = []\n",
    "        n_perm = 100\n",
    "        for i in range(n_perm):\n",
    "            perm = random.sample(list(true), k=len(true))\n",
    "            perm_cs.append(sklearn.metrics.homogeneity_score(labels_true=perm, labels_pred=cut))\n",
    "    \n",
    "        true_cs = sklearn.metrics.homogeneity_score(labels_true=true, labels_pred=cut)\n",
    "    \n",
    "        percentile = scipy.stats.percentileofscore(perm_cs, true_cs)\n",
    "        df_arr.append(x)\n",
    "        k_arr.append(k)\n",
    "        stat_arr.append('Diagnosis_homogeneity')\n",
    "        p_arr.append((100-percentile)/100) \n",
    "\n",
    "        for var in ['TJC','SJC']:\n",
    "            mdata['cluster'] = cut\n",
    "            vecs = []\n",
    "            for i in range(1, k+1):\n",
    "                vecs.append(list(mdata[mdata['cluster'] == i][var].values))\n",
    "            \n",
    "            H,p = scipy.stats.kruskal(*vecs)\n",
    "            df_arr.append(x)\n",
    "            k_arr.append(k)\n",
    "            stat_arr.append(var + '_KW')\n",
    "            p_arr.append((100-percentile)/100) \n",
    "\n",
    "df_results = pd.DataFrame.from_dict({'DataType':df_arr,\n",
    "                                     'n_clust':k_arr,\n",
    "                                     'statistic':stat_arr,\n",
    "                                     'pval':p_arr})\n",
    "\n",
    "df_results[df_results['pval'] < 0.05]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# inter intra dist, a similar idea to above about completeness/homogeneity\n",
    "str_to_df = {\n",
    "    'df_mb': df_mb,\n",
    "    'df_olink': df_olink,\n",
    "    'df_fa': df_fa,\n",
    "    'df_acpa_plasma': df_acpa_plasma,\n",
    "    'df_acpa_fecal': df_acpa_fecal,\n",
    "    'df_path_filt': df_path_filt\n",
    "}\n",
    "\n",
    "\n",
    "for name in str_to_df:\n",
    "    print(name)\n",
    "    meta_var = name\n",
    "    df_var = str_to_df[name].astype(float)\n",
    "\n",
    "    # subtract each col value by the mean and divide by standard dev\n",
    "    df_var_std = (df_var - df_var.mean())/df_var.std()\n",
    "    \n",
    "    # from earlier analyses, pair_to_ids maps 'T2': ('501A','500A')\n",
    "    # if name = 'df_fa' drop the T5 pair\n",
    "    pair_to_dist = {}\n",
    "    for p in pair_to_ids:\n",
    "        id_UA, id_RA = pair_to_ids[p]\n",
    "        if name == 'df_fa':\n",
    "            if p == 'T5':\n",
    "                continue\n",
    "        pair_to_dist[p] = np.linalg.norm(df_var_std.loc[id_UA,:].values-df_var_std.loc[id_RA,:].values)\n",
    "    \n",
    "    df_paired_var = pd.DataFrame.from_dict(pair_to_dist, orient='index', columns=['OLINK'])\n",
    "    \n",
    "    # merge with diagnosis data\n",
    "    df_var_meta = pd.concat([df_dx, df_var],axis=1)\n",
    "    \n",
    "    # get df_raw from distances\n",
    "    id_to_dist = {}\n",
    "    \n",
    "    samples = list(df_var.index.values)\n",
    "    SubjectID1 = []\n",
    "    SubjectID2 = []\n",
    "    Group1 = []\n",
    "    Group2 = []\n",
    "    Distance = []\n",
    "    \n",
    "    for i in range(len(samples)):\n",
    "        for j in range(i):\n",
    "            SubjectID1.append(samples[j])\n",
    "            SubjectID2.append(samples[i])\n",
    "            Group1.append(df_map.loc[samples[j],'Diagnosis'])\n",
    "            Group2.append(df_map.loc[samples[i],'Diagnosis'])\n",
    "            temp = df_var_std.loc[[samples[j],samples[i]],:]\n",
    "            temp = temp.dropna(how='any',axis=1)\n",
    "            Distance.append(np.linalg.norm(temp.loc[samples[j],:].values-temp.loc[samples[i],:].values))\n",
    "    \n",
    "    df_raw = pd.DataFrame.from_dict({'SubjectID1':SubjectID1, \n",
    "                                     'SubjectID2':SubjectID2, \n",
    "                                     'Group1':Group1, \n",
    "                                     'Group2': Group2, \n",
    "                                     'Distance': Distance}, orient='columns')#, columns=['OLINK'])\n",
    "    \n",
    "                        \n",
    "    # get split DFs\n",
    "    df_RA = df_raw[df_raw['Group1'] == 'RA']\n",
    "    df_RA = df_RA[df_RA['Group2'] == 'RA']\n",
    "    df_UA = df_raw[df_raw['Group1'] == 'Unaffected']\n",
    "    df_UA = df_UA[df_UA['Group2'] == 'Unaffected']\n",
    "    \n",
    "    # compare distances\n",
    "    inter_twin = df_paired_beta['Bray_Curtis'].values\n",
    "    inter_RA = df_RA['Distance'].values\n",
    "    inter_UA = df_UA['Distance'].values\n",
    "    \n",
    "    u, p = scipy.stats.mannwhitneyu(inter_twin, inter_RA)\n",
    "    print(u, p)\n",
    "    \n",
    "    t, p = scipy.stats.ttest_ind(inter_twin, inter_RA)\n",
    "    #print(t, p)\n",
    "    \n",
    "    t, p = scipy.stats.ttest_ind(inter_UA, inter_RA)\n",
    "    # print(t, p)\n",
    "    \n",
    "    f, p = scipy.stats.f_oneway(inter_UA, inter_RA, inter_twin)\n",
    "    print(f, p)\n",
    "    \n",
    "    category = ['intra_twin_pair']*len(inter_twin) + ['inter_RA_only']*len(inter_RA) + ['inter_UA_only']*len(inter_UA)\n",
    "    distances = list(inter_twin) + list(inter_RA) + list(inter_UA)\n",
    "    df_dist = pd.DataFrame(data=np.array([category,distances]).T, columns=['category',meta_var + ' distance'])\n",
    "    df_dist[meta_var + ' distance'] = df_dist[meta_var + ' distance'].astype(float)\n",
    "    df_dist.to_csv(path + 'outputs/jobs10/inter_intra_' + meta_var + '_dist.tsv',sep='\\t')\n",
    "                         \n",
    "    sns.boxplot(data=df_dist, x='category', y=meta_var + ' distance')\n",
    "    sns.swarmplot(data=df_dist, x='category', y=meta_var + ' distance', color='black')\n",
    "    plt.savefig(path + 'outputs/jobs10/plot_inter_intra_' + meta_var + '.pdf',  bbox_inches=\"tight\")\n",
    "    plt.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get # of TP correlations with and without FDR\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "# jobs17 UA, jobs18 RA taxa\n",
    "# jobs20 RA, jobs21 UA pathways\n",
    "var_to_job = {\n",
    "    'UA_taxa': 'jobs17',\n",
    "    'RA_taxa': 'jobs18',\n",
    "    'RA_path': 'jobs20',\n",
    "    'UA_path': 'jobs21'}\n",
    "\n",
    "\n",
    "\n",
    "# grab relevant correlations\n",
    "for v in var_to_job:\n",
    "    print(v)\n",
    "    for a in ['noFDR']:#,'FDR']:\n",
    "        print(a)\n",
    "        # load dataframe of interest\n",
    "        df = pd.read_csv(path + 'outputs/' + var_to_job[v] + '/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "\n",
    "        # focus on var of interest\n",
    "        if v.split('_')[-1] == 'taxa':\n",
    "            # grab taxa of interest and pathways of interest\n",
    "            taxa_df = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/15_taxa/15_taxa_stats_clean.csv')\n",
    "            var_int = list(taxa_df[taxa_df['wilcoxon'] < 0.05]['feature'].values)\n",
    "            var_int = var_int + ['Bilophila wadsworthia']\n",
    "            var_int = ['k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis',\n",
    "                       'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia']\n",
    "\n",
    "        elif v.split('_')[-1] == 'path':            \n",
    "            path_df = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/16_path/16_path_stats_clean.csv')\n",
    "            var_int = list(path_df[path_df['wilcoxon'] < 0.05]['feature'].values)\n",
    "            var_int = [y.replace('|', '_').replace('/', '__').replace('\\\\', '__') for y in var_int]\n",
    "            df['var1'] = [x.replace('|', '_').replace('/', '__').replace('\\\\', '__') for x in list(df['var1'])]\n",
    "\n",
    "        # for each var of interest\n",
    "        print(var_int)\n",
    "        counter=0\n",
    "        for f in var_int:\n",
    "            print(f)\n",
    "            # df_f = df[df['var1'].str.contains(f)]\n",
    "            df_f = df[df['var1'] == f]\n",
    "            print(len(df_f))\n",
    "\n",
    "            # FDR \n",
    "            if a == 'FDR':\n",
    "                df_f['padjust'] = multipletests(df_f['pvalues'], method='fdr_bh')[1]\n",
    "                df_f = df_f[df_f['padjust'] < 0.05]\n",
    "            # filter on TP\n",
    "            df_f = df_f[df_f['class'] == 'TP']\n",
    "            counter += len(df_f['var2'].values)\n",
    "\n",
    "            # summarize results\n",
    "            print(len(df_f['var2'].values))\n",
    "        print('Total')\n",
    "        print(counter)\n",
    "            # print(df_f['var2'].values)\n",
    "df_f.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# jobs17 UA, jobs18 RA\n",
    "df = pd.read_csv(path + 'outputs/jobs17/data_processing/summary_df_resample_1.txt', sep='\\t')\n",
    "df = df[df['var1'] == 'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis']\n",
    "# df = df[df['var1'] == 'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia']\n",
    "\n",
    "\n",
    "df['padjust'] = multipletests(df['pvalues'], method='fdr_bh')[1]\n",
    "df = df[df['class'] == 'TP']\n",
    "print(len(df))\n",
    "bf_var2 = df['var2'].values\n",
    "df2 = df[df['padjust'] < 0.05]\n",
    "print(len(df2))\n",
    "\n",
    "# find intersect with differential ACPAs from fecal and plasma\n",
    "df_acpa = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/10_acpa_fecal/acpa_fecal_stats_clean.csv')\n",
    "df_acpa = df_acpa[df_acpa['ttest'] < 0.05]\n",
    "diffs_fecal = list(df_acpa['acpa fecal analyte'].values)\n",
    "diffs_fecal = [x.replace(' I','_I') for x in diffs_fecal]\n",
    "diffs_fecal = [x + '_Fecal' for x in diffs_fecal]\n",
    "\n",
    "df_acpa = pd.read_csv(path + 'inputs/RA_twins/16S/jobs/11_acpa_plasma/acpa_plasma_stats_clean.csv')\n",
    "df_acpa = df_acpa[df_acpa['ttest'] < 0.05]\n",
    "diffs_plasma = list(df_acpa['acpa plasma analyte'].values)\n",
    "diffs_plasma = [x.replace(' I','_I') for x in diffs_plasma]\n",
    "diffs_plasma = [x + '_Plasma' for x in diffs_plasma]\n",
    "\n",
    "\n",
    "#df = df[df['var2'].isin(['SCF','Flt3L'] + diffs_fecal + diffs_plasma)]\n",
    "#print(['SCF','Flt3L'] + diffs_fecal + diffs_plasma)\n",
    "#print(df['var2'].values)\n",
    "df = df[df['correlations'] > 0]\n",
    "df\n",
    "# df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CUTIE plots\n",
    "x =  'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis'\n",
    "df = pd.concat([df_asv_filt[x], df_dx, df_quant.loc[:,bf_var2[0:2]]],axis=1)\n",
    "df.columns = ['Blautia_faecis','Diagnosis','Flt3L','SCF']#,'ApoE_277_296_Cit2_sm2_cyclic_IgG_fecal','FibrinogenB_54_72_IgA_fecal','Fibrinogen_IgA_fecal']\n",
    "df.to_csv(path + 'outputs/jobs02/df_bf.tsv', sep='\\t')\n",
    "df.head()\n",
    "\n",
    "x =  'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia'\n",
    "df = pd.concat([df_asv_filt[x], df_dx, df_quant.loc[:,bw_var2]],axis=1)\n",
    "df.columns = ['Bilophila_wadsworthia','Diagnosis','IL7','TNFB']\n",
    "df.to_csv(path + 'outputs/jobs02/df_bw.tsv', sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# perform cohen's D\n",
    "def cohens_d_2s(group1, group2):\n",
    "    \"\"\"\n",
    "    Calculate Cohen's d for two groups.\n",
    "\n",
    "    Parameters:\n",
    "    group1, group2 (array-like): The two groups for which to calculate Cohen's d.\n",
    "\n",
    "    Returns:\n",
    "    d (float): The computed Cohen's d value.\n",
    "    \"\"\"\n",
    "    mean1, mean2 = np.mean(group1), np.mean(group2)\n",
    "    var1, var2 = np.var(group1, ddof=1), np.var(group2, ddof=1)\n",
    "    n1, n2 = len(group1), len(group2)\n",
    "\n",
    "    pooled_std = np.sqrt(((n1 - 1) * var1 + (n2 - 1) * var2) / (n1 + n2 - 2))\n",
    "    d = (mean1 - mean2) / pooled_std\n",
    "\n",
    "    return d\n",
    "\n",
    "\n",
    "def cohens_d_1s(group):\n",
    "    \"\"\"\n",
    "    \"\"\"\n",
    "    mean = np.mean(group)\n",
    "    var = np.var(group, ddof=1)\n",
    "    n = len(group)\n",
    "    d = mean / np.sqrt(var)\n",
    "\n",
    "    return d\n",
    "\n",
    "def mad(data):\n",
    "    \"\"\"\n",
    "    Median absolute difference\n",
    "    \"\"\"\n",
    "    median = np.median(data)\n",
    "    deviations = np.abs(data - median)\n",
    "    mad = np.median(deviations)\n",
    "    return mad\n",
    "\n",
    "def mag_d_1s(B):\n",
    "    return np.median(B) / np.sqrt((mad(B) ** 2)/2)\n",
    "\n",
    "def mag_d_2s(A, B):\n",
    "    return (np.median(B) - np.median(A)) / np.sqrt((mad(A) ** 2 + mad(B) ** 2)/2)\n",
    "\n",
    "    \n",
    "# Example usage:\n",
    "# group1 = [1, 2, 3, 4, 5]\n",
    "# group2 = [3, 4, 5, 6, 7]\n",
    "\n",
    "# effect_size = cohens_d(group1, group2)\n",
    "# print(\"Cohen's d:\", effect_size)\n",
    "\n",
    "str_to_df = {\n",
    "    'df_mb': df_mb,\n",
    "    'df_olink': df_olink,\n",
    "    'df_fa': df_fa,\n",
    "    'df_acpa_plasma': df_acpa_plasma,\n",
    "    'df_acpa_fecal': df_acpa_fecal,\n",
    "    'df_asv': df_asv.drop('Diagnosis',axis=1),\n",
    "    'df_path': df_path.drop(['Diagnosis','TwinPair'],axis=1)\n",
    "}\n",
    "\n",
    "\n",
    "# for each dataframe / data type \n",
    "# compute the top differential features via t-test\n",
    "# except for asv and path, which you should take the top from wilcoxon\n",
    " \n",
    "var = []\n",
    "pv = []\n",
    "stat = []\n",
    "cd2s = []\n",
    "cd1s = []\n",
    "dtype = []\n",
    "md2s = []\n",
    "md1s = []\n",
    "\n",
    "for d in str_to_df:\n",
    "    print(d)\n",
    "    # grab df and merge with dx metadata\n",
    "    df_orig = str_to_df[d]\n",
    "    # d = 'df_fa'\n",
    "    # df = df_fa.copy()    \n",
    "    df = pd.concat([df_meta_edit, df_orig], axis=1)\n",
    "    if d == 'df_fa':\n",
    "        df = df.drop(['506A','507A'])\n",
    "    \n",
    "    features = list(df.columns.values)[4:]\n",
    "    pairs = list(set(df.TwinPair.values))\n",
    "    \n",
    "    df_RA = df[df['Diagnosis'] == 'RA']\n",
    "    df_UA = df[df['Diagnosis'] == 'Unaffected']\n",
    "    \n",
    "    \n",
    "    for f in features:\n",
    "        # get the vectors for each diagnosis\n",
    "        diffs = []\n",
    "        RA_vals = []\n",
    "        UA_vals = []\n",
    "        for t in pairs:\n",
    "            twin_df = df[df['TwinPair'] == t]\n",
    "            fRAt = twin_df[twin_df['Diagnosis'] == 'RA'][f].values.astype(float)[0]\n",
    "            fUAt = twin_df[twin_df['Diagnosis'] == 'Unaffected'][f].values.astype(float)[0]\n",
    "            diffs.append(fRAt - fUAt)\n",
    "            RA_vals.append(fRAt)\n",
    "            UA_vals.append(fUAt)\n",
    "    \n",
    "        # do paired testing\n",
    "        if d in ['df_asv','df_path']:\n",
    "            t,p = scipy.stats.wilcoxon(RA_vals, UA_vals)\n",
    "        else:\n",
    "            t,p = scipy.stats.ttest_rel(RA_vals, UA_vals)\n",
    "    \n",
    "        if p < 0.05:\n",
    "            stat.append(t)\n",
    "            pv.append(p)\n",
    "            var.append(f)\n",
    "            dtype.append(d)\n",
    "\n",
    "    \n",
    "        \n",
    "            # get unpaired vectors \n",
    "            fRA = df[df['Diagnosis'] == 'RA'][f].astype(float).values\n",
    "            fUA = df[df['Diagnosis'] == 'Unaffected'][f].astype(float).values\n",
    "            cd_up = cohens_d_2s(fRA, fUA)\n",
    "            cd_p = cohens_d_1s(diffs)\n",
    "    \n",
    "            cd2s.append(cd_up)\n",
    "            cd1s.append(cd_p)\n",
    "\n",
    "            md_up = mag_d_2s(fRA, fUA)\n",
    "            md_p = mag_d_1s(diffs)\n",
    "            \n",
    "            md2s.append(md_up)\n",
    "            md1s.append(md_p)\n",
    "        \n",
    "df_results = pd.DataFrame({'dtype': dtype,\n",
    "                           'feature': var,\n",
    "                           'pval': pv,\n",
    "                           'test_stat': stat,\n",
    "                           'cohens_d_unpaired': cd2s,\n",
    "                           'cohens_d_paired': cd1s,\n",
    "                           'mag_d_unpaired': md2s,\n",
    "                           'mag_d_paired': md1s})\n",
    "\n",
    "df_results['cohens_d_unpaired_abs'] = df_results['cohens_d_unpaired'].apply(lambda x: np.abs(x))\n",
    "df_results['cohens_d_paired_abs'] = df_results['cohens_d_paired'].apply(lambda x: np.abs(x))\n",
    "df_results['mag_d_unpaired_abs'] = df_results['mag_d_unpaired'].apply(lambda x: np.abs(x))\n",
    "df_results['mag_d_paired_abs'] = df_results['mag_d_paired'].apply(lambda x: np.abs(x))\n",
    "\n",
    "df_results.to_csv(path + 'inputs/cohensd_pretest.tsv', sep='\\t')\n",
    "df_results.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sort by unpaired\n",
    "metrics = ['Mag_D','Cohens_D']\n",
    "for m in metrics:\n",
    "    metric_p = m.lower() + '_paired_abs'\n",
    "    metric_up = m.lower() + '_unpaired_abs'\n",
    "    \n",
    "    sort_2s_df = df_results.sort_values(metric_p, ascending=False)\n",
    "    sort_1s_df = df_results.sort_values(metric_up, ascending=False)\n",
    "    \n",
    "    # get first three entires of each row\n",
    "    sort_2s_df = sort_2s_df.groupby('dtype').head(3)\n",
    "    sort_1s_df = sort_1s_df.groupby('dtype').head(3)\n",
    "    \n",
    "    # export to df\n",
    "    # sort_2s_df.to_csv(path + 'inputs/sort_2s_df.tsv', sep='\\t')\n",
    "    # sort_1s_df.to_csv(path + 'inputs/sort_1s_df.tsv', sep='\\t')\n",
    "    \n",
    "    # preview\n",
    "    df = sort_1s_df.copy()\n",
    "    # df = df[df['dtype'] != 'df_path']\n",
    "    df['Direction'] = df['cohens_d_paired'].apply(lambda x: 'Unaffected' if x < 0 else 'RA')\n",
    "    df = df.drop(['pval','test_stat','dtype'],axis=1)\n",
    "\n",
    "    # melt\n",
    "    df = pd.melt(df, id_vars=['feature','Direction'], value_vars=[metric_up, metric_p], var_name='Measure', value_name='Value')\n",
    "    \n",
    "    # rename for formatting\n",
    "    df['Measure'] = df['Measure'].replace({metric_up: m + '_Unpaired', metric_p: m + '_Paired'})\n",
    "    df = df[df['Measure'] == m + '_Paired']\n",
    "    # trim features\n",
    "    # df['feature'] = df['feature'].apply(lambda x: x[-25:])\n",
    "    a =  'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  Blautia|s  Blautia faecis'\n",
    "    b =  'k  Bacteria|p  Proteobacteria|c  Deltaproteobacteria|o  Desulfovibrionales|f  Desulfovibrionaceae|g  Bilophila|s  Bilophila wadsworthia'\n",
    "    mapping = {a: 'Blautia faecis', b: 'Bilophila wadsworthia'}\n",
    "    df['feature'] = df['feature'].replace(mapping)\n",
    "\n",
    "    a = 'Metabolism|Xenobiotics biodegradation and metabolism|Atrazine degradation',\n",
    "    b = 'Metabolism|Biosynthesis of other secondary metabolites|Biosynthesis of various plant secondary metabolites'\n",
    "    c = 'Metabolism|Amino acid metabolism|Lysine biosynthesis'\n",
    "    mapping = {a: 'Atrazine degradation', b: 'Biosynthesis of Plant Secondary Metabolites', c: 'Lysine biosynthesis'}\n",
    "    df['feature'] = df['feature'].replace(mapping)\n",
    "    \n",
    "    df = df[df['feature'] != 'k  Bacteria|p  Firmicutes|c  Clostridia|o  Eubacteriales|f  Lachnospiraceae|g  GGB3746|s  GGB3746 SGB5089']\n",
    "    df = df[df['feature'] != 'k  Bacteria|p  Firmicutes|c  CFGB3004|o  OFGB3004|f  FGB3004|g  GGB58158|s  GGB58158 SGB79798']\n",
    "    df = df[df['feature'] != 'Bilophila wadsworthia']\n",
    "    df.to_csv(path + 'inputs/' + m.lower() + '_1s_test_df.tsv', sep='\\t')\n",
    "\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[[16,25,26],:].feature.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# perform random forest \n",
    "from sklearn.model_selection import cross_val_score,cross_val_predict,  KFold,  LeaveOneOut, StratifiedKFold\n",
    "from sklearn.metrics import roc_curve, auc\n",
    "from sklearn import datasets\n",
    "from sklearn.datasets import load_wine\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn.metrics import RocCurveDisplay\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.svm import SVC\n",
    "\n",
    "diagnosis = df_num['BinDiag']\n",
    "\n",
    "# get X and y data\n",
    "X, y = df_num.drop('BinDiag',axis=1).values, df_num['BinDiag'].values\n",
    "\n",
    "str_to_df = {\n",
    "    'df_mb': df_mb,\n",
    "    'df_olink': df_olink,\n",
    "    'df_fa': df_fa,\n",
    "    'df_acpa_plasma': df_acpa_plasma,\n",
    "    'df_acpa_fecal': df_acpa_fecal,\n",
    "    'df_asv': df_asv_filt,\n",
    "    'df_path': df_path_filt\n",
    "}\n",
    "\n",
    "\n",
    "for name in str_to_df:\n",
    "    print(name)\n",
    "    df = str_to_df[name]\n",
    "    # get columns of interest\n",
    "    features = [x.strip() for x in list(df.columns.values)]\n",
    "    \n",
    "    # subset df_rf\n",
    "    df_test = df_rf[features]\n",
    "    \n",
    "    # merge with diagnosis binarized\n",
    "    df_test = pd.concat([df_test, diagnosis], axis=1)\n",
    "\n",
    "    # complete cases\n",
    "    df_test = df_test.dropna(axis=0)\n",
    "    \n",
    "    # separate data and labels\n",
    "    X, y = df_test.drop('BinDiag',axis=1).values, df_test['BinDiag'].values\n",
    "\n",
    "    # set fold split\n",
    "    kf = LeaveOneOut()\n",
    "    # kf = KFold(n_splits=3)\n",
    "    \n",
    "    # try SVC\n",
    "    # clf = SVC(kernel='linear', class_weight='balanced', probability=True, random_state=42)\n",
    "    clf = SVC(probability=True, random_state=42)\n",
    "    all_y = []\n",
    "    all_probs=[]\n",
    "    for train, test in kf.split(X, y):\n",
    "        all_y.append(y[test])\n",
    "        all_probs.append(clf.fit(X[train], y[train]).predict_proba(X[test])[:,1])\n",
    "    all_y = np.array(all_y)\n",
    "    all_probs = np.array(all_probs)\n",
    "\n",
    "    fpr, tpr, thresholds = roc_curve(all_y,all_probs)\n",
    "    roc_auc = auc(fpr, tpr)\n",
    "    print('SVC')\n",
    "    print(roc_auc)\n",
    "        \n",
    "    # try RF\n",
    "    # clf = RandomForestClassifier(n_estimators=100, random_state=42, max_depth=10)\n",
    "    clf = RandomForestClassifier(n_estimators=100, random_state=42)\n",
    "    all_y = []\n",
    "    all_probs=[]\n",
    "    for train, test in kf.split(X, y):\n",
    "        all_y.append(y[test])\n",
    "        all_probs.append(clf.fit(X[train], y[train]).predict_proba(X[test])[:,1])\n",
    "    all_y = np.array(all_y)\n",
    "    all_probs = np.array(all_probs)\n",
    "\n",
    "    fpr, tpr, thresholds = roc_curve(all_y,all_probs)\n",
    "    roc_auc = auc(fpr, tpr)\n",
    "    print('RF')\n",
    "    print(roc_auc)\n",
    "    #print(all_probs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load in the new dataframe of pts\n",
    "df_rb = pd.read_csv('/Users/KevinBu/Desktop/clemente_lab/Projects/twinsra/inputs/Patient_data.csv', skiprows=1)\n",
    "df_rb = df_rb.iloc[0:20,:]\n",
    "df_rb['Subject'] = df_rb['Subject'].map(lambda x: str(int(x)) + 'A')\n",
    "df_rb = df_rb.set_index('Subject')\n",
    "df_rb['DAS28-CRP'] = df_rb['DAS28-CRP'].astype(float)\n",
    "df_rb = df_rb.drop(['182A','183A'])\n",
    "df_rb = df_rb.rename(columns={'CRP (performed 12/2023 on serum)':'CRP'})\n",
    "df_rb['Sibling_pair'] = df_rb['Sibling_pair'].map(lambda x: 'T' + str(int(x.split('_')[1])+1))\n",
    "df_rb = df_rb[df_rb['Sibling_pair'] != 'T10']\n",
    "# create new col for ordinal disease activity based on CRP DAS28\n",
    "'''\n",
    "remission =< 2.6\n",
    "low dis activity >2.6 - 3.2\n",
    "mod dis activity > 3.2 - 5.1\n",
    "hi dis activity >5.1\n",
    "'''\n",
    "conditions = [\n",
    "    (df_rb['DAS28-CRP'] <= 2.6),\n",
    "    (df_rb['DAS28-CRP'] > 2.6) & (df_rb['DAS28-CRP'] <= 3.2),\n",
    "    (df_rb['DAS28-CRP'] > 3.2) & (df_rb['DAS28-CRP'] <= 5.1),\n",
    "    (df_rb['DAS28-CRP'] > 5.1)\n",
    "]\n",
    "choices = ['remission', 'low', 'moderate','high']\n",
    "df_rb['disease_severity'] = np.select(conditions, choices, default='nan')\n",
    "\n",
    "# subset columns we want\n",
    "df_rb = df_rb.loc[:,['disease_severity','CRP','DAS28-CRP','Sibling_pair','Diagnosis']]\n",
    "\n",
    "# get paired version\n",
    "pair_to_diff = {}\n",
    "for p in list(df_rb['Sibling_pair'].values):\n",
    "    df = df_rb[df_rb['Sibling_pair'] == p]\n",
    "    vals_UA = float(df[df['Diagnosis'] == '1_Unaffected']['CRP'].values)\n",
    "    vals_RA = float(df[df['Diagnosis'] == '2_RA']['CRP'].values)\n",
    "    pair_to_diff[p] = vals_RA - vals_UA\n",
    "\n",
    "df_paired_CRP = pd.DataFrame.from_dict(pair_to_diff, orient='index', columns=['CRP_diff'])\n",
    "\n",
    "pair_to_diff = {}\n",
    "for p in list(df_rb['Sibling_pair'].values):\n",
    "    df = df_rb[df_rb['Sibling_pair'] == p]\n",
    "    vals_UA = float(df[df['Diagnosis'] == '1_Unaffected']['DAS28-CRP'].values)\n",
    "    vals_RA = float(df[df['Diagnosis'] == '2_RA']['DAS28-CRP'].values)\n",
    "    pair_to_diff[p] = vals_RA - vals_UA\n",
    "\n",
    "df_paired_DC = pd.DataFrame.from_dict(pair_to_diff, orient='index', columns=['DAS28-CRP_diff'])\n",
    "\n",
    "\n",
    "df_paired_final = pd.concat([df_paired_alpha, df_paired_beta, df_paired_CRP, df_paired_DC], axis=1)\n",
    "df_paired_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# correlations non paired\n",
    "#x = 'Shannon_Diff'\n",
    "#x = 'Bray_Curtis'\n",
    "#y = 'CRP_diff'\n",
    "#y = 'DAS28-CRP_diff'\n",
    "# take df_alpha \n",
    "df_alpha_mod = df_alpha.copy()\n",
    "\n",
    "# add col with SJC TJC\n",
    "df_alpha_mod['SJC'] = df_alpha_mod.index.map(samp_to_SJC)\n",
    "df_alpha_mod['TJC'] = df_alpha_mod.index.map(samp_to_TJC)\n",
    "\n",
    "\n",
    "# df_beta tbd since it is a distance matrix not PCoA coords yet\n",
    "df_corr = pd.concat([df_alpha_mod, df_rb[['CRP','disease_severity','DAS28-CRP']]], axis=1)\n",
    "\n",
    "# df_corr.head()\n",
    "\n",
    "df_corr = df_corr[df_corr['Diagnosis'] == 'RA']\n",
    "for x in [alpha_metric]:\n",
    "    for y in ['SJC','TJC','CRP','DAS28-CRP']:\n",
    "        r, p = scipy.stats.spearmanr(a=df_corr[x],b=df_corr[y])\n",
    "        # sns.lmplot(data=df_corr, x=x, y=y)\n",
    "        print(r, p)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_corr2 = df_corr.drop(['disease_severity'],axis=1)\n",
    "df_corr2 = df_corr2.rename({'DAS28-CRP':'DAS28_CRP'},axis=1)\n",
    "df_corr2.to_csv(path + 'outputs/jobs02/df_corr.tsv',sep='\\t')\n",
    "df_corr2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_corr3 = pd.concat([df_corr2.set_index('TwinPair'), df_paired_beta['Bray_Curtis']], axis=1)\n",
    "for x in ['Bray_Curtis']:\n",
    "    for y in ['SJC','TJC','CRP','DAS28_CRP']:\n",
    "        r, p = scipy.stats.spearmanr(a=df_corr3[x],b=df_corr3[y])\n",
    "        # ns.lmplot(data=df_corr3, x=x, y=y)\n",
    "        print(r, p)\n",
    "df_corr3.to_csv(path + 'outputs/jobs02/df_corr3.tsv', sep='\\t')\n",
    "df_corr3\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# process pathway results into lefse able format\n",
    "df = pd.read_csv(path + '/inputs/RA_Twins/16S/jobs/16_path/16_path_stats_clean.csv')\n",
    "df = df[df['wilcoxon'] <= 0.05]\n",
    "df['Group'] = df['wilcox stat'].map(lambda x: 'RA' if x >= 18 else 'Unaffected')\n",
    "df['wilcoxshift'] = df['wilcox stat'].map(lambda x: x-18)\n",
    "df_lp = df[['feature', 'wilcox stat', 'Group', 'wilcoxshift', 'wilcoxon']].set_index('feature')\n",
    "df_lp.to_csv(path + 'outputs/jobs06/pathway_results.tsv', sep='\\t', header=False)\n",
    "df_lp\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# comparin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
